diff -Naur orig/mkinitramfs mod/mkinitramfs
--- orig/mkinitramfs	2014-08-18 13:55:06.000000000 +0200
+++ mod/mkinitramfs	2016-05-06 07:50:52.948649265 +0200
@@ -132,6 +132,7 @@
 copy /sbin/swapoff
 
 # module-init-tools
+#kmod
 copy /sbin/insmod
 copy /sbin/modprobe
 copy /sbin/rmmod
@@ -252,13 +253,12 @@
 copy /bin/kill
 copy /bin/ps
 copy /sbin/sysctl
-copy /usr/bin/top
-copy /usr/bin/uptime
-copy /usr/bin/vmstat
-copy /usr/bin/w
-copy /usr/bin/watch
-
-copy /usr/bin/pidof
+copy /bin/top
+copy /bin/uptime
+copy /bin/vmstat
+copy /bin/w
+copy /bin/watch
+copy /bin/pidof
 
 # psmisc
 copy /bin/fuser
@@ -351,7 +351,7 @@
 # joe
 copy /usr/bin/joe
 
-copypkg module-init-tools
+copypkg kmod
 copypkg systemd
 copypkg libatasmart
 copypkg lvm
@@ -362,6 +362,7 @@
 copypkg attr
 copypkg polkit
 copypkg cryptsetup
+copypkg btrfs-progs
 
 # cleanup
 rm -rf "$dir"/usr/share/doc
