diff -Naur orig/mkinitramfs patched/mkinitramfs
--- orig/mkinitramfs	2014-08-18 13:55:06.000000000 +0200
+++ patched/mkinitramfs	2016-03-28 18:21:25.440293548 +0200
@@ -131,7 +131,7 @@
 copy /usr/bin/setsid
 copy /sbin/swapoff
 
-# module-init-tools
+# kmod
 copy /sbin/insmod
 copy /sbin/modprobe
 copy /sbin/rmmod
@@ -351,7 +351,7 @@
 # joe
 copy /usr/bin/joe
 
-copypkg module-init-tools
+copypkg kmod
 copypkg systemd
 copypkg libatasmart
 copypkg lvm
@@ -362,6 +362,7 @@
 copypkg attr
 copypkg polkit
 copypkg cryptsetup
+copypkg btrfs-progs
 
 # cleanup
 rm -rf "$dir"/usr/share/doc
