diff -Naur amarok-2.1.80-orig/cmake/modules/FindQtScriptQtBindings.cmake amarok-2.1.80/cmake/modules/FindQtScriptQtBindings.cmake
--- amarok-2.1.80-orig/cmake/modules/FindQtScriptQtBindings.cmake	2009-09-01 19:19:31.000000000 +0200
+++ amarok-2.1.80/cmake/modules/FindQtScriptQtBindings.cmake	2009-09-05 10:12:40.000000000 +0200
@@ -33,7 +33,7 @@
         set(QTSCRIPTQTBINDINGS_FOUND TRUE)
     else(BINDINGS_RUN_RESULT EQUAL 1)
         message( STATUS "QtBindings not found. run `cd cmake/modules/bindingstest; mkdir build; cd build; cmake ..; make; ./bindingstest; echo $?` If it prints '0' then you're actually fine.")
-        set(QTSCRIPTQTBINDINGS_FOUND FALSE)
+        set(QTSCRIPTQTBINDINGS_FOUND TRUE)
     endif(BINDINGS_RUN_RESULT EQUAL 1)
 
     set(CMAKE_REQUIRED_DEFINTIONS "" )
