diff -Naur orig/configure mod/configure
--- orig/configure	2012-09-19 23:53:40.000000000 +0200
+++ mod/configure	2016-04-10 16:23:01.086649512 +0200
@@ -14931,7 +14931,7 @@
 
 # The format below is used so that the requirements appear nicely in the .pc
 # file.
-GSTREAMERMM_MODULES='giomm-2.4 >= 2.28.0 libxml++-2.6 >= 2.14 gstreamer-0.10 >= 0.10.36 gstreamer-base-0.10 >= 0.10.36 gstreamer-controller-0.10 >= 0.10.36 gstreamer-dataprotocol-0.10 >= 0.10.36 gstreamer-net-0.10 >= 0.10.36 gstreamer-plugins-base-0.10 >= 0.10.36 gstreamer-audio-0.10 >= 0.10.36 gstreamer-cdda-0.10 >= 0.10.36 gstreamer-interfaces-0.10 >= 0.10.36 gstreamer-netbuffer-0.10 >= 0.10.36 gstreamer-rtp-0.10 >= 0.10.36 gstreamer-video-0.10 >= 0.10.36 gstreamer-pbutils-0.10 >= 0.10.36'
+GSTREAMERMM_MODULES='giomm-2.4 >= 2.28.0 libxml++-3.0 >= 2.14 gstreamer-0.10 >= 0.10.36 gstreamer-base-0.10 >= 0.10.36 gstreamer-controller-0.10 >= 0.10.36 gstreamer-dataprotocol-0.10 >= 0.10.36 gstreamer-net-0.10 >= 0.10.36 gstreamer-plugins-base-0.10 >= 0.10.36 gstreamer-audio-0.10 >= 0.10.36 gstreamer-cdda-0.10 >= 0.10.36 gstreamer-interfaces-0.10 >= 0.10.36 gstreamer-netbuffer-0.10 >= 0.10.36 gstreamer-rtp-0.10 >= 0.10.36 gstreamer-video-0.10 >= 0.10.36 gstreamer-pbutils-0.10 >= 0.10.36'
 
 
 
