diff -Naur orig/src/CMakeLists.txt patched/src/CMakeLists.txt
--- orig/src/CMakeLists.txt	2015-12-20 14:38:33.000000000 +0100
+++ patched/src/CMakeLists.txt	2015-12-22 00:19:24.227407400 +0100
@@ -5,7 +5,7 @@
 
 set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR})
 
-set(MOZPLUGIN_INSTALL_DIR "lib${LIB_SUFFIX}/mozilla/plugins/" CACHE PATH "plugin install directory")
+set(MOZPLUGIN_INSTALL_DIR "lib${LIB_SUFFIX}/mozilla/plugins/" CACHE STRING "plugin install directory")
 
 add_subdirectory(uri_parser)
 add_subdirectory(config_parser)
