diff -Naur orig/drivers/net/ethernet/dec/tulip/xircom_cb.c patched/drivers/net/ethernet/dec/tulip/xircom_cb.c
--- orig/drivers/net/ethernet/dec/tulip/xircom_cb.c	2015-05-17 18:55:21.000000000 +0200
+++ patched/drivers/net/ethernet/dec/tulip/xircom_cb.c	2015-05-22 17:47:49.915117501 +0200
@@ -139,6 +139,12 @@
 
 static const struct pci_device_id xircom_pci_table[] = {
 	{ PCI_VDEVICE(XIRCOM, 0x0003), },
+	{ PCI_VDEVICE(XIRCOM, 0x0005), },
+	{ PCI_VDEVICE(XIRCOM, 0x0007), },
+	{ PCI_VDEVICE(XIRCOM, 0x000b), },
+	{ PCI_VDEVICE(XIRCOM, 0x000f), },
+	{ PCI_VDEVICE(XIRCOM, 0x0101), },
+	{ PCI_VDEVICE(XIRCOM, 0x0103), },
 	{0,},
 };
 MODULE_DEVICE_TABLE(pci, xircom_pci_table);
diff -Naur orig/drivers/net/ethernet/via/via-rhine.c patched/drivers/net/ethernet/via/via-rhine.c
--- orig/drivers/net/ethernet/via/via-rhine.c	2015-05-17 18:55:21.000000000 +0200
+++ patched/drivers/net/ethernet/via/via-rhine.c	2015-05-22 17:47:49.915117501 +0200
@@ -278,6 +278,7 @@
 	{ 0x1106, 0x3065, PCI_ANY_ID, PCI_ANY_ID, },	/* VT6102 */
 	{ 0x1106, 0x3106, PCI_ANY_ID, PCI_ANY_ID, },	/* 6105{,L,LOM} */
 	{ 0x1106, 0x3053, PCI_ANY_ID, PCI_ANY_ID, },	/* VT6105M */
+	{ 0x1106, 0x6100, PCI_ANY_ID, PCI_ANY_ID, },
 	{ }	/* terminate list */
 };
 MODULE_DEVICE_TABLE(pci, rhine_pci_tbl);
diff -Naur orig/drivers/scsi/aacraid/linit.c patched/drivers/scsi/aacraid/linit.c
--- orig/drivers/scsi/aacraid/linit.c	2015-05-17 18:55:21.000000000 +0200
+++ patched/drivers/scsi/aacraid/linit.c	2015-05-22 17:47:49.915117501 +0200
@@ -149,6 +149,7 @@
 	{ 0x1011, 0x0046, 0x9005, 0x0365, 0, 0, 53 }, /* Adaptec 5400S (Mustang)*/
 	{ 0x1011, 0x0046, 0x9005, 0x0364, 0, 0, 54 }, /* Adaptec 5400S (Mustang)*/
 	{ 0x1011, 0x0046, 0x9005, 0x1364, 0, 0, 55 }, /* Dell PERC2/QC */
+	{ 0x1011, 0x0046, 0x9005, 0x1365, 0, 0, 39 },
 	{ 0x1011, 0x0046, 0x103c, 0x10c2, 0, 0, 56 }, /* HP NetRAID-4M */
 
 	{ 0x9005, 0x0285, 0x1028, PCI_ANY_ID, 0, 0, 57 }, /* Dell Catchall */
diff -Naur orig/drivers/staging/i2o/i2o_block.c patched/drivers/staging/i2o/i2o_block.c
--- orig/drivers/staging/i2o/i2o_block.c	2015-05-17 18:55:21.000000000 +0200
+++ patched/drivers/staging/i2o/i2o_block.c	2015-05-22 17:47:49.915117501 +0200
@@ -1226,3 +1226,13 @@
 
 module_init(i2o_block_init);
 module_exit(i2o_block_exit);
+
+static struct pci_device_id i2o_block_pci_table[] = {
+	{0x8086, 0x3092, PCI_ANY_ID, PCI_ANY_ID},
+	{0x8086, 0x9621, PCI_ANY_ID, PCI_ANY_ID},
+	{0x8086, 0x9622, PCI_ANY_ID, PCI_ANY_ID},
+	{0x8086, 0x9641, PCI_ANY_ID, PCI_ANY_ID},
+	{0x8086, 0x96a1, PCI_ANY_ID, PCI_ANY_ID},
+	{0}
+};
+MODULE_DEVICE_TABLE(pci, i2o_block_pci_table);
diff -Naur orig/sound/pci/ymfpci/ymfpci.c patched/sound/pci/ymfpci/ymfpci.c
--- orig/sound/pci/ymfpci/ymfpci.c	2015-05-17 18:55:21.000000000 +0200
+++ patched/sound/pci/ymfpci/ymfpci.c	2015-05-22 17:47:49.915117501 +0200
@@ -67,12 +67,17 @@
 MODULE_PARM_DESC(rear_switch, "Enable shared rear/line-in switch");
 
 static const struct pci_device_id snd_ymfpci_ids[] = {
+	{ PCI_VDEVICE(YAMAHA, 0x0003), 0, },   /* YMF740 */
 	{ PCI_VDEVICE(YAMAHA, 0x0004), 0, },   /* YMF724 */
-	{ PCI_VDEVICE(YAMAHA, 0x000d), 0, },   /* YMF724F */
+	{ PCI_VDEVICE(YAMAHA, 0x0005), 0, },
+	{ PCI_VDEVICE(YAMAHA, 0x0006), 0, },
+	{ PCI_VDEVICE(YAMAHA, 0x0008), 0, },
 	{ PCI_VDEVICE(YAMAHA, 0x000a), 0, },   /* YMF740 */
 	{ PCI_VDEVICE(YAMAHA, 0x000c), 0, },   /* YMF740C */
+	{ PCI_VDEVICE(YAMAHA, 0x000d), 0, },   /* YMF724F */
 	{ PCI_VDEVICE(YAMAHA, 0x0010), 0, },   /* YMF744 */
 	{ PCI_VDEVICE(YAMAHA, 0x0012), 0, },   /* YMF754 */
+	{ PCI_VDEVICE(YAMAHA, 0x0020), 0, },
 	{ 0, }
 };
 
@@ -197,12 +202,17 @@
 		return err;
 
 	switch (pci_id->device) {
+	case 0x0003: str = "YMF740";  model = "DS-1"; break;
 	case 0x0004: str = "YMF724";  model = "DS-1"; break;
-	case 0x000d: str = "YMF724F"; model = "DS-1"; break;
+	case 0x0005: str = "YMF???";  model = "DS-1"; break;
+	case 0x0006: str = "YMF???";  model = "DS-1"; break;
+	case 0x0008: str = "YMF???";  model = "DS-1"; break;
 	case 0x000a: str = "YMF740";  model = "DS-1L"; break;
 	case 0x000c: str = "YMF740C"; model = "DS-1L"; break;
+	case 0x000d: str = "YMF724F"; model = "DS-1"; break;
 	case 0x0010: str = "YMF744";  model = "DS-1S"; break;
 	case 0x0012: str = "YMF754";  model = "DS-1E"; break;
+	case 0x0020: str = "YMF???";  model = "DS-1"; break;
 	default: model = str = "???"; break;
 	}
 
