diff -Naur orig/misc/syntax/php.syntax patched/misc/syntax/php.syntax
--- orig/misc/syntax/php.syntax	2010-09-07 09:16:43.000000000 +0200
+++ patched/misc/syntax/php.syntax	2010-09-14 13:21:11.000000000 +0200
@@ -3078,6 +3078,8 @@
 
     keyword wholeright $+ brightgreen
 
+    keyword " green
+    keyword '*' brightgreen
     keyword /\* brown
     keyword \*/ brown
     keyword // brown
@@ -3133,7 +3135,7 @@
 context linestart # \n brown
     spellcheck
 
-context " " green
+context exclusive " " green
     spellcheck
     keyword \\" brightgreen
     keyword \\\n brightgreen
@@ -3149,6 +3151,3 @@
     keyword \\' brightgreen
     keyword \\\{abtnvfr\} brightgreen
 
-context ' ' brightgreen
-    spellcheck
-    keyword \\' brightcyan
