diff -Naur orig/Source_Files/XML/InfoTree.cpp patched/Source_Files/XML/InfoTree.cpp
--- orig/Source_Files/XML/InfoTree.cpp	2015-08-31 22:35:35.000000000 +0200
+++ patched/Source_Files/XML/InfoTree.cpp	2016-07-16 01:43:50.739113550 +0200
@@ -26,6 +26,7 @@
 #include "shell.h"
 #include "TextStrings.h"
 
+#include <boost/version.hpp> 
 #include <boost/function.hpp>
 #include <boost/iterator/filter_iterator.hpp>
 #include <boost/iterator/transform_iterator.hpp>
@@ -77,14 +78,22 @@
 	boost::property_tree::write_xml(filename.GetPath(),
 									boost::property_tree::ptree(*this),
 									std::locale(),
+#if BOOST_VERSION >= 105600
+									boost::property_tree::xml_writer_make_settings<boost::property_tree::ptree::key_type>(' ', 2));
+#else
 									boost::property_tree::xml_writer_make_settings(' ', 2));
+#endif
 }
 
 void InfoTree::save_xml(std::ostringstream& stream) const
 {
 	boost::property_tree::write_xml(stream,
 									boost::property_tree::ptree(*this),
+#if BOOST_VERSION >= 105600
+									boost::property_tree::xml_writer_make_settings<boost::property_tree::ptree::key_type>(' ', 2));
+#else
 									boost::property_tree::xml_writer_make_settings(' ', 2));
+#endif
 }
 
 InfoTree InfoTree::load_ini(FileSpecifier filename)
