diff -Naur orig/Source_Files/FFmpeg/Movie.cpp patched/Source_Files/FFmpeg/Movie.cpp
--- orig/Source_Files/FFmpeg/Movie.cpp	2015-08-31 22:35:35.000000000 +0200
+++ patched/Source_Files/FFmpeg/Movie.cpp	2016-07-15 10:37:52.461999387 +0200
@@ -420,7 +420,7 @@
         video_stream->codec->width = view_rect.w;
         video_stream->codec->height = view_rect.h;
         video_stream->codec->time_base = (AVRational){1, TICKS_PER_SECOND};
-        video_stream->codec->pix_fmt = PIX_FMT_YUV420P;
+        video_stream->codec->pix_fmt = AV_PIX_FMT_YUV420P;
         video_stream->codec->flags |= CODEC_FLAG_CLOSED_GOP;
         video_stream->codec->thread_count = get_cpu_count();
         
@@ -539,7 +539,7 @@
     // initialize conversion context
     if (success)
     {
-        av->sws_ctx = sws_getContext(temp_surface->w, temp_surface->h, PIX_FMT_RGB32,
+        av->sws_ctx = sws_getContext(temp_surface->w, temp_surface->h, AV_PIX_FMT_RGB32,
                                      video_stream->codec->width,
                                      video_stream->codec->height,
                                      video_stream->codec->pix_fmt,
diff -Naur orig/Source_Files/FFmpeg/SDL_ffmpeg.c patched/Source_Files/FFmpeg/SDL_ffmpeg.c
--- orig/Source_Files/FFmpeg/SDL_ffmpeg.c	2015-09-01 00:18:43.000000000 +0200
+++ patched/Source_Files/FFmpeg/SDL_ffmpeg.c	2016-07-15 12:48:27.275013823 +0200
@@ -86,7 +86,7 @@
  *  Provide a fast way to get the correct context.
  *  \returns The context matching the input values.
  */
-struct SwsContext* getContext( SDL_ffmpegConversionContext **context, int inWidth, int inHeight, enum PixelFormat inFormat, int outWidth, int outHeight, enum PixelFormat outFormat )
+struct SwsContext* getContext( SDL_ffmpegConversionContext **context, int inWidth, int inHeight, enum AVPixelFormat inFormat, int outWidth, int outHeight, enum AVPixelFormat outFormat )
 {
     SDL_ffmpegConversionContext *ctx = *context;
 
@@ -621,7 +621,7 @@
     {
         case 24:
             sws_scale( getContext( &file->videoStream->conversionContext,
-                                   frame->w, frame->h, PIX_FMT_RGB24,
+                                   frame->w, frame->h, AV_PIX_FMT_RGB24,
                                    file->videoStream->_ffmpeg->codec->width,
                                    file->videoStream->_ffmpeg->codec->height,
                                    file->videoStream->_ffmpeg->codec->pix_fmt ),
@@ -634,7 +634,7 @@
             break;
         case 32:
             sws_scale( getContext( &file->videoStream->conversionContext,
-                                   frame->w, frame->h, PIX_FMT_BGR32,
+                                   frame->w, frame->h, AV_PIX_FMT_BGR32,
                                    file->videoStream->_ffmpeg->codec->width,
                                    file->videoStream->_ffmpeg->codec->height,
                                    file->videoStream->_ffmpeg->codec->pix_fmt ),
@@ -2223,7 +2223,7 @@
                                    file->videoStream->_ffmpeg->codec->height,
                                    file->videoStream->_ffmpeg->codec->pix_fmt,
                                    frame->overlay->w, frame->overlay->h,
-                                   PIX_FMT_YUYV422 ),
+                                   AV_PIX_FMT_YUYV422 ),
                        ( const uint8_t* const* )file->videoStream->decodeFrame->data,
                        file->videoStream->decodeFrame->linesize,
                        0,
@@ -2245,7 +2245,7 @@
                                            file->videoStream->_ffmpeg->codec->height,
                                            file->videoStream->_ffmpeg->codec->pix_fmt,
                                            frame->surface->w, frame->surface->h,
-                                           PIX_FMT_RGB32 ),
+                                           AV_PIX_FMT_RGB32 ),
                                ( const uint8_t* const* )file->videoStream->decodeFrame->data,
                                file->videoStream->decodeFrame->linesize,
                                0,
@@ -2259,7 +2259,7 @@
                                            file->videoStream->_ffmpeg->codec->height,
                                            file->videoStream->_ffmpeg->codec->pix_fmt,
                                            frame->surface->w, frame->surface->h,
-                                           PIX_FMT_RGB24 ),
+                                           AV_PIX_FMT_RGB24 ),
                                ( const uint8_t* const* )file->videoStream->decodeFrame->data,
                                file->videoStream->decodeFrame->linesize,
                                0,
