diff -Naur orig/qtbase/config.tests/unix/alsa/alsatest.cpp patched/qtbase/config.tests/unix/alsa/alsatest.cpp
--- orig/qtbase/config.tests/unix/alsa/alsatest.cpp	2016-03-03 12:35:22.000000000 +0100
+++ patched/qtbase/config.tests/unix/alsa/alsatest.cpp	2016-04-17 13:49:44.378258307 +0200
@@ -32,7 +32,8 @@
 ****************************************************************************/
 
 #include <alsa/asoundlib.h>
-#if(!(SND_LIB_MAJOR == 1 && SND_LIB_MINOR == 0 && SND_LIB_SUBMINOR >= 10))
+#if(!(SND_LIB_MAJOR == 1 && SND_LIB_MINOR == 1 && SND_LIB_SUBMINOR >= 0) && \
+	!(SND_LIB_MAJOR == 1 && SND_LIB_MINOR == 0 && SND_LIB_SUBMINOR >= 10))
 #error "Alsa version found too old, require >= 1.0.10"
 #endif
 
