diff -up a/qtbase/mkspecs/common/gcc-base.conf.than b/qtbase/mkspecs/common/gcc-base.conf
--- a/qtbase/mkspecs/common/gcc-base.conf	2016-06-02 17:30:07.249027901 +0200
+++ b/qtbase/mkspecs/common/gcc-base.conf	2016-06-02 17:30:14.681748012 +0200
@@ -32,7 +32,7 @@
 #
 
 QMAKE_CFLAGS_OPTIMIZE      = -O2
-QMAKE_CFLAGS_OPTIMIZE_FULL = -O3
+QMAKE_CFLAGS_OPTIMIZE_FULL = -O2
 
 QMAKE_CFLAGS               += -pipe
 QMAKE_CFLAGS_DEPS          += -M
