diff -Naur coreutils-8.25-orig/src/stty.c coreutils-8.25/src/stty.c
--- coreutils-8.25-orig/src/stty.c	2016-01-07 13:32:35.000000000 +0100
+++ coreutils-8.25/src/stty.c	2016-02-14 18:31:55.690873833 +0100
@@ -248,7 +248,7 @@
   {"imaxbel", input, SANE_SET | REV, IMAXBEL, 0},
 #endif
 #ifdef IUTF8
-  {"iutf8", input, SANE_UNSET | REV, IUTF8, 0},
+  {"iutf8", input, SANE_SET | REV, IUTF8, 0},
 #endif
 
   {"opost", output, SANE_SET | REV, OPOST, 0},
