diff -Naur orig/src/layout/drawabletuplet.cpp patched/src/layout/drawabletuplet.cpp
--- orig/src/layout/drawabletuplet.cpp	2015-04-11 11:06:45.000000000 +0200
+++ patched/src/layout/drawabletuplet.cpp	2016-11-24 17:25:15.774764730 +0100
@@ -10,13 +10,14 @@
 #include <QPen>
 #include <QPainter>
 #include <QFont>
+#include <math.h>
 
 CADrawableTuplet::CADrawableTuplet( CATuplet *tuplet, CADrawableContext *c, double x1, double y1, double x2, double y2 )
  : CADrawableMusElement( tuplet, c, x1, 0 ) {
 	setDrawableMusElementType( DrawableTuplet );
 
 	setWidth( x2-x1 );
-	setHeight( (abs(y2-y1)>5)?abs(y2-y1):8 );
+	setHeight( (abs(y2-y1)>5)?fabs(y2-y1):8 );
 	setYPos( (c && qMin(y1, y2)>c->yPos())?qMin(y1, y2):(qMin(y1,y2)-height()) );
 }
 
diff -Naur orig/src/score/sheet.cpp patched/src/score/sheet.cpp
--- orig/src/score/sheet.cpp	2015-04-11 11:06:44.000000000 +0200
+++ patched/src/score/sheet.cpp	2016-11-24 17:18:04.683130652 +0100
@@ -6,6 +6,7 @@
 */
 
 #include <QHash>
+#include <QObject>
 
 #include "score/document.h"
 #include "score/context.h"
