diff -Naur orig/configure.ac mod/configure.ac
--- orig/configure.ac	2015-12-15 12:29:06.000000000 +0100
+++ mod/configure.ac	2016-01-11 20:10:38.541410963 +0100
@@ -2212,7 +2212,7 @@
     dnl (See https://llvm.org/bugs/show_bug.cgi?id=6823)
     if test "x$enable_llvm_shared_libs" = xyes; then
         dnl We can't use $LLVM_VERSION because it has 'svn' stripped out,
-        LLVM_SO_NAME=LLVM-`$LLVM_CONFIG --version`
+        LLVM_SO_NAME=LLVM
         AS_IF([test -f "$LLVM_LIBDIR/lib$LLVM_SO_NAME.$IMP_LIB_EXT"], [llvm_have_one_so=yes])
 
         if test "x$llvm_have_one_so" = xyes; then
