diff -Naur orig/cpufreq/src/cpufreq-monitor-libcpufreq.c patched/cpufreq/src/cpufreq-monitor-libcpufreq.c
--- orig/cpufreq/src/cpufreq-monitor-libcpufreq.c	2017-01-14 21:07:32.000000000 +0100
+++ patched/cpufreq/src/cpufreq-monitor-libcpufreq.c	2017-03-18 04:05:26.766985429 +0100
@@ -25,6 +25,7 @@
 #include <stdlib.h>
 #include <linux/version.h>
 #include <cpufreq.h>
+#include <cpupower.h>
 #include "cpufreq-monitor-libcpufreq.h"
 #include "cpufreq-utils.h"
 
@@ -111,11 +112,7 @@
 		/* Check whether it failed because
 		 * cpu is not online.
 		 */
-#if LINUX_VERSION_CODE < KERNEL_VERSION(4, 7, 0)
-		if (!cpufreq_cpu_exists (cpu)) {
-#else
 		if (cpupower_is_cpu_online (cpu)) {
-#endif
 			g_object_set (G_OBJECT (monitor), "online", FALSE, NULL);
 			return TRUE;
 		}
