diff -Naur orig/ipc/chromium/src/base/atomicops_internals_mutex.cc patched/ipc/chromium/src/base/atomicops_internals_mutex.cc
--- orig/ipc/chromium/src/base/atomicops_internals_mutex.cc	2014-01-28 05:03:44.000000000 +0100
+++ patched/ipc/chromium/src/base/atomicops_internals_mutex.cc	2014-02-05 16:50:13.168717289 +0100
@@ -6,6 +6,7 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 #include "base/atomicops.h"
+#include "base/lock.h"
 
 namespace base {
 namespace subtle {
diff -Naur orig/ipc/chromium/src/base/file_util.cc patched/ipc/chromium/src/base/file_util.cc
--- orig/ipc/chromium/src/base/file_util.cc	2012-11-20 03:46:00.000000000 +0100
+++ patched/ipc/chromium/src/base/file_util.cc	2012-11-21 12:08:21.528937491 +0100
@@ -8,7 +8,7 @@
 #include <io.h>
 #endif
 #include <stdio.h>
-#if defined(ANDROID) || defined(OS_POSIX)
+#if defined(ANDROID) || defined(OS_POSIX) || defined(OS_LINUX)
 #include <unistd.h>
 #endif
 
diff -Naur orig/ipc/chromium/src/base/message_pump_libevent.cc patched/ipc/chromium/src/base/message_pump_libevent.cc
--- orig/ipc/chromium/src/base/message_pump_libevent.cc	2012-11-20 03:46:00.000000000 +0100
+++ patched/ipc/chromium/src/base/message_pump_libevent.cc	2012-11-21 12:07:18.812626506 +0100
@@ -6,7 +6,7 @@
 
 #include <errno.h>
 #include <fcntl.h>
-#if defined(ANDROID) || defined(OS_POSIX)
+#if defined(ANDROID) || defined(OS_POSIX) || defined(OS_LINUX)
 #include <unistd.h>
 #endif
 
diff -Naur orig/ipc/chromium/src/base/time_posix.cc patched/ipc/chromium/src/base/time_posix.cc
--- orig/ipc/chromium/src/base/time_posix.cc	2012-11-20 03:46:00.000000000 +0100
+++ patched/ipc/chromium/src/base/time_posix.cc	2012-11-21 12:07:18.812626506 +0100
@@ -13,7 +13,7 @@
 #else
 #include <time.h>
 #endif
-#if defined(ANDROID) || defined(OS_POSIX)
+#if defined(ANDROID) || defined(OS_POSIX) || defined(OS_LINUX)
 #include <unistd.h>
 #endif
 
