diff -Naur pmtools-orig/madt/Makefile pmtools/madt/Makefile
--- pmtools-orig/madt/Makefile	2005-11-11 18:07:04.000000000 +0100
+++ pmtools/madt/Makefile	2012-03-22 21:05:00.000000000 +0100
@@ -2,5 +2,6 @@
 	./madt < APIC
 madt: madt.c acpi.h tables.c
 	cc -o madt madt.c
+all: madt
 clean:
 	rm madt
diff -Naur pmtools-orig/Makefile pmtools/Makefile
--- pmtools-orig/Makefile	2010-01-23 19:18:18.000000000 +0100
+++ pmtools/Makefile	2012-03-22 21:05:21.000000000 +0100
@@ -1,4 +1,4 @@
-DIRS = acpidump acpixtract turbostat
+DIRS = acpidump acpixtract madt turbostat
 
 all:
 	for i in $(DIRS); do $(MAKE) -C $$i $@; done
diff -Naur pmtools-orig/turbostat/Makefile pmtools/turbostat/Makefile
--- pmtools-orig/turbostat/Makefile	2010-11-24 06:36:50.000000000 +0100
+++ pmtools/turbostat/Makefile	2012-03-22 21:09:46.000000000 +0100
@@ -1,5 +1,5 @@
 turbostat : turbostat.c
-
+all: turbostat
 clean :
 	rm -f turbostat
 
