diff -Naurdp dhcp-4.2.5-P1.orig/client/clparse.c dhcp-4.2.5-P1/client/clparse.c
--- dhcp-4.2.5-P1.orig/client/clparse.c	2013-03-04 19:35:08.000000000 +0100
+++ dhcp-4.2.5-P1/client/clparse.c	2013-03-27 15:12:09.360445375 +0100
@@ -132,10 +132,10 @@ isc_result_t read_client_conf ()
 	/* Set some defaults... */
 	top_level_config.timeout = 60;
 	top_level_config.select_interval = 0;
-	top_level_config.reboot_timeout = 10;
+	top_level_config.reboot_timeout = 5;
 	top_level_config.retry_interval = 300;
-	top_level_config.backoff_cutoff = 15;
-	top_level_config.initial_interval = 3;
+	top_level_config.backoff_cutoff = 10;
+	top_level_config.initial_interval = 1;
 
 	/*
 	 * RFC 2131, section 4.4.1 specifies that the client SHOULD wait a
