diff -Naur orig/setup.py patched/setup.py
--- orig/setup.py	2016-05-21 00:04:28.000000000 +0200
+++ patched/setup.py	2016-05-22 17:15:28.213233124 +0200
@@ -57,9 +57,9 @@
     ('{prefix}/share/applications'.format(prefix=sys.prefix), ['data/mate-tweak.desktop',]),
     ('{prefix}/share/applications'.format(prefix=sys.prefix), ['data/mate-tweak.desktop',]),
     ('{prefix}/share/mate/applications'.format(prefix=sys.prefix), ['data/marco-compton.desktop',]),
-    ('{prefix}/share/mate/applications'.format(prefix=sys.prefix), ['data/marco-xcompmgr.desktop',]),
+#    ('{prefix}/share/mate/applications'.format(prefix=sys.prefix), ['data/marco-xcompmgr.desktop',]),
     ('{prefix}/share/mate/applications'.format(prefix=sys.prefix), ['data/metacity-compton.desktop',]),
-    ('{prefix}/share/mate/applications'.format(prefix=sys.prefix), ['data/metacity-xcompmgr.desktop',]),
+#    ('{prefix}/share/mate/applications'.format(prefix=sys.prefix), ['data/metacity-xcompmgr.desktop',]),
     ('{prefix}/share/mate/applications'.format(prefix=sys.prefix), ['data/marco-no-composite.desktop',]),
     ('{prefix}/share/mate/applications'.format(prefix=sys.prefix), ['data/metacity-no-composite.desktop',]),
     ('{prefix}/share/mate-tweak'.format(prefix=sys.prefix), ['data/mate-volume-control-applet.desktop',]),
