diff -up gtksourceview-1.8.5/gtksourceview/gtksourceregex.c.use-gnu gtksourceview-1.8.5/gtksourceview/gtksourceregex.c
--- gtksourceview-1.8.5/gtksourceview/gtksourceregex.c.use-gnu	2007-01-03 10:40:28.000000000 -0500
+++ gtksourceview-1.8.5/gtksourceview/gtksourceregex.c	2008-02-15 00:29:54.000000000 -0500
@@ -18,6 +18,8 @@
  *  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
  */
 
+#define _GNU_SOURCE 1
+
 #ifdef HAVE_CONFIG_H
 #include <config.h>
 #endif
@@ -27,6 +29,7 @@
 #include <glib.h>
 
 #ifdef NATIVE_GNU_REGEX
+
 #include <sys/types.h>
 #include <regex.h>
 #else
