diff -Nur orig/src/translations/baka-mplayer_hu.ts mod/src/translations/baka-mplayer_hu.ts
--- orig/src/translations/baka-mplayer_hu.ts	1970-01-01 01:00:00.000000000 +0100
+++ mod/src/translations/baka-mplayer_hu.ts	2019-03-08 12:43:06.000000000 +0100
@@ -0,0 +1,2028 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!DOCTYPE TS>
+<TS version="2.1" language="hu_HU">
+<context>
+    <name>AboutDialog</name>
+    <message>
+        <location filename="../ui/aboutdialog.ui" line="14"/>
+        <source>About Baka MPlayer</source>
+        <translation>A Baka Mplayer</translation>
+    </message>
+    <message>
+        <location filename="../ui/aboutdialog.ui" line="132"/>
+        <source>About</source>
+        <translation>Röviden</translation>
+    </message>
+    <message>
+        <location filename="../ui/aboutdialog.ui" line="185"/>
+        <source>Baka MPlayer 0.0.0</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../ui/aboutdialog.ui" line="195"/>
+        <source>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;b&gt;Baka MPlayer&lt;/b&gt; is a free and open source, cross-platform, &lt;b&gt;libmpv&lt;/b&gt; based multimedia player. Its simple design reflects the idea for an uncluttered, simple, and enjoyable environment for watching tv shows.&lt;br&gt;&lt;br&gt;Copyright © 2015 godly-devotion, u8sand&lt;br&gt;&lt;br&gt;&lt;a href=&quot;https://github.com/u8sand/Baka-MPlayer&quot;&gt;&lt;span style=&quot; text-decoration: underline; color:#008098;&quot;&gt;https://github.com/u8sand/Baka-MPlayer&lt;/span&gt;&lt;/a&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</source>
+        <translation>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;b&gt;A Baka MPlayer&lt;/b&gt; szabad és nyílt forrású, &lt;b&gt;libmpv&lt;/b&gt; alapú multimédia lejátszó. Egyszerű kinézetű és csak az alapvető funkciókra koncentrál.&lt;br&gt;&lt;br&gt;Copyright © 2016 godly-devotion, u8sand&lt;br&gt;&lt;br&gt;&lt;a href=&quot;https://github.com/u8sand/Baka-MPlayer&quot;&gt;&lt;span style=&quot; text-decoration: underline; color:#008098;&quot;&gt;https://github.com/u8sand/Baka-MPlayer&lt;/span&gt;&lt;/a&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</translation>
+    </message>
+    <message>
+        <location filename="../ui/aboutdialog.ui" line="209"/>
+        <source>Credits</source>
+        <translation>Részletek</translation>
+    </message>
+    <message>
+        <location filename="../ui/aboutdialog.ui" line="233"/>
+        <source>mpv - video player based on MPlayer/mplayer2
+GPLv2 &lt;mpv.io&gt;
+
+Material Design icons
+Attribution 4.0 International &lt;Google&gt;
+
+Noto Sans -  font family with many supported characters
+Apache License, version 2.0 &lt;Google&gt;
+
+Retro Cassette image
+&lt;Lukas Troup&gt;
+
+Gesture icons
+CC License &lt;Jeff Portaro&gt;
+
+Download icon
+CC BY 3.0 &lt;Sasha Mescheryakov&gt;
+
+Translations
+- Chinese &lt;amazingfate, Antares95&gt;
+- Croatian &lt;gogo&gt;
+- Dutch &lt;robin007bond&gt;
+- French &lt;chapouvalpin&gt;
+- German: &lt;yiip87&gt;
+- Italian &lt;Aloysius&gt;
+- Korean &lt;godly-devotion&gt;
+- Portuguese &lt;u8sand&gt;
+- Russian &lt;suhr&gt;
+- Spanish &lt;Alfredo Ramos&gt;</source>
+        <translation>mpv - videó lejátszó MPlayer/mplayer2 alapon
+GPLv2 &lt;mpv.io&gt;
+
+Material Design icons
+Attribution 4.0 International &lt;Google&gt;
+
+Noto Sans -  betűkészlet sok karakterrel
+Apache License, version 2.0 &lt;Google&gt;
+
+Retro Cassette image
+&lt;Lukas Troup&gt;
+
+Gesture icons
+CC License &lt;Jeff Portaro&gt;
+
+Download icon
+CC BY 3.0 &lt;Sasha Mescheryakov&gt;
+
+Translations
+- Chinese &lt;amazingfate, Antares95&gt;
+- Croatian &lt;gogo&gt;
+- Dutch &lt;robin007bond&gt;
+- French &lt;chapouvalpin&gt;
+- German: &lt;yiip87&gt;
+- Italian &lt;Aloysius&gt;
+- Korean &lt;godly-devotion&gt;
+- Portuguese &lt;u8sand&gt;
+- Russian &lt;suhr&gt;
+- Spanish &lt;Alfredo Ramos&gt;
+- Huingarian &lt;Attila Kiss&gt;</translation>
+    </message>
+    <message>
+        <location filename="../ui/aboutdialog.ui" line="269"/>
+        <source>License</source>
+        <translation>Licensz</translation>
+    </message>
+    <message>
+        <location filename="../ui/aboutdialog.ui" line="293"/>
+        <source>                GNU GENERAL PUBLIC LICENSE
+                   Version 2, June 1991
+
+This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation; either version 2 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program; if not, write to the Free Software
+Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../ui/aboutdialog.ui" line="334"/>
+        <source>&amp;Close</source>
+        <translation>Be&amp;csuk</translation>
+    </message>
+</context>
+<context>
+    <name>BakaEngine</name>
+    <message>
+        <location filename="../bakacommands.cpp" line="101"/>
+        <source>Open Subtitle File</source>
+        <translation>Felirat fájl megnyitás</translation>
+    </message>
+    <message>
+        <location filename="../bakacommands.cpp" line="102"/>
+        <source>Subtitle Files</source>
+        <translation>Felirat fájlok</translation>
+    </message>
+    <message>
+        <location filename="../bakacommands.cpp" line="116"/>
+        <source>Open Audio File</source>
+        <translation>Hangfájl megyitás</translation>
+    </message>
+    <message>
+        <location filename="../bakacommands.cpp" line="117"/>
+        <location filename="../bakacommands.cpp" line="390"/>
+        <source>Audio Files</source>
+        <translation>Hangfájlok</translation>
+    </message>
+    <message>
+        <location filename="../bakacommands.cpp" line="162"/>
+        <source>Saved to &quot;%0&quot;, with subs</source>
+        <translation>Mentve mint &quot;%0&quot; felirattal</translation>
+    </message>
+    <message>
+        <location filename="../bakacommands.cpp" line="164"/>
+        <source>Saved to &quot;%0&quot;, without subs</source>
+        <translation>Mentve mint &quot;%0&quot; felirat nélkül</translation>
+    </message>
+    <message>
+        <location filename="../bakacommands.cpp" line="313"/>
+        <source>DimDialog not supported on this platform</source>
+        <translation>Asztal kioltás funció erre a plattformra nem elérhető</translation>
+    </message>
+    <message>
+        <location filename="../bakacommands.cpp" line="326"/>
+        <source>Dim Lights</source>
+        <translation>Asztal kioltás</translation>
+    </message>
+    <message>
+        <location filename="../bakacommands.cpp" line="326"/>
+        <source>In order to dim the lights, the desktop compositor has to be enabled. This can be done through Window Manager Desktop.</source>
+        <translation>Az asztal lekapcsolási effektushoz az asztalkezelő kompozitornak aktiváltnak kell lennie. Ez az ablakkezelő beállításainál lehetséges.</translation>
+    </message>
+    <message>
+        <location filename="../bakacommands.cpp" line="354"/>
+        <source>http://bakamplayer.u8sand.net/help.php</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../bakacommands.cpp" line="387"/>
+        <source>Open File</source>
+        <translation>Fájlnyitás</translation>
+    </message>
+    <message>
+        <location filename="../bakacommands.cpp" line="388"/>
+        <source>Media Files</source>
+        <translation>Médiafájlok</translation>
+    </message>
+    <message>
+        <location filename="../bakacommands.cpp" line="389"/>
+        <source>Video Files</source>
+        <translation>Videófájlok</translation>
+    </message>
+    <message>
+        <location filename="../bakacommands.cpp" line="391"/>
+        <source>All Files</source>
+        <translation>Fájlok</translation>
+    </message>
+    <message>
+        <location filename="../bakacommands.cpp" line="510"/>
+        <source>Fit Window: %0</source>
+        <translation>Ablak kitöltés: %0</translation>
+    </message>
+    <message>
+        <location filename="../bakacommands.cpp" line="510"/>
+        <source>To Current Size</source>
+        <translation>Aktuálisra</translation>
+    </message>
+    <message>
+        <location filename="../bakacommands.cpp" line="569"/>
+        <source>Speed: %0x</source>
+        <translation>Sebesség: %0x</translation>
+    </message>
+    <message>
+        <location filename="../bakacommands.cpp" line="584"/>
+        <source>Press ESC or double-click to leave full screen</source>
+        <translation>ESC vagy dupla klikk a telli képmód elhagyásához</translation>
+    </message>
+    <message>
+        <location filename="../bakacommands.cpp" line="604"/>
+        <source>Press %0 to show all controls again</source>
+        <translation>%0 nyomásra minden vezérlőelem látszik</translation>
+    </message>
+    <message>
+        <location filename="../bakacommands.cpp" line="637"/>
+        <source>usage: baka &lt;command&gt; [...]</source>
+        <translation>használat: baka &lt;parancs&gt; [...]</translation>
+    </message>
+    <message>
+        <location filename="../bakacommands.cpp" line="638"/>
+        <source>commands:</source>
+        <translation>parancsok:</translation>
+    </message>
+    <message>
+        <location filename="../bakacommands.cpp" line="659"/>
+        <source>usage: %0 %1</source>
+        <translation>használat: %0 %1</translation>
+    </message>
+    <message>
+        <location filename="../bakacommands.cpp" line="660"/>
+        <source>description:</source>
+        <translation>leírás:</translation>
+    </message>
+    <message>
+        <location filename="../bakacommands.cpp" line="664"/>
+        <source>advanced:</source>
+        <translation>haladó:</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.cpp" line="113"/>
+        <source>invalid command &apos;%0&apos;</source>
+        <translation>ismeretlen parancs &apos;%0&apos;</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.cpp" line="118"/>
+        <source>invalid parameter &apos;%0&apos;</source>
+        <translation>ismeretlen paraméter &apos;%0&apos;</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.cpp" line="123"/>
+        <source>&apos;%0&apos; requires parameters</source>
+        <translation>A &apos;%0&apos; paraméteres</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.h" line="48"/>
+        <source>Increase sub size</source>
+        <translation>Felirat nagyítás</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.h" line="49"/>
+        <source>Decrease sub size</source>
+        <translation>Felirat kicsinyítés</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.h" line="50"/>
+        <source>Toggle subtitle visibility</source>
+        <translation>Alcímki/be</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.h" line="51"/>
+        <source>Restart playback</source>
+        <translation>Újrajátszás</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.h" line="52"/>
+        <source>Go to next chapter</source>
+        <translation>Következő fejezetre</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.h" line="53"/>
+        <source>Go to previous chapter</source>
+        <translation>Előző fejezetre</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.h" line="54"/>
+        <source>Seek forwards by %0 sec</source>
+        <translation>Előre %0 másodperccel</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.h" line="55"/>
+        <source>Seek backwards by %0 sec</source>
+        <translation>Vissza %0 másodperccel</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.h" line="56"/>
+        <source>Frame step backwards</source>
+        <translation>Képváltásra vissza</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.h" line="57"/>
+        <source>Frame step</source>
+        <translation>Képváltásonkénti lép</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.h" line="58"/>
+        <source>Toggle mute audio</source>
+        <translation>Némítás ki/be</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.h" line="59"/>
+        <source>Take screenshot with subtitles</source>
+        <translation>Alfeliratos képernyőkép</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.h" line="60"/>
+        <source>Take screenshot without subtitles</source>
+        <translation>Alfelirat nélküli képernyőkép</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.h" line="61"/>
+        <source>Decrease volume</source>
+        <translation>Halkítás</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.h" line="62"/>
+        <source>Increase volume</source>
+        <translation>Hangnövelés</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.h" line="63"/>
+        <source>Increase playback speed by %0</source>
+        <translation>Gyorsítás ennyivel: %0</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.h" line="64"/>
+        <source>Decrease playback speed by %0</source>
+        <translation>Lassítás ennyivel: %0</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.h" line="65"/>
+        <source>Reset speed</source>
+        <translation>Sebesség reset</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.h" line="66"/>
+        <source>Toggle fullscreen</source>
+        <translation>Teli kép ki/be</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.h" line="67"/>
+        <source>Toggle hide all controls mode</source>
+        <translation>Minden vezérlés ki/be</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.h" line="68"/>
+        <source>Dim lights</source>
+        <translation>Asztal láthatóság</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.h" line="69"/>
+        <source>Show the file in its folder</source>
+        <translation>E mappa fájljai</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.h" line="70"/>
+        <source>View media information</source>
+        <translation>Média tulajdonságok</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.h" line="71"/>
+        <source>Show jump to time dialog</source>
+        <translation>Idóre ugratási dialógus be</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.h" line="72"/>
+        <source>Open a new window</source>
+        <translation>Új ablak</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.h" line="73"/>
+        <source>Show open file dialog</source>
+        <translation>Fájlnyitó dialógus</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.h" line="74"/>
+        <source>Quit</source>
+        <translation>KIlépés</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.h" line="75"/>
+        <source>Play next file</source>
+        <translation>Következő fájl lejátszása</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.h" line="76"/>
+        <source>Play previous file</source>
+        <translation>Előzőző fájl lejátszása</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.h" line="77"/>
+        <source>Stop playback</source>
+        <translation>Lejátszás leáll</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.h" line="78"/>
+        <source>Show location dialog</source>
+        <translation>Helydialógus</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.h" line="79"/>
+        <source>Open clipboard location</source>
+        <translation>Megnyitás vágaólapról</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.h" line="80"/>
+        <source>Toggle playlist visibility</source>
+        <translation>Lejátszólista ki/be</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.h" line="81"/>
+        <source>Open the last played file</source>
+        <translation>Legutóbbi megnyitása</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.h" line="82"/>
+        <source>Access command-line</source>
+        <translation>Parancssor</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.h" line="83"/>
+        <source>Launch online help</source>
+        <translation>Segítség hálózatról</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.h" line="84"/>
+        <source>Play/Pause</source>
+        <translation>Lejátszás/Szünet</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.h" line="85"/>
+        <source>Fit the window to the video</source>
+        <translation>A videó ablakkitöltő</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.h" line="86"/>
+        <location filename="../bakaengine.h" line="87"/>
+        <location filename="../bakaengine.h" line="88"/>
+        <location filename="../bakaengine.h" line="89"/>
+        <location filename="../bakaengine.h" line="90"/>
+        <source>Fit window to %0%</source>
+        <translation>%0% ablakkitöltés</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.h" line="91"/>
+        <source>Boss key</source>
+        <translation>Főnök gomb</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.h" line="92"/>
+        <source>Select next file on playlist</source>
+        <translation>Lejátszólista következője</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.h" line="93"/>
+        <source>Select previous file on playlist</source>
+        <translation>Lejátszólista előzője</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.h" line="94"/>
+        <source>Play selected file on playlist</source>
+        <translation>Lejátszólista kijelötje</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.h" line="95"/>
+        <source>Remove selected file from playlist</source>
+        <translation>Lejátszólistából a kijelölt eltávolítása</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.h" line="131"/>
+        <source>executes mpv command</source>
+        <translation>mpv parancs végrehajtás</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.h" line="139"/>
+        <source>executes system shell command</source>
+        <translation>Rendszerhéj parancs végrehajtás</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.h" line="148"/>
+        <source>creates a new instance of baka-mplayer</source>
+        <translation>Új baka-mplayer példány</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.h" line="157"/>
+        <source>shows the open location dialog</source>
+        <translation>Helymegnyitási dialógus</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.h" line="166"/>
+        <source>opens the clipboard</source>
+        <translation>Vágólap</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.h" line="175"/>
+        <source>shows the current file in folder</source>
+        <translation>A mappa fájljai</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.h" line="184"/>
+        <source>add subtitles dialog</source>
+        <translation>Felirat dialógus be</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.h" line="193"/>
+        <source>add audio track dialog</source>
+        <translation>Hangsáv dialógus be</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.h" line="201"/>
+        <source>[subs]</source>
+        <translation>[feliratok]</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.h" line="202"/>
+        <source>take a screenshot (with subtitles if specified)</source>
+        <translation>képernyőfotó (felirattal, ha megadott)</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.h" line="211"/>
+        <source>toggles media info display</source>
+        <translation>Médiainfó be/ki</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.h" line="220"/>
+        <source>stops the current playback</source>
+        <translation>Lejátszás stop</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.h" line="229"/>
+        <source>playlist options</source>
+        <translation>Lejátszólista lehetőségek</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.h" line="238"/>
+        <source>opens jump dialog</source>
+        <translation>Ugratási dialógus</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.h" line="247"/>
+        <source>toggles dim desktop</source>
+        <translation>Asztal ki/be</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.h" line="256"/>
+        <source>toggles output textbox</source>
+        <translation>Textbox kimenet ki/be</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.h" line="265"/>
+        <source>opens preferences dialog</source>
+        <translation>Beállítások</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.h" line="274"/>
+        <source>launches online help</source>
+        <translation>Online segítség</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.h" line="283"/>
+        <source>opens the update dialog or updates youtube-dl</source>
+        <translation>A lejátszó, vagy a youtube-dl frissíthetősége</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.h" line="291"/>
+        <source>[file]</source>
+        <translation>[fájl]</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.h" line="292"/>
+        <source>opens the open file dialog or the file specified</source>
+        <translation>Fájlnyitási dialógus</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.h" line="301"/>
+        <source>toggle play/pause state</source>
+        <translation>Lejászás/Szünet állapot átbillentése</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.h" line="309"/>
+        <source>[percent]</source>
+        <translation>[százalék]</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.h" line="310"/>
+        <source>fit the window</source>
+        <translation>Ablakkitöltés</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.h" line="319"/>
+        <source>toggle deinterlace</source>
+        <translation>Csíktalanítás ki/be</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.h" line="328"/>
+        <source>toggle motion interpolation</source>
+        <translation>Mozgásinterpoláció ki/be</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.h" line="337"/>
+        <source>mutes the audio</source>
+        <translation>Némítás</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.h" line="345"/>
+        <location filename="../bakaengine.h" line="417"/>
+        <source>[level]</source>
+        <translation>[fokozat]</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.h" line="346"/>
+        <source>adjusts the volume</source>
+        <translation>Hangerő</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.h" line="354"/>
+        <source>[ratio]</source>
+        <translation>[arány]</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.h" line="355"/>
+        <source>adjusts the speed</source>
+        <translation>Sebesség</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.h" line="364"/>
+        <source>toggles fullscreen state</source>
+        <translation>Teli kép állapot átváltása</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.h" line="373"/>
+        <source>toggles hide all controls state</source>
+        <translation>Vezérlőelemek láthatóságának átbillentése</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.h" line="382"/>
+        <source>pause and hide the window</source>
+        <translation>Szünet és az ablak elrejtése</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.h" line="391"/>
+        <source>clears the output textbox</source>
+        <translation>Kimeti textbox ürítés</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.h" line="399"/>
+        <source>[command]</source>
+        <translation>[parancs]</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.h" line="400"/>
+        <source>internal help menu</source>
+        <translation>Belső segítő menü</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.h" line="408"/>
+        <source>[qt]</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.h" line="409"/>
+        <source>open about dialog</source>
+        <translation>Ismertető</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.h" line="418"/>
+        <source>set mpv msg-level</source>
+        <translation>mpv üzenet fokozat</translation>
+    </message>
+    <message>
+        <location filename="../bakaengine.h" line="427"/>
+        <source>quit baka-mplayer</source>
+        <translation>Kilépés</translation>
+    </message>
+</context>
+<context>
+    <name>InputDialog</name>
+    <message>
+        <location filename="../ui/inputdialog.ui" line="14"/>
+        <source>Dialog</source>
+        <translation>Párbeszédablak</translation>
+    </message>
+    <message>
+        <location filename="../ui/inputdialog.ui" line="71"/>
+        <source>Please enter a value:</source>
+        <translation>Adjon meg egy értéket:</translation>
+    </message>
+    <message>
+        <location filename="../ui/inputdialog.ui" line="118"/>
+        <source>&amp;OK</source>
+        <translation>&amp;Rendben</translation>
+    </message>
+    <message>
+        <location filename="../ui/inputdialog.ui" line="134"/>
+        <source>&amp;Cancel</source>
+        <translation>&amp;Mégsem</translation>
+    </message>
+</context>
+<context>
+    <name>JumpDialog</name>
+    <message>
+        <location filename="../ui/jumpdialog.ui" line="14"/>
+        <source>Dialog</source>
+        <translation>Párbeszédablak</translation>
+    </message>
+    <message>
+        <location filename="../ui/jumpdialog.ui" line="112"/>
+        <source> hr</source>
+        <translation> ó</translation>
+    </message>
+    <message>
+        <location filename="../ui/jumpdialog.ui" line="130"/>
+        <location filename="../ui/jumpdialog.ui" line="161"/>
+        <source>:</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../ui/jumpdialog.ui" line="143"/>
+        <source> min</source>
+        <translation> perc</translation>
+    </message>
+    <message>
+        <location filename="../ui/jumpdialog.ui" line="174"/>
+        <source> sec</source>
+        <translation> mp.</translation>
+    </message>
+    <message>
+        <location filename="../ui/jumpdialog.ui" line="190"/>
+        <source>&amp;OK</source>
+        <translation>&amp;Rendben</translation>
+    </message>
+</context>
+<context>
+    <name>KeyDialog</name>
+    <message>
+        <location filename="../ui/keydialog.ui" line="14"/>
+        <source>Modify Shortcut Key</source>
+        <translation>Gyorsbillentyű módosítás</translation>
+    </message>
+    <message>
+        <location filename="../ui/keydialog.ui" line="75"/>
+        <source>(optional)</source>
+        <translation>(választható)</translation>
+    </message>
+    <message>
+        <location filename="../ui/keydialog.ui" line="85"/>
+        <source>Shortcut:</source>
+        <translation>Billentyű:</translation>
+    </message>
+    <message>
+        <location filename="../ui/keydialog.ui" line="92"/>
+        <source>Command:</source>
+        <translation>Parancs:</translation>
+    </message>
+    <message>
+        <location filename="../ui/keydialog.ui" line="99"/>
+        <source>Label:</source>
+        <translation>Címke:</translation>
+    </message>
+    <message>
+        <location filename="../ui/keydialog.ui" line="123"/>
+        <source>C&amp;lear Shorcut</source>
+        <translation>Tör&amp;lés</translation>
+    </message>
+    <message>
+        <location filename="../ui/keydialog.ui" line="130"/>
+        <source>Chang&amp;e</source>
+        <translation>Vált&amp;oztat</translation>
+    </message>
+    <message>
+        <location filename="../ui/keydialog.ui" line="137"/>
+        <source>&amp;Add</source>
+        <translation>Hozzá&amp;ad</translation>
+    </message>
+    <message>
+        <location filename="../ui/keydialog.ui" line="150"/>
+        <source>&amp;Cancel</source>
+        <translation>&amp;Mégse</translation>
+    </message>
+</context>
+<context>
+    <name>LocationDialog</name>
+    <message>
+        <location filename="../ui/locationdialog.ui" line="14"/>
+        <source>Open Location</source>
+        <translation>Hely megnyitás</translation>
+    </message>
+    <message>
+        <location filename="../ui/locationdialog.ui" line="85"/>
+        <source>Paste copied link</source>
+        <translation>Másolt link beillesztés</translation>
+    </message>
+    <message>
+        <location filename="../ui/locationdialog.ui" line="119"/>
+        <source>Type URL or location here</source>
+        <translation>URL, vagy hely</translation>
+    </message>
+    <message>
+        <location filename="../ui/locationdialog.ui" line="138"/>
+        <source>&amp;OK</source>
+        <translation>R&amp;enben</translation>
+    </message>
+</context>
+<context>
+    <name>MainWindow</name>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="20"/>
+        <source>Baka MPlayer</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="264"/>
+        <source>&gt;</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="306"/>
+        <source>Search Playlist</source>
+        <translation>Lejátszólista keresés</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="319"/>
+        <source>Click to select a specific file by index</source>
+        <translation>Klikkre  az index szerinti fájl</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="322"/>
+        <source>File 0 of 0</source>
+        <translation>Fájl 0 ebből: 0</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="375"/>
+        <source>Select the Current File</source>
+        <translation>Aktuális fájl</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="427"/>
+        <source>Hide files with different extensions</source>
+        <translation>Más kiterjesztésű fájlok elrejése</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="473"/>
+        <source>Refresh Playlist</source>
+        <translation>Lejátszólista frissítés</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="633"/>
+        <source>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Left Click to Open File&lt;br&gt;Mouse Wheel Click to Jump&lt;br&gt;Right Click to Open URL&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</source>
+        <translation>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Bal klikk fájlnyitás&lt;br&gt;Görgő klikk ugrás&lt;br&gt;Jobb klikk URL nyitás&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="659"/>
+        <source>0:00:00</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="669"/>
+        <source>/</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="685"/>
+        <source>Click to change between total and remaining time</source>
+        <translation>Klikk teljes/hátralévő időt vált</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="688"/>
+        <source>-0:00:00</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="1067"/>
+        <source>&amp;File</source>
+        <translation>&amp;Fájl</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="1071"/>
+        <source>&amp;Recently Opened</source>
+        <translation>Le&amp;gutóbbi</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="1091"/>
+        <source>&amp;View</source>
+        <translation>&amp;Nézet</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="1098"/>
+        <source>Fit &amp;Window</source>
+        <translation>Ablak &amp;igazítás</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="1112"/>
+        <source>Aspect &amp;Ratio</source>
+        <translation>Képa&amp;rány</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="1124"/>
+        <source>Subtitle &amp;Track</source>
+        <translation>Felira&amp;t sáv</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="1135"/>
+        <source>Font Si&amp;ze</source>
+        <translation>Betű&amp;méret</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="1146"/>
+        <source>Take &amp;Screenshot</source>
+        <translation>Ké&amp;pernyőfotó</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="1168"/>
+        <source>&amp;Playback</source>
+        <translation>&amp;Lejátszás</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="1172"/>
+        <source>R&amp;epeat</source>
+        <translation>Ism&amp;étlés</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="1188"/>
+        <source>Audio &amp;Tracks</source>
+        <translation>Ha&amp;ngsávok</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="1199"/>
+        <source>Spee&amp;d</source>
+        <translation>Se&amp;besség</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="1208"/>
+        <source>&amp;Volume</source>
+        <translation>&amp;Hangerő</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="1228"/>
+        <source>&amp;Navigate</source>
+        <translation>&amp;Navigálás</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="1235"/>
+        <source>&amp;Chapters</source>
+        <translation>&amp;Fejezetek</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="1250"/>
+        <source>&amp;Settings</source>
+        <translation>Beállítá&amp;s</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="1263"/>
+        <source>&amp;Help</source>
+        <translation>&amp;Súgó</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="1281"/>
+        <source>&amp;New Player</source>
+        <translation>&amp;Új lejátszó</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="1289"/>
+        <source>&amp;Open File...</source>
+        <translation>Fá&amp;jl...</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="1294"/>
+        <source>Open &amp;Location...</source>
+        <translation>He&amp;ly...</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="1299"/>
+        <source>Open &amp;Copied Link</source>
+        <translation>H&amp;ivatkozás</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="1307"/>
+        <source>Show in &amp;Folder</source>
+        <translation>&amp;Mappa tartalom</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="1315"/>
+        <source>Pla&amp;y Next File</source>
+        <translation>Kö&amp;vetkező</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="1323"/>
+        <source>Play &amp;Previous File</source>
+        <translation>Elő&amp;ző</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="1328"/>
+        <source>E&amp;xit</source>
+        <translation>Ki&amp;lépés</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="1336"/>
+        <source>&amp;Full Screen</source>
+        <translation>&amp;Teljes kép</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="1350"/>
+        <source>Show S&amp;ubtitles</source>
+        <translation>F&amp;eliratok</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="1361"/>
+        <source>Show Playback &amp;Info</source>
+        <translation>Lejátszási &amp;infó</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="1369"/>
+        <location filename="../ui/mainwindow.cpp" line="1274"/>
+        <source>&amp;Play</source>
+        <translation>&amp;Start</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="1377"/>
+        <source>&amp;Stop</source>
+        <translation>Sto&amp;p</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="1385"/>
+        <source>&amp;Restart</source>
+        <translation>Új&amp;ra</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="1397"/>
+        <source>Sh&amp;uffle</source>
+        <translation>&amp;Keverve</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="1405"/>
+        <source>Stop after &amp;Current</source>
+        <translation>A&amp;ktuális után stop</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="1413"/>
+        <source>&amp;Next Chapter</source>
+        <translation>Kö&amp;vetkező fejezet</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="1421"/>
+        <source>&amp;Previous Chapter</source>
+        <translation>El&amp;őző fejezet</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="1429"/>
+        <source>&amp;Frame Step</source>
+        <translation>Ké&amp;pváltásra előre</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="1437"/>
+        <source>Frame &amp;Back Step</source>
+        <translation>Képváltás&amp;ra vissza</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="1445"/>
+        <source>&amp;Jump to Time...</source>
+        <translation>Időre u&amp;grás...</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="1456"/>
+        <source>&amp;Show Playlist</source>
+        <translation>Leját&amp;szólista</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="1467"/>
+        <source>Hide &amp;Album Art</source>
+        <translation>&amp;Album fajta elrejtés</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="1475"/>
+        <source>&amp;Dim Lights</source>
+        <translation>Háttér&amp;fények</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="1480"/>
+        <source>Online &amp;Help</source>
+        <translation>&amp;Online súgó</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="1485"/>
+        <source>&amp;Check for Updates</source>
+        <translation>Frissítés &amp;keresés</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="1490"/>
+        <source>About &amp;Qt</source>
+        <translation>A &amp;QT</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="1495"/>
+        <source>About &amp;Baka MPlayer</source>
+        <translation>A &amp;Baka MPlayer</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="1500"/>
+        <source>To &amp;Current Size</source>
+        <translation>A&amp;ktuális méretre</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="1505"/>
+        <source>&amp;50%</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="1510"/>
+        <source>&amp;75%</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="1515"/>
+        <source>&amp;100%</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="1520"/>
+        <source>&amp;200%</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="1525"/>
+        <source>&amp;Auto Detect</source>
+        <translation>&amp;Automatikus</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="1530"/>
+        <source>Force &amp;4:3</source>
+        <translation>&amp;4:3</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="1535"/>
+        <source>Force 16:&amp;9</source>
+        <translation>16:&amp;9</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="1540"/>
+        <source>Force &amp;2.35:1</source>
+        <translation>&amp;2,35:1</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="1548"/>
+        <source>&amp;Add Subtitle File...</source>
+        <translation>Felir&amp;atfájl...</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="1556"/>
+        <location filename="../ui/mainwindow.ui" line="1606"/>
+        <location filename="../ui/mainwindow.ui" line="1647"/>
+        <location filename="../ui/mainwindow.ui" line="1670"/>
+        <source>[ &amp;none ]</source>
+        <translation>[ &amp;nincs ]</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="1561"/>
+        <source>&amp;Size +</source>
+        <translation>&amp;Nagyobb</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="1566"/>
+        <source>S&amp;ize -</source>
+        <translation>K&amp;isebb</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="1571"/>
+        <source>&amp;Reset Size</source>
+        <translation>E&amp;redeti méret</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="1582"/>
+        <source>&amp;Off</source>
+        <translation>&amp;Nincs</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="1590"/>
+        <source>&amp;Playlist</source>
+        <translation>Le&amp;játszólista</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="1598"/>
+        <source>&amp;This File</source>
+        <translation>Ez a fá&amp;jl</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="1614"/>
+        <source>Show &amp;Command Line</source>
+        <translation>Paran&amp;cssor</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="1619"/>
+        <source>&amp;Increase by 5%</source>
+        <translation>5%-a&amp;l nő</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="1624"/>
+        <source>&amp;Decrease by 5%</source>
+        <translation>5%-al ki&amp;sebb</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="1629"/>
+        <source>With &amp;Subtitles...</source>
+        <translation>Feli&amp;ratozva...</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="1634"/>
+        <source>&amp;Without Subtitles...</source>
+        <translation>Felirat &amp;nélkül...</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="1639"/>
+        <source>&amp;Preferences...</source>
+        <translation>&amp;Beállítások...</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="1652"/>
+        <source>&amp;Increase by 0.1%</source>
+        <translation>0,1%-a&amp;l nő</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="1657"/>
+        <source>&amp;Decrease by 0.1%</source>
+        <translation>0,1%-al ki&amp;sebb</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="1662"/>
+        <source>&amp;Reset</source>
+        <translation>&amp;Alapállapot</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="1675"/>
+        <source>15&amp;0%</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="1680"/>
+        <source>Update &amp;Streaming Support</source>
+        <translation>Adat&amp;szórási képesség frissítés</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="1688"/>
+        <source>&amp;Mute</source>
+        <translation>Né&amp;mítás</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="1699"/>
+        <source>&amp;Motion Interpolation</source>
+        <translation>&amp;Mozgás interpoláció</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="1702"/>
+        <source>Reduce video studder</source>
+        <translation>Videó akadozás csökkentés</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="1713"/>
+        <source>&amp;Deinterlace</source>
+        <translation>&amp;Csíktalanítás</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="1718"/>
+        <source>&amp;Add Audio File...</source>
+        <translation>H&amp;angfájl...</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.ui" line="1723"/>
+        <source>&amp;Hide All Controls</source>
+        <translation>Vezérlőelemek &amp;ki</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.cpp" line="207"/>
+        <source>Playing</source>
+        <translation>Lejátszás</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.cpp" line="209"/>
+        <source>Paused</source>
+        <translation>Szünet</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.cpp" line="352"/>
+        <source>Sub</source>
+        <translation>Felirat</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.cpp" line="364"/>
+        <source>Audio</source>
+        <translation>Hang</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.cpp" line="609"/>
+        <source>Speed: %0x</source>
+        <translation>Sebesség: %0x</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.cpp" line="656"/>
+        <source>Subtitles visible</source>
+        <translation>Feliratok be</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.cpp" line="656"/>
+        <source>Subtitles hidden</source>
+        <translation>Feliratok ki</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.cpp" line="666"/>
+        <source>Muted</source>
+        <translation>Némítva</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.cpp" line="666"/>
+        <source>Unmuted</source>
+        <translation>Némítás ki</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.cpp" line="777"/>
+        <source>Enter the file number you want to play:
+Note: Value must be from %0 - %1</source>
+        <translation>Lejátszandó fájlok darabszáma:
+A megadandó érték %0 - %1 közti</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.cpp" line="778"/>
+        <source>Enter File Number</source>
+        <translation>Fájlszám</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.cpp" line="869"/>
+        <source>Previous</source>
+        <translation>Előzőző</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.cpp" line="879"/>
+        <location filename="../ui/mainwindow.cpp" line="1276"/>
+        <source>Play</source>
+        <translation>Lejátszás</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.cpp" line="889"/>
+        <source>Next</source>
+        <translation>Következő</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.cpp" line="1110"/>
+        <source>No selection</source>
+        <translation>Nincs választott</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.cpp" line="1116"/>
+        <source>File %0 of %1</source>
+        <translation>Fájl %0 ebből: %1</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.cpp" line="1283"/>
+        <source>&amp;Pause</source>
+        <translation>Szüne&amp;t</translation>
+    </message>
+    <message>
+        <location filename="../ui/mainwindow.cpp" line="1285"/>
+        <source>Pause</source>
+        <translation>Szünet</translation>
+    </message>
+</context>
+<context>
+    <name>MpvHandler</name>
+    <message>
+        <location filename="../mpvhandler.cpp" line="91"/>
+        <source>File</source>
+        <translation>Fájl</translation>
+    </message>
+    <message>
+        <location filename="../mpvhandler.cpp" line="92"/>
+        <source>Title</source>
+        <translation>Cím</translation>
+    </message>
+    <message>
+        <location filename="../mpvhandler.cpp" line="93"/>
+        <source>File size</source>
+        <translation>Fájlméret</translation>
+    </message>
+    <message>
+        <location filename="../mpvhandler.cpp" line="94"/>
+        <source>Date created</source>
+        <translation>Létrehozva</translation>
+    </message>
+    <message>
+        <location filename="../mpvhandler.cpp" line="95"/>
+        <source>Media length</source>
+        <translation>Médiahossz</translation>
+    </message>
+    <message>
+        <location filename="../mpvhandler.cpp" line="97"/>
+        <source>Video (x%0)</source>
+        <translation>Videó (x%0)</translation>
+    </message>
+    <message>
+        <location filename="../mpvhandler.cpp" line="98"/>
+        <source>Video Output</source>
+        <translation>Videó kimenet</translation>
+    </message>
+    <message>
+        <location filename="../mpvhandler.cpp" line="99"/>
+        <source>Resolution</source>
+        <translation>Felbontás</translation>
+    </message>
+    <message>
+        <location filename="../mpvhandler.cpp" line="102"/>
+        <source>FPS</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../mpvhandler.cpp" line="103"/>
+        <source>A/V Sync</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../mpvhandler.cpp" line="104"/>
+        <location filename="../mpvhandler.cpp" line="110"/>
+        <source>Bitrate</source>
+        <translation>Bitsebesség</translation>
+    </message>
+    <message>
+        <location filename="../mpvhandler.cpp" line="104"/>
+        <location filename="../mpvhandler.cpp" line="110"/>
+        <source>%0 kbps</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../mpvhandler.cpp" line="106"/>
+        <source>Audio (x%0)</source>
+        <translation>Hang (x%0)</translation>
+    </message>
+    <message>
+        <location filename="../mpvhandler.cpp" line="107"/>
+        <source>Audio Output</source>
+        <translation>Hang kimenet</translation>
+    </message>
+    <message>
+        <location filename="../mpvhandler.cpp" line="108"/>
+        <source>Sample Rate</source>
+        <translation>Mintavételi sűrűség</translation>
+    </message>
+    <message>
+        <location filename="../mpvhandler.cpp" line="109"/>
+        <source>Channels</source>
+        <translation>Csatornák</translation>
+    </message>
+    <message>
+        <location filename="../mpvhandler.cpp" line="114"/>
+        <source>Chapters</source>
+        <translation>Fejezetek</translation>
+    </message>
+    <message>
+        <location filename="../mpvhandler.cpp" line="123"/>
+        <source>Metadata</source>
+        <translation>Meta adatok</translation>
+    </message>
+    <message>
+        <location filename="../mpvhandler.cpp" line="188"/>
+        <source>Buffering...</source>
+        <translation>Pufferelés...</translation>
+    </message>
+    <message>
+        <location filename="../mpvhandler.cpp" line="198"/>
+        <source>Your network is slow or stuck, please wait a bit</source>
+        <translation>A hálózat lassú, vagy akadozó, türelem</translation>
+    </message>
+    <message>
+        <location filename="../mpvhandler.cpp" line="227"/>
+        <source>File couldn&apos;t be opened</source>
+        <translation>A fájl megnyithatatlan</translation>
+    </message>
+    <message>
+        <location filename="../mpvhandler.cpp" line="313"/>
+        <source>File does not exist</source>
+        <translation>A fájl nincs</translation>
+    </message>
+    <message>
+        <location filename="../mpvhandler.cpp" line="352"/>
+        <source>File no longer exists</source>
+        <translation>A fájl már nincs</translation>
+    </message>
+    <message>
+        <location filename="../mpvhandler.cpp" line="511"/>
+        <source>Volume: %0%</source>
+        <translation>Hangerő: %0%</translation>
+    </message>
+    <message>
+        <location filename="../mpvhandler.cpp" line="626"/>
+        <source>Deinterlacing: %0</source>
+        <translation>Csíktalanítás: %0</translation>
+    </message>
+    <message>
+        <location filename="../mpvhandler.cpp" line="626"/>
+        <location filename="../mpvhandler.cpp" line="644"/>
+        <source>enabled</source>
+        <translation>Engedve</translation>
+    </message>
+    <message>
+        <location filename="../mpvhandler.cpp" line="626"/>
+        <location filename="../mpvhandler.cpp" line="644"/>
+        <source>disabled</source>
+        <translation>Tiltva</translation>
+    </message>
+    <message>
+        <location filename="../mpvhandler.cpp" line="644"/>
+        <source>Motion Interpolation: %0</source>
+        <translation>Mozgás interpoláció: %0</translation>
+    </message>
+</context>
+<context>
+    <name>PlaylistWidget</name>
+    <message>
+        <location filename="../widgets/playlistwidget.cpp" line="292"/>
+        <source>Delete sub-file?</source>
+        <translation>Feliratfájl törlés?</translation>
+    </message>
+    <message>
+        <location filename="../widgets/playlistwidget.cpp" line="293"/>
+        <location filename="../widgets/playlistwidget.cpp" line="307"/>
+        <source>Would you like to delete the associated sub file [%0]?</source>
+        <translation>Tényleg törli ezt a fájlt: [%0]?</translation>
+    </message>
+    <message>
+        <location filename="../widgets/playlistwidget.cpp" line="306"/>
+        <source>Delete external sub-file?</source>
+        <translation>Külső feliratfájl törlés?</translation>
+    </message>
+    <message>
+        <location filename="../widgets/playlistwidget.cpp" line="325"/>
+        <source>R&amp;emove from Playlist</source>
+        <translation>L&amp;ejátszólistából törlés</translation>
+    </message>
+    <message>
+        <location filename="../widgets/playlistwidget.cpp" line="330"/>
+        <source>&amp;Delete from Disk</source>
+        <translation>&amp;Lemezről törlés</translation>
+    </message>
+    <message>
+        <location filename="../widgets/playlistwidget.cpp" line="335"/>
+        <source>&amp;Refresh</source>
+        <translation>F&amp;rissítés</translation>
+    </message>
+</context>
+<context>
+    <name>PreferencesDialog</name>
+    <message>
+        <location filename="../ui/preferencesdialog.ui" line="14"/>
+        <source>Preferences</source>
+        <translation>Beállítások</translation>
+    </message>
+    <message>
+        <location filename="../ui/preferencesdialog.ui" line="250"/>
+        <source>Player</source>
+        <translation>Lejátszó</translation>
+    </message>
+    <message>
+        <location filename="../ui/preferencesdialog.ui" line="270"/>
+        <source>On Top</source>
+        <translation>Előtérben</translation>
+    </message>
+    <message>
+        <location filename="../ui/preferencesdialog.ui" line="276"/>
+        <source>A&amp;lways</source>
+        <translation>Á&amp;llandóan</translation>
+    </message>
+    <message>
+        <location filename="../ui/preferencesdialog.ui" line="283"/>
+        <source>&amp;When Playing</source>
+        <translation>Lejáts&amp;záskor</translation>
+    </message>
+    <message>
+        <location filename="../ui/preferencesdialog.ui" line="290"/>
+        <source>&amp;Never</source>
+        <translation>&amp;Soha</translation>
+    </message>
+    <message>
+        <location filename="../ui/preferencesdialog.ui" line="300"/>
+        <source>Tray Icon</source>
+        <translation>Tálcaikon</translation>
+    </message>
+    <message>
+        <location filename="../ui/preferencesdialog.ui" line="309"/>
+        <source>&amp;Hide Popup</source>
+        <translation>Felbu&amp;kkanók ki</translation>
+    </message>
+    <message>
+        <location filename="../ui/preferencesdialog.ui" line="319"/>
+        <source>Language</source>
+        <translation>Nyelv</translation>
+    </message>
+    <message>
+        <location filename="../ui/preferencesdialog.ui" line="342"/>
+        <source>Mouse Gesture</source>
+        <translation>Egér és tapipad</translation>
+    </message>
+    <message>
+        <location filename="../ui/preferencesdialog.ui" line="380"/>
+        <source>Change Volume</source>
+        <translation>Hangerő</translation>
+    </message>
+    <message>
+        <location filename="../ui/preferencesdialog.ui" line="406"/>
+        <source>Seek Position</source>
+        <translation>Lejátszási helyzet</translation>
+    </message>
+    <message>
+        <location filename="../ui/preferencesdialog.ui" line="427"/>
+        <source>&amp;Use Mouse &amp;&amp; Touch Gestures</source>
+        <translation>Egé&amp;r &amp;&amp; Tapipad használata</translation>
+    </message>
+    <message>
+        <location filename="../ui/preferencesdialog.ui" line="450"/>
+        <source>History</source>
+        <translation>Előzmények</translation>
+    </message>
+    <message>
+        <location filename="../ui/preferencesdialog.ui" line="458"/>
+        <source>Remember &amp;recently opened files up to</source>
+        <translation>Megjegyezendő megnyi&amp;tott fájlok</translation>
+    </message>
+    <message>
+        <location filename="../ui/preferencesdialog.ui" line="477"/>
+        <source>Remember playback &amp;position</source>
+        <translation>Lejátszá&amp;si helyzet megjegyzése</translation>
+    </message>
+    <message>
+        <location filename="../ui/preferencesdialog.ui" line="490"/>
+        <source>Key Bindings</source>
+        <translation>Gyorsbillentyűk</translation>
+    </message>
+    <message>
+        <location filename="../ui/preferencesdialog.ui" line="513"/>
+        <source>Prefix a command with &apos;mpv&apos; to make it run as an input command.</source>
+        <translation>Az &apos;mpv&apos; parancshoz szükséges kiegészítések.</translation>
+    </message>
+    <message>
+        <location filename="../ui/preferencesdialog.ui" line="581"/>
+        <source>Key</source>
+        <translation>Billentyű</translation>
+    </message>
+    <message>
+        <location filename="../ui/preferencesdialog.ui" line="586"/>
+        <source>Command</source>
+        <translation>Parancs</translation>
+    </message>
+    <message>
+        <location filename="../ui/preferencesdialog.ui" line="591"/>
+        <source>Label</source>
+        <translation>Címke</translation>
+    </message>
+    <message>
+        <location filename="../ui/preferencesdialog.ui" line="673"/>
+        <source>Rese&amp;t All</source>
+        <translation>Alapállapo&amp;tba</translation>
+    </message>
+    <message>
+        <location filename="../ui/preferencesdialog.ui" line="683"/>
+        <source>Video</source>
+        <translation>Videó</translation>
+    </message>
+    <message>
+        <location filename="../ui/preferencesdialog.ui" line="703"/>
+        <source>&amp;Auto-Fit Window:</source>
+        <translation>&amp;Automata ablakkitöltés:</translation>
+    </message>
+    <message>
+        <location filename="../ui/preferencesdialog.ui" line="714"/>
+        <source>50%</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../ui/preferencesdialog.ui" line="719"/>
+        <source>75%</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../ui/preferencesdialog.ui" line="724"/>
+        <source>100%</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../ui/preferencesdialog.ui" line="729"/>
+        <source>200%</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../ui/preferencesdialog.ui" line="752"/>
+        <source>Screenshot</source>
+        <translation>Képfotó</translation>
+    </message>
+    <message>
+        <location filename="../ui/preferencesdialog.ui" line="758"/>
+        <source>File Name Template:</source>
+        <translation>Fájlnév minta:</translation>
+    </message>
+    <message>
+        <location filename="../ui/preferencesdialog.ui" line="769"/>
+        <source>jpg</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../ui/preferencesdialog.ui" line="774"/>
+        <source>png</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../ui/preferencesdialog.ui" line="779"/>
+        <source>tga</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../ui/preferencesdialog.ui" line="787"/>
+        <source>Chang&amp;e...</source>
+        <translation>&amp;Változtatás...</translation>
+    </message>
+    <message>
+        <location filename="../ui/preferencesdialog.ui" line="794"/>
+        <source>File Format:</source>
+        <translation>Fájltípus:</translation>
+    </message>
+    <message>
+        <location filename="../ui/preferencesdialog.ui" line="804"/>
+        <source>screenshot%#04n</source>
+        <translation>kepfoto%x04n</translation>
+    </message>
+    <message>
+        <location filename="../ui/preferencesdialog.ui" line="811"/>
+        <source>Default Save Location:</source>
+        <translation>Mentési hely:</translation>
+    </message>
+    <message>
+        <location filename="../ui/preferencesdialog.ui" line="860"/>
+        <source>Always show the &amp;save window</source>
+        <translation>A mentési ablak mindig lát&amp;szik</translation>
+    </message>
+    <message>
+        <location filename="../ui/preferencesdialog.ui" line="884"/>
+        <source>Advanced</source>
+        <translation>Haladó</translation>
+    </message>
+    <message>
+        <location filename="../ui/preferencesdialog.ui" line="890"/>
+        <source>Command Line</source>
+        <translation>Parancs sor</translation>
+    </message>
+    <message>
+        <location filename="../ui/preferencesdialog.ui" line="896"/>
+        <source>mpv msg-level:</source>
+        <translation>mpv üzenet fokozat:</translation>
+    </message>
+    <message>
+        <location filename="../ui/preferencesdialog.ui" line="907"/>
+        <source>none</source>
+        <translation>nincs</translation>
+    </message>
+    <message>
+        <location filename="../ui/preferencesdialog.ui" line="912"/>
+        <source>fatal</source>
+        <translation>végzetes</translation>
+    </message>
+    <message>
+        <location filename="../ui/preferencesdialog.ui" line="917"/>
+        <source>error</source>
+        <translation>hiba</translation>
+    </message>
+    <message>
+        <location filename="../ui/preferencesdialog.ui" line="922"/>
+        <source>warn</source>
+        <translation>figyelmeztetés</translation>
+    </message>
+    <message>
+        <location filename="../ui/preferencesdialog.ui" line="927"/>
+        <source>info</source>
+        <translation>infó</translation>
+    </message>
+    <message>
+        <location filename="../ui/preferencesdialog.ui" line="932"/>
+        <source>status</source>
+        <translation>állapot</translation>
+    </message>
+    <message>
+        <location filename="../ui/preferencesdialog.ui" line="937"/>
+        <source>verbose</source>
+        <translation>szószátyár</translation>
+    </message>
+    <message>
+        <location filename="../ui/preferencesdialog.ui" line="942"/>
+        <source>debug</source>
+        <translation>hibakereső</translation>
+    </message>
+    <message>
+        <location filename="../ui/preferencesdialog.ui" line="947"/>
+        <source>trace</source>
+        <translation>léptető</translation>
+    </message>
+    <message>
+        <location filename="../ui/preferencesdialog.ui" line="1010"/>
+        <source>&amp;OK</source>
+        <translation>&amp;Rendben</translation>
+    </message>
+    <message>
+        <location filename="../ui/preferencesdialog.ui" line="1017"/>
+        <source>&amp;Cancel</source>
+        <translation>&amp;Mégse</translation>
+    </message>
+    <message>
+        <location filename="../ui/preferencesdialog.cpp" line="63"/>
+        <source>Choose screenshot directory</source>
+        <translation>Képfotó mappa</translation>
+    </message>
+    <message>
+        <location filename="../ui/preferencesdialog.cpp" line="90"/>
+        <source>Reset All Key Bindings?</source>
+        <translation>Minden alapállapotba?</translation>
+    </message>
+    <message>
+        <location filename="../ui/preferencesdialog.cpp" line="90"/>
+        <source>Are you sure you want to reset all shortcut keys to its original bindings?</source>
+        <translation>Biztosan alapállapotúak legyenek a gyorsbillentyűk?</translation>
+    </message>
+    <message>
+        <location filename="../ui/preferencesdialog.cpp" line="259"/>
+        <source>Existing keybinding</source>
+        <translation>Létező billentyű hozzárendelés</translation>
+    </message>
+    <message>
+        <location filename="../ui/preferencesdialog.cpp" line="260"/>
+        <source>%0 is already being used. Would you like to change its function?</source>
+        <translation>A %0 mér használatos. Új funkciója legyen?</translation>
+    </message>
+</context>
+<context>
+    <name>ScreenshotDialog</name>
+    <message>
+        <location filename="../ui/screenshotdialog.ui" line="14"/>
+        <source>Take Screenshot</source>
+        <translation>Képfotó készítés</translation>
+    </message>
+    <message>
+        <location filename="../ui/screenshotdialog.ui" line="118"/>
+        <source>...</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../ui/screenshotdialog.ui" line="128"/>
+        <source>Save Location</source>
+        <translation>Mentési hely</translation>
+    </message>
+    <message>
+        <location filename="../ui/screenshotdialog.ui" line="135"/>
+        <source>File Name Template</source>
+        <translation>Fájlnév minta</translation>
+    </message>
+    <message>
+        <location filename="../ui/screenshotdialog.ui" line="155"/>
+        <source>jpg</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../ui/screenshotdialog.ui" line="160"/>
+        <source>png</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../ui/screenshotdialog.ui" line="165"/>
+        <source>tga</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../ui/screenshotdialog.ui" line="177"/>
+        <source>Sho&amp;w subtitles</source>
+        <translation>&amp;Feliratokkal</translation>
+    </message>
+    <message>
+        <location filename="../ui/screenshotdialog.ui" line="184"/>
+        <source>Always show this dialog</source>
+        <translation>Ez a dialóg ablak mindig látszik</translation>
+    </message>
+    <message>
+        <location filename="../ui/screenshotdialog.ui" line="213"/>
+        <source>&amp;Save</source>
+        <translation>&amp;Ment</translation>
+    </message>
+    <message>
+        <location filename="../ui/screenshotdialog.cpp" line="26"/>
+        <source>Choose screenshot directory</source>
+        <translation>Képfotó mappa</translation>
+    </message>
+</context>
+<context>
+    <name>UpdateDialog</name>
+    <message>
+        <location filename="../ui/updatedialog.ui" line="14"/>
+        <source>Check for Updates</source>
+        <translation>Frissíthetőségek</translation>
+    </message>
+    <message>
+        <location filename="../ui/updatedialog.ui" line="134"/>
+        <source>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Checking for updates...&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</source>
+        <translation>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Frissíthetőségek keresése...&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</translation>
+    </message>
+    <message>
+        <location filename="../ui/updatedialog.ui" line="157"/>
+        <source>About 0 second(s) remaining</source>
+        <translation>Még 0 mp. van hátra</translation>
+    </message>
+    <message>
+        <location filename="../ui/updatedialog.ui" line="205"/>
+        <source>&amp;Update</source>
+        <translation>F&amp;rissít</translation>
+    </message>
+    <message>
+        <location filename="../ui/updatedialog.ui" line="224"/>
+        <source>&amp;Cancel</source>
+        <translation>&amp;Mégse</translation>
+    </message>
+    <message>
+        <location filename="../ui/updatedialog.cpp" line="22"/>
+        <source>&amp;CLOSE</source>
+        <translation>&amp;BEZÁR</translation>
+    </message>
+    <message>
+        <location filename="../ui/updatedialog.cpp" line="32"/>
+        <source>Download Complete</source>
+        <translation>Letöltés kész</translation>
+    </message>
+    <message>
+        <location filename="../ui/updatedialog.cpp" line="67"/>
+        <source>About %0 second(s) remaining</source>
+        <translation>Még %0 mp. van hátra</translation>
+    </message>
+    <message>
+        <location filename="../ui/updatedialog.cpp" line="69"/>
+        <source>Calculating...</source>
+        <translation>Számolás...</translation>
+    </message>
+    <message>
+        <location filename="../ui/updatedialog.cpp" line="129"/>
+        <source>You have the latest version!</source>
+        <translation>Már minden a legfrissebb!</translation>
+    </message>
+    <message>
+        <location filename="../ui/updatedialog.cpp" line="133"/>
+        <source>Update Available!
+Version: %0</source>
+        <translation>Frissítés elérhető!
+Verzió: %0</translation>
+    </message>
+</context>
+<context>
+    <name>UpdateManager</name>
+    <message>
+        <location filename="../updatemanager.cpp" line="40"/>
+        <source>Checking for updates...</source>
+        <translation>Frissítés keresés...</translation>
+    </message>
+    <message>
+        <location filename="../updatemanager.cpp" line="86"/>
+        <source>Downloading update...</source>
+        <translation>Frissítés letöltése...</translation>
+    </message>
+    <message>
+        <location filename="../updatemanager.cpp" line="93"/>
+        <source>fopen error
+</source>
+        <translation>fájlnyitási hiba
+</translation>
+    </message>
+    <message>
+        <location filename="../updatemanager.cpp" line="113"/>
+        <source>write error</source>
+        <translation>írási hiba</translation>
+    </message>
+    <message>
+        <location filename="../updatemanager.cpp" line="134"/>
+        <source>Download complete</source>
+        <translation>Letöltés kész</translation>
+    </message>
+    <message>
+        <location filename="../updatemanager.cpp" line="140"/>
+        <source>Redirected...</source>
+        <translation>Átirányítva...</translation>
+    </message>
+    <message>
+        <location filename="../updatemanager.cpp" line="152"/>
+        <source>Extracting...</source>
+        <translation>Kibontás...</translation>
+    </message>
+    <message>
+        <location filename="../updatemanager.cpp" line="181"/>
+        <source>Creating updater script...</source>
+        <translation>Frissítő szkript készül...</translation>
+    </message>
+    <message>
+        <location filename="../updatemanager.cpp" line="185"/>
+        <source>Could not open file for writing...</source>
+        <translation>Nem lehet írni a fájl...</translation>
+    </message>
+    <message>
+        <location filename="../updatemanager.cpp" line="201"/>
+        <source>Updating...</source>
+        <translation>Frissítés...</translation>
+    </message>
+    <message>
+        <location filename="../updatemanager.cpp" line="209"/>
+        <source>Done. Restarting...</source>
+        <translation>Kész. Újraindítás...</translation>
+    </message>
+</context>
+</TS>
diff -Nur orig/src/translations.pri mod/src/translations.pri
--- orig/src/translations.pri	2015-12-01 17:03:48.000000000 +0100
+++ mod/src/translations.pri	2019-03-07 09:45:45.000000000 +0100
@@ -8,4 +8,5 @@
     translations/baka-mplayer_it.ts \
     translations/baka-mplayer_es.ts \
     translations/baka-mplayer_fr.ts \
-    translations/baka-mplayer_de.ts
+    translations/baka-mplayer_de.ts \
+    translations/baka-mplayer_hu.ts
