diff -Naur orig/vuze mod/vuze
--- orig/vuze	2015-11-14 00:14:08.000000000 +0100
+++ mod/vuze	2016-04-09 09:24:39.854898004 +0200
@@ -3,7 +3,7 @@
 ######## CONFIGURATION OPTIONS ########
 AUTOUPDATE_SCRIPT=1 # change to 0 if you don't want your changes overwritten on next SCRIPT_VERSION change
 JAVA_PROGRAM_DIR=""	# use full path to java bin dir, ex. "/usr/java/j2sdk1.4.2/bin/"
-#PROGRAM_DIR="/home/username/apps/azureus"	# use full path to Azureus bin dir
+PROGRAM_DIR="/usr/lib/azureus"	# use full path to Azureus bin dir
 JAVA_ARGS="-Xmx256m"
 
 #export MOZILLA_FIVE_HOME="/path/to/gre"	# Full path to GRE/Mozilla. When commenting out this line, also comment out the next line
