diff -Naur vuze-orig/vuze.desktop vuze/vuze.desktop
--- vuze-orig/vuze.desktop	2010-04-09 09:55:14.000000000 +0200
+++ vuze/vuze.desktop	2016-07-30 13:04:14.229440552 +0200
@@ -1,10 +1,10 @@
-[Desktop Entry]
-Encoding=UTF-8
-Categories=Java;Network;FileTransfer;P2P
-Comment=Multimedia Bittorrent Client 
-Exec=vuze %f
-GenericName=Multimedia Bittorrent Client
-Icon=vuze.png
-MimeType=application/x-bittorrent
-Name=Vuze
-Type=Application
+[Desktop Entry]
+Encoding=UTF-8
+Categories=Java;Network;FileTransfer;P2P
+Comment=Multimedia Bittorrent Client 
+Exec=vuze %f
+GenericName=Multimedia Bittorrent Client
+Icon=vuze
+MimeType=application/x-bittorrent
+Name=Vuze
+Type=Application
