--- a/data/gnome-calculator.desktop.in.orig	2013-02-26 21:13:07.000000000 +0100
+++ a/data/gnome-calculator.desktop.in	2013-02-26 21:13:45.000000000 +0100
@@ -7,7 +7,7 @@
 Terminal=false
 Type=Application
 StartupNotify=true
-Categories=GNOME;GTK;Utility;Calculator;
+Categories=GNOME;GTK;Utility;Calculator;X-UHU-Menu/Tools/Calculators;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=gnome-calculator
 X-GNOME-Bugzilla-Component=general
