diff -Naur orig/client/CMakeLists.txt mod/client/CMakeLists.txt
--- orig/client/CMakeLists.txt	2015-10-31 01:10:45.000000000 +0100
+++ mod/client/CMakeLists.txt	2015-11-03 11:36:30.206445942 +0100
@@ -145,7 +145,7 @@
     ${${PROJECT_NAME}_GENERATED_FILES}
 )
 
-set(${PROJECT_NAME}_PKG_CONFIG_REQUIRES "Qt5Widget, Qt5DBus")
+set(${PROJECT_NAME}_PKG_CONFIG_REQUIRES "Qt5Widgets, Qt5DBus")
 
 add_library(${PROJECT_NAME} SHARED ${${PROJECT_NAME}_ALL_FILES})
 target_link_libraries(${PROJECT_NAME}
