diff -Naur orig/lib/getline.c patched/lib/getline.c
--- orig/lib/getline.c	2005-04-04 22:46:05.000000000 +0200
+++ patched/lib/getline.c	2017-03-01 00:48:15.469326457 +0100
@@ -155,7 +155,7 @@
 }
 
 int
-getline (lineptr, n, stream)
+get_line (lineptr, n, stream)
      char **lineptr;
      size_t *n;
      FILE *stream;
diff -Naur orig/lib/getline.h patched/lib/getline.h
--- orig/lib/getline.h	2005-04-04 22:46:05.000000000 +0200
+++ patched/lib/getline.h	2017-03-01 00:48:15.469326457 +0100
@@ -12,7 +12,7 @@
 #define GETLINE_NO_LIMIT -1
 
 int
-  getline __PROTO ((char **_lineptr, size_t *_n, FILE *_stream));
+  get_line __PROTO ((char **_lineptr, size_t *_n, FILE *_stream));
 int
   getline_safe __PROTO ((char **_lineptr, size_t *_n, FILE *_stream,
                          int limit));
