diff -Naur orig/SConstruct mod/SConstruct
--- orig/SConstruct	2010-12-26 14:40:14.000000000 +0100
+++ mod/SConstruct	2015-01-09 20:48:59.459298247 +0100
@@ -4,7 +4,7 @@
 
 win32CrossCompile = 0
 linux32CrossCompile = 0
-disableSDL = 0          # set this to 1 on Linux with SDL version >= 1.2.10
+disableSDL = 1          # set this to 1 on Linux with SDL version >= 1.2.10
 disableLua = 0
 enableGLShaders = 1
 enableDebug = 0
@@ -22,7 +22,7 @@
     compilerFlags = compilerFlags + ' -fno-inline-functions '
     compilerFlags = compilerFlags + ' -fomit-frame-pointer -ffast-math '
 
-fltkConfig = 'fltk-config'
+fltkConfig = 'fltk1-config'
 
 # -----------------------------------------------------------------------------
 
@@ -291,7 +291,7 @@
             cppName = flName[:-3] + '_fl.cpp'
             hppName = flName[:-3] + '_fl.hpp'
             Command([cppName, hppName], flName,
-                    'fluid -c -o %s -h %s $SOURCES' % (cppName, hppName))
+                    'fluid1 -c -o %s -h %s $SOURCES' % (cppName, hppName))
             cppNames += [cppName]
     return cppNames
 
