diff -Naur ORIG/data/exaile.desktop UHU/data/exaile.desktop
--- ORIG/data/exaile.desktop	2010-04-07 22:09:52.000000000 +0200
+++ UHU/data/exaile.desktop	2010-04-25 13:03:24.000000000 +0200
@@ -7,7 +7,7 @@
 GenericName[fi]=Musiikkisoitin
 GenericName[en_GB]=Music Player
 GenericName[fr]=Lecteur de musique
-GenericName[hu]=Zene lejátszó
+GenericName[hu]=Zenelejátszó
 GenericName[it_IT]=Riproduttore Audio
 GenericName[nn]=Musikkavspelar
 GenericName[pl]=Odtwarzacz muzyki
