diff -Naur orig/Makefile patched/Makefile
--- orig/Makefile	2013-10-26 23:05:52.000000000 +0200
+++ patched/Makefile	2015-10-28 05:28:11.201116231 +0100
@@ -120,7 +120,7 @@
 	tools/generate-launcher "${DESTDIR}" "${PREFIX}" "${LIBINSTALLDIR}" \
 		"${PYTHON2_CMD}" && \
 	  chmod 755 ${DESTDIR}${PREFIX}/bin/exaile
-	sed "s|\@bindir\@|${EXAILEBINDIR}|" data/org.exaile.Exaile.service.in > \
+	sed "s|\@bindir\@|${PREFIX}/bin|" data/org.exaile.Exaile.service.in > \
 		${DESTDIR}${PREFIX}/share/dbus-1/services/org.exaile.Exaile.service
 	${MAKE} -C plugins install
 
