diff -Naur gnome-u2ps-0.0.4.orig/configure gnome-u2ps-0.0.4/configure
--- gnome-u2ps-0.0.4.orig/configure	2004-08-17 07:07:04.000000000 +0200
+++ gnome-u2ps-0.0.4/configure	2005-09-05 12:49:48.000000000 +0200
@@ -20834,30 +20834,6 @@
   moralaa=""
 fi
 
-if test $iconv_test = "no"; then
-  { echo "$as_me:$LINENO: WARNING: *** iconv implemantation is broken ***
-  $moralaa
-***********************************************************
-   You are using messy crappy broken iconv library.
-   Your system must go home and be kicked out from Japan
-   right now.
-***********************************************************
-
-Continuing...
-" >&5
-echo "$as_me: WARNING: *** iconv implemantation is broken ***
-  $moralaa
-***********************************************************
-   You are using messy crappy broken iconv library.
-   Your system must go home and be kicked out from Japan
-   right now.
-***********************************************************
-
-Continuing...
-" >&2;}
-  sleep 5
-fi
-
 u2ps_reqs="
 glib-2.0		>= 2.0.0
 libgnome-2.0		>= 2.0.0
