http://comments.gmane.org/gmane.linux.lfs.beyond.devel/24641
diff -urN webkitgtk-1.10.2.orig/Source/ThirdParty/ANGLE/src/compiler/glslang.y webkitgtk-1.10.2/Source/ThirdParty/ANGLE/src/compiler/glslang.y
--- webkitgtk-1.10.2.orig/Source/ThirdParty/ANGLE/src/compiler/glslang.y	2012-11-23 21:12:16.000000000 +0100
+++ webkitgtk-1.10.2/Source/ThirdParty/ANGLE/src/compiler/glslang.y	2013-09-26 15:52:38.470444817 +0200
@@ -47,6 +47,7 @@
 %expect 1 /* One shift reduce conflict because of if | else */
 %pure-parser
 %parse-param {TParseContext* context}
+%lex-param {YYLEX_PARAM}
 
 %union {
     struct {
