diff -urN stunnel-5.02.orig/tools/stunnel.conf-sample.in stunnel-5.02/tools/stunnel.conf-sample.in
--- stunnel-5.02.orig/tools/stunnel.conf-sample.in	2014-05-08 10:31:52.000000000 +0200
+++ stunnel-5.02/tools/stunnel.conf-sample.in	2014-06-10 09:08:27.105398465 +0200
@@ -9,7 +9,7 @@
 
 ; A copy of some devices and system files is needed within the chroot jail
 ; Chroot conflicts with configuration file reload and many other features
-;chroot = @prefix@/var/lib/stunnel/
+chroot = /var/lib/stunnel/
 ; Chroot jail can be escaped if setuid option is not used
 setuid = nobody
 setgid = @DEFAULT_GROUP@
@@ -26,8 +26,8 @@
 ; **************************************************************************
 
 ; Certificate/key is needed in server mode and optional in client mode
-cert = @prefix@/etc/stunnel/mail.pem
-;key = @prefix@/etc/stunnel/mail.pem
+cert = /etc/stunnel/mail.pem
+;key = /etc/stunnel/mail.pem
 
 ; Authentication stuff needs to be configured to prevent MITM attacks
 ; It is not enabled by default!
@@ -58,17 +58,17 @@
 
 ; Example SSL server mode services
 
-[pop3s]
-accept  = 995
-connect = 110
-
-[imaps]
-accept  = 993
-connect = 143
-
-[ssmtp]
-accept  = 465
-connect = 25
+;[pop3s]
+;accept  = 995
+;connect = 110
+
+;[imaps]
+;accept  = 993
+;connect = 143
+
+;[ssmtp]
+;accept  = 465
+;connect = 25
 
 ; Example SSL client mode services
 
