diff -Naurd a/lex.c b/lex.c
--- a/lex.c	2011-02-21 06:22:16.000000000 +0100
+++ b/lex.c	2011-11-24 05:50:46.000000000 +0100
@@ -26,7 +26,7 @@
 #define MAX_CHAR_VALUE 255      /* maximum value which can be represented by a "char" data type */
 
 static union AnyValue LexAnyValue;
-static struct Value LexValue = { TypeVoid, &LexAnyValue, FALSE, FALSE };
+static struct Value LexValue = { TypeVoid, &LexAnyValue, NULL, FALSE, FALSE, FALSE };
 
 struct ReservedWord
 {
@@ -197,8 +197,12 @@
             LEXER_INC(Lexer);
         }
         
-        for (Result = 0; Lexer->Pos != Lexer->End && IS_BASE_DIGIT(*Lexer->Pos, Base); LEXER_INC(Lexer))
-            Result = Result * (double)Base + GET_BASE_DIGIT(*Lexer->Pos);
+        Result = 0;
+        while (Lexer->Pos != Lexer->End && IS_BASE_DIGIT(*Lexer->Pos, Base))
+        {
+        	Result = Result * Base + GET_BASE_DIGIT(*Lexer->Pos);
+        	LEXER_INC(Lexer);
+        }
             
         FPResult *= pow((double)Base, (double)Result * ExponentMultiplier);
     }
diff -Naurd a/Makefile b/Makefile
--- a/Makefile	2011-02-21 06:27:03.000000000 +0100
+++ b/Makefile	2011-04-07 20:08:41.000000000 +0200
@@ -1,6 +1,6 @@
 CC=gcc
 CFLAGS=-Wall -pedantic -g -DUNIX_HOST
-LIBS=-lm -lreadline
+LIBS=-lm -lreadline -lncurses
 
 TARGET	= picoc
 SRCS	= picoc.c table.c lex.c parse.c expression.c heap.c type.c \
diff -Naurd a/picoc.c b/picoc.c
--- a/picoc.c	2011-02-21 06:22:16.000000000 +0100
+++ b/picoc.c	2011-04-07 20:10:30.000000000 +0200
@@ -82,7 +82,7 @@
     }
 
     PicocExitBuf[40] = 0;
-    setjmp(PicocExitBuf);
+    PicocPlatformSetExitPoint();
     if (PicocExitBuf[40]) {
         printf("Leaving PicoC\n\r");
         PicocCleanup();
diff -Naurd a/platform/library_surveyor.c b/platform/library_surveyor.c
--- a/platform/library_surveyor.c	2011-02-21 06:22:13.000000000 +0100
+++ b/platform/library_surveyor.c	2011-04-07 20:12:24.000000000 +0200
@@ -851,8 +851,7 @@
         if (getchar(&ch)) {
             if (ch == 0x1B) {  // if ESC found, exit picoC
                 printf("found ESC\r\n");
-                PicocExitBuf[40] = 1;
-                longjmp(PicocExitBuf, 1);
+                PlatformExit(0);
             }
         }
     }
