diff -Naur xf86-video-tdfx-1.4.5-orig/src/tdfx_driver.c xf86-video-tdfx-1.4.5/src/tdfx_driver.c
--- xf86-video-tdfx-1.4.5-orig/src/tdfx_driver.c	2012-07-17 07:21:16.000000000 +0200
+++ xf86-video-tdfx-1.4.5/src/tdfx_driver.c	2013-03-31 12:13:52.000000000 +0200
@@ -62,10 +62,6 @@
 
 #include "compiler.h"
 
-/* Drivers using the mi implementation of backing store need: */
-
-#include "mibstore.h"
-
 /* All drivers using the vgahw module need this */
 /* This driver needs to be modified to not use vgaHW for multihead operation */
 #include "vgaHW.h"
