diff -Naur orig/desktop/scripts/soffice.sh patched/desktop/scripts/soffice.sh
--- orig/desktop/scripts/soffice.sh	2016-12-23 21:08:05.000000000 +0100
+++ patched/desktop/scripts/soffice.sh	2017-01-11 11:17:23.709119111 +0100
@@ -137,6 +137,9 @@
     exit 1;
 fi
 
+PYTHONPATH=$sd_prog${PYTHONPATH+:$PYTHONPATH}
+export PYTHONPATH
+
 case "`uname -s`" in
 NetBSD|OpenBSD|DragonFly)
 # this is a temporary hack until we can live with the default search paths
diff -Naur orig/pyuno/source/module/uno.py patched/pyuno/source/module/uno.py
--- orig/pyuno/source/module/uno.py	2016-12-23 21:08:05.000000000 +0100
+++ patched/pyuno/source/module/uno.py	2017-01-11 11:20:04.494782743 +0100
@@ -17,6 +17,7 @@
 #   the License at http://www.apache.org/licenses/LICENSE-2.0 .
 #
 import pyuno
+import os
 import sys
 import traceback
 import warnings
@@ -24,6 +25,10 @@
 # since on Windows sal3.dll no longer calls WSAStartup
 import socket
 
+sys.path.append('/usr/lib/libreoffice/program')
+if getattr(os.environ, 'URE_BOOTSTRAP', None) is None:
+    os.environ['URE_BOOTSTRAP'] = "vnd.sun.star.pathname:/usr/lib/libreoffice/program/fundamentalrc"
+
 # Some Python 2/3 compatibility code copied from the six library
 PY2 = sys.version_info[0] == 2
 
diff -Naur orig/pyuno/source/officehelper.py patched/pyuno/source/officehelper.py
--- orig/pyuno/source/officehelper.py	2016-12-23 21:08:05.000000000 +0100
+++ patched/pyuno/source/officehelper.py	2017-01-11 11:17:23.709119111 +0100
@@ -45,7 +45,7 @@
         if "UNO_PATH" in os.environ:
             sOffice = os.environ["UNO_PATH"]
         else:
-            sOffice = "" # lets hope for the best
+            sOffice = "/usr/lib/libreoffice/program"
         sOffice = os.path.join(sOffice, "soffice")
         if platform.startswith("win"):
             sOffice += ".exe"
