diff -Naur orig/misc/mc.keymap patched/misc/mc.keymap
--- orig/misc/mc.keymap	2016-03-12 16:45:47.000000000 +0100
+++ patched/misc/mc.keymap	2016-03-28 12:40:06.320783533 +0200
@@ -119,10 +119,10 @@
 # SortPrev =
 # SortNext =
 # SortReverse =
-# SortByName =
-# SortByExt =
-# SortBySize =
-# SortByMTime =
+SortByName = alt-shift-n
+SortByExt = alt-shift-e
+SortBySize = alt-shift-s
+SortByMTime = alt-shift-t
 # ScrollLeft =
 # ScrollRight =
 
