diff -Naur wxmaxima-16.04.1-orig/data/wxMaxima.desktop wxmaxima-16.04.1/data/wxMaxima.desktop
--- wxmaxima-16.04.1-orig/data/wxMaxima.desktop	2016-04-06 12:11:47.000000000 +0200
+++ wxmaxima-16.04.1/data/wxMaxima.desktop	2016-04-29 17:30:06.585617390 +0200
@@ -6,6 +6,7 @@
 Comment=A graphical user interface that allows symbolic and numeric calculations using Maxima
 Comment[de]=Eine grafische Oberfläche für Maxima, ein Programm für Symbolische und Numerische Berechnungen
 Comment[ru]=Графический интерфейс для символьных и числовых вычислений с помощью Maxima
+Comment[hu]=Matematikai programcsomag
 X-GNOME-FullName=wxMaxima - GUI for Maxima
 X-GNOME-FullName[ru]=wxMaxima - графический интерфейс для Maxima
 X-GNOME-FullName[de]=wxMaxima - eine grafische Oberfläche für Maxima
