diff -Naur pykde4-4.12.3-orig/sip/phonon/backendcapabilities.sip pykde4-4.12.3/sip/phonon/backendcapabilities.sip
--- pykde4-4.12.3-orig/sip/phonon/backendcapabilities.sip	2014-01-18 19:18:35.000000000 +0000
+++ pykde4-4.12.3/sip/phonon/backendcapabilities.sip	2014-03-24 19:43:22.790926057 +0000
@@ -26,7 +26,7 @@
 {
 namespace BackendCapabilities
 {
-class Notifier : QObject /NoDefaultCtors,PyQt4NoQMetaObject/
+class Notifier : QObject /NoDefaultCtors,PyQtNoQMetaObject/
 {
 
 signals:
