diff -urN grub-0.92.orig/util/grub-install.in grub-0.92/util/grub-install.in
--- grub-0.92.orig/util/grub-install.in	2003-01-05 18:09:42.000000000 +0100
+++ grub-0.92/util/grub-install.in	2003-01-05 18:13:15.000000000 +0100
@@ -45,9 +45,9 @@
 if test -x /bin/tempfile; then
     mklog="/bin/tempfile --prefix=grub"
     mkimg="/bin/tempfile --prefix=grub"
-elif test -x /bin/mktemp; then
-    mklog="/bin/mktemp /tmp/grub-install.log.XXXXXX"
-    mkimg="/bin/mktemp /tmp/grub-install.img.XXXXXX"
+elif test -x /usr/bin/mktemp; then
+    mklog="/usr/bin/mktemp /tmp/grub-install.log.XXXXXX"
+    mkimg="/usr/bin/mktemp /tmp/grub-install.img.XXXXXX"
 else
     mklog=""
     mkimg=""
