diff -Nur orig/util/grub-mkconfig_lib.in mod/util/grub-mkconfig_lib.in
--- orig/util/grub-mkconfig_lib.in	2016-09-21 19:43:34.000000000 +0200
+++ mod/util/grub-mkconfig_lib.in	2016-09-21 19:57:53.157971830 +0200
@@ -240,7 +240,7 @@
 {
   version_test_gt_a="`echo "$1" | sed -e "s/[^-]*-//"`"
   version_test_gt_b="`echo "$2" | sed -e "s/[^-]*-//"`"
-  version_test_gt_cmp=gt
+  version_test_gt_cmp=lt
   if [ "x$version_test_gt_b" = "x" ] ; then
     return 0
   fi
