diff -Naur orig/Makefile mod/Makefile
--- orig/Makefile	2011-03-16 22:45:13.000000000 +0100
+++ mod/Makefile	2016-03-25 09:13:36.754117900 +0100
@@ -7,9 +7,9 @@
 all: $(MAN1)
 
 install: $(MAN1)
-	mkdir -p $(prefix)/bin $(prefix)/man/man1
+	mkdir -p $(prefix)/bin $(prefix)/share/man/man1
 	cp $(BIN) $(prefix)/bin/
-	cp $(MAN1) $(prefix)/man/man1
+	cp $(MAN1) $(prefix)/share/man/man1
 
 clean:
 	rm -f *.1 *.txt *.ps *.pdf *.html
