--- f-spot.orig/data/desktop-files/f-spot.desktop.in.in	2009-05-02 14:15:48.000000000 +0200
+++ f-spot/data/desktop-files/f-spot.desktop.in.in	2009-08-26 19:40:42.000000000 +0200
@@ -9,7 +9,7 @@
 StartupNotify=true
 Terminal=false
 Type=Application
-Categories=Graphics;Photography;GNOME;GTK;
+Categories=GTK;GNOME;Graphics;Photography;X-UHU-Menu/Graphics;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=f-spot
 X-GNOME-Bugzilla-Component=General
