diff -Naur orig/wscript mod/wscript
--- orig/wscript	2011-04-18 09:42:29.000000000 +0200
+++ mod/wscript	2016-02-28 20:25:22.368347216 +0100
@@ -13,18 +13,18 @@
 
 def options(ctx):
   print('  %s/options()' %d)
-  ctx.tool_options('gnu_dirs')
-  ctx.tool_options('compiler_cc')
-  ctx.tool_options('python') # options for disabling pyc or pyo compilation
+  ctx.load('gnu_dirs')
+  ctx.load('compiler_cc')
+  ctx.load('python') # options for disabling pyc or pyo compilation
 
 
 def configure(ctx):
   print('  %s/configure()' %d)
 
   env = ctx.env
-  ctx.check_tool('gnu_dirs')
-  ctx.check_tool('compiler_cc')
-  ctx.check_tool('python')
+  ctx.load('gnu_dirs')
+  ctx.load('compiler_cc')
+  ctx.load('python')
   ctx.check_python_version((3,1,0))
   ctx.check_python_headers()
   ctx.check_cfg(package='cairo', atleast_version=cairo_version_required,
