diff -Naur orig/data/org.gnome.Shell.desktop.in.in mod/data/org.gnome.Shell.desktop.in.in
--- orig/data/org.gnome.Shell.desktop.in.in	2015-12-02 10:36:16.000000000 +0100
+++ mod/data/org.gnome.Shell.desktop.in.in	2016-02-13 18:48:09.074174192 +0100
@@ -2,7 +2,7 @@
 Type=Application
 _Name=GNOME Shell
 _Comment=Window management and application launching
-Exec=@bindir@/gnome-shell
+Exec=gnome-shell
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=gnome-shell
 X-GNOME-Bugzilla-Component=general
