diff -Naur orig/data/gnome-shell-extension-prefs.desktop.in.in mod/data/gnome-shell-extension-prefs.desktop.in.in
--- orig/data/gnome-shell-extension-prefs.desktop.in.in	2015-12-17 01:23:51.000000000 +0100
+++ mod/data/gnome-shell-extension-prefs.desktop.in.in	2016-02-13 18:51:59.915289788 +0100
@@ -2,7 +2,7 @@
 Type=Application
 _Name=GNOME Shell Extension Preferences
 _Comment=Configure GNOME Shell Extensions
-Exec=@bindir@/gnome-shell-extension-prefs %u
+Exec=gnome-shell-extension-prefs %u
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=gnome-shell
 X-GNOME-Bugzilla-Component=extensions
