diff -Naurdp mercurial-2.4/mercurial/cmdutil.py mercurial-2.4-closedbranch/mercurial/cmdutil.py
--- mercurial-2.4/mercurial/cmdutil.py	2012-11-01 22:09:16.000000000 +0100
+++ mercurial-2.4-closedbranch/mercurial/cmdutil.py	2013-01-25 10:03:09.436240424 +0100
@@ -1461,6 +1461,8 @@ def displaygraph(ui, dag, displayer, sho
             char = '@'
         elif ctx.obsolete():
             char = 'x'
+        elif ctx.closesbranch():
+            char = '_'
         copies = None
         if getrenamed and ctx.rev():
             copies = []
