diff -Naur ORIG/setup.py UHU/setup.py
--- ORIG/setup.py	2014-06-01 21:15:14.000000000 +0000
+++ UHU/setup.py	2014-06-03 02:34:33.404281250 +0000
@@ -255,9 +255,9 @@
             mofile = join(modir, 'hg.mo')
             mobuildfile = join('mercurial', mofile)
             cmd = ['msgfmt', '-v', '-o', mobuildfile, pofile]
-            if sys.platform != 'sunos5':
+#            if sys.platform != 'sunos5':
                 # msgfmt on Solaris does not know about -c
-                cmd.append('-c')
+#                cmd.append('-c')
             self.mkpath(join('mercurial', modir))
             self.make_file([pofile], mobuildfile, spawn, (cmd,))
 
