diff -Naur orig/app/core/gimpparamspecs-duplicate.c patched/app/core/gimpparamspecs-duplicate.c
--- orig/app/core/gimpparamspecs-duplicate.c	2015-08-21 00:57:25.000000000 +0200
+++ patched/app/core/gimpparamspecs-duplicate.c	2016-07-16 14:17:11.315294971 +0200
@@ -53,28 +53,6 @@
                                               spec->default_value,
                                               pspec->flags);
         }
-      else
-        {
-          static GQuark  multiline_quark = 0;
-          GParamSpec    *new;
-
-          if (! multiline_quark)
-            multiline_quark = g_quark_from_static_string ("multiline");
-
-          new = g_param_spec_string (pspec->name,
-                                     g_param_spec_get_nick (pspec),
-                                     g_param_spec_get_blurb (pspec),
-                                     spec->default_value,
-                                     pspec->flags);
-
-          if (GEGL_IS_PARAM_SPEC_MULTILINE (pspec))
-            {
-              g_param_spec_set_qdata (new, multiline_quark,
-                                      GINT_TO_POINTER (TRUE));
-            }
-
-          return new;
-        }
     }
   else if (G_IS_PARAM_SPEC_BOOLEAN (pspec))
     {
diff -Naur orig/app/widgets/gimppropwidgets.c patched/app/widgets/gimppropwidgets.c
--- orig/app/widgets/gimppropwidgets.c	2015-08-21 00:57:26.000000000 +0200
+++ patched/app/widgets/gimppropwidgets.c	2016-07-16 14:25:30.347191122 +0200
@@ -1431,11 +1431,6 @@
 
       if (G_IS_PARAM_SPEC_STRING (pspec))
         {
-          static GQuark multiline_quark = 0;
-
-          if (! multiline_quark)
-            multiline_quark = g_quark_from_static_string ("multiline");
-
           if (GIMP_IS_PARAM_SPEC_CONFIG_PATH (pspec))
             {
               widget = gimp_prop_file_chooser_button_new (config,
@@ -1443,20 +1438,6 @@
                                                           g_param_spec_get_nick (pspec),
                                                           GTK_FILE_CHOOSER_ACTION_OPEN);
             }
-          else if (g_param_spec_get_qdata (pspec, multiline_quark))
-            {
-              GtkTextBuffer *buffer;
-              GtkWidget     *view;
-
-              buffer = gimp_prop_text_buffer_new (config, pspec->name, -1);
-              view = gtk_text_view_new_with_buffer (buffer);
-
-              widget = gtk_scrolled_window_new (NULL, NULL);
-              gtk_scrolled_window_set_shadow_type (GTK_SCROLLED_WINDOW (widget),
-                                                   GTK_SHADOW_IN);
-              gtk_container_add (GTK_CONTAINER (widget), view);
-              gtk_widget_show (view);
-            }
           else
             {
               widget = gimp_prop_entry_new (config, pspec->name, -1);
