diff -Naur orig/src/vteapp.c patched/src/vteapp.c
--- orig/src/vteapp.c	2015-10-14 07:56:24.000000000 +0200
+++ patched/src/vteapp.c	2015-12-13 02:06:25.793838563 +0100
@@ -573,7 +573,7 @@
 	char *geometry = NULL;
 	gint lines = -1;
 	const char *message = "Launching interactive shell...\r\n";
-	const char *font = NULL;
+	const char *font = "Fixed Bold 12";
 	const char *command = NULL;
 	const char *working_directory = NULL;
 	const char *output_file = NULL;
