diff -Naurdp a/src/vte-private.h b/src/vte-private.h
--- a/src/vte-private.h	2014-06-23 19:26:54.000000000 +0200
+++ b/src/vte-private.h	2014-06-23 23:00:21.433927676 +0200
@@ -101,9 +101,9 @@ G_BEGIN_DECLS
 #define VTE_INPUT_CHUNK_SIZE		0x2000
 #define VTE_MAX_INPUT_READ		0x1000
 #define VTE_INVALID_BYTE		'?'
-#define VTE_DISPLAY_TIMEOUT		10
-#define VTE_UPDATE_TIMEOUT		15
-#define VTE_UPDATE_REPEAT_TIMEOUT	30
+#define VTE_DISPLAY_TIMEOUT		5
+#define VTE_UPDATE_TIMEOUT		5
+#define VTE_UPDATE_REPEAT_TIMEOUT	10
 #define VTE_MAX_PROCESS_TIME		100
 #define VTE_CELL_BBOX_SLACK		1
 
