diff -Naur orig/telepathy-glib/telepathy-glib-1-dbus.pc.in patched/telepathy-glib/telepathy-glib-1-dbus.pc.in
--- orig/telepathy-glib/telepathy-glib-1-dbus.pc.in	2014-04-04 13:12:45.000000000 +0200
+++ patched/telepathy-glib/telepathy-glib-1-dbus.pc.in	2016-03-31 01:59:08.383402765 +0200
@@ -6,7 +6,6 @@
 Name: Telepathy-GLib-DBus
 Description: D-Bus interfaces from the Telepathy framework
 Version: @VERSION@
-Requires: pkg-config >= 0.21
 Requires.private: dbus-1 >= 0.95, dbus-glib-1 >= 0.90, glib-2.0 >= 2.30, gobject-2.0 >= 2.30, gio-2.0 >= 2.30
 Libs: -L${libdir} -ltelepathy-glib-1-dbus
 Cflags: -I${includedir}/telepathy-glib-1-dbus -I${includedir}/telepathy-glib-1
diff -Naur orig/telepathy-glib/telepathy-glib-1-dbus-uninstalled.pc.in patched/telepathy-glib/telepathy-glib-1-dbus-uninstalled.pc.in
--- orig/telepathy-glib/telepathy-glib-1-dbus-uninstalled.pc.in	2014-04-04 13:12:45.000000000 +0200
+++ patched/telepathy-glib/telepathy-glib-1-dbus-uninstalled.pc.in	2016-03-31 01:58:59.003356252 +0200
@@ -6,7 +6,6 @@
 Name: Telepathy-GLib-DBus (uninstalled copy)
 Description: D-Bus interfaces from the Telepathy framework
 Version: @VERSION@
-Requires: pkg-config >= 0.21
 Requires.private: dbus-1 >= 0.95, dbus-glib-1 >= 0.90, glib-2.0 >= 2.30, gobject-2.0 >= 2.30, gio-2.0 >= 2.30
 Libs: ${abs_top_builddir}/telepathy-glib/libtelepathy-glib-1-dbus.la
 Cflags: -I${abs_top_srcdir} -I${abs_top_builddir}
