diff -urN banshee.orig/data/desktop-files/common.desktop.in banshee/data/desktop-files/common.desktop.in
--- banshee.orig/data/desktop-files/common.desktop.in	2009-03-04 21:58:51.000000000 +0100
+++ banshee/data/desktop-files/common.desktop.in	2009-03-22 17:58:55.000000000 +0100
@@ -6,7 +6,7 @@
 StartupNotify=true
 Terminal=false
 Type=Application
-Categories=GTK;GNOME;Audio;Music;Player;AudioVideo;X-Ximian-Main;X-Novell-Main;X-Red-Hat-Base;
+Categories=GTK;GNOME;Audio;Music;Player;AudioVideo;X-Ximian-Main;X-Novell-Main;X-Red-Hat-Base;X-UHU-Menu/Multimedia/MusicPlayers;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=banshee
 X-GNOME-Bugzilla-Component=General
