diff -Naurd a/data/bs-ng.desktop b/data/bs-ng.desktop
--- a/data/bs-ng.desktop	2013-02-10 19:00:08.000000000 +0100
+++ b/data/bs-ng.desktop	2013-02-16 07:57:01.000000000 +0100
@@ -1,9 +1,11 @@
 [Desktop Entry]
-Version=1.0
+Version=1.1
 Type=Application
 Terminal=false
 Icon=bs-ng
 Name=BashStyle-NG
 Exec=bashstyle
 Comment=Configure Bash, ReadLine, Vim and Nano
-Categories=GTK;GNOME;Utility;
+Comment[hu]=Bash, ReadLine, Vim és Nano beállító
+Categories=GTK;GNOME;Utility;X-UHU-Menu/Tools/Utilities;
+
