diff -Naur orig/configure.ac patched/configure.ac
--- orig/configure.ac	2015-11-13 15:32:40.000000000 +0100
+++ patched/configure.ac	2015-11-15 22:55:44.097743782 +0100
@@ -1149,7 +1149,13 @@
 
         if test "x$with_osmesa" != "xno"
         then
-            WINE_CHECK_SONAME(OSMesa,glAccum,,,[$X_LIBS -lm $X_EXTRA_LIBS])
+            WINE_CHECK_SONAME(OSMesa,OSMesaCreateContext,,,[$X_LIBS -lm $X_EXTRA_LIBS])
+            if test "x$ac_cv_lib_soname_OSMesa" = "x"; then
+                osmesa_save_CC=$CC
+                CC=$CXX
+                WINE_CHECK_SONAME(OSMesa,OSMesaCreateContext,,,[-lglapi -lpthread -ldl $X_LIBS -lm $X_EXTRA_LIBS])
+                CC=$osmesa_save_CC
+            fi
             WINE_NOTICE_WITH(osmesa,[test "x$ac_cv_lib_soname_OSMesa" = "x"],
                              [libOSMesa ${notice_platform}development files not found (or too old), OpenGL rendering in bitmaps won't be supported.])
         fi
