diff -Naurdp LVM2.2.02.105.orig/lib/filters/filter-partitioned.c LVM2.2.02.105/lib/filters/filter-partitioned.c
--- LVM2.2.02.105.orig/lib/filters/filter-partitioned.c	2014-01-20 20:25:30.000000000 +0100
+++ LVM2.2.02.105/lib/filters/filter-partitioned.c	2014-02-26 13:14:52.022748188 +0100
@@ -15,6 +15,11 @@
 
 #include "lib.h"
 #include "filter.h"
+#include <sys/ioctl.h>
+#include <linux/fs.h>
+#undef MAJOR
+#undef MINOR
+#undef MKDEV
 
 static int _passes_partitioned_filter(struct dev_filter *f, struct device *dev)
 {
@@ -24,10 +29,17 @@ static int _passes_partitioned_filter(st
 	uint64_t size;
 
 	/* Check it's accessible */
-	if (!dev_open_readonly_quiet(dev)) {
+	if (!dev_open_flags(dev, O_RDONLY|O_NONBLOCK, 0, 1)) {
 		log_debug_devs("%s: Skipping: open failed", name);
 		return 0;
 	}
+	
+	/* Skip cdrom device */
+	#define CDROM_GET_CAPABILITY   0x5331
+	if (ioctl(dev->fd, CDROM_GET_CAPABILITY) >= 0) {
+		log_debug_devs("%s: Skipping: cdrom device", name );
+		goto out;
+	}
 
 	/* Check it's not too small */
 	if (!dev_get_size(dev, &size)) {
