diff -Naur nail-11.25.orig/nail.rc nail-11.25/nail.rc
--- nail-11.25.orig/nail.rc	2005-01-15 15:33:16.000000000 +0100
+++ nail-11.25/nail.rc	2005-09-06 17:49:33.000000000 +0200
@@ -38,7 +38,7 @@
 
 # Outgoing messages are sent in ISO-8859-1 if all their characters are
 # representable in it, otherwise in UTF-8.
-set sendcharsets=iso-8859-1,utf-8
+set sendcharsets=iso-8859-1,iso-8859-2,utf-8
 
 # Display sender's real names in header summaries.
 set showname
