diff -Naur orig/src/org.mate.applets.UptimeApplet.mate-panel-applet.in patched/src/org.mate.applets.UptimeApplet.mate-panel-applet.in
--- orig/src/org.mate.applets.UptimeApplet.mate-panel-applet.in	2016-01-06 23:30:48.000000000 +0100
+++ patched/src/org.mate.applets.UptimeApplet.mate-panel-applet.in	2017-01-08 23:34:02.362954555 +0100
@@ -4,20 +4,24 @@
 Name=Uptime applet
 Name[en_GB]=Uptime applet
 Name[bg_BG]=Индикатор на uptime
+Name[hu_HU]=Uptime applet
 Name[ru_RU]=Индикатор uptime
 Description=Display system uptime
 Description[en_GB]=Display system uptime
 Description[bg_BG]=Показва uptime на машината
+Description[hu_HU]=A rendszer futásidejét kijelző applet
 Description[ru_RU]=Показывает uptime
 
 [UptimeApplet]
 Name=Uptime applet
 Name[en_GB]=Uptime applet
 Name[bg_BG]=Индикатор на uptime
+Name[hu_HU]=Uptime applet
 Name[ru_RU]=Индикатор uptime
 Description=Display system uptime
 Description[en_GB]=Display system uptime
 Description[bg_BG]=Показва uptime на машината
+Description[hu_HU]=A rendszer futásidejét kijelző applet
 Description[ru_RU]=Показывает uptime
 Icon=applet_uptime
 MateComponentId=OAFIID:MATE_UptimeApplet
