diff -Naur AdobeReader-orig/Adobe/Reader9/Resource/Support/AdobeReader.desktop AdobeReader/Adobe/Reader9/Resource/Support/AdobeReader.desktop
--- AdobeReader-orig/Adobe/Reader9/Resource/Support/AdobeReader.desktop	2009-10-08 02:31:40.000000000 +0200
+++ AdobeReader/Adobe/Reader9/Resource/Support/AdobeReader.desktop	2009-10-25 17:28:47.000000000 +0100
@@ -4,9 +4,10 @@
 Exec=acroread 
 Type=Application
 GenericName=PDF Viewer
+GenericName[hu]=PDF-megjelenítő
 Terminal=false
 Icon=AdobeReader9
 Caption=PDF Viewer
 X-KDE-StartupNotify=false
-Categories=Application;Office;Viewer;X-Red-Hat-Base;
+Categories=Application;Office;Viewer;X-Red-Hat-Base;X-UHU-Menu/Tools/Viewers;
 InitialPreference=9
