diff -Naur orig/mozilla/build/autoconf/icu.m4 patched/mozilla/build/autoconf/icu.m4
--- orig/mozilla/build/autoconf/icu.m4	2016-12-14 03:09:42.000000000 +0100
+++ patched/mozilla/build/autoconf/icu.m4	2017-02-27 17:10:59.000000000 +0100
@@ -15,12 +15,11 @@
     MOZ_SYSTEM_ICU=1)
 
 if test -n "$MOZ_SYSTEM_ICU"; then
-    PKG_CHECK_MODULES(MOZ_ICU, icu-i18n >= 50.1)
-else
-    MOZ_ICU_INCLUDES="/intl/icu/source/common /intl/icu/source/i18n"
+    PKG_CHECK_MODULES(MOZ_ICU, icu-i18n >= 58.1)
+    CFLAGS="$CFLAGS $MOZ_ICU_CFLAGS"
+    CXXFLAGS="$CXXFLAGS $MOZ_ICU_CFLAGS"
 fi
 
-AC_SUBST_LIST(MOZ_ICU_INCLUDES)
 AC_SUBST(MOZ_SYSTEM_ICU)
 
 MOZ_ARG_WITH_STRING(intl-api,
@@ -70,7 +69,7 @@
         fi
     fi
 
-    version=`sed -n 's/^[[:space:]]*#[[:space:]]*define[[:space:]][[:space:]]*U_ICU_VERSION_MAJOR_NUM[[:space:]][[:space:]]*\([0-9][0-9]*\)[[:space:]]*$/\1/p' "$icudir/common/unicode/uvernum.h"`
+    version=`sed -n 's/^[[[:space:]]]*#[[:space:]]*define[[:space:]][[:space:]]*U_ICU_VERSION_MAJOR_NUM[[:space:]][[:space:]]*\([0-9][0-9]*\)[[:space:]]*$/\1/p' "$icudir/common/unicode/uvernum.h"`
     if test x"$version" = x; then
        AC_MSG_ERROR([cannot determine icu version number from uvernum.h header file $lineno])
     fi
@@ -86,7 +85,7 @@
     dnl We also don't do it on Windows because sometimes the file goes
     dnl missing -- possibly due to overzealous antivirus software? --
     dnl which prevents the browser from starting up :(
-    if test -z "$JS_STANDALONE" -a -z "$MOZ_SYSTEM_ICU" -a "$OS_TARGET" != WINNT; then
+    if test -z "$JS_STANDALONE" -a -z "$MOZ_SYSTEM_ICU" -a "$OS_TARGET" != WINNT -a "$MOZ_WIDGET_TOOLKIT" != "android"; then
         MOZ_ICU_DATA_ARCHIVE=1
     else
         MOZ_ICU_DATA_ARCHIVE=
diff -Naur orig/mozilla/dom/indexedDB/moz.build patched/mozilla/dom/indexedDB/moz.build
--- orig/mozilla/dom/indexedDB/moz.build	2016-12-14 03:09:31.000000000 +0100
+++ patched/mozilla/dom/indexedDB/moz.build	2017-03-15 16:04:25.537634576 +0100
@@ -116,4 +116,3 @@
 
 if CONFIG['ENABLE_INTL_API']:
     CXXFLAGS += CONFIG['MOZ_ICU_CFLAGS']
-    LOCAL_INCLUDES += CONFIG['MOZ_ICU_INCLUDES']
diff -Naur orig/mozilla/gfx/thebes/moz.build patched/mozilla/gfx/thebes/moz.build
--- orig/mozilla/gfx/thebes/moz.build	2016-12-14 03:09:40.000000000 +0100
+++ patched/mozilla/gfx/thebes/moz.build	2017-03-15 16:01:22.819476837 +0100
@@ -251,7 +251,6 @@
 if CONFIG['ENABLE_INTL_API']:
     CXXFLAGS += CONFIG['MOZ_ICU_CFLAGS']
     CFLAGS += CONFIG['MOZ_ICU_CFLAGS']
-    LOCAL_INCLUDES += CONFIG['MOZ_ICU_INCLUDES']
     USE_LIBS += [
         'icu',
     ]
diff -Naur orig/mozilla/intl/build/moz.build patched/mozilla/intl/build/moz.build
--- orig/mozilla/intl/build/moz.build	2016-12-14 03:09:42.000000000 +0100
+++ patched/mozilla/intl/build/moz.build	2017-03-15 15:54:55.950882218 +0100
@@ -10,7 +10,6 @@
 
 FINAL_LIBRARY = 'xul'
 CXXFLAGS += CONFIG['MOZ_ICU_CFLAGS']
-LOCAL_INCLUDES += CONFIG['MOZ_ICU_INCLUDES']
 LOCAL_INCLUDES += [
     '../locale',
     '../lwbrk',
diff -Naur orig/mozilla/intl/unicharutil/util/internal/moz.build patched/mozilla/intl/unicharutil/util/internal/moz.build
--- orig/mozilla/intl/unicharutil/util/internal/moz.build	2016-12-14 03:09:42.000000000 +0100
+++ patched/mozilla/intl/unicharutil/util/internal/moz.build	2017-03-15 15:33:54.000000000 +0100
@@ -20,7 +20,6 @@
 
 if CONFIG['ENABLE_INTL_API']:
     CXXFLAGS += CONFIG['MOZ_ICU_CFLAGS']
-    LOCAL_INCLUDES += CONFIG['MOZ_ICU_INCLUDES']
 
 DIST_INSTALL = True
 
diff -Naur orig/mozilla/intl/unicharutil/util/moz.build patched/mozilla/intl/unicharutil/util/moz.build
--- orig/mozilla/intl/unicharutil/util/moz.build	2016-12-14 03:09:42.000000000 +0100
+++ patched/mozilla/intl/unicharutil/util/moz.build	2017-03-15 15:34:43.824108154 +0100
@@ -41,7 +41,6 @@
 
 if CONFIG['ENABLE_INTL_API']:
     CXXFLAGS += CONFIG['MOZ_ICU_CFLAGS']
-    LOCAL_INCLUDES += CONFIG['MOZ_ICU_INCLUDES']
     USE_LIBS += ['icu']
 
 DIST_INSTALL = True
diff -Naur orig/mozilla/js/src/moz.build patched/mozilla/js/src/moz.build
--- orig/mozilla/js/src/moz.build	2016-12-14 03:09:44.000000000 +0100
+++ patched/mozilla/js/src/moz.build	2017-03-15 15:12:55.908119239 +0100
@@ -724,7 +724,6 @@
 
 CFLAGS += CONFIG['MOZ_ICU_CFLAGS']
 CXXFLAGS += CONFIG['MOZ_ICU_CFLAGS']
-LOCAL_INCLUDES += CONFIG['MOZ_ICU_INCLUDES']
 
 NO_EXPAND_LIBS = True
 
diff -Naur orig/mozilla/layout/forms/moz.build patched/mozilla/layout/forms/moz.build
--- orig/mozilla/layout/forms/moz.build	2016-12-14 03:09:46.000000000 +0100
+++ patched/mozilla/layout/forms/moz.build	2017-03-15 16:01:55.419862172 +0100
@@ -57,7 +57,6 @@
     # nsNumberControlFrame.cpp requires ICUUtils.h which in turn requires
     # i18n/unum.h
     CXXFLAGS += CONFIG['MOZ_ICU_CFLAGS']
-    LOCAL_INCLUDES += CONFIG['MOZ_ICU_INCLUDES']
 
 if CONFIG['GNU_CXX']:
     CXXFLAGS += ['-Wno-error=shadow']
diff -Naur orig/mozilla/netwerk/dns/moz.build patched/mozilla/netwerk/dns/moz.build
--- orig/mozilla/netwerk/dns/moz.build	2016-12-14 03:10:00.000000000 +0100
+++ patched/mozilla/netwerk/dns/moz.build	2017-03-15 16:02:18.844138943 +0100
@@ -73,7 +73,6 @@
     DEFINES['IDNA2008'] = True
     CXXFLAGS += CONFIG['MOZ_ICU_CFLAGS']
     CFLAGS += CONFIG['MOZ_ICU_CFLAGS']
-    LOCAL_INCLUDES += CONFIG['MOZ_ICU_INCLUDES']
     USE_LIBS += ['icu']
 else:
     UNIFIED_SOURCES += [
diff -Naur orig/mozilla/xpcom/build/moz.build patched/mozilla/xpcom/build/moz.build
--- orig/mozilla/xpcom/build/moz.build	2016-12-14 03:10:01.000000000 +0100
+++ patched/mozilla/xpcom/build/moz.build	2017-03-15 15:24:24.372768858 +0100
@@ -102,4 +102,3 @@
     CXXFLAGS += CONFIG['TK_CFLAGS']
 
 CXXFLAGS += CONFIG['MOZ_ICU_CFLAGS']
-LOCAL_INCLUDES += CONFIG['MOZ_ICU_INCLUDES']
