diff -Naur orig/eric/eric5.desktop mod/eric/eric5.desktop
--- orig/eric/eric5.desktop	2013-02-03 14:01:19.000000000 +0100
+++ mod/eric/eric5.desktop	2016-04-30 23:29:40.398869086 +0200
@@ -1,5 +1,4 @@
 [Desktop Entry]
-Version=1.0
 Type=Application
 Exec=eric5
 MimeType=text/x-python;
@@ -9,7 +8,9 @@
 Name[de]=eric5
 Comment=Integrated Development Environment for Python
 Comment[de]=Integrierte Entwicklungsumgebung für Python
+Comment[hu]=Python fejlesztő felület
 GenericName=Integrated Development Environment for Python
 GenericName[de]=Integrierte Entwicklungsumgebung für Python
-Categories=Qt;Python;Development;IDE;
+GenericName[hu]=Python alkalmazás fejlesztő
+Categories=Qt;Development;IDE;
 StartupNotify=true
diff -Naur orig/eric/eric5_webbrowser.desktop mod/eric/eric5_webbrowser.desktop
--- orig/eric/eric5_webbrowser.desktop	2013-02-03 14:01:19.000000000 +0100
+++ mod/eric/eric5_webbrowser.desktop	2016-04-30 23:27:54.038873312 +0200
@@ -11,5 +11,5 @@
 Comment[de]=Web Browser für PyQt4
 GenericName=Web Browser
 GenericName[de]=Web Browser
-Categories=Qt;Python;Network;WebBrowser;
+Categories=Qt;Network;WebBrowser;
 StartupNotify=true
