diff -Naur orig/data/moovida.desktop mod/data/moovida.desktop
--- orig/data/moovida.desktop	2009-11-23 11:49:40.000000000 +0100
+++ mod/data/moovida.desktop	2016-04-30 22:20:30.600073928 +0200
@@ -1,5 +1,4 @@
 [Desktop Entry]
-Version=1.0
 Name=Moovida Media Center
 Comment=Play movies and songs on TV with remote
 Exec=elisa %U
diff -Naur orig/data/moovida-mobile.desktop mod/data/moovida-mobile.desktop
--- orig/data/moovida-mobile.desktop	2009-11-23 11:49:40.000000000 +0100
+++ mod/data/moovida-mobile.desktop	2016-04-30 22:22:00.332070363 +0200
@@ -1,5 +1,4 @@
 [Desktop Entry]
-Version=1.0
 Name=Moovida Media Center
 Comment=Play movies and songs on TV with remote
 Exec=elisa %U
@@ -8,4 +7,4 @@
 Type=Application
 Categories=GNOME;GTK;AudioVideo;Player;X-Ximian-Main;X-Red-Hat-Base;
 X-Osso-Service=com.fluendo.elisa
-OnlyShowIn=Mobile;
+OnlyShowIn=X-Mobile;
