diff -Nur orig/usr/bin/build mod/usr/bin/build
--- orig/usr/bin/build	2017-04-05 14:45:14.000000000 +0200
+++ mod/usr/bin/build	2017-04-16 15:40:47.000000000 +0200
@@ -74,13 +74,14 @@
 export UHUBUILD_COMPILESNAPSHOTDIR="$UHUBUILD_PATH"/misc-${UHUBUILD_DISTR}/snapshot/compile
 export UHUBUILD_INSTALLSNAPSHOTDIR="$UHUBUILD_PATH"/misc-${UHUBUILD_DISTR}/snapshot/install
 export UB_UHUCHECK=1
+export UB_ARCH="i386"
 
 for i in "$@"; do
 	case $1 in
 		-64)
 			yellowecho "64 bites fordítás."
 			echo
-			export UB_ARCH="x86_64"
+			export UB_ARCH="amd64"
 			shift;;
 		-pv|--pack-version)
 			yellowecho "Az UB környezet becsomagolása verziókövetéssel fog megtörténni."
@@ -482,6 +483,7 @@
 	# export UHUBUILD_EXTRARELEASE=`date +'%Y%m%h%H%M%S'`
 
 	# Hajrá, akkor innentől fordítunk...
+	export UHUBUILD_ARCH="${UB_ARCH}"
 	uhubuild-chrootbuild "$UHUBUILD_PATH"/UB-${UHUBUILD_DISTR}/"$i" || {
 
 	# Hiba történt!
