diff -Naur orig/eaccelerator.c mod/eaccelerator.c
--- orig/eaccelerator.c	2014-03-02 23:57:02.000000000 +0100
+++ mod/eaccelerator.c	2014-03-02 23:59:36.000000000 +0100
@@ -63,7 +63,9 @@
 
 #include "php.h"
 #include "php_ini.h"
-#include "php_logos.h"
+#if PHP_VERSION_ID < 50500
+#  include "php_logos.h"
+#endif
 #include "main/fopen_wrappers.h"
 #include "ext/standard/info.h"
 #include "ext/standard/php_incomplete_class.h"
@@ -2269,8 +2271,10 @@
         }
     }
 
-    php_register_info_logo(EACCELERATOR_VERSION_GUID, "text/plain", (unsigned char*)EACCELERATOR_VERSION_STRING, sizeof(EACCELERATOR_VERSION_STRING));
-    php_register_info_logo(EACCELERATOR_LOGO_GUID,    "image/gif",  (unsigned char*)eaccelerator_logo, sizeof(eaccelerator_logo));
+    #if PHP_VERSION_ID < 50500
+      php_register_info_logo(EACCELERATOR_VERSION_GUID, "text/plain", (unsigned char*)EACCELERATOR_VERSION_STRING, sizeof(EACCELERATOR_VERSION_STRING));
+      php_register_info_logo(EACCELERATOR_LOGO_GUID,    "image/gif",  (unsigned char*)eaccelerator_logo, sizeof(eaccelerator_logo));
+    #endif
 
     return SUCCESS;
 }
