diff -Naur orig/config.c mod/config.c
--- orig/config.c	2008-01-30 12:01:34.000000000 +0100
+++ mod/config.c	2012-11-26 22:00:11.000000000 +0100
@@ -13,7 +13,7 @@
 #include <sys/types.h>
 #include <dirent.h>
 #include "vdr/plugin.h"
-#include "tinystr.h"
+#include <tinystr.h>
 #include "config.h"
 #include "setupsetup.h"
 #include "sysconfig.h"
diff -Naur orig/config.h mod/config.h
--- orig/config.h	2008-01-29 14:25:51.000000000 +0100
+++ mod/config.h	2012-11-26 22:01:13.000000000 +0100
@@ -11,7 +11,7 @@
 
 #ifndef CONFIG_H
 #define CONFIG_H
-#include "tinystr.h"
+#include <tinystr.h>
 #include "plugins.h"
 #include "sysconfig.h"
 #include "menus.h"
diff -Naur orig/menus.c mod/menus.c
--- orig/menus.c	2006-03-05 10:47:25.000000000 +0100
+++ mod/menus.c	2012-11-26 22:07:29.000000000 +0100
@@ -344,11 +344,11 @@
   bool ok = true;
   TiXmlElement *elem = NULL;
 
-  if(xmlNode != NULL && xmlNode->Type() == TiXmlNode::ELEMENT)
+  if(xmlNode != NULL && xmlNode->Type() == TiXmlNode::TINYXML_ELEMENT)
   {
     do
     {
-        if(xmlNode->Type() == TiXmlNode::ELEMENT)
+        if(xmlNode->Type() == TiXmlNode::TINYXML_ELEMENT)
         {
           elem = xmlNode->ToElement ();
           if( strcmp(xmlNode->Value(), "menu")== 0)
@@ -810,7 +810,7 @@
      }
     }
     else
-        if( node != NULL && node->Type() == TiXmlNode::ELEMENT)
+        if( node != NULL && node->Type() == TiXmlNode::TINYXML_ELEMENT)
         {
             do
             {
@@ -991,7 +991,7 @@
     bool          ok   = true;
     TiXmlElement *elem = NULL;
 
-    if(node != NULL && node->Type() == TiXmlNode::ELEMENT)
+    if(node != NULL && node->Type() == TiXmlNode::TINYXML_ELEMENT)
     {
 
       elem = node->ToElement ();
@@ -1183,11 +1183,11 @@
 {
    bool ok = true;
    TiXmlElement *elem = NULL;
-   if(node != NULL && node->Type() == TiXmlNode::ELEMENT && (node=node->FirstChild())!=NULL)
+   if(node != NULL && node->Type() == TiXmlNode::TINYXML_ELEMENT && (node=node->FirstChild())!=NULL)
    {
         do
         {
-           if(node->Type() == TiXmlNode::ELEMENT && strcmp(node->Value(), "menu")== 0)
+           if(node->Type() == TiXmlNode::TINYXML_ELEMENT && strcmp(node->Value(), "menu")== 0)
            {
             elem = node->ToElement ();
             const char* name = elem->Attribute("name");
diff -Naur orig/menus.h mod/menus.h
--- orig/menus.h	2008-01-08 19:53:01.000000000 +0100
+++ mod/menus.h	2012-11-26 22:08:09.000000000 +0100
@@ -13,7 +13,7 @@
 #define MENUS_H
 #include <vdr/tools.h>
 #include "util.h"
-#include "tinystr.h"
+#include <tinyxml.h>
 #include "sysconfig.h"
 
 
