diff -Naurd a/stable/Makefile b/stable/Makefile
--- a/stable/Makefile	2013-05-04 17:36:15.000000000 +0200
+++ b/stable/Makefile	2013-06-18 19:54:08.000000000 +0200
@@ -28,9 +28,8 @@
 	rm -f *.so
 
 install:
-	cp -f vterm.h $(includedir)
+	cp -f vterm.h $(DESTDIR)$(includedir)
 	chmod 755 libvterm.so
-	cp -f libvterm.so ${prefix}/${LIB_ARCH}
-	ldconfig	
+	cp -f libvterm.so $(DESTDIR)${prefix}/${LIB_ARCH}
 	
 
