diff -Naurd a/LXDE.desktop.in b/LXDE.desktop.in
--- a/LXDE.desktop.in	2011-07-29 16:04:03.000000000 +0200
+++ b/LXDE.desktop.in	2011-07-30 17:25:17.000000000 +0200
@@ -1,13 +1,14 @@
 [Desktop Entry]
-Encoding=UTF-8
+Version=1.1
 # The names/descriptions should really be better
 Name=LXDE
 Comment=LXDE - Lightweight X11 desktop environment
 Comment[de]=LXDE - Leichtgewichtige X11 Desktop-Umgebung
 Comment[fi]=LXDE - kevyt X11-työpöytäympäristö
+Comment[hu]=LXDE - Pehelysúlyú X11 környezet
 Comment[ja]=LXDE - 軽量な X11 デスクトップ環境
 Comment[pt]=LXDE - Ambiente de trabalho X11 leve
 Comment[zh_TW]=LXDE - 輕量級的 X11 桌面環境
 Exec=@prefix@/bin/startlxde
 # Icon=
-Type=Application
+Type=XSession
