diff -Nur orig/src/druntime/posix.mak mod/src/druntime/posix.mak
--- orig/src/druntime/posix.mak	2016-03-03 03:42:14.000000000 +0100
+++ mod/src/druntime/posix.mak	2017-06-04 19:16:14.251229684 +0200
@@ -44,7 +44,7 @@
 ifeq ($(BUILD),debug)
 	CFLAGS += -g
 else
-	CFLAGS += -O3
+	CFLAGS += -O2
 endif
 ifeq (solaris,$(OS))
 	CFLAGS+=-D_REENTRANT  # for thread-safe errno
diff -Nur orig/src/druntime/test/common.mak mod/src/druntime/test/common.mak
--- orig/src/druntime/test/common.mak	2016-03-03 03:42:14.000000000 +0100
+++ mod/src/druntime/test/common.mak	2017-06-04 19:16:50.095175459 +0200
@@ -24,5 +24,5 @@
 	CFLAGS += -g
 else
 	DFLAGS += -O -release
-	CFLAGS += -O3
+	CFLAGS += -O2
 endif
diff -Nur orig/src/phobos/posix.mak mod/src/phobos/posix.mak
--- orig/src/phobos/posix.mak	2016-03-03 03:42:16.000000000 +0100
+++ mod/src/phobos/posix.mak	2017-06-04 19:17:07.755148742 +0200
@@ -96,7 +96,7 @@
 ifeq ($(BUILD),debug)
 	CFLAGS += -g
 else
-	CFLAGS += -O3
+	CFLAGS += -O2
 endif
 
 # Set DFLAGS
