diff -Naurd gtk-sharp.orig//glib/glue/list.c gtk-sharp//glib/glue/list.c
--- gtk-sharp.orig//glib/glue/list.c	2004-06-25 20:42:19.000000000 +0200
+++ gtk-sharp//glib/glue/list.c	2012-06-04 19:13:10.000000000 +0200
@@ -20,7 +20,7 @@
  */
 
 
-#include <glib/glist.h>
+#include <glib.h>
 
 /* Forward declarations */
 gpointer gtksharp_list_get_data (GList *l);
diff -Naurd gtk-sharp.orig//glib/glue/slist.c gtk-sharp//glib/glue/slist.c
--- gtk-sharp.orig//glib/glue/slist.c	2004-06-25 20:42:19.000000000 +0200
+++ gtk-sharp//glib/glue/slist.c	2012-06-04 19:13:18.000000000 +0200
@@ -20,7 +20,7 @@
  */
 
 
-#include <glib/gslist.h>
+#include <glib.h>
 
 /* Forward declarations */
 gpointer gtksharp_slist_get_data (GSList *l);
