diff -Naurd a/Makefile b/Makefile
--- a/Makefile	2011-05-16 06:13:06.000000000 +0200
+++ b/Makefile	2011-11-30 20:58:45.000000000 +0100
@@ -1,5 +1,5 @@
-CC  = ccache gcc
-CXX = ccache g++
+CC  = gcc
+CXX = g++
 
 OPT = -O2 -fno-inline-functions
 
@@ -16,10 +16,10 @@
 	$(CXX) -DIMPLEMENT_SAVE_PNG -fno-inline-functions -g $(OPT) `sdl-config --cflags` -D_GNU_SOURCE=1 -D_REENTRANT -Wnon-virtual-dtor -Wreturn-type -fthreadsafe-statics -c $<
 
 game: $(objects)
-	$(CXX) -g $(OPT) -L. -L/sw/lib -L. -D_GNU_SOURCE=1 -D_REENTRANT -Wnon-virtual-dtor -Wreturn-type -L/usr/lib `sdl-config --libs` -lSDLmain -lSDL -lGL -lGLU -lGLEW -lSDL_image -lSDL_ttf -lSDL_mixer -lpng -lboost_regex-mt -lboost_system-mt -lpthread -fthreadsafe-statics $(objects) -o game
+	$(CXX) -g $(OPT) -L. -L/sw/lib -L. -D_GNU_SOURCE=1 -D_REENTRANT -Wnon-virtual-dtor -Wreturn-type -L/usr/lib `sdl-config --libs` -lSDLmain -lSDL -lGL -lGLU -lGLEW -lSDL_image -lSDL_ttf -lSDL_mixer -lpng -lboost_regex -lboost_system -lpthread -fthreadsafe-statics $(objects) -o game
 
 server: $(server_objects)
-	$(CXX) -fno-inline-functions -g $(OPT) -L/sw/lib -D_GNU_SOURCE=1 -D_REENTRANT -Wnon-virtual-dtor -Wreturn-type -L/usr/lib `sdl-config --libs` -lSDLmain -lSDL -lGL -lGLU -lSDL_image -lSDL_ttf -lSDL_mixer -lboost_regex-mt -lboost_system-mt -lboost_thread-mt -lboost_iostreams-mt -fthreadsafe-statics $(server_objects) -o server
+	$(CXX) -fno-inline-functions -g $(OPT) -L/sw/lib -D_GNU_SOURCE=1 -D_REENTRANT -Wnon-virtual-dtor -Wreturn-type -L/usr/lib `sdl-config --libs` -lSDLmain -lSDL -lGL -lGLU -lSDL_image -lSDL_ttf -lSDL_mixer -lboost_regex -lboost_system -lboost_thread -lboost_iostreams -fthreadsafe-statics $(server_objects) -o server
 
 poolalloc.o: src/poolalloc.c
 	$(CC) -fno-inline-functions -g $(OPT) `sdl-config --cflags` -D_GNU_SOURCE=1 -D_REENTRANT -Wreturn-type -c src/poolalloc.c
diff -Naurd a/src/IMG_savepng.h b/src/IMG_savepng.h
--- a/src/IMG_savepng.h	2010-03-13 06:15:48.000000000 +0100
+++ b/src/IMG_savepng.h	2011-11-26 20:40:54.000000000 +0100
@@ -25,6 +25,7 @@
 #define __IMG_SAVETOPNG_H__
 
 #include <SDL.h>
+#include <zlib.h>
 
 #define IMG_COMPRESS_OFF 0
 #define IMG_COMPRESS_MAX 9
diff -Naurd a/src/server.cpp b/src/server.cpp
--- a/src/server.cpp	2010-10-16 05:56:52.000000000 +0200
+++ b/src/server.cpp	2011-11-30 21:02:39.000000000 +0100
@@ -37,7 +37,7 @@
 private:
 	void start_accept()
 	{
-		socket_ptr socket(new tcp::socket(acceptor_.io_service()));
+		socket_ptr socket(new tcp::socket(acceptor_.get_io_service()));
 		acceptor_.async_accept(*socket, boost::bind(&server::handle_accept, this, socket, boost::asio::placeholders::error));
 	}
 
