diff -Naur iproute2-4.4.0-orig/Makefile iproute2-4.4.0/Makefile
--- iproute2-4.4.0-orig/Makefile	2016-01-11 17:33:03.000000000 +0100
+++ iproute2-4.4.0/Makefile	2016-02-21 22:08:18.117075645 +0100
@@ -3,7 +3,7 @@
 SBINDIR?=/sbin
 CONFDIR?=/etc/iproute2
 DATADIR?=$(PREFIX)/share
-DOCDIR?=$(DATADIR)/doc/iproute2
+DOCDIR?=$(DATADIR)/doc//Packages/iproute2
 MANDIR?=$(DATADIR)/man
 ARPDDIR?=/var/lib/arpd
 KERNEL_INCLUDE?=/usr/include
