diff -Naurd slmodem-2.9.10/drivers/amrmo_init.c sl/drivers/amrmo_init.c
--- slmodem-2.9.10/drivers/amrmo_init.c	2005-09-20 17:16:30.000000000 +0200
+++ sl/drivers/amrmo_init.c	2005-09-20 17:20:58.245915144 +0200
@@ -250,6 +250,8 @@
 	 PCI_ANY_ID, PCI_ANY_ID, 0, 0, ICH4_CARD},
        	{PCI_VENDOR_ID_INTEL, PCI_DEVICE_ID_ICH5,          /* 8086:24d6 */
 	 PCI_ANY_ID, PCI_ANY_ID, 0, 0, ICH4_CARD},
+       	{PCI_VENDOR_ID_INTEL, 0x266d,                      /* 8086:266d */
+	 PCI_ANY_ID, PCI_ANY_ID, 0, 0, ICH4_CARD},
 	{PCI_VENDOR_ID_VIA, PCI_DEVICE_ID_VIA,             /* 1106:3068 */
 	 PCI_ANY_ID, PCI_ANY_ID, 0, 0, VIA3058_CARD},
 	{PCI_VENDOR_ID_SIS, PCI_DEVICE_ID_SIS630,          /* 1039:7013 */
