diff -Naur orig/example.jwmrc mod/example.jwmrc
--- orig/example.jwmrc	2015-06-29 00:21:20.000000000 +0200
+++ mod/example.jwmrc	2015-07-08 14:11:07.635250069 +0200
@@ -3,17 +3,18 @@
 
     <!-- The root menu. -->
     <RootMenu onroot="12">
-        <Program icon="terminal.png" label="Terminal">xterm</Program>
-        <Menu icon="folder.png" label="Applications">
-            <Program icon="music.png" label="Audacious">audacious</Program>
-            <Program icon="calculator.png" label="Calculator">xcalc</Program>
+      <Include>$HOME/.jwmrc-mjwm</Include>
+        <Program icon="xterm-color_48x48.xpm" label="xterm">xterm</Program>
+        <Menu icon="folder.png" label="Alapok">
+            <Program icon="calc.png" label="Calculator">xcalc</Program>
             <Program icon="gimp.png" label="Gimp">gimp</Program>
-            <Program icon="www.png" label="Iceweasel">iceweasel</Program>
-            <Program icon="editor.png" label="XEdit">xedit</Program>
+            <Program icon="firefox.png" label="Firefox">firefox</Program>
+            <Program icon="MidnightCommander.png" label="mc">xterm -e mc</Program>
         </Menu>
-        <Menu icon="folder.png" label="Utilities">
-            <Program icon="font.png" label="Fonts">xfontsel</Program>
-            <Program icon="window.png" label="Window Properties">
+        <Menu icon="folder.png" label="JWM eszközök">
+            <Program icon="uhu-64x64.png" label="Menügenerátor">jwmuhumenu</Program>
+            <Program icon="fonts.png" label="Fonts">xfontsel</Program>
+            <Program icon="preferences-system-windows.png" label="Window Properties">
                 xprop | xmessage -file -
             </Program>
         </Menu>
@@ -22,8 +23,10 @@
             xscreensaver-command -lock
         </Program>
         <Separator/>
-        <Restart label="Restart" icon="restart.png"/>
-        <Exit label="Exit" confirm="true" icon="quit.png"/>
+        <Restart label="JWM restart" icon="reload.png"/>
+        <Program icon="system-shutdown.png" label="Kikapcsolás">systemctl poweroff</Program>
+        <Program icon="redo.png" label="Újraindítás">systemctl reboot</Program>
+        <Exit label="Kilépés" confirm="true" icon="system-log-out.png"/>
     </RootMenu>
 
     <!-- Options for program groups. -->
@@ -122,9 +125,11 @@
          IconPath can be listed multiple times to allow searching
          for icons in multiple paths.
       -->
-    <IconPath>
-        /usr/share/icons/wm-icons/32x32-gant
-    </IconPath>
+    <IconPath>/usr/share/pixmaps</IconPath>
+    <IconPath>/usr/share/icons/gnome/48x48/actions</IconPath>
+    <IconPath>/usr/share/icons/gnome/48x48/places</IconPath>
+    <IconPath>/usr/share/icons/gnome/48x48/apps</IconPath>
+    <IconPath>/usr/share/icons/gnome/48x48/devices</IconPath>
 
     <!-- Virtual Desktops -->
     <!-- Desktop tags can be contained within Desktops for desktop names. -->
@@ -177,4 +182,9 @@
     <Key mask="A" key="Up">udesktop</Key>
     <Key mask="A" key="Down">ddesktop</Key>
 
+    <!-- AUTOSTART -->
+    <StartupCommand>nm-applet</StartupCommand>
+    <StartupCommand>retrovol -hide</StartupCommand>
+    <StartupCommand>synclient MaxTapTime=0</StartupCommand>
+
 </JWM>
