diff -Nur orig/source/Irrlicht/Makefile mod/source/Irrlicht/Makefile
--- orig/source/Irrlicht/Makefile	2015-09-12 18:15:36.000000000 +0200
+++ mod/source/Irrlicht/Makefile	2016-08-12 13:53:51.414977997 +0200
@@ -86,7 +86,7 @@
 staticlib sharedlib install: SYSTEM = Linux
 STATIC_LIB = libIrrlicht.a
 LIB_PATH = ../../lib/$(SYSTEM)
-INSTALL_DIR = /usr/local/lib
+INSTALL_DIR = $(DESTDIR)/usr/lib
 sharedlib install: SHARED_LIB = libIrrlicht.so
 sharedlib: LDFLAGS += -L/usr/X11R6/lib$(LIBSELECT) -lGL -lXxf86vm
 staticlib sharedlib: CXXINCS += -I/usr/X11R6/include
