diff -Nur orig/include/IrrCompileConfig.h mod/include/IrrCompileConfig.h
--- orig/include/IrrCompileConfig.h	2015-09-12 18:15:36.000000000 +0200
+++ mod/include/IrrCompileConfig.h	2016-08-12 13:48:35.153689015 +0200
@@ -194,7 +194,7 @@
 /** On some systems there is no support for the dynamic extension of OpenGL
 	via function pointers such that this has to be undef'ed. */
 #if !defined(_IRR_OSX_PLATFORM_) && !defined(_IRR_SOLARIS_PLATFORM_)
-#define _IRR_OPENGL_USE_EXTPOINTER_
+//#define _IRR_OPENGL_USE_EXTPOINTER_
 #endif
 
 //! On some Linux systems the XF86 vidmode extension or X11 RandR are missing. Use these flags
