diff -Naur orig/config.h.in patched/config.h.in
--- orig/config.h.in	2012-09-20 03:11:03.000000000 +0200
+++ patched/config.h.in	2014-10-19 21:47:50.810182912 +0200
@@ -57,8 +57,8 @@
 /* Define if your <locale.h> file defines LC_MESSAGES. */
 #undef HAVE_LC_MESSAGES
 
-/* Have libcpufreq. */
-#undef HAVE_LIBCPUFREQ
+/* Have libcpupower. */
+#undef HAVE_LIBCPUPOWER
 
 /* libnotify available */
 #undef HAVE_LIBNOTIFY
diff -Naur orig/configure.in patched/configure.in
--- orig/configure.in	2012-09-20 03:07:34.000000000 +0200
+++ patched/configure.in	2014-10-19 21:47:09.857979841 +0200
@@ -518,14 +518,14 @@
    suid=no
 fi
 
-AC_CHECK_HEADER(cpufreq.h, have_libcpufreq=yes, have_libcpufreq=no)
-LIBCPUFREQ_LIBS=
-if test "x$have_libcpufreq" = "xyes"; then
-   AC_DEFINE([HAVE_LIBCPUFREQ], [1], [Have libcpufreq.])
-   LIBCPUFREQ_LIBS=-lcpufreq
+AC_CHECK_HEADER(cpufreq.h, have_libcpupower=yes, have_libcpupower=no)
+LIBCPUPOWER_LIBS=
+if test "x$have_libcpupower" = "xyes"; then
+   AC_DEFINE([HAVE_LIBCPUPOWER], [1], [Have libcpupower.])
+   LIBCPUPOWER_LIBS=-lcpupower
 fi
-AM_CONDITIONAL(HAVE_LIBCPUFREQ, test x$have_libcpufreq = xyes)
-AC_SUBST(LIBCPUFREQ_LIBS)
+AM_CONDITIONAL(HAVE_LIBCPUPOWER, test x$have_libcpupower = xyes)
+AC_SUBST(LIBCPUPOWER_LIBS)
 
 build_cpufreq_applet=no
 
diff -Naur orig/cpufreq/src/cpufreq-monitor-factory.c patched/cpufreq/src/cpufreq-monitor-factory.c
--- orig/cpufreq/src/cpufreq-monitor-factory.c	2012-09-19 23:46:37.000000000 +0200
+++ patched/cpufreq/src/cpufreq-monitor-factory.c	2014-10-19 21:49:14.290596869 +0200
@@ -31,7 +31,7 @@
 #include "cpufreq-monitor-sysfs.h"
 #include "cpufreq-monitor-procfs.h"
 #include "cpufreq-monitor-cpuinfo.h"
-#ifdef HAVE_LIBCPUFREQ
+#ifdef HAVE_LIBCPUPOWER
 #include "cpufreq-monitor-libcpufreq.h"
 #endif
 #include "cpufreq-monitor-factory.h"
@@ -41,7 +41,7 @@
 {
 	   CPUFreqMonitor *monitor = NULL;
 
-#ifdef HAVE_LIBCPUFREQ
+#ifdef HAVE_LIBCPUPOWER
 	   monitor = cpufreq_monitor_libcpufreq_new (cpu);
 	   return monitor;
 #endif	
diff -Naur orig/cpufreq/src/cpufreq-selector/cpufreq-selector-factory.c patched/cpufreq/src/cpufreq-selector/cpufreq-selector-factory.c
--- orig/cpufreq/src/cpufreq-selector/cpufreq-selector-factory.c	2012-08-28 16:55:02.000000000 +0200
+++ patched/cpufreq/src/cpufreq-selector/cpufreq-selector-factory.c	2014-10-19 21:49:46.690757533 +0200
@@ -24,7 +24,7 @@
 #include "cpufreq-selector-factory.h"
 #include "cpufreq-selector-sysfs.h"
 #include "cpufreq-selector-procfs.h"
-#ifdef HAVE_LIBCPUFREQ
+#ifdef HAVE_LIBCPUPOWER
 #include "cpufreq-selector-libcpufreq.h"
 #endif
 
@@ -33,7 +33,7 @@
 {
 	CPUFreqSelector *selector = NULL;
 	
-#ifdef HAVE_LIBCPUFREQ
+#ifdef HAVE_LIBCPUPOWER
 	selector = cpufreq_selector_libcpufreq_new (cpu);
 #else
 	if (g_file_test ("/sys/devices/system/cpu/cpu0/cpufreq", G_FILE_TEST_EXISTS)) { /* 2.6 kernel */
@@ -41,7 +41,7 @@
 	} else if (g_file_test ("/proc/cpufreq", G_FILE_TEST_EXISTS)) { /* 2.4 kernel */
 		selector = cpufreq_selector_procfs_new (cpu);
 	}
-#endif /* HAVE_LIBCPUFREQ */
+#endif /* HAVE_LIBCPUPOWER */
 
 	return selector;
 }
diff -Naur orig/cpufreq/src/cpufreq-selector/Makefile.am patched/cpufreq/src/cpufreq-selector/Makefile.am
--- orig/cpufreq/src/cpufreq-selector/Makefile.am	2012-08-28 16:55:02.000000000 +0200
+++ patched/cpufreq/src/cpufreq-selector/Makefile.am	2014-10-19 21:51:29.723268443 +0200
@@ -16,7 +16,7 @@
 endif
 
 
-if HAVE_LIBCPUFREQ
+if HAVE_LIBCPUPOWER
 cpufreq_files=cpufreq-selector-libcpufreq.c        cpufreq-selector-libcpufreq.h
 endif
 
@@ -36,7 +36,7 @@
 
 cpufreq_selector_LDADD = \
 	$(CPUFREQ_SELECTOR_LIBS) 	\
-	$(LIBCPUFREQ_LIBS)
+	$(LIBCPUPOWER_LIBS)
 
 if HAVE_POLKIT
 cpufreq_selector_LDADD += \
diff -Naur orig/cpufreq/src/Makefile.am patched/cpufreq/src/Makefile.am
--- orig/cpufreq/src/Makefile.am	2012-09-19 23:46:37.000000000 +0200
+++ patched/cpufreq/src/Makefile.am	2014-10-19 21:50:44.291043157 +0200
@@ -13,7 +13,7 @@
 
 libexec_PROGRAMS = cpufreq-applet
 
-if HAVE_LIBCPUFREQ
+if HAVE_LIBCPUPOWER
 cpufreq_files=cpufreq-monitor-libcpufreq.c    cpufreq-monitor-libcpufreq.h
 endif
 
@@ -33,7 +33,7 @@
 cpufreq_applet_LDADD =  \
 	$(GNOME_APPLETS_LIBS)   	\
 	$(LIBGLADE_LIBS)		\
-	$(LIBCPUFREQ_LIBS)		\
+	$(LIBCPUPOWER_LIBS)		\
 	$(DBUS_LIBS)
 
 	
