diff -Naur orig/configure.ac patched/configure.ac
--- orig/configure.ac	2014-08-27 23:24:20.000000000 +0200
+++ patched/configure.ac	2014-08-27 23:56:32.145052395 +0200
@@ -4,7 +4,7 @@
 AC_CONFIG_HEADERS([config.h])
 AC_CONFIG_SRCDIR([src/core/main.cpp])
 AC_CONFIG_MACRO_DIR([m4])
-AM_INIT_AUTOMAKE([1.11 foreign])
+AM_INIT_AUTOMAKE([1.11 foreign subdir-objects])
 AM_SILENT_RULES([yes])
 
 ### Check for programs ###
