diff -urN nmap-5.21.orig/libdnet-stripped/configure nmap-5.21/libdnet-stripped/configure
--- nmap-5.21.orig/libdnet-stripped/configure	2010-01-14 17:47:28.000000000 +0100
+++ nmap-5.21/libdnet-stripped/configure	2010-01-27 14:45:17.000000000 +0100
@@ -7061,28 +7061,7 @@
 
 
 
-  # Check whether --enable-static was given.
-if test "${enable_static+set}" = set; then
-  enableval=$enable_static; p=${PACKAGE-default}
-    case $enableval in
-    yes) enable_static=yes ;;
-    no) enable_static=no ;;
-    *)
-     enable_static=no
-      # Look at the argument we got.  We use all the common list separators.
-      lt_save_ifs="$IFS"; IFS="${IFS}$PATH_SEPARATOR,"
-      for pkg in $enableval; do
-	IFS="$lt_save_ifs"
-	if test "X$pkg" = "X$p"; then
-	  enable_static=yes
-	fi
-      done
-      IFS="$lt_save_ifs"
-      ;;
-    esac
-else
-  enable_static=yes
-fi
+enable_static=yes
 
 
 
