diff -urN compile.orig/BitTorrent/GUI_wx/__init__.py compile/BitTorrent/GUI_wx/__init__.py
--- compile.orig/BitTorrent/GUI_wx/__init__.py	2009-02-04 23:42:12.000000000 +0100
+++ compile/BitTorrent/GUI_wx/__init__.py	2010-02-21 14:05:27.847878294 +0100
@@ -25,7 +25,7 @@
 else:
     # doesn't work in py2exe
     try:
-        wxversion.select('2.6')
+        wxversion.select('2.8')
     except:
         pass
     
