--- libpano13/CMakeLists.txt.orig	2011-02-22 20:06:38.000000000 +0100
+++ libpano13/CMakeLists.txt	2011-09-19 22:16:02.000000000 +0200
@@ -236,13 +236,6 @@
 # install documentation expected by hugin install script
 INSTALL(
   FILES
-	AUTHORS
-	gpl.txt
-	README
-  DESTINATION "include/pano13"
-        )
-INSTALL(
-  FILES
         doc/Optimize.txt
         doc/PTblender.readme
         doc/PTmender.readme
@@ -307,7 +300,7 @@
 set(CPACK_PACKAGE_CONTACT "${PACKAGE_BUGREPORT}")
 set(CPACK_SOURCE_PACKAGE_FILE_NAME "libpano13")
 set(CPACK_PACKAGE_DESCRIPTION_FILE "${TOP_SRC_DIR}/README")
-set(CPACK_RESOURCE_FILE_LICENSE "${TOP_SRC_DIR}/gpl.txt")
+set(CPACK_RESOURCE_FILE_LICENSE "${TOP_SRC_DIR}/COPYING")
 
 # Find the revision number and use it as the release in rpm-package-build.
 # This way we may omit the otherwise needed "--force" parameter when
