diff -Naur orig/src/Makefile.am patched/src/Makefile.am
--- orig/src/Makefile.am	2006-09-26 21:41:56.000000000 +0200
+++ patched/src/Makefile.am	2013-10-06 17:47:36.136271180 +0200
@@ -26,21 +26,6 @@
 	$(srcdir)/Utility.cs		\
 	$(srcdir)/Mono.Zeroconf/*.cs
 
-if HAVE_MONODOC
-docdir = $(MONODOC_DIR)
-doc_DATA = daap-sharp-docs.zip daap-sharp-docs.tree daap-sharp-docs.source
-DISTCLEANFILES = daap-sharp-docs.zip daap-sharp-docs.tree
-endif
-
-docs2html:
-	monodocs2html -source:en -dest:html
-
-update-docs: $(DAAP_ASSEMBLY)
-	$(MONODOCER) -delete -assembly:$(DAAP_ASSEMBLY) -path:en
-
-daap-sharp-docs.zip: $(srcdir)/en/*/*
-	$(MDASSEMBLER) --out daap-sharp-docs --ecma $(srcdir)/en
-
 all: $(ASSEMBLIES)
 
 $(DAAP_ASSEMBLY): $(DAAPSOURCES)
