diff -Naur orig/krita/plugins/formats/jpeg/kis_jpeg_converter.cc mod/krita/plugins/formats/jpeg/kis_jpeg_converter.cc
--- orig/krita/plugins/formats/jpeg/kis_jpeg_converter.cc	2014-03-02 15:53:40.000000000 +0100
+++ mod/krita/plugins/formats/jpeg/kis_jpeg_converter.cc	2014-03-24 22:14:42.000000000 +0100
@@ -153,7 +153,7 @@
 
 //     setup_read_icc_profile(&cinfo);
     // read header
-    jpeg_read_header(&cinfo, true);
+    jpeg_read_header(&cinfo, (boolean)true);
 
     // start reading
     jpeg_start_decompress(&cinfo);
@@ -500,16 +500,16 @@
     // Set default compression parameters
     jpeg_set_defaults(&cinfo);
     // Customize them
-    jpeg_set_quality(&cinfo, options.quality, options.baseLineJPEG);
+    jpeg_set_quality(&cinfo, options.quality, (boolean)options.baseLineJPEG);
 
     if (options.progressive) {
         jpeg_simple_progression(&cinfo);
     }
     // Optimize ?
-    cinfo.optimize_coding = options.optimize;
+    cinfo.optimize_coding = (boolean)options.optimize;
 
     // Smoothing
-    cinfo.smoothing_factor = options.smooth;
+    cinfo.smoothing_factor = (boolean)options.smooth;
 
     // Subsampling
     switch (options.subsampling) {
@@ -557,10 +557,10 @@
     cinfo.X_density = INCH_TO_POINT(image->xRes()); // It is the "invert" macro because we convert from pointer-per-inchs to points
     cinfo.Y_density = INCH_TO_POINT(image->yRes()); // It is the "invert" macro because we convert from pointer-per-inchs to points
     cinfo.density_unit = 1;
-    cinfo.write_JFIF_header = 1;
+    cinfo.write_JFIF_header = (boolean)true;
 
     // Start compression
-    jpeg_start_compress(&cinfo, true);
+    jpeg_start_compress(&cinfo, (boolean)true);
     // Save exif and iptc information if any available
 
     if (metaData && !metaData->empty()) {
diff -Naur orig/krita/plugins/formats/jpeg/kis_jpeg_destination.cpp mod/krita/plugins/formats/jpeg/kis_jpeg_destination.cpp
--- orig/krita/plugins/formats/jpeg/kis_jpeg_destination.cpp	2014-03-02 15:46:08.000000000 +0100
+++ mod/krita/plugins/formats/jpeg/kis_jpeg_destination.cpp	2014-03-24 22:39:17.000000000 +0100
@@ -68,7 +68,7 @@
     dest->next_output_byte = dest->buffer;
     dest->free_in_buffer = OUTPUT_BUFFER_SIZE;
     
-    return true;
+    return (boolean)true;
 }
 
 void term_destination(j_compress_ptr cinfo)
diff -Naur orig/krita/plugins/formats/jpeg/kis_jpeg_source.cpp mod/krita/plugins/formats/jpeg/kis_jpeg_source.cpp
--- orig/krita/plugins/formats/jpeg/kis_jpeg_source.cpp	2014-03-02 15:46:08.000000000 +0100
+++ mod/krita/plugins/formats/jpeg/kis_jpeg_source.cpp	2014-03-24 22:35:23.000000000 +0100
@@ -70,7 +70,7 @@
     src->bytes_in_buffer = numBytesRead;
     src->anyDataReceived = true;
 
-    return true;
+    return (boolean)true;
 }
 
 void skip_input_data(j_decompress_ptr cinfo, long numBytes)
