--- a/data/alacarte.desktop.in.in.orig	2013-03-01 19:28:20.000000000 +0100
+++ a/data/alacarte.desktop.in.in	2013-03-01 19:28:37.000000000 +0100
@@ -5,7 +5,7 @@
 Terminal=false
 Type=Application
 StartupNotify=true
-Categories=GNOME;Settings;DesktopSettings;Utility;
+Categories=GNOME;Settings;DesktopSettings;Utility;X-UHU-Menu/Tools/Utilities;
 NotShowIn=KDE;
 Icon=alacarte
 X-GNOME-Bugzilla-Bugzilla=GNOME
