diff -urN libindi-0.9.8.orig/CMakeLists.txt libindi-0.9.8/CMakeLists.txt
--- libindi-0.9.8.orig/CMakeLists.txt	2014-03-30 17:20:24.000000000 +0200
+++ libindi-0.9.8/CMakeLists.txt	2014-06-11 09:55:28.718151620 +0200
@@ -26,7 +26,7 @@
 ## the following are directories where stuff will be installed to
 set(INCLUDE_INSTALL_DIR      "${CMAKE_INSTALL_PREFIX}/include/" CACHE PATH "The subdirectory to the header prefix")
 set(PKGCONFIG_INSTALL_PREFIX "${LIB_DESTINATION}/pkgconfig/" CACHE STRING "Base directory for pkgconfig files")
-set(UDEVRULES_INSTALL_DIR "/lib/udev/rules.d" CACHE STRING "Base directory for udev rules")
+set(UDEVRULES_INSTALL_DIR "/usr/lib/udev/rules.d" CACHE STRING "Base directory for udev rules")
 
 ##################  Includes  ################################
 Include (CheckCXXSourceCompiles)
