diff -Naur ORIG/wxPython/config.py UHU/wxPython/config.py
--- ORIG/wxPython/config.py	2013-12-16 13:39:50.000000000 +0000
+++ UHU/wxPython/config.py	2014-05-19 11:04:49.153906752 +0000
@@ -170,7 +170,7 @@
                      # build_options.py, if it is different to that.  The default
                      # is to use the value of WX_CONFIG.
 
-WXPORT = 'gtk2'    # On Linux/Unix there are several ports of wxWidgets available.
+WXPORT = 'gtk3'    # On Linux/Unix there are several ports of wxWidgets available.
                    # Setting this value lets you select which will be used for
                    # the wxPython build.  Possibilites are 'gtk', 'gtk2' and
                    # 'x11'.  Currently only gtk and gtk2 works.
@@ -903,7 +903,7 @@
     BUILD_ACTIVEX = 0
 
 
-if UNICODE and WXPORT not in ['msw', 'gtk2', 'osx_carbon', 'osx_cocoa']:
+if UNICODE and WXPORT not in ['msw', 'gtk3', 'osx_carbon', 'osx_cocoa']:
     raise SystemExit, "UNICODE mode not currently supported on this WXPORT: "+WXPORT
 
 
@@ -1097,10 +1097,10 @@
             WXPLAT = '__WXGTK__'
             portcfg = os.popen('gtk-config --cflags', 'r').read()[:-1]
             BUILD_BASE = BUILD_BASE + '-' + WXPORT
-        elif WXPORT == 'gtk2':
+        elif WXPORT == 'gtk3':
             WXPLAT = '__WXGTK__'
             GENDIR = 'gtk' # no code differences so use the same generated sources
-            portcfg = os.popen('pkg-config gtk+-2.0 --cflags', 'r').read()[:-1]
+            portcfg = os.popen('pkg-config gtk+-3.0 --cflags', 'r').read()[:-1]
         elif WXPORT == 'x11':
             WXPLAT = '__WXX11__'
             portcfg = ''
diff -Naur ORIG/wxPython/src/_colour.i UHU/wxPython/src/_colour.i
--- ORIG/wxPython/src/_colour.i	2013-12-16 13:43:31.000000000 +0000
+++ UHU/wxPython/src/_colour.i	2014-05-20 06:46:24.112329744 +0000
@@ -204,14 +204,6 @@
     // static void ChangeLightness(byte* r, byte* g, byte* b, int ialpha);
     // wxColour ChangeLightness(int ialpha) const;
 
-    
-    DocDeclStr(
-        long , GetPixel() const,
-        "Returns a pixel value which is platform-dependent. On Windows, a
-COLORREF is returned. On X, an allocated pixel value is returned.  -1
-is returned if the pixel is invalid (on X, unallocated).", "");
-    
-    
     %extend {
         KeepGIL(__eq__);
         DocStr(__eq__, "Compare colours for equality.", "");
@@ -293,7 +285,6 @@
         def __reduce__(self):               return (Colour, self.Get(True))
         }
 
-    %property(Pixel, GetPixel, doc="See `GetPixel`");
     %property(RGB, GetRGB, SetRGB, doc="See `GetRGB` and `SetRGB`");
     %property(red,   Red);
     %property(green, Green);
diff -Naur ORIG/wxPython/src/gtk/_gdi_wrap.cpp UHU/wxPython/src/gtk/_gdi_wrap.cpp
--- ORIG/wxPython/src/gtk/_gdi_wrap.cpp	2013-12-16 13:51:16.000000000 +0000
+++ UHU/wxPython/src/gtk/_gdi_wrap.cpp	2014-05-20 06:45:29.821012279 +0000
@@ -5316,34 +5316,6 @@
 }
 
 
-SWIGINTERN PyObject *_wrap_Colour_GetPixel(PyObject *SWIGUNUSEDPARM(self), PyObject *args) {
-  PyObject *resultobj = 0;
-  wxColour *arg1 = (wxColour *) 0 ;
-  long result;
-  void *argp1 = 0 ;
-  int res1 = 0 ;
-  PyObject *swig_obj[1] ;
-  
-  if (!args) SWIG_fail;
-  swig_obj[0] = args;
-  res1 = SWIG_ConvertPtr(swig_obj[0], &argp1,SWIGTYPE_p_wxColour, 0 |  0 );
-  if (!SWIG_IsOK(res1)) {
-    SWIG_exception_fail(SWIG_ArgError(res1), "in method '" "Colour_GetPixel" "', expected argument " "1"" of type '" "wxColour const *""'"); 
-  }
-  arg1 = reinterpret_cast< wxColour * >(argp1);
-  {
-    PyThreadState* __tstate = wxPyBeginAllowThreads();
-    result = (long)((wxColour const *)arg1)->GetPixel();
-    wxPyEndAllowThreads(__tstate);
-    if (PyErr_Occurred()) SWIG_fail;
-  }
-  resultobj = SWIG_From_long(static_cast< long >(result));
-  return resultobj;
-fail:
-  return NULL;
-}
-
-
 SWIGINTERN PyObject *_wrap_Colour___eq__(PyObject *SWIGUNUSEDPARM(self), PyObject *args, PyObject *kwargs) {
   PyObject *resultobj = 0;
   wxColour *arg1 = (wxColour *) 0 ;
@@ -43877,7 +43849,6 @@
 	 { (char *)"Colour_SetRGB", (PyCFunction) _wrap_Colour_SetRGB, METH_VARARGS | METH_KEYWORDS, NULL},
 	 { (char *)"Colour_SetRGBA", (PyCFunction) _wrap_Colour_SetRGBA, METH_VARARGS | METH_KEYWORDS, NULL},
 	 { (char *)"Colour_GetRGBA", (PyCFunction)_wrap_Colour_GetRGBA, METH_O, NULL},
-	 { (char *)"Colour_GetPixel", (PyCFunction)_wrap_Colour_GetPixel, METH_O, NULL},
 	 { (char *)"Colour___eq__", (PyCFunction) _wrap_Colour___eq__, METH_VARARGS | METH_KEYWORDS, NULL},
 	 { (char *)"Colour___ne__", (PyCFunction) _wrap_Colour___ne__, METH_VARARGS | METH_KEYWORDS, NULL},
 	 { (char *)"Colour_Get", (PyCFunction) _wrap_Colour_Get, METH_VARARGS | METH_KEYWORDS, NULL},
diff -Naur ORIG/wxPython/src/helpers.cpp UHU/wxPython/src/helpers.cpp
--- ORIG/wxPython/src/helpers.cpp	2013-12-23 22:40:20.000000000 +0000
+++ UHU/wxPython/src/helpers.cpp	2014-05-20 06:03:47.260473725 +0000
@@ -29,8 +29,8 @@
 #include <gtk/gtk.h>
 #include <gdk/gdkx.h>
 #define GetXWindow(wxwin) (wxwin)->m_wxwindow ? \
-                          GDK_WINDOW_XWINDOW((wxwin)->m_wxwindow->window) : \
-                          GDK_WINDOW_XWINDOW((wxwin)->m_widget->window)
+                          GDK_WINDOW_XID(gtk_widget_get_window((wxwin)->m_wxwindow)) : \
+                          GDK_WINDOW_XID(gtk_widget_get_window((wxwin)->m_widget))
 #include <locale.h>
 #endif
 
