--- a/PyBinding/Makefile.orig	2011-10-06 02:43:53.000000000 +0200
+++ a/PyBinding/Makefile	2011-10-31 19:00:17.000000000 +0100
@@ -47,8 +47,8 @@
 
 RESGEN=resgen2
 
-INSTALL_DIR = $(DESTDIR)$(prefix)/lib/monodevelop/AddIns/PyBinding
-
+INSTALL_DIR = $(prefix)/lib/monodevelop/AddIns/PyBinding
+PKGCONFIG_DIR = $(prefix)/lib/pkgconfig
 PYBINDING_PC = $(BUILD_DIR)/monodevelop-pybinding.pc
 
 FILES =  \
@@ -166,20 +166,18 @@
 
 install-local: $(ASSEMBLY) $(ASSEMBLY_MDB)
 	make pre-install-local-hook prefix=$(prefix)
-	mkdir -p $(INSTALL_DIR)
-	$(call cp,$(ASSEMBLY),$(INSTALL_DIR))
-	$(call cp,$(ASSEMBLY_MDB),$(INSTALL_DIR))
-	mkdir -p '$(DESTDIR)$(libdir)/pkgconfig'
-	$(call cp,$(PYBINDING_PC),$(DESTDIR)$(libdir)/pkgconfig)
-	$(call cp,$(PYBINDING_DLL_MDB),$(INSTALL_DIR))
+	mkdir -p $(DESTDIR)$(INSTALL_DIR)
+	$(call cp,$(ASSEMBLY),$(DESTDIR)$(INSTALL_DIR))
+	$(call cp,$(ASSEMBLY_MDB),$(DESTDIR)$(INSTALL_DIR))
+	mkdir -p $(DESTDIR)$(PKGCONFIG_DIR)
+	$(call cp,$(PYBINDING_PC),$(DESTDIR)$(PKGCONFIG_DIR))
 	make post-install-local-hook prefix=$(prefix)
 
 uninstall-local: $(ASSEMBLY) $(ASSEMBLY_MDB)
 	make pre-uninstall-local-hook prefix=$(prefix)
-	$(call rm,$(ASSEMBLY),$(INSTALL_DIR))
-	$(call rm,$(ASSEMBLY_MDB),$(INSTALL_DIR))
-	$(call rm,$(PYBINDING_PC),$(DESTDIR)$(libdir)/pkgconfig)
-	$(call rm,$(PYBINDING_DLL_MDB),$(INSTALL_DIR))
+	$(call rm,$(ASSEMBLY),$(DESTDIR)$(INSTALL_DIR))
+	$(call rm,$(ASSEMBLY_MDB),$(DESTDIR)$(INSTALL_DIR))
+	$(call rm,$(PYBINDING_PC),$(DESTDIR)$(PKGCONFIG_DIR))
 	make post-uninstall-local-hook prefix=$(prefix)
 
 install: install-local
