From 16ea3b40b4d36bce5345e053d3c674bafa8d7ffa Mon Sep 17 00:00:00 2001
From: Christoph Lipka <c-lipka@users.noreply.github.com>
Date: Sat, 4 Jan 2014 15:30:38 +0100
Subject: [PATCH] Fix for configure attempting to link boost_system even with
 boost 1.49 or earlier.

---
 unix/configure.ac | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/unix/configure.ac b/unix/configure.ac
index f62f5d4..fae3e2c 100644
--- a/unix/configure.ac
+++ b/unix/configure.ac
@@ -326,6 +326,9 @@ AC_RUN_IFELSE(
   ],,
   [AC_MSG_RESULT([cross-compiling])]  # FIXME
 )
+  if test $boost_thread_links == '1'; then
+    break
+  fi
 done
 if test $boost_thread_links != '1'; then
   AC_MSG_RESULT([no])
-- 
1.8.5.2

