Description: do not create user directories
Author: Andreas Beckmann <anbe@debian.org>

--- a/unix/prebuild.sh
+++ b/unix/prebuild.sh
@@ -642,6 +642,8 @@ install-data-local:
 	for f in \$\$filelist ; do \\
 	  \$(INSTALL_DATA) \$(top_srcdir)/doc/\$\$f \$(DESTDIR)\$(povdocdir)/\$\$f && echo "\$(DESTDIR)\$(povdocdir)/\$\$f" >> \$(povinstall); \\
 	done
+
+install-user-data-local:
 	@echo "Creating user directories..."; \\
 	for p in \$(povuser) \$(povconfuser) ; do \\
 	  \$(mkdir_p) \$\$p && chown \$(povowner) \$\$p && chgrp \$(povgroup) \$\$p && printf "%s\\n" "\$\$p" "\`cat \$(povinstall)\`" > \$(povinstall); \\
