diff -Naur orig/pcmanfm/settings.cpp patched/pcmanfm/settings.cpp
--- orig/pcmanfm/settings.cpp	2017-01-14 03:13:45.134741323 +0100
+++ patched/pcmanfm/settings.cpp	2017-05-06 23:10:06.622655505 +0200
@@ -172,11 +172,11 @@
   if(fallbackIconThemeName_.isEmpty()) {
     // FIXME: we should choose one from installed icon themes or get
     // the value from XSETTINGS instead of hard code a fallback value.
-    fallbackIconThemeName_ = "oxygen"; // fallback icon theme name
+    fallbackIconThemeName_ = "gnome"; // fallback icon theme name
   }
   suCommand_ = settings.value("SuCommand", "lxqt-sudo %s").toString();
-  setTerminal(settings.value("Terminal", "xterm").toString());
-  setArchiver(settings.value("Archiver", "file-roller").toString());
+  setTerminal(settings.value("Terminal", "qterminal").toString());
+  setArchiver(settings.value("Archiver", "xarchiver").toString());
   setSiUnit(settings.value("SIUnit", false).toBool());
 
   setOnlyUserTemplates(settings.value("OnlyUserTemplates", false).toBool());
