diff -urN dia-0.97.2.orig/plug-ins/postscript/diapsft2renderer.c dia-0.97.2/plug-ins/postscript/diapsft2renderer.c
--- dia-0.97.2.orig/plug-ins/postscript/diapsft2renderer.c	2009-11-07 18:13:53.000000000 +0100
+++ dia-0.97.2/plug-ins/postscript/diapsft2renderer.c	2014-02-17 13:30:36.966379189 +0100
@@ -29,8 +29,8 @@
 #include <pango/pangoft2.h>
 #include <pango/pango-engine.h>
 /* I'd really rather avoid this */
-#include <freetype/ftglyph.h>
-#include <freetype/ftoutln.h>
+#include <freetype2/ftglyph.h>
+#include <freetype2/ftoutln.h>
 
 #define DPI 300
 
