--- celestia.orig/src/celestia/gtk/data/celestia.desktop	2009-06-22 17:44:10.000000000 +0200
+++ celestia/src/celestia/gtk/data/celestia.desktop	2009-07-09 15:31:59.000000000 +0200
@@ -1,12 +1,11 @@
 [Desktop Entry]
-Encoding=UTF-8
-Version=1.4.0
+Version=1.1
 Type=Application
 Exec=celestia
 Icon=celestia
-X-GNOME-DocPath=
 Terminal=false
 Name=Celestia
 Comment=Space Simulator
+Comment[hu]=Űrszimulátor
 StartupNotify=true
-Categories=Astronomy;Science;Application;
+Categories=GTK;GNOME;Science;Astronomy;X-UHU-Menu/Edutainment/Science;
