diff -urN nfs-utils-1.3.0.orig/systemd/nfs-server.service nfs-utils-1.3.0/systemd/nfs-server.service
--- nfs-utils-1.3.0.orig/systemd/nfs-server.service	2014-03-25 16:12:07.000000000 +0100
+++ nfs-utils-1.3.0/systemd/nfs-server.service	2014-05-29 09:59:52.947546433 +0200
@@ -1,11 +1,11 @@
 [Unit]
 Description=NFS server and services
-Requires= network.target proc-fs-nfsd.mount rpcbind.target
+Requires= network.target proc-fs-nfsd.mount rpcbind.target rpcbind.service
 Requires= nfs-mountd.service
 Wants=rpc-statd.service nfs-idmapd.service rpc-gssd.service rpc-svcgssd.service
 Wants=rpc-statd-notify.service
 
-After= network.target proc-fs-nfsd.mount rpcbind.target nfs-mountd.service
+After= network.target proc-fs-nfsd.mount rpcbind.target nfs-mountd.service rpcbind.service
 After= nfs-idmapd.service rpc-statd.service
 After= rpc-gssd.service rpc-svcgssd.service
 Before= rpc-statd-notify.service
diff -urN nfs-utils-1.3.0.orig/systemd/rpc-statd.service nfs-utils-1.3.0/systemd/rpc-statd.service
--- nfs-utils-1.3.0.orig/systemd/rpc-statd.service	2014-03-25 16:12:07.000000000 +0100
+++ nfs-utils-1.3.0/systemd/rpc-statd.service	2014-05-29 10:00:05.639545867 +0200
@@ -2,8 +2,8 @@
 Description=NFS status monitor for NFSv2/3 locking.
 DefaultDependencies=no
 Conflicts=umount.target
-Requires=nss-lookup.target rpcbind.target
-After=network.target nss-lookup.target rpcbind.target
+Requires=nss-lookup.target rpcbind.target rpcbind.service
+After=network.target nss-lookup.target rpcbind.target rpcbind.service
 
 PartOf=nfs-utils.service
 
