diff -Nur orig/gparted.desktop.in.in mod/gparted.desktop.in.in
--- orig/gparted.desktop.in.in	2015-10-14 19:42:48.000000000 +0200
+++ mod/gparted.desktop.in.in	2017-01-03 13:25:02.774913033 +0100
@@ -3,7 +3,7 @@
 _GenericName=Partition Editor
 _X-GNOME-FullName=GParted Partition Editor
 _Comment=Create, reorganize, and delete partitions
-Exec=@gksuprog@ @installdir@/gparted %f
+Exec=gparted %f
 Icon=gparted
 Terminal=false
 Type=Application
