diff -Naurd gparted.orig/gparted.desktop.in.in gparted/gparted.desktop.in.in
--- gparted.orig/gparted.desktop.in.in	2009-04-02 17:29:01.000000000 +0200
+++ gparted/gparted.desktop.in.in	2009-04-08 19:19:48.000000000 +0200
@@ -3,9 +3,9 @@
 _GenericName=Partition Editor
 _X-GNOME-FullName=GParted Partition Editor
 _Comment=Create, reorganize, and delete partitions
-Exec=@gksuprog@ @installdir@/gparted %f
+Exec=gparted %f
 Icon=gparted
 Terminal=false
 Type=Application
-Categories=GNOME;System;Filesystem;
+Categories=GTK;GNOME;System;Filesystem;X-UHU-Menu/Tools/Utilities;
 StartupNotify=true
