diff -Naur util-linux-2.27.1-orig/misc-utils/whereis.1 util-linux-2.27.1/misc-utils/whereis.1
--- util-linux-2.27.1-orig/misc-utils/whereis.1	2015-10-06 08:59:51.587589616 +0200
+++ util-linux-2.27.1/misc-utils/whereis.1	2016-02-16 16:37:35.165714611 +0100
@@ -130,13 +130,13 @@
 .B /usr/\:bin
 which are not documented
 in
-.B /usr/\:man/\:man1
+.B /usr/share/\:man/\:man1
 or have no source in
 .BR /usr/\:src :
 .IP
 .B cd /usr/bin
 .br
-.B whereis \-u \-ms \-M /usr/man/man1 \-S /usr/src \-f *
+.B whereis \-u \-ms \-M /usr/share/man/man1 \-S /usr/src \-f *
 .SH "FILE SEARCH PATHS"
 By default
 .B whereis
diff -Naur util-linux-2.27.1-orig/misc-utils/whereis.c util-linux-2.27.1/misc-utils/whereis.c
--- util-linux-2.27.1-orig/misc-utils/whereis.c	2015-10-06 08:59:51.587589616 +0200
+++ util-linux-2.27.1/misc-utils/whereis.c	2016-02-16 16:40:16.297714506 +0100
@@ -96,75 +96,21 @@
 static const char *bindirs[] = {
 	"/usr/bin",
 	"/usr/sbin",
-	"/usr/lib",
-	"/usr/lib64",
 	"/bin",
 	"/sbin",
-	"/etc",
-	"/usr/etc",
-	"/lib",
-	"/lib64",
-	"/usr/games",
-	"/usr/games/bin",
-	"/usr/games/lib",
-	"/usr/emacs/etc",
-	"/usr/lib/emacs/*/etc",
-	"/usr/TeX/bin",
-	"/usr/tex/bin",
-	"/usr/interviews/bin/LINUX",
-
-	"/usr/X11R6/bin",
-	"/usr/X386/bin",
-	"/usr/bin/X11",
-	"/usr/X11/bin",
-	"/usr/X11R5/bin",
-
 	"/usr/local/bin",
 	"/usr/local/sbin",
-	"/usr/local/etc",
-	"/usr/local/lib",
-	"/usr/local/games",
-	"/usr/local/games/bin",
-	"/usr/local/emacs/etc",
-	"/usr/local/TeX/bin",
-	"/usr/local/tex/bin",
-	"/usr/local/bin/X11",
-
-	"/usr/contrib",
-	"/usr/hosts",
-	"/usr/include",
-
-	"/usr/g++-include",
-
-	"/usr/ucb",
-	"/usr/old",
-	"/usr/new",
-	"/usr/local",
-	"/usr/libexec",
-	"/usr/share",
-
 	"/opt/*/bin",
 	NULL
 };
 
 static const char *mandirs[] = {
-	"/usr/man/*",
 	"/usr/share/man/*",
-	"/usr/X386/man/*",
-	"/usr/X11/man/*",
-	"/usr/TeX/man/*",
-	"/usr/interviews/man/mann",
-	"/usr/share/info",
 	NULL
 };
 
 static const char *srcdirs[] = {
 	"/usr/src/*",
-	"/usr/src/lib/libc/*",
-	"/usr/src/lib/libc/net/*",
-	"/usr/src/ucb/pascal",
-	"/usr/src/ucb/pascal/utilities",
-	"/usr/src/undoc",
 	NULL
 };
 
