diff -Naurd a/classes/ttermina.cc b/classes/ttermina.cc
--- a/classes/ttermina.cc	2008-07-11 13:20:27.000000000 +0200
+++ b/classes/ttermina.cc	2011-01-13 17:22:18.000000000 +0100
@@ -235,7 +235,7 @@
          }
       }
    }
- while (--pos>=queBack);
+ while (--pos>queBack);
 
  return queBack;
 }
diff -Naurd a/classes/x11/x11src.cc b/classes/x11/x11src.cc
--- a/classes/x11/x11src.cc	2009-12-29 16:52:24.000000000 +0100
+++ b/classes/x11/x11src.cc	2012-10-19 17:22:07.000000000 +0200
@@ -87,6 +87,8 @@
  #include <pthread.h>
 #endif
 
+#include <sys/stat.h>
+
 #ifdef TVOSf_Solaris
  // At least in the Solaris 7 box I tested looks like ITIMER_REAL is broken
  // and behaves like ITIMER_VIRTUAL
diff -Naurd a/Makefile.in b/Makefile.in
--- a/Makefile.in	2004-10-18 21:14:14.000000000 +0200
+++ b/Makefile.in	2011-01-13 17:24:00.000000000 +0100
@@ -31,7 +31,7 @@
 	$(MAKE) -C examples
 
 rhtv-config$(EXE_EXT): rhtv-config.c include/tv/configtv.h
-	@GCC@ -o rhtv-config$(EXE_EXT) -Iinclude rhtv-config.c
+	@GCC@ -o rhtv-config$(EXE_EXT) $(CFLAGS) $(LDFLAGS) -Iinclude rhtv-config.c
 
 install-headers:
 	@install_headers@
