From dffeb373e2116d3486169872116fd0b2dd39885a Mon Sep 17 00:00:00 2001
From: Alec Leamas <leamas.alec@gmail.com>
Date: Wed, 9 Sep 2015 14:24:30 +0200
Subject: [PATCH] lib: Fix missing --lirc flag.

---
 lib/Makefile.am | 1 +
 1 file changed, 1 insertion(+)

diff --git a/lib/Makefile.am b/lib/Makefile.am
index 9e14641..4f9435a 100644
--- a/lib/Makefile.am
+++ b/lib/Makefile.am
@@ -30,6 +30,7 @@ liblirc_la_SOURCES          = config_file.c \
                               transmit.c \
                               util.c

+libirrecord_la_LDFLAGS      = -llirc
 libirrecord_la_SOURCES      = irrecord.c

 liblirc_client_la_LDFLAGS   = -version-info 3:0:3
--
2.4.2

