From f1c2f41a04589e019973e06e446205a9a4bfea9f Mon Sep 17 00:00:00 2001
From: Alec Leamas <leamas.alec@gmail.com>
Date: Tue, 24 May 2016 21:55:06 +0200
Subject: [PATCH] plugins/devinput: Make the list-devices support available.

---
 plugins/devinput.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/plugins/devinput.c b/plugins/devinput.c
index df97ffd..25b21ae 100644
--- a/plugins/devinput.c
+++ b/plugins/devinput.c
@@ -89,7 +89,11 @@ const struct driver hw_devinput = {
 	.api_version	= 4,
 	.driver_version = "0.9.3",
 	.info		= "See file://" PLUGINDOCS "/devinput.html",
+#ifdef HAVE_LIBUDEV_H
+	.device_hint    = "drvctl",
+#else
 	.device_hint    = "/dev/input/event*",
+#endif
 };
 
 
-- 
2.5.5

