Description: link against xpcomglue_s_nomozalloc
Author: Mike Hommey <glandium@debian.org>
Last-Update: 2012-04-07

diff --git a/configure.in b/configure.in
index 8d0c87d..271a4f7 100644
--- a/configure.in
+++ b/configure.in
@@ -42,7 +42,7 @@ AC_ARG_WITH([xulrunner-sdk],
         [
 	XULRUNNER_SDK_PREFIX=$with_xulrunner_sdk
 	GECKO_CFLAGS="-I$XULRUNNER_SDK_PREFIX/include -I$XULRUNNER_SDK_PREFIX/include/plugin -I$XULRUNNER_SDK_PREFIX/include/nspr -I$XULRUNNER_SDK_PREFIX/include/xpcom -I$XULRUNNER_SDK_PREFIX/include/string -I$XULRUNNER_SDK_PREFIX/include/java"
-	GECKO_LIBS="-L$XULRUNNER_SDK_PREFIX/lib -lxpcomglue_s -lxul -lxpcom -lplds4 -lplc4 -lnspr4 -lpthread -ldl"
+	GECKO_LIBS="-L$XULRUNNER_SDK_PREFIX/lib -lxpcomglue_s_nomozalloc -lxul -lxpcom -lplds4 -lplc4 -lnspr4 -lpthread -ldl"
 	],
         [])
 
