diff -Nur orig/converter/ppm/ppmtopict.c mod/converter/ppm/ppmtopict.c
--- orig/converter/ppm/ppmtopict.c	2021-02-15 18:32:22.000000000 +0100
+++ mod/converter/ppm/ppmtopict.c	2021-02-15 18:37:14.340694500 +0100
@@ -208,7 +208,7 @@
 
    Use buffer 'outBuf'.
 -----------------------------------------------------------------------------*/
-    unsigned int i;
+    int i;
     unsigned int count;
     unsigned int run;
     unsigned int rep;
