diff --git a/converter/pbm/pbmtoepson.c b/converter/pbm/pbmtoepson.c
index 122a438..5d67746 100644
--- a/converter/pbm/pbmtoepson.c
+++ b/converter/pbm/pbmtoepson.c
@@ -75,7 +75,7 @@ parseCommandLine(int                  argc,
             &dpiSpec,                         0);
     OPTENT3(0, "adjacent",     OPT_FLAG,     NULL,
             &adjacentSpec,                    0);
-    OPTENT3(0, "nonadjacent",  OPT_FLAG,     NULL,
+    OPTENT3(0, "noadjacent",  OPT_FLAG,     NULL,
             &nonadjacentSpec,                 0);
 
     opt.opt_table = option_def;
