diff -Nur orig/GNUmakefile mod/GNUmakefile
--- orig/GNUmakefile	2021-02-15 18:08:03.000000000 +0100
+++ mod/GNUmakefile	2021-02-15 18:11:54.886337829 +0100
@@ -1,4 +1,4 @@
-# Makefile for Netpbm
+## Makefile for Netpbm
 
 # Configuration should normally be done in the included file config.mk.
 
@@ -394,16 +394,12 @@
 endif
 
 .PHONY: install.manwebmain
-install.manwebmain: $(PKGDIR)/$(PKGMANDIR)/web/netpbm.url $(PKGDIR)/bin/doc.url
+install.manwebmain: $(PKGDIR)/$(PKGMANDIR)/web/netpbm.url
 
 $(PKGDIR)/$(PKGMANDIR)/web/netpbm.url: $(PKGDIR)/$(PKGMANDIR)/web
 	echo "$(NETPBM_DOCURL)" > $@
 	chmod $(INSTALL_PERM_MAN) $@
 
-$(PKGDIR)/bin/doc.url: $(PKGDIR)/bin
-	echo "$(NETPBM_DOCURL)" > $@
-	chmod $(INSTALL_PERM_MAN) $@
-
 .PHONY: install-dev
 # Note that you might install the development package and NOT the runtime
 # package.  If you have a special system for building stuff, maybe for 
