diff -Naur orig/mozilla/toolkit/mozapps/extensions/internal/XPIProvider.jsm patched/mozilla/toolkit/mozapps/extensions/internal/XPIProvider.jsm
--- orig/mozilla/toolkit/mozapps/extensions/internal/XPIProvider.jsm	2014-07-18 02:05:55.000000000 +0200
+++ patched/mozilla/toolkit/mozapps/extensions/internal/XPIProvider.jsm	2014-07-29 00:15:10.858529761 +0200
@@ -1894,8 +1894,8 @@
         this.showUpgradeUI();
         flushCaches = true;
       }
-      else if (aAppChanged === undefined) {
-        // For new profiles we will never need to show the add-on selection UI
+      else {
+        // For all profiles we will never need to show the add-on selection UI
         Services.prefs.setBoolPref(PREF_SHOWN_SELECTION_UI, true);
       }
 
