diff -Naur orig/mozilla/config/baseconfig.mk patched/mozilla/config/baseconfig.mk
--- orig/mozilla/config/baseconfig.mk	2015-06-08 19:49:23.000000000 +0200
+++ patched/mozilla/config/baseconfig.mk	2015-06-11 18:15:45.273073146 +0200
@@ -4,7 +4,7 @@
 # whether a normal build is happening or whether the check is running.
 includedir := $(includedir)/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
 idldir = $(datadir)/idl/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
-installdir = $(libdir)/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
+installdir = $(libdir)/$(MOZ_APP_NAME)
 sdkdir = $(libdir)/$(MOZ_APP_NAME)-devel-$(MOZ_APP_VERSION)
 ifndef TOP_DIST
 TOP_DIST = dist
