diff -Naur orig/src/lib/libKActivities.pc.cmake patched/src/lib/libKActivities.pc.cmake
--- orig/src/lib/libKActivities.pc.cmake	2018-01-07 00:17:41.000000000 +0100
+++ patched/src/lib/libKActivities.pc.cmake	2018-02-10 17:09:11.519532189 +0100
@@ -7,6 +7,6 @@
 Description: libKActivities is a C++ library for using KDE activities
 URL: http://www.kde.org
 Requires: Qt5Core
-Version: ${KACTIVITIES_LIB_VERSION_STRING}
+Version: ${KACTIVITIES_VERSION_STRING}
 Libs: -L${LIB_INSTALL_DIR} -lKF5Activities
 Cflags: -I${INCLUDE_INSTALL_DIR}
