diff -Naur orig/lib/auth_unix.c patched/lib/auth_unix.c
--- orig/lib/auth_unix.c	2016-05-20 07:52:44.000000000 +0200
+++ patched/lib/auth_unix.c	2016-07-04 11:47:42.220955043 +0200
@@ -150,9 +150,11 @@
 static const char *mycanonifyid(const char *identifier, size_t len)
 {
     static char retbuf[81];
+    char backup[81];
     struct group *grp;
     char *p;
     int username_tolower = 0;
+    int ic, rbc;
 
     if (!len) len = strlen(identifier);
     if (len >= sizeof(retbuf)) return NULL;
@@ -194,6 +196,22 @@
 	}
     }
 
+    if( (libcyrus_config_getswitch(CYRUSOPT_NORMALIZEUID) == 1) ) {
+        strcpy(backup,retbuf);
+       /* remove leading blanks */
+       for(ic=0; isblank(backup[ic]); ic++);
+       for(rbc=0; backup[ic]; ic++) {
+            retbuf[rbc] = ( isalpha(backup[ic]) ?
+                 tolower(backup[ic]) : backup[ic] );
+            rbc++;
+       }
+       retbuf[rbc] = '\0';
+       /* remove trailing blanks */
+       for(--rbc; isblank(retbuf[rbc]); rbc--) {
+            retbuf[rbc] = '\0';
+       }
+    }
+
     return retbuf;
 }
 
diff -Naur orig/lib/imapoptions patched/lib/imapoptions
--- orig/lib/imapoptions	2016-05-20 07:52:44.000000000 +0200
+++ patched/lib/imapoptions	2016-07-04 11:45:12.466884889 +0200
@@ -2001,6 +2001,11 @@
 /* The absolute path to the zoneinfo db file.  If not specified,
    will be confdir/zoneinfo.db */
 
+{ "normalizeuid", 0, SWITCH }
+/* Lowercase uid and strip leading and trailing blanks. It is recommended
+   to set this to yes, especially if OpenLDAP is used as authentication
+   source. */
+
 /*
 .SH SEE ALSO
 .PP
diff -Naur orig/lib/libcyr_cfg.c patched/lib/libcyr_cfg.c
--- orig/lib/libcyr_cfg.c	2016-05-20 07:52:44.000000000 +0200
+++ patched/lib/libcyr_cfg.c	2016-07-04 11:45:12.466884889 +0200
@@ -156,6 +156,10 @@
       CFGVAL(long, 1),
       CYRUS_OPT_SWITCH },
 
+    { CYRUSOPT_NORMALIZEUID,
+      CFGVAL(long, 1),
+      CYRUS_OPT_SWITCH },
+
     { CYRUSOPT_LAST, { NULL }, CYRUS_OPT_NOTOPT }
 };
 
diff -Naur orig/lib/libcyr_cfg.h patched/lib/libcyr_cfg.h
--- orig/lib/libcyr_cfg.h	2016-05-20 07:52:44.000000000 +0200
+++ patched/lib/libcyr_cfg.h	2016-07-04 11:45:12.470884951 +0200
@@ -111,6 +111,8 @@
     CYRUSOPT_SQL_USESSL,
     /* Checkpoint after every recovery (OFF) */
     CYRUSOPT_SKIPLIST_ALWAYS_CHECKPOINT,
+    /* Lowercase uid and strip leading and trailing blanks (OFF) */
+    CYRUSOPT_NORMALIZEUID,
 
     CYRUSOPT_LAST
     
