diff -Naur orig/master/conf/normal.conf patched/master/conf/normal.conf
--- orig/master/conf/normal.conf	2016-05-20 07:52:44.000000000 +0200
+++ patched/master/conf/normal.conf	2016-07-04 11:42:58.597034315 +0200
@@ -26,11 +26,11 @@
 #  https		cmd="httpd -s" listen="https" prefork=0
 
   # at least one LMTP is required for delivery
-#  lmtp		cmd="lmtpd" listen="lmtp" prefork=0
-  lmtpunix	cmd="lmtpd" listen="/var/imap/socket/lmtp" prefork=0
+  lmtp		cmd="lmtpd" listen="lmtp" prefork=0
+  lmtpunix	cmd="lmtpd" listen="/var/lib/imap/socket/lmtp" prefork=0
 
   # this is required if using notifications
-#  notify	cmd="notifyd" listen="/var/imap/socket/notify" proto="udp" prefork=1
+#  notify	cmd="notifyd" listen="/var/lib/imap/socket/notify" proto="udp" prefork=1
 }
 
 EVENTS {
diff -Naur orig/master/conf/prefork.conf patched/master/conf/prefork.conf
--- orig/master/conf/prefork.conf	2016-05-20 07:52:44.000000000 +0200
+++ patched/master/conf/prefork.conf	2016-07-04 11:42:58.597034315 +0200
@@ -26,11 +26,11 @@
 #  https		cmd="httpd -s" listen="https" prefork=1
 
   # at least one LMTP is required for delivery
-#  lmtp		cmd="lmtpd" listen="lmtp" prefork=0
-  lmtpunix	cmd="lmtpd" listen="/var/imap/socket/lmtp" prefork=1
+  lmtp		cmd="lmtpd" listen="lmtp" prefork=0
+  lmtpunix	cmd="lmtpd" listen="/var/lib/imap/socket/lmtp" prefork=1
 
   # this is only necessary if using notifications
-#  notify	cmd="notifyd" listen="/var/imap/socket/notify" proto="udp" prefork=1
+#  notify	cmd="notifyd" listen="/var/lib/imap/socket/notify" proto="udp" prefork=1
 }
 
 EVENTS {
