diff -Naurdp a/master/conf/normal.conf b/master/conf/normal.conf
--- a/master/conf/normal.conf	2006-03-30 18:01:39.000000000 +0200
+++ b/master/conf/normal.conf	2006-09-04 16:19:18.000000000 +0200
@@ -22,11 +22,11 @@ SERVICES {
 #  nntps		cmd="nntpd -s" listen="nntps" prefork=0
 
   # at least one LMTP is required for delivery
-#  lmtp		cmd="lmtpd" listen="lmtp" prefork=0
-  lmtpunix	cmd="lmtpd" listen="/var/imap/socket/lmtp" prefork=0
+  lmtp		cmd="lmtpd" listen="lmtp" prefork=0
+  lmtpunix	cmd="lmtpd" listen="/var/lib/imap/socket/lmtp" prefork=0
 
   # this is required if using notifications
-#  notify	cmd="notifyd" listen="/var/imap/socket/notify" proto="udp" prefork=1
+#  notify	cmd="notifyd" listen="/var/lib/imap/socket/notify" proto="udp" prefork=1
 }
 
 EVENTS {
diff -Naurdp a/master/conf/prefork.conf b/master/conf/prefork.conf
--- a/master/conf/prefork.conf	2003-10-22 20:03:10.000000000 +0200
+++ b/master/conf/prefork.conf	2006-09-04 16:18:28.000000000 +0200
@@ -22,11 +22,11 @@ SERVICES {
 #  nntps		cmd="nntpd -s" listen="nntps" prefork=1
 
   # at least one LMTP is required for delivery
-#  lmtp		cmd="lmtpd" listen="lmtp" prefork=0
-  lmtpunix	cmd="lmtpd" listen="/var/imap/socket/lmtp" prefork=1
+  lmtp		cmd="lmtpd" listen="lmtp" prefork=0
+  lmtpunix	cmd="lmtpd" listen="/var/lib/imap/socket/lmtp" prefork=1
 
   # this is only necessary if using notifications
-#  notify	cmd="notifyd" listen="/var/imap/socket/notify" proto="udp" prefork=1
+#  notify	cmd="notifyd" listen="/var/lib/imap/socket/notify" proto="udp" prefork=1
 }
 
 EVENTS {
