diff -Naur orig/src/ipelib/ipebitmap_unix.cpp mod/src/ipelib/ipebitmap_unix.cpp
--- orig/src/ipelib/ipebitmap_unix.cpp	2016-01-24 12:16:26.000000000 +0100
+++ mod/src/ipelib/ipebitmap_unix.cpp	2016-05-03 15:38:57.123706094 +0200
@@ -106,7 +106,7 @@
 {
   jpegErrorManager *myerr = (jpegErrorManager*) cinfo->err;
   (*(cinfo->err->format_message)) (cinfo, jpegLastErrorMsg);
-  longjmp(myerr->setjmp_buffer, 1);
+  longjmp(myerr->setjmp_buffer, TRUE);
 }
 
 bool dctDecode(Buffer dctData, Buffer pixelData, int components)
@@ -125,7 +125,7 @@
   // Decompression:
   jpeg_create_decompress(&cinfo);
   jpeg_mem_src(&cinfo, (unsigned char *) dctData.data(), dctData.size());
-  jpeg_read_header(&cinfo, 1);
+  jpeg_read_header(&cinfo, TRUE);
   cinfo.out_color_space = ((components == 3) ? JCS_RGB : JCS_GRAYSCALE);
   jpeg_start_decompress(&cinfo);
   while (cinfo.output_scanline < cinfo.output_height) {
@@ -133,7 +133,7 @@
     int index = cinfo.output_scanline * row_stride;
     unsigned char *buffer[1];
     buffer[0] = (unsigned char *) &(pixelData[index]);
-    jpeg_read_scanlines(&cinfo, buffer, 1);
+    jpeg_read_scanlines(&cinfo, buffer, TRUE);
   }
   jpeg_finish_decompress(&cinfo);
   jpeg_destroy_decompress(&cinfo);
