diff -urN xca-0.8.1.orig/misc/xca.desktop xca-0.8.1/misc/xca.desktop
--- xca-0.8.1.orig/misc/xca.desktop	2010-01-05 07:52:03.000000000 +0100
+++ xca-0.8.1/misc/xca.desktop	2010-03-03 15:05:48.716988847 +0100
@@ -9,5 +9,5 @@
 Exec=xca %F
 Icon=xca-32x32.xpm
 Terminal=false
-Categories=Application;Utility;Qt;
+Categories=Application;Utility;Qt;X-UHU-Menu/Tools/Utilities;
 MimeType=application/x-xca-database;application/x-xca-template;application/x-x509-ca-cert;application/pkcs10;application/x-pkcs7-certificates;application/x-pkcs12;
