diff -Naur orig/uzbl-core.desktop mod/uzbl-core.desktop
--- orig/uzbl-core.desktop	2016-04-08 21:27:27.000000000 +0200
+++ mod/uzbl-core.desktop	2016-04-08 21:42:57.457717994 +0200
@@ -3,7 +3,7 @@
 Comment=Uzbl - web interface tools which adhere to the unix philosophy.
 Terminal=false
 Type=Application
-Icon=uzbl.png
+Icon=uzbl
 Exec=uzbl-browser %u
 StartupNotify=false
 Categories=Application;Network;WebBrowser;
diff -Naur orig/uzbl-tabbed.desktop mod/uzbl-tabbed.desktop
--- orig/uzbl-tabbed.desktop	2016-04-08 21:27:27.000000000 +0200
+++ mod/uzbl-tabbed.desktop	2016-04-08 21:43:10.793717789 +0200
@@ -3,7 +3,7 @@
 Comment=Uzbl - web interface tools which adhere to the unix philosophy.
 Terminal=false
 Type=Application
-Icon=uzbl.png
+Icon=uzbl
 Exec=uzbl-tabbed %U
 StartupNotify=false
 Categories=Application;Network;WebBrowser;
