diff -Nur orig/src/themesmodel.cpp mod/src/themesmodel.cpp
--- orig/src/themesmodel.cpp	2016-08-23 15:01:27.000000000 +0200
+++ mod/src/themesmodel.cpp	2016-09-13 13:19:30.000000000 +0200
@@ -84,10 +84,10 @@
 
 void ThemesModel::populate()
 {
-    QString themesBaseDir = KSharedConfig::openConfig(SDDM_CONFIG_FILE, KConfig::SimpleConfig)->group("General").readEntry("ThemesDir");
+    QString themesBaseDir = KSharedConfig::openConfig(SDDM_CONFIG_FILE, KConfig::SimpleConfig)->group("Theme").readEntry("ThemeDir");
 
     if (themesBaseDir.isEmpty()) {
-        themesBaseDir = QStandardPaths::locate(QStandardPaths::GenericDataLocation, "sddm", QStandardPaths::LocateDirectory) + "/themes";
+        themesBaseDir = QStandardPaths::locate(QStandardPaths::GenericDataLocation, "sddm/themes", QStandardPaths::LocateDirectory);
     }
 
     QDir dir(themesBaseDir);
