diff -Naur orig/src/Core/Scripting/OOJSPlayerShip.m mod/src/Core/Scripting/OOJSPlayerShip.m
--- orig/src/Core/Scripting/OOJSPlayerShip.m	2016-04-24 12:50:33.000000000 +0200
+++ mod/src/Core/Scripting/OOJSPlayerShip.m	2016-04-24 13:59:02.614012961 +0200
@@ -1522,7 +1522,7 @@
 
 	if (!isCargo)
 	{
-		if (argc > offset+5 && JS_ValueToECMAUint32(context, OOJS_ARGV[offset + 5], &uValue) && isfinite(uValue))
+		if (argc > offset+5 && JS_ValueToECMAUint32(context, OOJS_ARGV[offset + 5], &uValue) && isfinite((double)uValue))
 		{
 			*risk = uValue;
 		}
