diff -urN gsettings-desktop-schemas-3.12.2.orig/schemas/org.gnome.desktop.background.gschema.xml.in.in gsettings-desktop-schemas-3.12.2/schemas/org.gnome.desktop.background.gschema.xml.in.in
--- gsettings-desktop-schemas-3.12.2.orig/schemas/org.gnome.desktop.background.gschema.xml.in.in	2013-04-17 11:01:52.000000000 +0200
+++ gsettings-desktop-schemas-3.12.2/schemas/org.gnome.desktop.background.gschema.xml.in.in	2014-07-15 09:37:42.809472919 +0200
@@ -10,7 +10,7 @@
       </_description>
     </key>
     <key name="picture-uri" type="s">
-      <default>'file://@datadir@/themes/Adwaita/backgrounds/adwaita-timed.xml'</default>
+      <default>'file:///usr/share/wallpapers/UHU/uhu-3-bg.png'</default>
       <_summary>Picture URI</_summary>
       <_description>
         URI to use for the background image. Note that the backend only supports
diff -urN gsettings-desktop-schemas-3.12.2.orig/schemas/org.gnome.desktop.screensaver.gschema.xml.in.in gsettings-desktop-schemas-3.12.2/schemas/org.gnome.desktop.screensaver.gschema.xml.in.in
--- gsettings-desktop-schemas-3.12.2.orig/schemas/org.gnome.desktop.screensaver.gschema.xml.in.in	2013-10-12 18:10:05.000000000 +0200
+++ gsettings-desktop-schemas-3.12.2/schemas/org.gnome.desktop.screensaver.gschema.xml.in.in	2014-07-15 09:39:12.815616391 +0200
@@ -94,7 +94,7 @@
       </_description>
     </key>
     <key name="picture-uri" type="s">
-      <default>'file://@datadir@/backgrounds/gnome/Locked.jpg'</default>
+      <default>'file:///usr/share/wallpapers/UHU/uhu-3-bg.png'</default>
       <_summary>Picture URI</_summary>
       <_description>
         URI to use for the background image. Not that the backend only supports
