diff -Naur orig/schemas/org.gnome.desktop.screensaver.gschema.xml.in mod/schemas/org.gnome.desktop.screensaver.gschema.xml.in
--- orig/schemas/org.gnome.desktop.screensaver.gschema.xml.in	2015-08-16 22:07:26.000000000 +0200
+++ mod/schemas/org.gnome.desktop.screensaver.gschema.xml.in	2016-01-30 11:43:21.097150204 +0100
@@ -1,7 +1,7 @@
 <schemalist>
   <schema gettext-domain="@GETTEXT_PACKAGE@" id="org.gnome.desktop.screensaver" path="/org/gnome/desktop/screensaver/">
     <key name="idle-activation-enabled" type="b">
-      <default>true</default>
+      <default>false</default>
       <summary>Activate when idle</summary>
       <description>
         Set this to TRUE to activate the screensaver when the session is idle.
