diff -Naurdp NetworkManager-0.9.10.0.orig/src/nm-ip4-config.c NetworkManager-0.9.10.0/src/nm-ip4-config.c
--- NetworkManager-0.9.10.0.orig/src/nm-ip4-config.c	2014-07-04 02:44:19.000000000 +0200
+++ NetworkManager-0.9.10.0/src/nm-ip4-config.c	2014-07-11 08:08:06.925868104 +0200
@@ -434,7 +434,7 @@ nm_ip4_config_create_setting (const NMIP
 
 	/* Use 'disabled' if the method wasn't previously set */
 	if (!method)
-		method = NM_SETTING_IP4_CONFIG_METHOD_DISABLED;
+		method = NM_SETTING_IP4_CONFIG_METHOD_AUTO;
 	g_object_set (s_ip4, NM_SETTING_IP4_CONFIG_METHOD, method, NULL);
 
 	/* Routes */
