diff -Naur e16-0.16.8.9.orig/src/menus-misc.c e16-0.16.8.9/src/menus-misc.c
--- e16-0.16.8.9.orig/src/menus-misc.c	2007-08-11 13:31:20.000000000 +0200
+++ e16-0.16.8.9/src/menus-misc.c	2007-08-13 12:11:42.000000000 +0200
@@ -408,15 +408,15 @@
 
    if (isabspath(file))
      {
-	ff = FindFile(file, NULL, 0);
+	ff = FindFile(file, NULL, 1);
      }
    else
      {
 	/* Check menus subdir first */
 	Esnprintf(buf, sizeof(buf), "menus/%s", file);
-	ff = FindFile(buf, NULL, 0);
+	ff = FindFile(buf, NULL, 1);
 	if (!ff)
-	   ff = FindFile(file, NULL, 0);
+	   ff = FindFile(file, NULL, 1);
      }
    if (!ff)
       goto done;
