diff -Naur ORIG/amsn.desktop UHU/amsn.desktop
--- ORIG/amsn.desktop	2008-10-27 10:18:47.000000000 +0100
+++ UHU/amsn.desktop	2009-08-26 21:21:10.000000000 +0200
@@ -2,7 +2,7 @@
 Name=aMSN
 Exec=amsn
 Icon=amsn
-Categories=Network;InstantMessaging;
+Categories=Network;InstantMessaging;X-UHU-Menu/Internet/Messengers;
 GenericName=MSN Messenger for Linux
 GenericName[de]=MSN Messenger für Linux
 GenericName[es]=MSN Messenger para Linux
