diff -Naur orig/lives-plugins/plugins/decoders/mpegts_decoder.c patched/lives-plugins/plugins/decoders/mpegts_decoder.c
--- orig/lives-plugins/plugins/decoders/mpegts_decoder.c	2016-03-08 16:00:09.000000000 +0100
+++ patched/lives-plugins/plugins/decoders/mpegts_decoder.c	2016-05-02 10:38:28.516783783 +0200
@@ -1023,8 +1023,8 @@
 static void new_pes_packet(PESContext *pes, AVPacket *pkt) {
   av_init_packet(pkt);
 
-#if !HAVE_AVPACKET_UNREF
-  pkt->destruct = av_destruct_packet;
+#if FF_API_DESTRUCT_PACKET
+  pkt->destruct = av_packet_unref;
 #endif
 
   pkt->data = pes->buffer;
