diff -Naurd a/edit.h b/edit.h
--- a/edit.h	2005-07-07 03:53:09.000000000 +0200
+++ b/edit.h	2011-10-29 10:56:49.000000000 +0200
@@ -9,7 +9,7 @@
                      Header file for FK-editor
 */
 
-#define VERSION "1.5.29a"
+#define VERSION "1.5.30a"
 
 #include "Xwpe.h"
 #include "WeString.h"
diff -Naurd a/Makefile.in b/Makefile.in
--- a/Makefile.in	2005-07-07 03:53:09.000000000 +0200
+++ b/Makefile.in	2011-10-29 10:32:05.000000000 +0200
@@ -9,7 +9,7 @@
 #Destination Directory
 prefix=@prefix@
 exec_prefix=@exec_prefix@
-DESTDIR=@bindir@
+BINDIR=@bindir@
 LIBDIR=@libdir@
 XWPELIBDIR=	$(LIBDIR)/xwpe
 MANDIR=@mandir@
@@ -115,24 +115,18 @@
 # Bug Note: Currently xwpe and xwe are installed even if X windows versions
 # are not compiled.  (They will function as wpe and we respectively.)
 install_fst:	xwpe
-	if test ! -d $(DESTDIR); then mkdir -p $(DESTDIR); fi
+	if test ! -d $(BINDIR); then mkdir -p $(BINDIR); fi
 	if test ! -d $(XWPELIBDIR); then mkdir -p $(XWPELIBDIR); fi
-	rm -f $(DESTDIR)/wpe $(DESTDIR)/xwpe $(DESTDIR)/we $(DESTDIR)/xwe
-	$(INSTALL) xwpe $(DESTDIR)/we
-	chmod ugo+x $(DESTDIR)/we
-	(cd $(DESTDIR) && $(LN_S) we xwe)
-	(cd $(DESTDIR) && $(LN_S) we xwpe)
-	(cd $(DESTDIR) && $(LN_S) we wpe)
+	rm -f $(BINDIR)/wpe $(BINDIR)/xwpe $(BINDIR)/we $(BINDIR)/xwe
+	$(INSTALL) xwpe $(BINDIR)/we
+	chmod ugo+x $(BINDIR)/we
+	(cd $(BINDIR) && $(LN_S) we xwe)
+	(cd $(BINDIR) && $(LN_S) we xwpe)
+	(cd $(BINDIR) && $(LN_S) we wpe)
 	rm -f $(XWPELIBDIR)/help.xwpe
 	if test ! -f $(XWPELIBDIR)/syntax_def; then  \
 	  $(INSTALL) syntax_def $(XWPELIBDIR)/syntax_def; \
 	  chmod ugo=r $(XWPELIBDIR)/syntax_def; fi
-	if test ! -d $(MANDIR)/man1; then mkdir -p $(MANDIR)/man1; fi
-	rm -f $(MANDIR)/man1/xwpe.1 $(MANDIR)/man1/xwe.1 
-	rm -f $(MANDIR)/man1/wpe.1 $(MANDIR)/man1/we.1 
-	$(INSTALL) xwe.1 wpe.1 we.1 $(MANDIR)/man1
-	chmod ugo=r $(MANDIR)/man1/xwe.1 
-	chmod ugo=r $(MANDIR)/man1/wpe.1 $(MANDIR)/man1/we.1 
 	if test -f libxwpe-x11.so; then \
 	  $(INSTALL) libxwpe-x11.so $(XWPELIBDIR)/libxwpe-x11.so; fi
 	if test -f libxwpe-term.so; then \
@@ -143,8 +137,6 @@
 	chmod ugo=r $(XWPELIBDIR)/help.xwpe
 	$(INSTALL) help.key_eng $(XWPELIBDIR)/help.key
 	chmod ugo=r $(XWPELIBDIR)/help.key
-	$(INSTALL) xwpe.1_eng $(MANDIR)/man1/xwpe.1
-	chmod ugo=r $(MANDIR)/man1/xwpe.1
 
 # German install uses old documention as no updated version is available
 install_german:	install_fst
diff -Naurd a/we_debug.c b/we_debug.c
--- a/we_debug.c	2005-07-07 03:53:09.000000000 +0200
+++ b/we_debug.c	2011-10-29 10:42:54.000000000 +0200
@@ -63,7 +63,7 @@
 extern char *e_tmp_dir;
 
 #ifdef NOTPARM
-char *tparm();
+/*char *tparm();*/
 char *tgoto();
 #endif
 #ifdef DEFTPUTS
diff -Naurd a/we_wind.c b/we_wind.c
--- a/we_wind.c	2005-07-07 03:53:09.000000000 +0200
+++ b/we_wind.c	2011-10-29 10:40:49.000000000 +0200
@@ -1372,7 +1372,7 @@
 				&& e_mouse.x <= xa+n-1) ret = CDO;
 #endif
       if (ret == CDO && e_sv_window(xa+n, ya, &num, *df, f) == WPE_CR)
-          strcpy(str, (*df)->name[num]);
+          (strcpy)(str, (*df)->name[num]);
    }  while(ret == CDO);
    return(ret);
 }
diff -Naurd a/xwpe.1_eng b/xwpe.1_eng
--- a/xwpe.1_eng	2005-07-07 03:53:09.000000000 +0200
+++ b/xwpe.1_eng	2011-10-29 10:37:52.000000000 +0200
@@ -82,7 +82,7 @@
 Start \fIxwpe\fP in the \fIrecover mode\fP. The internal buffers
 are saved in files with the postfix `.ESV' in the name
 if a signal is caught by \fIxwpe\fP (except for SIGKILL).
-If the editor is invoked with the `-r' option is
+If the editor is invoked with the `\-r' option is
 specified in the command line, the old session will be
 recovered.
 
@@ -181,8 +181,8 @@
 "string:  Characters used for comments depending on the
           column."
 "string:  Special single characters."
-"integer: Column for comments (-1 for none)."
-"integer: Column for continuing line (-1 for none)"
+"integer: Column for comments (\-1 for none)."
+"integer: Column for continuing line (\-1 for none)"
 "Integer: Column for comment till the end of line (1000
           for none)"
 .EE
@@ -217,7 +217,7 @@
  struct switch typedef union unsigned void volatile
  while
  0
- ~^()[]{}<>+-/*%=|&!.?:,; /* */ NULL NULL "'#\\  -1 -1 1000
+ ~^()[]{}<>+\-/*%=|&!.?:,; /* */ NULL NULL "'#\\  \-1 \-1 1000
 
  .f
  64
@@ -257,7 +257,7 @@
 
 Documentation hasn't been updated.
 
-Compilers are assumed to have a -c and -o option.  This causes problems
+Compilers are assumed to have a \-c and \-o option.  This causes problems
 for those who wish to use java, perl, or other languages.
 
 \fBmake install\fP installs x versions even if not compiled in.
