diff -Naur orig/sample-config/nrpe.xinetd.in patched/sample-config/nrpe.xinetd.in
--- orig/sample-config/nrpe.xinetd.in	2013-09-06 17:27:13.000000000 +0200
+++ patched/sample-config/nrpe.xinetd.in	2016-07-11 10:09:36.388303996 +0200
@@ -2,15 +2,15 @@
 # description: NRPE (Nagios Remote Plugin Executor)
 service nrpe
 {
+        disable         = yes
        	flags           = REUSE
         socket_type     = stream    
 	port		= @nrpe_port@    
        	wait            = no
         user            = @nrpe_user@
 	group		= @nrpe_group@
-       	server          = @bindir@/nrpe
+       	server          = @sbindir@/nrpe
         server_args     = -c @sysconfdir@/nrpe.cfg --inetd
        	log_on_failure  += USERID
-        disable         = no
 	only_from       = 127.0.0.1
 }
