diff -Naurd a/include/linux/config.h b/include/linux/config.h
--- a/include/linux/config.h	1970-01-01 01:00:00.000000000 +0100
+++ b/include/linux/config.h	2006-10-14 05:34:03.000000000 +0200
@@ -0,0 +1,8 @@
+#ifndef _LINUX_CONFIG_H
+#define _LINUX_CONFIG_H
+/* This file is no longer in use and kept only for backward compatibility.
+ * autoconf.h is now included via -imacros on the commandline
+ */
+#include <linux/autoconf.h>
+
+#endif
