diff -Naurdp a/fs/fat/inode.c b/fs/fat/inode.c
--- a/fs/fat/inode.c	2011-08-29 06:16:01.000000000 +0200
+++ b/fs/fat/inode.c	2011-10-14 15:38:18.092738080 +0200
@@ -893,7 +893,7 @@ static int fat_show_options(struct seq_f
 enum {
 	Opt_check_n, Opt_check_r, Opt_check_s, Opt_uid, Opt_gid,
 	Opt_umask, Opt_dmask, Opt_fmask, Opt_allow_utime, Opt_codepage,
-	Opt_usefree, Opt_nocase, Opt_quiet, Opt_showexec, Opt_debug,
+	Opt_usefree, Opt_nocase, Opt_quiet, Opt_noquiet, Opt_showexec, Opt_debug,
 	Opt_immutable, Opt_dots, Opt_nodots,
 	Opt_charset, Opt_shortname_lower, Opt_shortname_win95,
 	Opt_shortname_winnt, Opt_shortname_mixed, Opt_utf8_no, Opt_utf8_yes,
@@ -919,6 +919,7 @@ static const match_table_t fat_tokens =
 	{Opt_usefree, "usefree"},
 	{Opt_nocase, "nocase"},
 	{Opt_quiet, "quiet"},
+	{Opt_noquiet, "noquiet"},
 	{Opt_showexec, "showexec"},
 	{Opt_debug, "debug"},
 	{Opt_immutable, "sys_immutable"},
@@ -1003,8 +1004,8 @@ static int parse_options(struct super_bl
 		opts->rodir = 1;
 	}
 	opts->name_check = 'n';
-	opts->quiet = opts->showexec = opts->sys_immutable = opts->dotsOK =  0;
-	opts->utf8 = opts->unicode_xlate = 0;
+	opts->unicode_xlate = opts->showexec = opts->sys_immutable = opts->dotsOK = 0;
+	opts->utf8 = opts->quiet = 1;
 	opts->numtail = 1;
 	opts->usefree = opts->nocase = 0;
 	opts->tz_utc = 0;
@@ -1051,6 +1052,9 @@ static int parse_options(struct super_bl
 		case Opt_quiet:
 			opts->quiet = 1;
 			break;
+		case Opt_noquiet:
+			opts->quiet = 0;
+			break;
 		case Opt_showexec:
 			opts->showexec = 1;
 			break;
@@ -1186,13 +1190,6 @@ static int parse_options(struct super_bl
 	}
 
 out:
-	/* UTF-8 doesn't provide FAT semantics */
-	if (!strcmp(opts->iocharset, "utf8")) {
-		fat_msg(sb, KERN_WARNING, "utf8 is not a recommended IO charset"
-		       " for FAT filesystems, filesystem will be "
-		       "case sensitive!");
-	}
-
 	/* If user doesn't specify allow_utime, it's initialized from dmask. */
 	if (opts->allow_utime == (unsigned short)-1)
 		opts->allow_utime = ~opts->fs_dmask & (S_IWGRP | S_IWOTH);
