diff -Naurdp a/net/core/dev.c b/net/core/dev.c
--- a/net/core/dev.c	2013-03-12 11:18:23.989247969 +0100
+++ b/net/core/dev.c	2013-03-12 11:22:05.957619770 +0100
@@ -1107,6 +1107,7 @@ int dev_change_name(struct net_device *d
 		write_seqcount_end(&devnet_rename_seq);
 		return err;
 	}
+	printk(KERN_INFO "%s renamed to %s\n", oldname, newname);
 
 rollback:
 	ret = device_rename(&dev->dev, dev->name);
