diff -Naur orig/Makefile patched/Makefile
--- orig/Makefile	2011-12-05 09:46:23.000000000 +0100
+++ patched/Makefile	2018-03-04 11:06:12.338313026 +0100
@@ -3,7 +3,8 @@
 #
 # 0.1.13: Added -I.
 
-LOCATION=/usr/local
+DESTDIR=
+LOCATION=/usr
 VERSION=0.1.14
 # VERSION changes
 #---------------------
@@ -73,8 +74,8 @@
 
 install: ${OBJ}
 	strip ripmime
-	cp ripmime ${LOCATION}/bin/
-	cp ripmime.1  ${LOCATION}/man/man1
+	cp ripmime ${DESTDIR}/${LOCATION}/bin/
+	cp ripmime.1  ${DESTDIR}/${LOCATION}/share/man/man1
 
 ffget_test: ffget_mmap_test.c ffget_mmap.[ch] logger.o ffget_mmap.o
 	${CC} ${CFLAGS} ffget_mmap_test.c logger.o ffget_mmap.o -o ffgt
