diff -Naur orig/src/ui/dialog/layer-properties.cpp patched/src/ui/dialog/layer-properties.cpp
--- orig/src/ui/dialog/layer-properties.cpp	2014-11-30 19:45:32.000000000 +0100
+++ patched/src/ui/dialog/layer-properties.cpp	2016-11-20 03:50:48.409142745 +0100
@@ -146,7 +146,7 @@
     destroy_();
     Glib::signal_idle().connect(
         sigc::bind_return(
-            sigc::bind(sigc::ptr_fun(&::operator delete), this),
+            sigc::bind(sigc::ptr_fun<void*, void>(&::operator delete), this),
             false 
         )
     );
