diff -Naur orig/avogadro/src/avogadro.desktop patched/avogadro/src/avogadro.desktop
--- orig/avogadro/src/avogadro.desktop	2013-12-06 16:50:04.000000000 +0100
+++ patched/avogadro/src/avogadro.desktop	2016-02-13 19:24:19.813146984 +0100
@@ -2,6 +2,7 @@
 Version=1.0
 Name=Avogadro
 Comment=Advanced molecular editor
+Comment[hu]=Molekula szerkesztő
 Exec=avogadro %f
 Icon=avogadro-icon
 Terminal=false
