diff -Naur avogadro-1.0.3-orig/libavogadro/src/camera.cpp avogadro-1.0.3/libavogadro/src/camera.cpp
--- avogadro-1.0.3-orig/libavogadro/src/camera.cpp	2011-04-25 07:22:18.000000000 +0200
+++ avogadro-1.0.3/libavogadro/src/camera.cpp	2012-04-02 10:40:32.000000000 +0200
@@ -26,6 +26,8 @@
 #include "glwidget.h"
 #include <avogadro/molecule.h>
 
+#include <GL/glu.h>
+
 using namespace Eigen;
 
 namespace Avogadro
diff -Naur avogadro-1.0.3-orig/libavogadro/src/glpainter_p.cpp avogadro-1.0.3/libavogadro/src/glpainter_p.cpp
--- avogadro-1.0.3-orig/libavogadro/src/glpainter_p.cpp	2011-04-25 07:22:18.000000000 +0200
+++ avogadro-1.0.3/libavogadro/src/glpainter_p.cpp	2012-04-02 10:41:01.000000000 +0200
@@ -43,6 +43,8 @@
 #include <QVarLengthArray>
 #include <Eigen/Geometry>
 
+#include <GL/glu.h>
+
 namespace Avogadro
 {
 
diff -Naur avogadro-1.0.3-orig/libavogadro/src/glwidget.cpp avogadro-1.0.3/libavogadro/src/glwidget.cpp
--- avogadro-1.0.3-orig/libavogadro/src/glwidget.cpp	2011-04-25 07:22:18.000000000 +0200
+++ avogadro-1.0.3/libavogadro/src/glwidget.cpp	2012-04-02 10:41:12.000000000 +0200
@@ -70,6 +70,8 @@
 
 #include <openbabel/mol.h>
 
+#include <GL/glu.h>
+
 using namespace OpenBabel;
 using namespace Eigen;
 
diff -Naur avogadro-1.0.3-orig/libavogadro/src/tools/selectrotatetool.cpp avogadro-1.0.3/libavogadro/src/tools/selectrotatetool.cpp
--- avogadro-1.0.3-orig/libavogadro/src/tools/selectrotatetool.cpp	2011-04-25 07:22:18.000000000 +0200
+++ avogadro-1.0.3/libavogadro/src/tools/selectrotatetool.cpp	2012-04-02 10:45:10.000000000 +0200
@@ -44,6 +44,8 @@
 #include <QComboBox>
 #include <QDebug>
 
+#include <GL/glu.h>
+
 using namespace std;
 using namespace OpenBabel;
 using namespace Eigen;
