diff -Naur orig/librecad/src/lib/engine/rs_ellipse.h patched/librecad/src/lib/engine/rs_ellipse.h
--- orig/librecad/src/lib/engine/rs_ellipse.h	2013-06-09 03:05:04.000000000 +0200
+++ patched/librecad/src/lib/engine/rs_ellipse.h	2013-08-05 01:25:41.060423916 +0200
@@ -29,9 +29,11 @@
 #ifndef RS_ELLIPSE_H
 #define RS_ELLIPSE_H
 
+#ifndef Q_MOC_RUN
 #include <boost/version.hpp>
 #include <boost/math/tools/roots.hpp>
 #include <boost/math/special_functions/ellint_2.hpp>
+#endif
 
 class LC_Quadratic;
 #include "rs_atomicentity.h"
