diff -Naur orig/Makefile mod/Makefile
--- orig/Makefile	2013-08-06 12:42:38.000000000 +0200
+++ mod/Makefile	2014-03-23 15:16:04.000000000 +0100
@@ -126,7 +126,7 @@
 # FFdeCSA
 CPUOPT     ?= pentium
 PARALLEL   ?= PARALLEL_32_INT
-CSAFLAGS   ?= -Wall -fPIC -g -O3 -mmmx -fomit-frame-pointer -fexpensive-optimizations -funroll-loops
+CSAFLAGS   ?= -Wall -fPIC -g -mmmx -fomit-frame-pointer -fexpensive-optimizations -funroll-loops
 FFDECSADIR  = FFdecsa
 FFDECSA     = $(FFDECSADIR)/FFdecsa.o
 FFDECSATEST = $(FFDECSADIR)/FFdecsa_test.done
