diff -Naur orig/config.template mod/config.template
--- orig/config.template	2014-11-16 00:13:53.000000000 +0100
+++ mod/config.template	2016-01-04 17:27:06.803004358 +0100
@@ -8,18 +8,18 @@
 #BACKEND = opengl
 #BACKEND = sdl
 #BACKEND = sdl2
-#BACKEND = mixer_sdl
+BACKEND = mixer_sdl
 #BACKEND = posix
 
 #COLOUR_DEPTH = 0
-#COLOUR_DEPTH = 16
+COLOUR_DEPTH = 16
 
 #OSTYPE = amiga
 #OSTYPE = beos
 #OSTYPE = cygwin
 #OSTYPE = freebsd
 #OSTYPE = haiku
-#OSTYPE = linux
+OSTYPE = linux
 #OSTYPE = mingw
 #OSTYPE = mac
 
@@ -41,7 +41,7 @@
 #SDL_CONFIG     = sdl-config
 #SDL2_CONFIG    = sdl2-config
 
-#VERBOSE = 1
+VERBOSE = 1
 
 # The following useful conditional compilation flags exist
 #
