diff -Naur orig/Makefile patched/Makefile
--- orig/Makefile	2016-01-09 22:36:24.000000000 +0100
+++ patched/Makefile	2016-07-23 00:51:24.836852283 +0200
@@ -71,17 +71,6 @@
 SDL_CONFIG     ?= sdl-config
 SDL2_CONFIG    ?= sdl2-config
 
-ifneq ($(OPTIMISE),)
-  CFLAGS += -O3
-  ifeq ($(findstring $(OSTYPE), amiga),)
-	ifneq ($(findstring $(CXX), clang),)
-		CFLAGS += -minline-all-stringops
-	endif
-  endif
-else
-  CFLAGS += -O
-endif
-
 ifdef DEBUG
   ifeq ($(shell expr $(DEBUG) \>= 1), 1)
     CFLAGS += -g -DDEBUG
diff -Naur orig/makeobj/Makefile patched/makeobj/Makefile
--- orig/makeobj/Makefile	2015-01-12 17:09:48.000000000 +0100
+++ patched/makeobj/Makefile	2016-07-23 00:52:30.821938219 +0200
@@ -42,17 +42,6 @@
   CXXFLAGS   += -DUSE_HW -DUSE_C
 endif
 
-ifneq ($(OPTIMISE),)
-  CFLAGS   += -O3 -fno-schedule-insns
-  CXXFLAGS += -O3 -fno-schedule-insns
-  ifneq ($(OSTYPE),mac)
-    CFLAGS  += -minline-all-stringops
-  endif
-else
-  CFLAGS   += -O
-  CXXFLAGS += -O
-endif
-
 ifdef DEBUG
   ifeq ($(shell expr $(DEBUG) \>= 1), 1)
     CFLAGS   += -g -DDEBUG
