diff -Naur orig/simsys.cc mod/simsys.cc
--- orig/simsys.cc	2015-11-05 00:20:56.000000000 +0100
+++ mod/simsys.cc	2016-01-04 17:40:05.067393242 +0100
@@ -168,7 +168,7 @@
 	find_directory(B_USER_DIRECTORY, &userDir);
 	sprintf(buffer, "%s/simutrans", userDir.Path());
 #else
-	sprintf(buffer, "%s/simutrans", getenv("HOME"));
+	sprintf(buffer, "%s/.simutrans", getenv("HOME"));
 #endif
 
 	dr_mkdir(buffer);
