diff -Naur orig/configure.ac patched/configure.ac
--- orig/configure.ac	2017-01-19 00:44:44.000000000 +0100
+++ patched/configure.ac	2017-01-23 04:14:23.910715863 +0100
@@ -37,17 +37,7 @@
 AC_SUBST(APPLET_CFLAGS)
 AC_SUBST(APPLET_LIBS)
 
-PKG_CHECK_EXISTS(indicator3-0.4 >= $INDICATOR_NG_VERSION,
-                 [have_indicator_ng="yes"],
-                 [have_indicator_ng="no"])
-
-if test "x$have_indicator_ng" = "xyes"; then
-    PKG_CHECK_MODULES(INDICATOR, indicator3-0.4 >= $INDICATOR_NG_VERSION
-                      libido3-0.1 >= 13.10,
-                      [AC_DEFINE(HAVE_INDICATOR_NG, 1, "New style indicators support")])
-else
-    PKG_CHECK_MODULES(INDICATOR, indicator3-0.4 >= $INDICATOR_REQUIRED_VERSION)
-fi
+PKG_CHECK_MODULES(INDICATOR, indicator3-0.4 >= $INDICATOR_REQUIRED_VERSION)
 
 AC_SUBST(INDICATOR_CFLAGS)
 AC_SUBST(INDICATOR_LIBS)
