diff -Naur orig/configure.ac patched/configure.ac
--- orig/configure.ac	2016-06-21 12:18:03.000000000 +0200
+++ patched/configure.ac	2016-07-08 23:21:08.566605219 +0200
@@ -57,17 +57,8 @@
 AC_SUBST(APPLET_CFLAGS)
 AC_SUBST(APPLET_LIBS)
 
-PKG_CHECK_EXISTS($INDICATOR_PKG >= $INDICATOR_NG_VERSION,
-                 [have_indicator_ng="yes"],
-                 [have_indicator_ng="no"])
+PKG_CHECK_MODULES(INDICATOR, $INDICATOR_PKG >= $INDICATOR_REQUIRED_VERSION)
 
-if test "x$with_gtk" = "x3.0" && test "x$have_indicator_ng" = "xyes"; then
-    PKG_CHECK_MODULES(INDICATOR, $INDICATOR_PKG >= $INDICATOR_NG_VERSION
-                      libido3-0.1 >= 13.10,
-                      [AC_DEFINE(HAVE_INDICATOR_NG, 1, "New style indicators support")])
-else
-    PKG_CHECK_MODULES(INDICATOR, $INDICATOR_PKG >= $INDICATOR_REQUIRED_VERSION)
-fi
 
 AC_SUBST(INDICATOR_CFLAGS)
 AC_SUBST(INDICATOR_LIBS)
