diff -Naur orig/lib_pypy/_tkinter/tklib_build.py patched/lib_pypy/_tkinter/tklib_build.py
--- orig/lib_pypy/_tkinter/tklib_build.py	2015-11-17 13:10:14.000000000 +0100
+++ patched/lib_pypy/_tkinter/tklib_build.py	2016-03-01 01:26:39.832955270 +0100
@@ -22,12 +22,9 @@
     linklibs = ['tcl', 'tk']
     libdirs = []
 else:
-    for _ver in ['', '8.6', '8.5', '']:
-        incdirs = ['/usr/include/tcl' + _ver]
-        linklibs = ['tcl' + _ver, 'tk' + _ver]
-        libdirs = []
-        if os.path.isdir(incdirs[0]):
-            break
+    incdirs = ['/usr/include/tcl8.6']
+    linklibs = ['tcl8.6', 'tk8.6']
+    libdirs = []
 
 config_ffi = FFI()
 config_ffi.cdef("""
