--- a/monodevelop.desktop.orig	2013-08-01 18:59:41.000000000 +0200
+++ a/monodevelop.desktop	2013-08-01 19:00:48.000000000 +0200
@@ -4,6 +4,7 @@
 Name=MonoDevelop
 GenericName=Integrated Development Environment
 Comment=Develop .NET applications in an Integrated Development Environment
+Comment[hu]=Integrált Fejlesztői Környezet a .NET keretrendszerhez
 Exec=monodevelop %F
 TryExec=monodevelop
 Icon=monodevelop
