diff -Naur orig/src/install.pri patched/src/install.pri
--- orig/src/install.pri	2015-10-08 13:08:40.000000000 +0200
+++ patched/src/install.pri	2015-10-10 11:38:48.084011378 +0200
@@ -44,7 +44,7 @@
     ico256.path = $$hicolor_folder/256x256/apps
 
     bashcompletion.files = $$PWD/../linux/completion/qupzilla
-    bashcompletion.path = /usr/share/bash-completion/completions
+    bashcompletion.path = /etc/bash_completion.d
 
     appdata.files = $$PWD/../linux/appdata/qupzilla.appdata.xml
     appdata.path = /usr/share/appdata
