--- a/data/quadrapassel.desktop.in.orig	2011-08-22 00:39:28.000000000 +0200
+++ a/data/quadrapassel.desktop.in	2011-10-20 18:32:30.000000000 +0200
@@ -5,7 +5,7 @@
 Exec=quadrapassel
 Terminal=false
 Type=Application
-Categories=GNOME;GTK;Game;BlocksGame;
+Categories=GNOME;GTK;Game;BlocksGame;X-UHU-Menu/Games/TacticStrategy;
 _Keywords=tetris;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=gnome-games
