diff -Naur orig/src/validate.c patched/src/validate.c
--- orig/src/validate.c	2016-02-15 21:18:53.259765911 +0100
+++ patched/src/validate.c	2016-02-15 21:21:08.008434094 +0100
@@ -363,7 +363,7 @@
 };
 
 static const char *show_in_registered[] = {
-  "GNOME", "KDE", "LXDE", "MATE", "Razor", "ROX", "TDE", "Unity", "XFCE", "Cinnamon", "EDE", "Old"
+  "GNOME", "KDE", "LXDE", "LXQt", "MATE", "Razor", "ROX", "TDE", "Unity", "XFCE", "Cinnamon", "EDE", "Lumina", "Old"
 };
 
 static struct {
