diff -Naur orig/src/validate.c patched/src/validate.c
--- orig/src/validate.c	2016-12-15 10:55:19.670305925 +0100
+++ patched/src/validate.c	2016-12-15 11:01:10.569449957 +0100
@@ -525,6 +525,18 @@
   { "KDE",                    FALSE, FALSE, FALSE, { NULL }, { "Qt", NULL } },
   { "GNOME",                  FALSE, FALSE, FALSE, { NULL }, { "GTK", NULL } },
   { "XFCE",                   FALSE, FALSE, FALSE, { NULL }, { "GTK", NULL } },
+  { "LXDE",                   FALSE, FALSE, FALSE, { NULL }, { "GTK", NULL } },
+  { "LXQt",                   FALSE, FALSE, FALSE, { NULL }, { "Qt", NULL } },
+  { "MATE",                   FALSE, FALSE, FALSE, { NULL }, { "GTK", NULL } },
+  { "Razor",                  FALSE, FALSE, FALSE, { NULL }, { NULL } },
+  { "ROX",                    FALSE, FALSE, FALSE, { NULL }, { NULL } },
+  { "TDE",                    FALSE, FALSE, FALSE, { NULL }, { NULL } },
+  { "Unity",                  FALSE, FALSE, FALSE, { NULL }, { "GTK", NULL } },
+  { "EDE",                    FALSE, FALSE, FALSE, { NULL }, { NULL } },
+  { "Cinnamon",               FALSE, FALSE, FALSE, { NULL }, { "GTK", NULL } },
+  { "Pantheon",               FALSE, FALSE, FALSE, { NULL }, { "GTK", NULL } },
+  { "Lumina",                 FALSE, FALSE, FALSE, { NULL }, { "Qt", NULL } },
+  { "Budgie",                 FALSE, FALSE, FALSE, { NULL }, { "GTK", NULL } },
   { "GTK",                    FALSE, FALSE, FALSE, { NULL }, { NULL } },
   { "Qt",                     FALSE, FALSE, FALSE, { NULL }, { NULL } },
   { "Motif",                  FALSE, FALSE, FALSE, { NULL }, { NULL } },
