diff -Naur orig/config.c patched/config.c
--- orig/config.c	2017-05-02 13:12:11.134741323 +0200
+++ patched/config.c	2017-05-16 03:04:33.309331884 +0200
@@ -129,7 +129,8 @@
 static const char *defTabooExts[] = { ".rpmsave", ".rpmorig", "~", ",v",
     ".disabled", ".dpkg-old", ".dpkg-dist", ".dpkg-new", ".cfsaved",
     ".ucf-old", ".ucf-dist", ".ucf-new",
-    ".rpmnew", ".swp", ".cfsaved", ".rhn-cfg-tmp-*"
+    ".rpmnew", ".swp", ".cfsaved", ".rhn-cfg-tmp-*",
+    ".dpkg-old", ".dpkg-new", ".dpkg-dist", ".dpkg-tmp"
 };
 static int defTabooCount = sizeof(defTabooExts) / sizeof(char *);
 
