diff -Naur ORIG/Alacarte_menu/gufw.desktop UHU/Alacarte_menu/gufw.desktop
--- ORIG/Alacarte_menu/gufw.desktop	2010-04-11 09:53:33.000000000 +0200
+++ UHU/Alacarte_menu/gufw.desktop	2010-06-02 11:17:06.000000000 +0200
@@ -2,7 +2,7 @@
 Version=1.0
 Type=Application
 Terminal=false
-Exec=su-to-root -X -c /usr/share/gufw/gufw.py
+Exec=gufw
 Name=Firewall configuration
 Name[zh_CN]=防火墙配置
 Name[uk]=Налаштування фаєрволу
