diff -Naur orig/data/filezilla.desktop patched/data/filezilla.desktop
--- orig/data/filezilla.desktop	2014-05-01 21:29:18.000000000 +0200
+++ patched/data/filezilla.desktop	2014-09-22 22:39:35.733053072 +0200
@@ -3,9 +3,11 @@
 GenericName=FTP client
 GenericName[de]=FTP-Client
 GenericName[fr]=Client FTP
+GenericName[hu]=FTP kliens
 Comment=Download and upload files via FTP, FTPS and SFTP
 Comment[de]=Dateien über FTP, FTPS und SFTP übertragen
 Comment[fr]=Transférer des fichiers via FTP, FTPS et SFTP
+Comment[hu]=Adatok fel és letöltésére FTP, FTPS és SFTP protokollokkal
 Exec=filezilla
 Terminal=false
 Icon=filezilla
