diff -Naur orig/bin/genesis.desktop patched/bin/genesis.desktop
--- orig/bin/genesis.desktop	2009-07-28 22:24:57.000000000 +0200
+++ patched/bin/genesis.desktop	2016-02-19 01:43:56.379550509 +0100
@@ -6,7 +6,8 @@
 Name=Genesis
 Name[de_DE]=Genesis
 Comment[de_DE]=Evolution synchronisieren
+Comment[hu]=SyncML szinkronizálás Evolutionhöz
 Comment=Synchronize Evolution
 Exec=genesis
 Icon=genesis
-Categories=Application;Utility;
+Categories=Utility;
