diff -Naur orig/data/midori.desktop.in patched/data/midori.desktop.in
--- orig/data/midori.desktop.in	2011-12-17 23:47:28.000000000 +0100
+++ patched/data/midori.desktop.in	2011-12-18 12:52:19.000000000 +0100
@@ -6,7 +6,7 @@
 _Comment=Lightweight web browser
 _X-GNOME-Keywords=Internet;WWW;Explorer
 _X-AppInstall-Keywords=Internet;WWW;Explorer
-Categories=GTK;Network;WebBrowser;
+Categories=GTK;Network;WebBrowser;X-UHU-Menu/Internet/Browsers;
 MimeType=text/html;application/xhtml+xml;x-scheme-handler/http;x-scheme-handler/https;x-scheme-handler/geo;image/svg+xml;
 Exec=midori %U
 Icon=midori
