diff -Naur ORIG/builds/install/arch-specific/linux/Makefile.in UHU/builds/install/arch-specific/linux/Makefile.in
--- ORIG/builds/install/arch-specific/linux/Makefile.in	2013-03-19 09:37:35.000000000 +0100
+++ UHU/builds/install/arch-specific/linux/Makefile.in	2013-10-18 20:19:59.332268121 +0200
@@ -158,7 +158,7 @@
 
 install: buildTarDir
 	(cd $(GEN_ROOT)/$(TarDir); FIREBIRD= FIREBIRD_LOCK= ./install.sh)
-	cp -r $(GEN_ROOT)/$(DebugDir)/* /
+#	cp -r $(GEN_ROOT)/$(DebugDir)/* /
 
 
 silent_install: buildTarDir
diff -Naur ORIG/builds/install/arch-specific/linux/misc/tarinstall.sh.in UHU/builds/install/arch-specific/linux/misc/tarinstall.sh.in
--- ORIG/builds/install/arch-specific/linux/misc/tarinstall.sh.in	2013-03-19 09:37:35.000000000 +0100
+++ UHU/builds/install/arch-specific/linux/misc/tarinstall.sh.in	2013-10-18 20:20:47.724269361 +0200
@@ -28,7 +28,8 @@
 
 OrigDir=`pwd`
 
-cd /
+#cd /
+cd /${UB_INSTALLDIR}
 
 tar -xzof "$OrigDir/buildroot.tar.gz"
 
diff -Naur ORIG/builds/install/arch-specific/linux/misc/tarMainInstall.sh.in UHU/builds/install/arch-specific/linux/misc/tarMainInstall.sh.in
--- ORIG/builds/install/arch-specific/linux/misc/tarMainInstall.sh.in	2013-03-19 09:37:35.000000000 +0100
+++ UHU/builds/install/arch-specific/linux/misc/tarMainInstall.sh.in	2013-10-18 20:26:06.992277535 +0200
@@ -35,6 +35,8 @@
 
 Version="$BuildVersion-$PackageVersion.$CpuType"
 
+InteractiveInstall=
+export InteractiveInstall
 if [ ! -z "$InteractiveInstall" ]
 then
     cat <<EOF
@@ -50,9 +52,9 @@
 
 if [ -e scripts ]; then
     displayMessage "Extracting install data"
-    runAndCheckExit "./scripts/preinstall.sh"
+#    runAndCheckExit "./scripts/preinstall.sh"
     runAndCheckExit "./scripts/tarinstall.sh"
-    runAndCheckExit "./scripts/postinstall.sh"
+#    runAndCheckExit "./scripts/postinstall.sh"
 fi
 
 displayMessage "Install completed"
