diff -Naur orig/configure.ac mod/configure.ac
--- orig/configure.ac	2014-08-30 14:29:09.000000000 +0200
+++ mod/configure.ac	2014-12-31 20:11:16.276077802 +0100
@@ -374,7 +374,7 @@
 supported_locales="be bg bs ca cs da de el es et fi fr gl hr hu hy it ja ko ms nl no pl pt ro ru sk sv tr zh_CN zh_TW"
 supported_wprefs_locales="bg ca cs de es et fi fr hr hu it ja ko nl pt ru sk zh_CN zh_TW"
 supported_wings_locales="bg ca cs de fr hu nl sk"
-supported_util_locales="de es fr nl pt"
+supported_util_locales="de es fr nl pt hu"
 
 for lang in $LINGUAS; do
 	ok=0
