diff -Naur orig/configure mod/configure
--- orig/configure	2014-04-27 12:31:56.000000000 +0200
+++ mod/configure	2015-03-08 18:21:08.929720831 +0100
@@ -14930,7 +14930,7 @@
 {
 
   struct ftdi_context *p;
-  p = ftdi_new();
+  p = (void*)12345;
   if( p != NULL ){
       return 0;
   } else {
diff -Naur orig/configure.ac mod/configure.ac
--- orig/configure.ac	2014-04-27 12:27:15.000000000 +0200
+++ mod/configure.ac	2015-03-08 18:21:08.933720831 +0100
@@ -1044,7 +1044,7 @@
 #include <ftdi.h>
   ]], [[
   struct ftdi_context *p;
-  p = ftdi_new();
+  p = (void*)12345;
   if( p != NULL ){
       return 0;
   } else {
