diff -Naurd a/src/sakura.c b/src/sakura.c
--- a/src/sakura.c	2014-06-23 08:58:00.000000000 +0200
+++ b/src/sakura.c	2014-06-27 16:32:57.000000000 +0200
@@ -224,6 +224,7 @@
 	guint height;
 	glong columns;
 	glong rows;
+	gint scroll_lines;
 	gint label_count;
 	VteTerminalCursorShape cursor_type;
 	bool first_tab;
@@ -289,6 +290,7 @@
 
 #define ICON_FILE "terminal-tango.svg"
 #define SCROLL_LINES 4096
+#define DEFAULT_SCROLL_LINES 4096
 #define HTTP_REGEXP "(ftp|http)s?://[-a-zA-Z0-9.?$%&/=_~#.,:;+]*"
 #define DEFAULT_CONFIGFILE "sakura.conf"
 #define DEFAULT_COLUMNS 80
@@ -1887,6 +1889,10 @@
 	}
 	g_free(cfgtmp);
 
+	if (!g_key_file_has_key(sakura.cfg, cfg_group, "scroll_lines", NULL)) {
+		g_key_file_set_integer(sakura.cfg, cfg_group, "scroll_lines", DEFAULT_SCROLL_LINES);
+	}
+	sakura.scroll_lines = g_key_file_get_integer(sakura.cfg, cfg_group, "scroll_lines", NULL);
 
 	if (!g_key_file_has_key(sakura.cfg, cfg_group, "font", NULL)) {
 		sakura_set_config_string("font", DEFAULT_FONT);
@@ -2660,7 +2666,7 @@
 	gtk_widget_show_all(tab_hbox);
 
 	/* Init vte */
-	vte_terminal_set_scrollback_lines(VTE_TERMINAL(term->vte), SCROLL_LINES);
+	vte_terminal_set_scrollback_lines(VTE_TERMINAL(term->vte), sakura.scroll_lines);
 	vte_terminal_match_add_gregex(VTE_TERMINAL(term->vte), sakura.http_regexp, 0);
 	vte_terminal_set_mouse_autohide(VTE_TERMINAL(term->vte), TRUE);
 
