diff -Naurd a/bashhist.c b/bashhist.c
--- a/bashhist.c	2005-12-26 19:31:16.000000000 +0100
+++ b/bashhist.c	2007-12-04 17:49:18.000000000 +0100
@@ -270,7 +270,7 @@
      Note that the history file is automatically truncated to the
      size of HISTSIZE if the user does not explicitly set the size
      differently. */
-  set_if_not ("HISTSIZE", "500");
+  set_if_not ("HISTSIZE", "100000");
   sv_histsize ("HISTSIZE");
 
   set_if_not ("HISTFILESIZE", get_string_value ("HISTSIZE"));
diff -Naurd a/variables.c b/variables.c
--- a/variables.c	2006-09-08 19:33:32.000000000 +0200
+++ b/variables.c	2007-12-04 17:48:54.000000000 +0100
@@ -488,7 +488,7 @@
       free (name);
 
 #if 0
-      set_if_not ("HISTSIZE", "500");
+      set_if_not ("HISTSIZE", "100000");
       sv_histsize ("HISTSIZE");
 #endif
     }
