diff -Naurdp bash-4.0.orig/config-top.h bash-4.0/config-top.h
--- bash-4.0.orig/config-top.h	2009-01-04 20:32:23.000000000 +0100
+++ bash-4.0/config-top.h	2009-10-28 20:06:45.000000000 +0100
@@ -75,7 +75,7 @@
 #define KSH_COMPATIBLE_SELECT
 
 /* System-wide .bashrc file for interactive shells. */
-/* #define SYS_BASHRC "/etc/bash.bashrc" */
+#define SYS_BASHRC "/etc/bashrc"
 
 /* System-wide .bash_logout for login shells. */
 /* #define SYS_BASH_LOGOUT "/etc/bash.bash_logout" */
