diff -Naur orig/channellists.c mod/channellists.c
--- orig/channellists.c	2008-01-21 22:08:22.000000000 +0100
+++ mod/channellists.c	2012-03-31 20:20:32.000000000 +0200
@@ -10,7 +10,6 @@
 #include <vdr/timers.h>
 #include <vdr/interface.h>
 
-#include "i18n.h"
 
 static const char *VERSION        = "0.0.4";
 static const char *DESCRIPTION    = trNOOP("Manage your channellists");
@@ -248,7 +247,6 @@
 bool cPluginChannellists::Initialize(void) {
   // Initialize any background activities the plugin shall perform.
 #if VDRVERSNUM < 10507
-  RegisterI18n(Phrases);
 #endif
   return true;
   }
diff -Naur orig/Makefile mod/Makefile
--- orig/Makefile	2008-01-21 18:12:35.000000000 +0100
+++ mod/Makefile	2012-03-31 20:18:39.000000000 +0200
@@ -47,7 +47,7 @@
 
 ### The object files (add further files here):
 
-OBJS = $(PLUGIN).o i18n.o
+OBJS = $(PLUGIN).o
 
 ### The main target:
 
