diff -Naur orig/src/CMakeLists.txt patched/src/CMakeLists.txt
--- orig/src/CMakeLists.txt	2016-12-27 08:22:24.000000000 +0100
+++ patched/src/CMakeLists.txt	2017-01-29 15:00:08.389602123 +0100
@@ -18,7 +18,7 @@
 add_subdirectory(views/collectionview)
 add_subdirectory(kconf_update)
 
-set(PERL ${PERL_EXECUTABLE})
+set(PERL "/usr/bin/perl")
 configure_file(
     ${CMAKE_CURRENT_SOURCE_DIR}/data/ical2vcal.cmake ${CMAKE_CURRENT_BINARY_DIR}/ical2vcal @ONLY
     )
