diff -Naur orig/fusion-icon.desktop patched/fusion-icon.desktop
--- orig/fusion-icon.desktop	2008-02-16 18:50:44.000000000 +0100
+++ patched/fusion-icon.desktop	2016-02-13 19:27:19.198036508 +0100
@@ -1,7 +1,8 @@
 [Desktop Entry]
 Name=Compiz Fusion Icon
 Comment=Start and manage Compiz Fusion
-Exec=fusion-icon
+Comment[hu]=Kompiz gyorsindító és konfiguráló
+Exec=fusion-icon -n
 Icon=fusion-icon
 Terminal=false
 Type=Application
