diff -Naur orig/data/stellarium.desktop patched/data/stellarium.desktop
--- orig/data/stellarium.desktop	2013-08-04 08:20:27.000000000 +0200
+++ patched/data/stellarium.desktop	2016-02-20 13:18:39.769165579 +0100
@@ -14,6 +14,7 @@
 Comment[es]=Planetario
 Comment[fr]=Planétarium
 Comment[ru]=Планетарий
+Comment[hu]=Égboltnézegető
 Keywords=astronomy;planetarium;
 
 Actions=Window;Fullscreen;Safemode;
