diff -Naur Scribus-orig/scribus.desktop Scribus/scribus.desktop
--- Scribus-orig/scribus.desktop	2012-05-05 11:09:15.000000000 +0200
+++ Scribus/scribus.desktop	2012-05-14 08:29:27.000000000 +0200
@@ -94,7 +94,7 @@
 Exec=scribus %f
 Terminal=false
 MimeType=application/vnd.scribus;
-Categories=Qt;Graphics;Publishing;
+Categories=Qt;Graphics;Publishing;X-UHU-Menu/Tools/DesktopPublishing;
 StartupWMClass=scribus
 StartupNotify=false
 X-AppInstall-Package=scribus
