diff -up syncevolution-1.5.1/src/syncevo/Logging.h.gcc6 syncevolution-1.5.1/src/syncevo/Logging.h
--- syncevolution-1.5.1/src/syncevo/Logging.h.gcc6	2014-04-25 09:55:47.000000000 +0200
+++ syncevolution-1.5.1/src/syncevo/Logging.h	2016-02-08 16:23:54.988845287 +0100
@@ -255,7 +255,7 @@ class Logger
         Handle &operator = (const Handle &other) throw ();
         ~Handle() throw ();
 
-        operator bool () const { return m_logger; }
+        operator bool () const { return m_logger.get() != 0; }
         bool operator == (Logger *logger) const { return m_logger.get() == logger; }
         Logger *get() const { return m_logger.get(); }
 
@@ -398,7 +398,7 @@ template<class L> class PushLogger : boo
         }
     }
 
-    operator bool () const { return m_logger; }
+    operator bool () const { return m_logger.get() != 0; }
 
     void reset(const Logger::Handle &logger)
     {
diff -up syncevolution-1.5.1/src/syncevo/StringDataBlob.h.gcc6 syncevolution-1.5.1/src/syncevo/StringDataBlob.h
--- syncevolution-1.5.1/src/syncevo/StringDataBlob.h.gcc6	2014-04-25 09:55:47.000000000 +0200
+++ syncevolution-1.5.1/src/syncevo/StringDataBlob.h	2016-02-08 16:23:54.988845287 +0100
@@ -53,7 +53,7 @@ class StringDataBlob : public DataBlob
 
     virtual boost::shared_ptr<std::string> getData() { return m_data; }
     virtual std::string getName() const { return m_name; }
-    virtual bool exists() const { return m_data; }
+    virtual bool exists() const { return m_data.get() != 0; }
     virtual bool isReadonly() const { return m_readonly; }
 };
 
diff -up syncevolution-1.5.1/src/synthesis/src/sysync/itemfield.cpp.gcc6 syncevolution-1.5.1/src/synthesis/src/sysync/itemfield.cpp
--- syncevolution-1.5.1/src/synthesis/src/sysync/itemfield.cpp.gcc6	2015-03-03 09:06:40.000000000 +0100
+++ syncevolution-1.5.1/src/synthesis/src/sysync/itemfield.cpp	2016-02-08 16:23:54.989845287 +0100
@@ -1488,7 +1488,7 @@ void TTimestampField::getAsRFC822date(st
       " %c%02hd%02hd",
       moffs>=0 ? '+' : '-',
       (uInt16)(abs(moffs) / MinsPerHour),
-      (uInt16)(abs(moffs) % MinsPerHour)
+      (uInt16)(((uInt16)abs(moffs)) % MinsPerHour)
     );
   }
 } // TTimestampField::getAsRFC822date
