diff -Naur orig/configure patched/configure
--- orig/configure	2014-05-12 22:13:37.000000000 +0200
+++ patched/configure	2015-12-13 04:05:19.101210724 +0100
@@ -4667,7 +4667,7 @@
 XML2_DEPS="libxml-2.0 >= 2.6.31"
 
 # Whether to enable support for SSH.
-SSH_DEPS="vte-2.90 >= 0.20 $XML2_DEPS"
+SSH_DEPS="vte-2.91 >= 0.38 $XML2_DEPS"
 
 # Check whether --enable-ssh was given.
 if test "${enable_ssh+set}" = set; then :
diff -Naur orig/plugins/ssh/vinagre-ssh-tab.c patched/plugins/ssh/vinagre-ssh-tab.c
--- orig/plugins/ssh/vinagre-ssh-tab.c	2013-07-02 16:14:35.000000000 +0200
+++ patched/plugins/ssh/vinagre-ssh-tab.c	2015-12-13 04:03:21.008625135 +0100
@@ -91,16 +91,17 @@
   arg[i++] = g_strdup (vinagre_connection_get_host (conn));
   arg[i++] = NULL;
 
-  vte_terminal_fork_command_full (VTE_TERMINAL (ssh_tab->priv->vte),
-				  0,
-				  NULL,
-				  arg,
-				  NULL,
-				  G_SPAWN_SEARCH_PATH,
-				  NULL,
-				  NULL,
-				  NULL,
-				  NULL);
+  vte_terminal_spawn_sync (VTE_TERMINAL (ssh_tab->priv->vte),
+                           0,
+                           NULL,
+                           arg,
+                           NULL,
+                           G_SPAWN_SEARCH_PATH,
+                           NULL,
+                           NULL,
+                           NULL,
+                           NULL,
+                           NULL);
   g_strfreev (arg);
   gtk_widget_show_all (GTK_WIDGET (ssh_tab));
 
@@ -123,7 +124,7 @@
 }
 
 static void
-ssh_disconnected_cb (VteTerminal *ssh, VinagreSshTab *tab)
+ssh_disconnected_cb (VteTerminal *ssh, gint exit_code, VinagreSshTab *tab)
 {
   g_signal_emit_by_name (G_OBJECT (tab), "tab-disconnected");
 }
