diff -Naur orig/etc/icons/octave.desktop.in patched/etc/icons/octave.desktop.in
--- orig/etc/icons/octave.desktop.in	2015-05-23 16:21:53.000000000 +0200
+++ patched/etc/icons/octave.desktop.in	2016-02-19 03:25:10.817672021 +0100
@@ -1,9 +1,10 @@
 [Desktop Entry]
 Name=GNU Octave
 Comment=Scientific Computing using GNU Octave
-TryExec=%OCTAVE_PREFIX%/bin/octave
-Exec=%OCTAVE_PREFIX%/bin/octave --force-gui
-Icon=%OCTAVE_IMAGEDIR%/octave-logo.svg
+Comment[hu]=Matematikai programnyelv
+TryExec=octave
+Exec=octave --force-gui
+Icon=octave-logo.svg
 Terminal=false
 Type=Application
 Categories=Education;Science;Math;
