diff -Naur octave-4.2.0-orig/etc/icons/octave.desktop.in octave-4.2.0/etc/icons/octave.desktop.in
--- octave-4.2.0-orig/etc/icons/octave.desktop.in	2016-11-13 16:16:10.000000000 +0100
+++ octave-4.2.0/etc/icons/octave.desktop.in	2016-11-19 19:55:59.120256331 +0100
@@ -1,9 +1,10 @@
 [Desktop Entry]
 Name=GNU Octave
 Comment=Scientific Computing using GNU Octave
-TryExec=%OCTAVE_PREFIX%/bin/octave
-Exec=%OCTAVE_PREFIX%/bin/octave --force-gui %f
-Icon=octave
+Comment[hu]=Matematikai programnyelv
+TryExec=octave
+Exec=octave --force-gui
+Icon=/usr/share/octave/4.0.0/imagelib/octave-logo.svg
 Terminal=false
 Type=Application
 Categories=Education;Science;Math;
