diff -urN octave-3.8.0.orig/doc/icons/octave.desktop.in octave-3.8.0/doc/icons/octave.desktop.in
--- octave-3.8.0.orig/doc/icons/octave.desktop.in	2013-12-27 22:57:41.000000000 +0100
+++ octave-3.8.0/doc/icons/octave.desktop.in	2014-02-18 12:42:08.294046840 +0100
@@ -1,10 +1,12 @@
 [Desktop Entry]
+Version=1.1
 Name=GNU Octave
 Comment=Scientific Computing using GNU Octave
-TryExec=%OCTAVE_PREFIX%/bin/octave
-Exec=%OCTAVE_PREFIX%/bin/octave
-Icon=%OCTAVE_IMAGEDIR%/octave-logo.svg
+Comment[hu]=Matematikai programnyelv
+TryExec=octave
+Exec=octave
+Icon=octave-logo.svg
 Terminal=true
 Type=Application
-Categories=Education;Science;Math;
+Categories=Education;Science;Math;X-UHU-Menu/Edutainment/Mathematics;
 StartupNotify=false
