diff -Naur orig/poedit.desktop patched/poedit.desktop
--- orig/poedit.desktop	2017-02-18 20:16:40.134741323 +0100
+++ patched/poedit.desktop	2017-03-31 23:33:37.354884307 +0200
@@ -7,6 +7,7 @@
 GenericName[es]=Editor de catálogos de Gettext
 GenericName[eu]=Gettext katalogo editatzailea
 GenericName[fr]=Outil de traduction
+GenericName[hu]=Gettext fájlokat szerkesztő program
 GenericName[lt]=Gettext katalogų redaktorius
 GenericName[nl]=Gettext-vertaalhulpmiddel
 GenericName[pl]=Edytor pakietów gettext
