diff -Naur orig/poedit.desktop patched/poedit.desktop
--- orig/poedit.desktop	2017-03-14 15:47:27.134741323 +0100
+++ patched/poedit.desktop	2017-03-31 21:07:15.281767315 +0200
@@ -9,6 +9,7 @@
 GenericName[es]=Editor de catálogos de Gettext
 GenericName[eu]=Gettext katalogo editatzailea
 GenericName[fr]=Éditeur de catalogues Gettext
+GenericName[hu]=Gettext fájlokat szerkesztő program
 GenericName[it]=Editor di cataloghi Gettext
 GenericName[kk]=Gettext каталогтар түзетушісі
 GenericName[ky]=Gettext каталогдор редактору
