diff -urN dpkg-1.17.9.orig/configure.ac dpkg-1.17.9/configure.ac
--- dpkg-1.17.9.orig/configure.ac	2014-04-30 06:47:33.000000000 +0200
+++ dpkg-1.17.9/configure.ac	2014-06-05 16:09:39.726731120 +0200
@@ -32,7 +32,7 @@
               [system logging directory [LOCALSTATEDIR/log]])
 
 # Set default dpkg-deb values
-DPKG_DEB_COMPRESSOR([xz])
+DPKG_DEB_COMPRESSOR([gzip])
 DPKG_DEB_PROG_TAR
 
 # Checks for programs.
