diff -Naurdp a/lib/dpkg/dpkg.h b/lib/dpkg/dpkg.h
--- a/lib/dpkg/dpkg.h	2014-05-30 18:30:50.000000000 +0200
+++ b/lib/dpkg/dpkg.h	2014-06-10 11:06:57.437081456 +0200
@@ -73,6 +73,7 @@ DPKG_BEGIN_DECLS
 #define POSTINSTFILE       "postinst"
 #define PRERMFILE          "prerm"
 #define POSTRMFILE         "postrm"
+#define POSTRMCUSTOMFILE   "postrm-custom"
 #define TRIGGERSCIFILE     "triggers"
 
 #define STATUSFILE        "status"
diff -Naurdp a/src/remove.c b/src/remove.c
--- a/src/remove.c	2014-06-05 19:56:33.000000000 +0200
+++ b/src/remove.c	2014-06-10 11:07:47.397079228 +0200
@@ -207,7 +207,8 @@ removal_bulk_remove_file(const char *fil
 {
   /* We need the postrm and list files for --purge. */
   if (strcmp(filetype, LISTFILE) == 0 ||
-      strcmp(filetype, POSTRMFILE) == 0)
+      strcmp(filetype, POSTRMFILE) == 0 ||
+      strcmp(filetype, POSTRMCUSTOMFILE) == 0)
     return;
 
   debug(dbg_stupidlyverbose, "removal_bulk info not postrm or list");
@@ -640,6 +641,12 @@ void removal_bulk(struct pkginfo *pkg) {
           filename);
     if (unlink(filename) && errno != ENOENT)
       ohshite(_("can't remove old postrm script"));
+      
+    filename = pkg_infodb_get_file(pkg, &pkg->installed, POSTRMCUSTOMFILE);
+    debug(dbg_general, "removal_bulk purge done, removing postrm-custom `%s'",
+          filename);
+    if (unlink(filename) && errno != ENOENT)
+      ohshite(_("can't remove old postrm-custom script"));
 
     pkg_set_status(pkg, PKG_STAT_NOTINSTALLED);
     pkg_set_want(pkg, PKG_WANT_UNKNOWN);
