diff -Naur orig/lib/dpkg/dpkg.h patched/lib/dpkg/dpkg.h
--- orig/lib/dpkg/dpkg.h	2016-06-29 02:24:39.000000000 +0200
+++ patched/lib/dpkg/dpkg.h	2016-07-21 10:03:29.565564013 +0200
@@ -71,6 +71,7 @@
 #define POSTINSTFILE       "postinst"
 #define PRERMFILE          "prerm"
 #define POSTRMFILE         "postrm"
+#define POSTRMCUSTOMFILE   "postrm-custom"
 #define TRIGGERSCIFILE     "triggers"
 
 #define STATUSFILE        "status"
diff -Naur orig/src/remove.c patched/src/remove.c
--- orig/src/remove.c	2016-07-03 21:02:19.000000000 +0200
+++ patched/src/remove.c	2016-07-21 10:08:59.575528104 +0200
@@ -206,7 +206,8 @@
 {
   /* We need the postrm and list files for --purge. */
   if (strcmp(filetype, LISTFILE) == 0 ||
-      strcmp(filetype, POSTRMFILE) == 0)
+      strcmp(filetype, POSTRMFILE) == 0 ||
+      strcmp(filetype, POSTRMCUSTOMFILE) == 0)
     return;
 
   debug(dbg_stupidlyverbose, "removal_bulk info not postrm or list");
@@ -656,6 +657,12 @@
           filename);
     if (unlink(filename) && errno != ENOENT)
       ohshite(_("can't remove old postrm script"));
+      
+    filename = pkg_infodb_get_file(pkg, &pkg->installed, POSTRMCUSTOMFILE);
+    debug(dbg_general, "removal_bulk purge done, removing postrm-custom '%s'",
+          filename);
+    if (unlink(filename) && errno != ENOENT)
+      ohshite(_("can't remove old postrm-custom script"));
 
     pkg_set_status(pkg, PKG_STAT_NOTINSTALLED);
     pkg_set_want(pkg, PKG_WANT_UNKNOWN);
