diff -Naur orig/dpkg-deb/build.c patched/dpkg-deb/build.c
--- orig/dpkg-deb/build.c	2015-09-17 16:58:38.000000000 +0200
+++ patched/dpkg-deb/build.c	2015-09-26 16:56:00.277219323 +0200
@@ -484,8 +484,8 @@
     pkg = check_control_area(dir);
     if (debar == NULL)
       debar = gen_dest_pathname_from_pkg(dest, pkg);
-    printf(_("dpkg-deb: building package '%s' in '%s'.\n"),
-           pkg->set->name, debar);
+    //printf(_("dpkg-deb: building package '%s' in '%s'.\n"),
+    //       pkg->set->name, debar);
   }
   m_output(stdout, _("<standard output>"));
 
