diff -Naur dpkg-1.18.4-orig/dpkg-deb/build.c dpkg-1.18.4/dpkg-deb/build.c
--- dpkg-1.18.4-orig/dpkg-deb/build.c	2015-12-12 21:49:37.000000000 +0100
+++ dpkg-1.18.4/dpkg-deb/build.c	2016-02-14 16:40:54.370648450 +0100
@@ -554,7 +554,7 @@
     m_dup2(p1[1],1); close(p1[0]); close(p1[1]);
     if (chdir(ctrldir))
       ohshite(_("failed to chdir to '%.255s'"), ctrldir);
-    execlp(TAR, "tar", "-cf", "-", "--format=gnu", ".", NULL);
+    execlp(TAR, "tar", "-cf", "-", "--format=gnu", "--owner=0", "--group=0", ".", NULL);
     ohshite(_("unable to execute %s (%s)"), "tar -cf", TAR);
   }
   close(p1[1]);
