diff -Naur orig/dpkg-deb/build.c patched/dpkg-deb/build.c
--- orig/dpkg-deb/build.c	2015-09-26 16:56:00.277219323 +0200
+++ patched/dpkg-deb/build.c	2015-09-26 16:57:08.833559276 +0200
@@ -504,7 +504,7 @@
       ohshite(_("failed to chdir to '%.255s'"), dir);
     if (chdir(BUILDCONTROLDIR))
       ohshite(_("failed to chdir to '%.255s'"), ".../DEBIAN");
-    execlp(TAR, "tar", "-cf", "-", "--format=gnu", ".", NULL);
+    execlp(TAR, "tar", "-cf", "-", "--format=gnu", "--owner=0", "--group=0", ".", NULL);
     ohshite(_("unable to execute %s (%s)"), "tar -cf", TAR);
   }
   close(p1[1]);
