diff -Naur orig/lib/dpkg/parse.c patched/lib/dpkg/parse.c
--- orig/lib/dpkg/parse.c	2015-09-16 14:50:15.000000000 +0200
+++ patched/lib/dpkg/parse.c	2015-09-26 17:02:59.859299918 +0200
@@ -188,7 +188,7 @@
       (pkg->status != PKG_STAT_NOTINSTALLED &&
        pkg->status != PKG_STAT_HALFINSTALLED)) {
     parse_ensure_have_field(ps, &pkgbin->description, "description");
-    parse_ensure_have_field(ps, &pkgbin->maintainer, "maintainer");
+    //parse_ensure_have_field(ps, &pkgbin->maintainer, "maintainer");
     parse_must_have_field(ps, pkgbin->version.version, "version");
   }
 
