diff -Naur timeslide-0.4-orig/Makefile timeslide-0.4/Makefile
--- timeslide-0.4-orig/Makefile	2003-09-01 17:32:12.000000000 +0200
+++ timeslide-0.4/Makefile	2017-04-18 20:48:43.820859318 +0200
@@ -1,5 +1,5 @@
 libtimeslide.so: libtimeslide.c
-	gcc -Wall -shared -ldl -o libtimeslide.so libtimeslide.c
+	gcc -fPIC -Wall -shared -ldl -o libtimeslide.so libtimeslide.c
 
 install: libtimeslide.so
 	mkdir -p $(DESTDIR)/usr/{bin,lib}
