diff -Naur orig/shell/containmentconfigview.cpp patched/shell/containmentconfigview.cpp
--- orig/shell/containmentconfigview.cpp	2016-03-02 13:17:41.000000000 +0100
+++ patched/shell/containmentconfigview.cpp	2016-03-15 13:04:06.906990954 +0100
@@ -121,7 +121,7 @@
 {
     if (!m_wallpaperConfigModel) {
         m_wallpaperConfigModel = new PlasmaQuick::ConfigModel(this);
-        QStringList dirs(QStandardPaths::locateAll(QStandardPaths::GenericDataLocation, PLASMA_RELATIVE_DATA_INSTALL_DIR "/wallpapers", QStandardPaths::LocateDirectory));
+        QStringList dirs(QStandardPaths::locateAll(QStandardPaths::GenericDataLocation, PLASMA_RELATIVE_DATA_INSTALL_DIR "/backgrounds", QStandardPaths::LocateDirectory));
         KPackage::Package pkg = KPackage::PackageLoader::self()->loadPackage(QStringLiteral("KPackage/Generic"));
         QStringList platform = KDeclarative::KDeclarative::runtimePlatform();
         if (!platform.isEmpty()) {
@@ -199,7 +199,7 @@
 
         //we have to construct an independent ConfigPropertyMap when we want to configure wallpapers that are not the current one
         KPackage::Package pkg = KPackage::PackageLoader::self()->loadPackage(QStringLiteral("Plasma/Generic"));
-        pkg.setDefaultPackageRoot(PLASMA_RELATIVE_DATA_INSTALL_DIR "/wallpapers");
+        pkg.setDefaultPackageRoot(PLASMA_RELATIVE_DATA_INSTALL_DIR "/backgrounds");
         pkg.setPath(wallpaper);
         QFile file(pkg.filePath("config", QStringLiteral("main.xml")));
         KConfigGroup cfg = m_containment->config();
diff -Naur orig/wallpapers/image/backgroundlistmodel.cpp patched/wallpapers/image/backgroundlistmodel.cpp
--- orig/wallpapers/image/backgroundlistmodel.cpp	2016-03-02 13:17:41.000000000 +0100
+++ patched/wallpapers/image/backgroundlistmodel.cpp	2016-03-15 13:06:25.647678932 +0100
@@ -121,7 +121,7 @@
         processPaths(selected);
     }
 
-    const QStringList dirs = QStandardPaths::locateAll(QStandardPaths::GenericDataLocation, QStringLiteral("wallpapers/"), QStandardPaths::LocateDirectory);
+    const QStringList dirs = QStandardPaths::locateAll(QStandardPaths::GenericDataLocation, QStringLiteral("backgrounds/"), QStandardPaths::LocateDirectory);
     qDebug() << " WP : -------" << dirs;
 
     BackgroundFinder *finder = new BackgroundFinder(m_wallpaper.data(), dirs);
@@ -387,7 +387,7 @@
     break;
 
     case RemovableRole: {
-        QString localWallpapers = QStandardPaths::writableLocation(QStandardPaths::GenericDataLocation) + "/wallpapers/";
+        QString localWallpapers = QStandardPaths::writableLocation(QStandardPaths::GenericDataLocation) + "/backgrounds/";
         QString path = b.filePath("preferred");
         return path.startsWith(localWallpapers) || m_removableWallpapers.contains(path);
     }
diff -Naur orig/wallpapers/image/image.cpp patched/wallpapers/image/image.cpp
--- orig/wallpapers/image/image.cpp	2016-03-02 13:17:41.000000000 +0100
+++ patched/wallpapers/image/image.cpp	2016-03-15 13:09:26.836577399 +0100
@@ -136,7 +136,7 @@
 
     if (m_mode == SlideShow) {
         if (m_slidePaths.isEmpty()) {
-            m_slidePaths << QStandardPaths::locateAll(QStandardPaths::GenericDataLocation, QStringLiteral("share/wallpapers"), QStandardPaths::LocateDirectory);
+            m_slidePaths << QStandardPaths::locateAll(QStandardPaths::GenericDataLocation, QStringLiteral("share/backgrounds"), QStandardPaths::LocateDirectory);
         }
 
         QTimer::singleShot(200, this, &Image::startSlideshow);
@@ -334,7 +334,7 @@
     m_slidePaths.removeAll(QString());
 
     if (m_slidePaths.isEmpty()) {
-        m_slidePaths << QStandardPaths::locateAll(QStandardPaths::GenericDataLocation, QStringLiteral("share/wallpapers"), QStandardPaths::LocateDirectory);
+        m_slidePaths << QStandardPaths::locateAll(QStandardPaths::GenericDataLocation, QStringLiteral("share/backgrounds"), QStandardPaths::LocateDirectory);
     }
 
     if (m_mode == SlideShow) {
@@ -443,7 +443,7 @@
         }
     } else {
         //if it's not an absolute path, check if it's just a wallpaper name
-        const QString path = QStandardPaths::locate(QStandardPaths::GenericDataLocation, QLatin1String("wallpapers/") + QString(m_wallpaper + QString::fromLatin1("/metadata.desktop")));
+        const QString path = QStandardPaths::locate(QStandardPaths::GenericDataLocation, QLatin1String("backgrounds/") + QString(m_wallpaper + QString::fromLatin1("/metadata.desktop")));
 
         if (!path.isEmpty()) {
             QDir dir(path);
@@ -485,7 +485,7 @@
             path = url.path();
         } else {
             path = QStandardPaths::locate(QStandardPaths::GenericDataLocation,
-                                          QLatin1String("wallpapers/") + url.path(),
+                                          QLatin1String("backgrounds/") + url.path(),
                                           QStandardPaths::LocateDirectory);
         }
 
@@ -493,7 +493,7 @@
             return;
         }
     } else {
-        QString wallpaperPath = QStandardPaths::writableLocation(QStandardPaths::GenericDataLocation) + QLatin1String("wallpapers/") + url.path();
+        QString wallpaperPath = QStandardPaths::writableLocation(QStandardPaths::GenericDataLocation) + QLatin1String("backgrounds/") + url.path();
 
         if (!wallpaperPath.isEmpty()) {
             KIO::FileCopyJob *job = KIO::file_copy(url, QUrl(wallpaperPath), -1, KIO::HideProgressInfo);
@@ -792,7 +792,7 @@
 //FIXME: we have to save the configuration also when the dialog cancel button is clicked.
 void Image::removeWallpaper(QString name)
 {
-    QString localWallpapers = QStandardPaths::writableLocation(QStandardPaths::GenericDataLocation) + "/wallpapers/";
+    QString localWallpapers = QStandardPaths::writableLocation(QStandardPaths::GenericDataLocation) + "/backgrounds/";
     QUrl nameUrl(name);
 
     //Package plugin name
