diff -Naur orig/shell/shellcorona.cpp mod/shell/shellcorona.cpp
--- orig/shell/shellcorona.cpp	2016-09-29 13:35:01.000000000 +0200
+++ mod/shell/shellcorona.cpp	2016-10-11 15:47:31.455150390 +0200
@@ -309,7 +309,7 @@
 
     connect(m_activityController, &KActivities::Controller::serviceStatusChanged, this, &ShellCorona::load, Qt::UniqueConnection);
 
-    if (m_activityController->serviceStatus() == KActivities::Controller::Running) {
+    if (m_activityController->serviceStatus() != KActivities::Controller::Running) {
         load();
     }
 }
@@ -611,7 +611,7 @@
 void ShellCorona::load()
 {
     if (m_shell.isEmpty() ||
-        m_activityController->serviceStatus() != KActivities::Controller::Running) {
+        m_activityController->serviceStatus() == KActivities::Controller::Running) {
         return;
     }
 
