diff -Naur ORIG/build.sh UHU/build.sh
--- ORIG/build.sh	2012-11-10 10:11:39.000000000 +0000
+++ UHU/build.sh	2014-05-04 09:49:00.094127244 +0000
@@ -13,7 +13,7 @@
 # default - compiling DC only (using by default)
 
 # path to lazbuild
-export lazbuild=$(which lazbuild)
+export lazbuild=$(which lazbuild)" "$(echo "--lazarusdir=$UB_COMPILEDIR")
 
 # Set up widgetset: gtk or gtk2 or qt
 # Set up processor architecture: i386 or x86_64
