--- a/data/sol.desktop.in.in.orig	2013-03-01 19:19:37.000000000 +0100
+++ a/data/sol.desktop.in.in	2013-03-01 19:19:53.000000000 +0100
@@ -5,7 +5,7 @@
 Icon=gnome-aisleriot
 Terminal=false
 Type=Application
-Categories=GNOME;GTK;Game;CardGame;
+Categories=GNOME;GTK;Game;CardGame;X-UHU-Menu/Games/Cards;
 _Keywords=solitaire;cards;klondike;spider;freecell;patience;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=aisleriot
