diff -Naur fpc-3.0.0.i386-linux-orig/install.sh fpc-3.0.0.i386-linux/install.sh
--- fpc-3.0.0.i386-linux-orig/install.sh	2015-11-24 22:22:21.000000000 +0100
+++ fpc-3.0.0.i386-linux/install.sh	2016-04-02 21:29:30.175524609 +0200
@@ -208,70 +208,6 @@
 
 OSNAME=`uname -s | tr A-Z a-z`
 
-case "$OSNAME" in
-  haiku)
-     # Install in /boot/common or /boot/home/config ?
-     if checkpath /boot/common/bin; then
-         PREFIX=/boot/common
-     else
-         PREFIX=/boot/home/config
-     fi
-     # If we can't write on prefix, we are probably 
-     # on Haiku with package management system.
-     # In this case, we have to install fpc in the non-packaged subdir
-     if [ ! -w "$PREFIX" ]; then
-     	PREFIX="$PREFIX/non-packaged"
-     fi
-  ;;
-  freebsd)
-     PREFIX=/usr/local
-  ;;
-  sunos)
-     # Check if GNU llinker is recent enough, version 2.21 is needed at least
-     GNU_LD=`which gld`
-     supported_emulations=`"$GNU_LD" --target-help | sed -n "s|.*supported emulations:||p" `
-     supports_elf_i386_sol2=`echo $supported_emulations | grep -w elf_i386_sol2 `
-     supports_elf_x86_64_sol2=`echo $supported_emulations | grep -w elf_x86_64_sol2 `
-     if [ "$supports_elf_i386_sol2" = "" ]; then
-       echo -n "GNU linker $GNU_LD does not support elf_i386_sol2 emulation, please consider "
-       echo "upgrading binutils package to at least version 2.21"
-     elif [ "$supports_elf_x86_64_sol2" = "" ]; then
-       echo -n "GNU linker $GNU_LD does not support elf_x86_64_sol2 emulation, please consider "
-       echo "upgrading binutils package to at least version 2.21"
-     fi
-     PREFIX=/usr/local
-     # Use GNU tar if present
-     if [ "`which gtar`" != "" ]; then
-       TAR=`which gtar`
-     fi
-  ;;
-  *)
-     # Install in /usr/local or /usr ?
-     if checkpath /usr/local/bin; then
-         PREFIX=/usr/local
-     else
-         PREFIX=/usr
-     fi
-  ;;
-esac
-
-# If we can't write on prefix, select subdir of home dir
-if [ ! -w "$PREFIX" ]; then
-  PREFIX="$HOME/fpc-$VERSION"
-fi
-
-case "$OSNAME" in
-  haiku)
-     ask "Install prefix (/boot/common or /boot/home/config) " PREFIX
-  ;;
-  *)
-     ask "Install prefix (/usr or /usr/local) " PREFIX
-  ;;
-esac
-
-# Support ~ expansion
-PREFIX=`eval echo $PREFIX`
-export PREFIX
 makedirhierarch "$PREFIX"
 
 # Set some defaults.
