diff --git a/src/mlib/filesystem.cpp b/src/mlib/filesystem.cpp
index 7ae7162..774bd9d 100644
--- a/src/mlib/filesystem.cpp
+++ b/src/mlib/filesystem.cpp
@@ -32,7 +32,7 @@
 #include <boost/filesystem/path_traits.hpp> // boost::filesystem::convert()
 #else
 #define BOOST_FS_3 boost::filesystem3
-#include <boost/filesystem/v3/path_traits.hpp>
+#include <boost/filesystem/path_traits.hpp>
 #endif 
 
 
