--- x.orig/src/mgui/bombono-dvd.desktop	2010-04-16 00:42:06.000000000 +0200
+++ x/src/mgui/bombono-dvd.desktop	2010-05-04 19:47:58.000000000 +0200
@@ -2,8 +2,10 @@
 Type=Application
 Name=Bombono DVD
 Comment=DVD authoring program
+Comment[hu]=DVD készítő program
 Icon=bombono-dvd
 Exec=bombono-dvd
-Categories=Video;AudioVideo;DiscBurning;GTK;
+Categories=GTK;AudioVideo;Video;DiscBurning;
 Terminal=false
 MimeType=application/x-bombono-dvd;
+StartupNotify=true
