diff -Naur orig/SConstruct patched/SConstruct
--- orig/SConstruct	2013-03-07 22:20:45.000000000 +0100
+++ patched/SConstruct	2014-03-23 18:00:19.917966619 +0100
@@ -565,14 +565,6 @@
 bin_prefix  = MakeEndPrefix('BIN_PREFIX',  prefix, 'bin')
 mgui_env.Alias('install', bin_prefix)
 
-def InstallDir(rel_dst_path, src_dir):
-    data_prefix = MakeEndPrefix('', *[prefix, rel_dst_path], add_to_dict=0)
-    mgui_env.Alias('install', data_prefix)
-    BV.InstallDir(mgui_env, data_prefix, src_dir)
-
-InstallDir('share/bombono', "resources")
-InstallDir('share', "docs/man")
-
 user_options_dict['XGETTEXT_SOURCES'] = []
 def AddSourcesForXgettext(src_files):
     user_options_dict['XGETTEXT_SOURCES'] += [ File(src) for src in src_files ]
