--- a/fc-cache/fc-cache.c.orig	2013-08-26 04:46:48.000000000 +0200
+++ a/fc-cache/fc-cache.c	2013-10-02 19:10:33.000000000 +0200
@@ -69,6 +69,7 @@
     {"really-force", 0, 0, 'r'},
     {"sysroot", 0, 0, 'y'},
     {"system-only", 0, 0, 's'},
+    {"no-sleep", 0, 0, 'n'},
     {"version", 0, 0, 'V'},
     {"verbose", 0, 0, 'v'},
     {"help", 0, 0, 'h'},
@@ -86,10 +87,10 @@
 {
     FILE *file = error ? stderr : stdout;
 #if HAVE_GETOPT_LONG
-    fprintf (file, "usage: %s [-frsvVh] [-y SYSROOT] [--force|--really-force] [--sysroot=SYSROOT] [--system-only] [--verbose] [--version] [--help] [dirs]\n",
+    fprintf (file, "usage: %s [-frsnvVh] [-y SYSROOT] [--force|--really-force] [--sysroot=SYSROOT] [--system-only] [--no-sleep] [--verbose] [--version] [--help] [dirs]\n",
 	     program);
 #else
-    fprintf (file, "usage: %s [-frsvVh] [-y SYSROOT] [dirs]\n",
+    fprintf (file, "usage: %s [-frsnvVh] [-y SYSROOT] [dirs]\n",
 	     program);
 #endif
     fprintf (file, "Build font information caches in [dirs]\n"
@@ -99,6 +100,7 @@
     fprintf (file, "  -f, --force              scan directories with apparently valid caches\n");
     fprintf (file, "  -r, --really-force       erase all existing caches, then rescan\n");
     fprintf (file, "  -s, --system-only        scan system-wide directories only\n");
+    fprintf (file, "  -n, --no-sleep           do not sleep before exit\n");
     fprintf (file, "  -y, --sysroot=SYSROOT    prepend SYSROOT to all paths for scanning\n");
     fprintf (file, "  -v, --verbose            display status information while busy\n");
     fprintf (file, "  -V, --version            display font config version and exit\n");
@@ -107,6 +109,7 @@
     fprintf (file, "  -f         (force)   scan directories with apparently valid caches\n");
     fprintf (file, "  -r,   (really force) erase all existing caches, then rescan\n");
     fprintf (file, "  -s         (system)  scan system-wide directories only\n");
+    fprintf (file, "  -n        (no-sleep) do not sleep beofre exit");
     fprintf (file, "  -y SYSROOT (sysroot) prepend SYSROOT to all paths for scanning\n");
     fprintf (file, "  -v         (verbose) display status information while busy\n");
     fprintf (file, "  -V         (version) display font config version and exit\n");
@@ -272,6 +275,7 @@
     FcBool	force = FcFalse;
     FcBool	really_force = FcFalse;
     FcBool	systemOnly = FcFalse;
+    FcBool	nosleep = FcFalse;
     FcConfig	*config;
     FcChar8     *sysroot = NULL;
     int		i;
@@ -281,9 +285,9 @@
     int		c;
 
 #if HAVE_GETOPT_LONG
-    while ((c = getopt_long (argc, argv, "frsy:Vvh", longopts, NULL)) != -1)
+    while ((c = getopt_long (argc, argv, "frsny:Vvh", longopts, NULL)) != -1)
 #else
-    while ((c = getopt (argc, argv, "frsy:Vvh")) != -1)
+    while ((c = getopt (argc, argv, "frsny:Vvh")) != -1)
 #endif
     {
 	switch (c) {
@@ -296,6 +300,9 @@
 	case 's':
 	    systemOnly = FcTrue;
 	    break;
+	case 'n':
+	    nosleep = FcTrue;
+	    break;
 	case 'y':
 	    sysroot = FcStrCopy ((const FcChar8 *)optarg);
 	    break;
@@ -389,7 +396,8 @@
     FcConfigDestroy (config);
     FcFini ();
     if (changed)
-	sleep (2);
+        if (!nosleep)
+	    sleep (2);
     if (verbose)
 	printf ("%s: %s\n", argv[0], ret ? "failed" : "succeeded");
     return ret;
