diff -Naur orig/src/common patched/src/common
--- orig/src/common	2015-06-15 23:58:10.288720326 +0200
+++ patched/src/common	2015-06-15 23:58:41.908877121 +0200
@@ -10,6 +10,8 @@
    red="$(echo -ne '\e[01;31m')"
   blue="$(echo -ne '\e[01;34m')"
  green="$(echo -ne '\e[00;32m')"
+ apple="$(echo -ne '\e[00;36m')"
+yellow="$(echo -ne '\e[01;33m')"
 normal="$(echo -ne '\e[00;00m')"
 
 # Azert jobb igy ellenorizni, mert lehet hogy a kimenet at van iranyitva,
@@ -29,6 +31,8 @@
 redecho   () { colorecho "$red"   "$@"; }
 blueecho  () { colorecho "$blue"  "$@"; }
 greenecho () { colorecho "$green" "$@"; }
+appleecho ()  { colorecho "$apple" "$@"; }
+yellowecho () { colorecho "$yellow" "$@"; }
 
 # Hibakezeles -----------------------------------------------------------------
 errors=0
