diff -Naur orig/src/uhubuild-check patched/src/uhubuild-check
--- orig/src/uhubuild-check	2015-12-29 01:47:08.415005833 +0100
+++ patched/src/uhubuild-check	2015-12-29 01:50:18.591948869 +0100
@@ -287,7 +287,7 @@
 if check distribution; then
 reformat 0 < distribution > $UHUBUILD_SRC2DIR/distribution
 	if [ $(wc -l < $UHUBUILD_SRC2DIR/distribution) -ne 1 ]; then
-		error "Hibás distribution fájl!"
+		error "A distribution fájl nem egysoros!"
 	fi
 fi
 
@@ -382,13 +382,13 @@
 		reformat 0 < $UHUBUILD_SRC2DIR/homepage.tmp > "$UHUBUILD_SRC2DIR/homepage"
 		rm $UHUBUILD_SRC2DIR/homepage.tmp
 		if [ "$(wc -l < "$UHUBUILD_SRC2DIR/homepage")" -ne 1 ]; then
-			error "a homepage nem egysoros!"
+			error "A homepage nem egysoros!"
 		fi
 		case "$(cat homepage)" in
 			ftp://*|http://*|https://*)
 				;;
 			*)
-				error "ismeretlen URL formátum a homepage fileban."
+				error "Ismeretlen URL formátum a homepage fájlban."
 				;;
 		esac
 	else
@@ -405,7 +405,7 @@
 		reformat 0 < $UHUBUILD_SRC2DIR/maintainer.tmp > $UHUBUILD_SRC2DIR/maintainer
 		rm $UHUBUILD_SRC2DIR/maintainer.tmp
 		if [ $(wc -l < $UHUBUILD_SRC2DIR/maintainer) -ne 1 ]; then
-			error "Hibás maintainer fájl!"
+			error "A maintainer fájl nem egysoros!"
 		fi
 	else
 		mv $UHUBUILD_SRC2DIR/maintainer.tmp $UHUBUILD_SRC2DIR/maintainer
@@ -440,10 +440,10 @@
 if check release; then
 	reformat 0 < release > $UHUBUILD_SRC2DIR/release
 	if [ "$(wc -l < "$UHUBUILD_SRC2DIR"/release)" -ne 1 ]; then
-		error "Hibás release fájl!"
+		error "A release fájl nem egysoros!"
 	fi
 	if ! grep -q '^[0-9]*$' < $UHUBUILD_SRC2DIR/release; then
-		error "Hibás release fájl!"
+		error "A release fájl csak számot tartalmazhat!"
 	fi
 fi
 
@@ -453,7 +453,7 @@
 if check sourcename; then
 	reformat 0 < sourcename > $UHUBUILD_SRC2DIR/sourcename
 	if [ "$(wc -l < "$UHUBUILD_SRC2DIR"/sourcename)" -ne 1 ]; then
-		error "Hibás sourcename fájl!"
+		error "A sourcename fájl nem egysoros!"
 	fi
 fi
 
@@ -463,11 +463,18 @@
 if check version; then
 	reformat 0 < version > $UHUBUILD_SRC2DIR/version
 	if [ $(wc -l < $UHUBUILD_SRC2DIR/version) -ne 1 ]; then
-		error "Hibás version fájl!"
+		error "A version fájl nem egysoros!"
 	fi
-	if grep -q -- '-' < $UHUBUILD_SRC2DIR/version; then
-		error "Nem lehet '-' jel a version fájlban!"
+	if ! grep -q '^[0-9]' < $UHUBUILD_SRC2DIR/version; then
+		error "A verzió csak számmal kezdődhet!"
 	fi
+#	allowed=".+~:"
+	cp $UHUBUILD_SRC2DIR/version $UHUBUILD_SRC2DIR/version.tmp
+	sed -i -e 's/[A-Za-z0-9.+~:]//g' $UHUBUILD_SRC2DIR/version.tmp
+	if [[ `stat -c%s $UHUBUILD_SRC2DIR/version.tmp` > 1 ]]; then
+		error "Nem megengedett karakter(ek) a version fájlban!"
+	fi
+	rm -f $UHUBUILD_SRC2DIR/version.tmp
 fi
 
 ##############
@@ -479,7 +486,7 @@
 		reformat 0 < $UHUBUILD_SRC2DIR/vendor.tmp > $UHUBUILD_SRC2DIR/vendor
 		rm $UHUBUILD_SRC2DIR/vendor.tmp
 		if [ $(wc -l < $UHUBUILD_SRC2DIR/vendor) -ne 1 ]; then
-			error "Hibás vendor fájl!"
+			error "A vendor fájl nem egysoros!"
 		fi
 	else
 		mv $UHUBUILD_SRC2DIR/vendor.tmp $UHUBUILD_SRC2DIR/vendor
@@ -629,7 +636,7 @@
 				""|"<"|"<<"|"<="|"="|">="|">>"|">")
 					;;
 				*)
-					error "Ismeretlen verzió-operátor ($op) a $pkg/depends fileban."
+					error "Ismeretlen verzió-operátor ($op) a $pkg/depends fájlban."
 					;;
 			esac
 		done < $UHUBUILD_SRC2DIR/$pkg/depends
