diff -Naur orig/src/uhubuild-chrootbuild patched/src/uhubuild-chrootbuild
--- orig/src/uhubuild-chrootbuild	2014-04-14 13:33:44.000000000 +0200
+++ patched/src/uhubuild-chrootbuild	2014-05-24 18:50:25.841321119 +0200
@@ -211,6 +211,7 @@
 echo -n 'dev '
 mkdir -p "$UHUBUILD_CHROOTDIR/dev"
 mount --bind /dev  "$UHUBUILD_CHROOTDIR/dev"
+mount -t tmpfs -o rw,nosuid,nodev,mode=1777 none "$UHUBUILD_CHROOTDIR"/dev/shm
 
 echo -n $'\b, proc '
 mkdir -p "$UHUBUILD_CHROOTDIR/proc"
