diff -Naur orig/src/steps/menu patched/src/steps/menu
--- orig/src/steps/menu	2014-05-07 09:58:01.000000000 +0200
+++ patched/src/steps/menu	2014-05-31 01:48:00.565100395 +0200
@@ -26,7 +26,8 @@
 			fgrep -q '${' "$f" \
 			  && error "$f: maradt benne '\${', szar a configure vagy a Makefile!" \
 			  || true
-			desktop-file-validate "$f" || warn "$f: hibás desktop fájl!" \
+			desktop-file-validate "$f" | grep 'error:' \
+			  && error "$f: hibás desktop fájl!" \
 			  || true
 		done
 	done
