diff -Naur orig/src/steps/control patched/src/steps/control
--- orig/src/steps/control	2016-01-31 18:20:33.342494681 +0100
+++ patched/src/steps/control	2016-01-31 19:46:16.071996092 +0100
@@ -87,7 +87,11 @@
 			echo
 		fi
 
-		echo "Distribution: ${UB_DISTRIBUTION/ dev/ 3}"
+		if [ "$UHUBUILD_DISTR" = "dev" ]; then
+			echo "Distribution: ${UB_DISTRIBUTION/ dev/ 3.1}"
+		else
+			echo "Distribution: UHU Linux ${UHUBUILD_DISTR}"
+		fi
 		if [ -f "$UB_SRCDIR/vendor" ]; then
 			echo "Vendor: $(cat "$UB_SRCDIR/vendor")"
 		fi
