diff -Naur orig/src/steps/cleanup patched/src/steps/cleanup
--- orig/src/steps/cleanup	2018-01-27 20:55:45.000000000 +0100
+++ patched/src/steps/cleanup	2018-01-28 17:04:23.900151896 +0100
@@ -8,14 +8,5 @@
 	find "$UB_INSTALLDIR/usr/lib/perl5" -name "perllocal.pod" -type f -delete
 fi
 
-# gtk-doc
-if [ ! -f "$UB_INSTALLDIR/usr/bin/gtkdocize" ] ; then
-if [ -d "$UB_INSTALLDIR/usr/share/gtk-doc" ]; then
-	rm -Rf "$UB_INSTALLDIR/usr/share/gtk-doc"
-fi
-fi
 # xvpics
 find "$UB_INSTALLDIR" -name .xvpics -delete
-
-# .la
-find "$UB_INSTALLDIR" -name *.la -delete
