diff -Naur orig/demos/Makefile.am patched/demos/Makefile.am
--- orig/demos/Makefile.am	2005-07-25 13:23:12.000000000 +0200
+++ patched/demos/Makefile.am	2013-03-30 21:33:48.307572442 +0100
@@ -13,7 +13,7 @@
 
 if INSTALL_TSE3_DOC
 docs_DATA = $(SONG_FILES)
-docsdir = ${prefix}/doc/${PACKAGE}-${VERSION}/songs
+docsdir = ${datadir}/doc/Packages/${PACKAGE}/songs
 endif
 
 ###############################################################################
diff -Naur orig/demos/Makefile.in patched/demos/Makefile.in
--- orig/demos/Makefile.in	2005-08-23 15:27:36.000000000 +0200
+++ patched/demos/Makefile.in	2013-03-30 21:33:34.523445213 +0100
@@ -200,7 +200,7 @@
 	Demo.tse2
 
 @INSTALL_TSE3_DOC_TRUE@docs_DATA = $(SONG_FILES)
-@INSTALL_TSE3_DOC_TRUE@docsdir = ${prefix}/doc/${PACKAGE}-${VERSION}/songs
+@INSTALL_TSE3_DOC_TRUE@docsdir = ${datadir}/doc/Packages/${PACKAGE}/songs
 
 ###############################################################################
 # Distribute them
diff -Naur orig/doc/Makefile.am patched/doc/Makefile.am
--- orig/doc/Makefile.am	2005-07-25 13:23:06.000000000 +0200
+++ patched/doc/Makefile.am	2013-03-30 21:35:31.916528718 +0100
@@ -118,6 +118,6 @@
 
 if INSTALL_TSE3_DOC
 docs_DATA = $(HTML_FILES) $(VERBATIM_FILES)
-docsdir = ${prefix}/doc/${PACKAGE}-${VERSION}
+docsdir = ${datadir}/doc/Packages/${PACKAGE}
 endif
 
diff -Naur orig/doc/Makefile.in patched/doc/Makefile.in
--- orig/doc/Makefile.in	2005-08-23 15:27:36.000000000 +0200
+++ patched/doc/Makefile.in	2013-03-30 21:35:09.816324737 +0100
@@ -270,7 +270,7 @@
 ###############################################################################
 # And this is how they are installed
 @INSTALL_TSE3_DOC_TRUE@docs_DATA = $(HTML_FILES) $(VERBATIM_FILES)
-@INSTALL_TSE3_DOC_TRUE@docsdir = ${prefix}/doc/${PACKAGE}-${VERSION}
+@INSTALL_TSE3_DOC_TRUE@docsdir = ${datadir}/doc/Packages/${PACKAGE}
 all: all-am
 
 .SUFFIXES:
diff -Naur orig/Makefile.am patched/Makefile.am
--- orig/Makefile.am	2005-07-25 13:23:12.000000000 +0200
+++ patched/Makefile.am	2013-03-30 21:36:36.561125368 +0100
@@ -46,6 +46,6 @@
 
 if INSTALL_TSE3_DOC
 docs_DATA = AUTHORS COPYING
-docsdir = ${prefix}/doc/${PACKAGE}-${VERSION}
+docsdir = ${datadir}/doc/Packages/${PACKAGE}
 endif
 
diff -Naur orig/Makefile.in patched/Makefile.in
--- orig/Makefile.in	2005-08-23 15:27:40.000000000 +0200
+++ patched/Makefile.in	2013-03-30 21:36:20.824980123 +0100
@@ -222,7 +222,7 @@
 ###############################################################################
 # Documentation
 @INSTALL_TSE3_DOC_TRUE@docs_DATA = AUTHORS COPYING
-@INSTALL_TSE3_DOC_TRUE@docsdir = ${prefix}/doc/${PACKAGE}-${VERSION}
+@INSTALL_TSE3_DOC_TRUE@docsdir = ${datadir}/doc/Packages/${PACKAGE}
 all: config.h
 	$(MAKE) $(AM_MAKEFLAGS) all-recursive
 
