diff -Nur orig/installwatch/create-localdecls mod/installwatch/create-localdecls
--- orig/installwatch/create-localdecls	2008-11-16 17:20:53.000000000 +0100
+++ mod/installwatch/create-localdecls	2017-06-04 14:21:53.205390280 +0200
@@ -65,9 +65,9 @@
 	2)
 		# 2 is the glibc version
 		case "$OsLibcMinor" in
-		0)
-			echo '#define GLIBC_MINOR 0' >> $OUTFILE
-			SUBVERSION='glibc-2.0' ;;
+		25)
+			echo '#define GLIBC_MINOR 25' >> $OUTFILE
+			SUBVERSION='glibc-2.25' ;;
 		1)
 			echo '#define GLIBC_MINOR 1' >> $OUTFILE
 			SUBVERSION='glibc-2.1' ;;
