diff -Naur orig/applets/clock/clock-location.c patched/applets/clock/clock-location.c
--- orig/applets/clock/clock-location.c	2013-09-25 16:10:00.000000000 +0200
+++ patched/applets/clock/clock-location.c	2013-10-21 01:39:06.309997563 +0200
@@ -542,8 +542,7 @@
 	}
 
 	priv->weather_info =
-		gweather_info_new_for_world (priv->world,
-					     priv->loc,
+		gweather_info_new (priv->loc,
 					     GWEATHER_FORECAST_LIST);
 	g_signal_connect (priv->weather_info, "updated",
 			  G_CALLBACK (weather_info_updated), loc);
