diff --git a/gtk/rglogview.cc b/gtk/rglogview.cc
index b55bc8da..317457d2 100644
--- a/gtk/rglogview.cc
+++ b/gtk/rglogview.cc
@@ -23,6 +23,7 @@
 #include <cassert>
 #include <cstring>
 #include <map>
+#include <utility>
 #include <apt-pkg/fileutl.h>

 #include "config.h"
@@ -86,7 +87,7 @@ void RGLogView::readLogs()
 			     COLUMN_LOG_TYPE, LOG_TYPE_TOPLEVEL,
 			     -1);
 	 g_free(sort_key);
-	 history_map.insert(make_pair<int,GtkTreeIter>(history_key,month_iter));
+	 history_map.insert(std::make_pair(history_key,month_iter));
       } else {
 	 month_iter = history_map[history_key];
       }
