diff -Naur gwenview-4.12.3-orig/lib/CMakeLists.txt gwenview-4.12.3/lib/CMakeLists.txt
--- gwenview-4.12.3-orig/lib/CMakeLists.txt	2014-02-24 07:25:57.000000000 +0000
+++ gwenview-4.12.3/lib/CMakeLists.txt	2014-03-10 18:25:34.627903762 +0000
@@ -32,6 +32,10 @@
     set(GV_JPEG_DIR libjpeg-80)
 endif()
 
+if ("${jpeglib_version}" EQUAL 90)
+    set(GV_JPEG_DIR libjpeg-80)
+endif()
+
 if ("${GV_JPEG_DIR}" STREQUAL "")
     message(FATAL_ERROR "Unknown libjpeg version: ${jpeglib_version}")
 endif()
