diff -Nur orig/CMakeLists.txt mod/CMakeLists.txt
--- orig/CMakeLists.txt	2021-03-17 09:51:48.000000000 +0100
+++ mod/CMakeLists.txt	2021-03-17 15:43:05.952882285 +0100
@@ -5,7 +5,7 @@
 #
 # Please use TABs to indent! (x4)
 #------------------------------------------------------------------------
-cmake_minimum_required(VERSION 3.10.3 FATAL_ERROR)
+cmake_minimum_required(VERSION 3.9.0 FATAL_ERROR)
 
 project(ETLEGACY C CXX)
 
