diff -Naur orig/cups/usersys.c patched/cups/usersys.c
--- orig/cups/usersys.c	2015-02-03 13:45:14.000000000 +0100
+++ patched/cups/usersys.c	2015-08-02 16:34:54.539915289 +0200
@@ -1017,7 +1017,7 @@
     struct stat	sockinfo;		/* Domain socket information */
 
     if (!stat(CUPS_DEFAULT_DOMAINSOCKET, &sockinfo) &&
-	(sockinfo.st_mode & S_IRWXO) == S_IRWXO)
+	(sockinfo.st_mode & (S_IROTH | S_IWOTH)) == (S_IROTH | S_IWOTH))
       cups_set_server_name(cc, CUPS_DEFAULT_DOMAINSOCKET);
     else
 #endif /* CUPS_DEFAULT_DOMAINSOCKET */
diff -Naur orig/scheduler/main.c patched/scheduler/main.c
--- orig/scheduler/main.c	2014-11-12 17:32:57.000000000 +0100
+++ patched/scheduler/main.c	2015-08-02 16:34:54.539915289 +0200
@@ -648,8 +648,15 @@
 
 #if defined(HAVE_LAUNCHD) || defined(HAVE_SYSTEMD)
   if (OnDemand)
+  {
     cupsdAddEvent(CUPSD_EVENT_SERVER_STARTED, NULL, NULL, "Scheduler started on demand.");
-  else
+# ifdef HAVE_SYSTEMD
+    sd_notifyf(0, "READY=1\n"
+               "STATUS=Scheduler is running...\n"
+               "MAINPID=%lu",
+               (unsigned long) getpid());
+# endif /* HAVE_SYSTEMD */
+  } else
 #endif /* HAVE_LAUNCHD || HAVE_SYSTEMD */
   if (fg)
     cupsdAddEvent(CUPSD_EVENT_SERVER_STARTED, NULL, NULL, "Scheduler started in foreground.");
diff -Naur orig/scheduler/org.cups.cupsd.path.in patched/scheduler/org.cups.cupsd.path.in
--- orig/scheduler/org.cups.cupsd.path.in	2014-03-21 15:50:24.000000000 +0100
+++ patched/scheduler/org.cups.cupsd.path.in	2015-08-02 16:40:25.641557133 +0200
@@ -2,7 +2,8 @@
 Description=CUPS Scheduler
 
 [Path]
-PathExists=@CUPS_CACHEDIR@/org.cups.cupsd
+PathExistsGlob=@CUPS_REQUESTS@/d*
 
 [Install]
 WantedBy=multi-user.target
+Alias=cups.path
diff -Naur orig/scheduler/org.cups.cupsd.service.in patched/scheduler/org.cups.cupsd.service.in
--- orig/scheduler/org.cups.cupsd.service.in	2014-10-21 13:55:01.000000000 +0200
+++ patched/scheduler/org.cups.cupsd.service.in	2015-08-02 16:40:39.785627270 +0200
@@ -1,11 +1,13 @@
 [Unit]
 Description=CUPS Scheduler
 Documentation=man:cupsd(8)
+After=network.target
 
 [Service]
 ExecStart=@sbindir@/cupsd -l
-Type=simple
+Type=notify
 
 [Install]
 Also=org.cups.cupsd.socket org.cups.cupsd.path
 WantedBy=printer.target
+Alias=cups.service
diff -Naur orig/scheduler/org.cups.cupsd.socket.in patched/scheduler/org.cups.cupsd.socket.in
--- orig/scheduler/org.cups.cupsd.socket.in	2014-10-20 20:54:52.000000000 +0200
+++ patched/scheduler/org.cups.cupsd.socket.in	2015-08-02 16:37:34.464708313 +0200
@@ -6,3 +6,4 @@
 
 [Install]
 WantedBy=sockets.target
+Alias=cups.socket
