diff -Naur sympa-5.3.orig/src/etc/script/create_db.mysql sympa-5.3/src/etc/script/create_db.mysql
--- sympa-5.3.orig/src/etc/script/create_db.mysql	2006-08-02 15:56:22.000000000 +0200
+++ sympa-5.3/src/etc/script/create_db.mysql	2007-05-31 15:14:19.000000000 +0200
@@ -8,7 +8,7 @@
 CREATE TABLE user_table (
   	email_user          	varchar (100) NOT NULL,
   	gecos_user          	varchar (150),
-  	password_user		varchar (40),
+  	password_user		varchar (80),
 	cookie_delay_user	int,
 	lang_user		varchar (10),
 	attributes_user		varchar(255),
diff -Naur sympa-5.3.orig/src/etc/script/create_db.Oracle sympa-5.3/src/etc/script/create_db.Oracle
--- sympa-5.3.orig/src/etc/script/create_db.Oracle	2006-08-02 15:56:22.000000000 +0200
+++ sympa-5.3/src/etc/script/create_db.Oracle	2007-05-31 15:14:19.000000000 +0200
@@ -18,7 +18,7 @@
 CREATE TABLE user_table (
         email_user              varchar2(100) NOT NULL,
         gecos_user              varchar2(150),
-        password_user           varchar2(40),
+        password_user           varchar2(80),
         cookie_delay_user       number,
         lang_user               varchar2(10),
 	attributes_user		varchar2(500),
diff -Naur sympa-5.3.orig/src/etc/script/create_db.Pg sympa-5.3/src/etc/script/create_db.Pg
--- sympa-5.3.orig/src/etc/script/create_db.Pg	2006-08-02 15:56:22.000000000 +0200
+++ sympa-5.3/src/etc/script/create_db.Pg	2007-05-31 15:14:19.000000000 +0200
@@ -10,7 +10,7 @@
   	email_user          	varchar (100) NOT NULL,
   	gecos_user          	varchar (150),
 	cookie_delay_user       int4,
-        password_user		varchar (40),
+        password_user		varchar (80),
         lang_user               varchar (10),
 	attributes_user		varchar (255),
 	CONSTRAINT ind_user PRIMARY KEY (email_user)
diff -Naur sympa-5.3.orig/src/etc/script/create_db.SQLite sympa-5.3/src/etc/script/create_db.SQLite
--- sympa-5.3.orig/src/etc/script/create_db.SQLite	2006-08-02 15:56:22.000000000 +0200
+++ sympa-5.3/src/etc/script/create_db.SQLite	2007-05-31 15:14:19.000000000 +0200
@@ -2,7 +2,7 @@
 CREATE TABLE user_table (
   	email_user          	varchar (100) NOT NULL,
   	gecos_user          	varchar (150),
-  	password_user		varchar (40),
+  	password_user		varchar (80),
 	cookie_delay_user	integer,
 	lang_user		varchar (10),
 	attributes_user		varchar(255),
diff -Naur sympa-5.3.orig/src/etc/script/create_db.Sybase sympa-5.3/src/etc/script/create_db.Sybase
--- sympa-5.3.orig/src/etc/script/create_db.Sybase	2006-08-02 15:56:22.000000000 +0200
+++ sympa-5.3/src/etc/script/create_db.Sybase	2007-05-31 15:14:19.000000000 +0200
@@ -11,7 +11,7 @@
 (
     email_user             varchar(100)           not null,
     gecos_user             varchar(150)           null    ,
-    password_user          varchar(40)            null    ,
+    password_user          varchar(80)            null    ,
     cookie_delay_user      numeric                null    ,
     lang_user              varchar(10)            null    ,
     attributes_user	   varchar(255)           null    ,
