diff -Naur sympa-5.3.orig/web_tt2/css.tt2 sympa-5.3/web_tt2/css.tt2
--- sympa-5.3.orig/web_tt2/css.tt2	2007-04-18 12:13:59.000000000 +0200
+++ sympa-5.3/web_tt2/css.tt2	2007-06-05 19:09:23.000000000 +0200
@@ -12,8 +12,8 @@
 background: [% color_4 %];
 }
 
-ul { margin: 0 0 0 4em }
-ol { margin: 0 0 0 4em }
+ul { margin: 0 0 0 2em }
+ol { margin: 0 0 0 2em }
 dl { margin: 0 0 0 1em }
 
 /* ########## Blocks, id and class ########## */
