--- a/data/gnome-shell-classic.desktop.in.in.orig	2013-02-15 21:16:50.000000000 +0100
+++ a/data/gnome-shell-classic.desktop.in.in	2013-02-15 21:17:05.000000000 +0100
@@ -2,8 +2,8 @@
 Type=Application
 _Name=GNOME Shell Classic
 _Comment=Window management and application launching
-Exec=@bindir@/gnome-shell
-TryExec=@bindir@/gnome-shell
+Exec=gnome-shell
+TryExec=gnome-shell
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=gnome-shell
 X-GNOME-Bugzilla-Component=general
