diff -Naurd a/scripts/squid_redirect b/scripts/squid_redirect
--- a/scripts/squid_redirect	2004-10-30 09:16:37.000000000 +0200
+++ b/scripts/squid_redirect	2004-11-19 18:23:53.130678746 +0100
@@ -108,6 +108,7 @@
 }
 
 # use the "clear" versions if ZAP_MODE is "CLEAR"
+$ENV{ZAP_MODE} |= "CLEAR";
 if (defined $ENV{ZAP_MODE} && $ENV{ZAP_MODE} eq CLEAR)
 { for my $type (keys %::StubURLs)
   { $::StubURLs{$type} =~ s/\.[^.]+$/-clear$&/;
