diff -Naur orig/chrome/browser/ui/startup/bad_flags_prompt.cc patched/chrome/browser/ui/startup/bad_flags_prompt.cc
--- orig/chrome/browser/ui/startup/bad_flags_prompt.cc	2015-08-22 21:01:52.000000000 +0200
+++ patched/chrome/browser/ui/startup/bad_flags_prompt.cc	2015-09-06 18:49:50.177571638 +0200
@@ -88,7 +88,6 @@
     // These flags disable sandbox-related security.
     switches::kDisableGpuSandbox,
     switches::kDisableSeccompFilterSandbox,
-    switches::kDisableSetuidSandbox,
     switches::kDisableWebSecurity,
 #if !defined(DISABLE_NACL)
     switches::kNaClDangerousNoSandboxNonSfi,
