diff -Naur orig/configure.ac mod/configure.ac
--- orig/configure.ac	2016-02-20 00:35:15.000000000 +0100
+++ mod/configure.ac	2016-02-26 16:22:53.494622804 +0100
@@ -2310,7 +2310,7 @@
   LCMS_CFLAGS="$LCMS2_CFLAGS"
   LCMS_LIBS="$LCMS2_LIBS"
   CFLAGS="$LCMS2_CFLAGS $CFLAGS"
-  AC_CHECK_HEADER(lcms2/lcms2.h,have_lcms_header='yes',,)
+  AC_CHECK_HEADER(lcms2.h,have_lcms_header='yes',,)
   if test "$have_lcms_header" = 'yes'; then
     AC_DEFINE(HAVE_LCMS2_LCMS2_H,1,Define if you have the <lcms2/lcms2.h> header file.)
   else
diff -Naur orig/magick/property.c mod/magick/property.c
--- orig/magick/property.c	2016-02-20 00:35:15.000000000 +0100
+++ mod/magick/property.c	2016-02-26 16:28:46.844746950 +0100
@@ -82,7 +82,7 @@
 #include "magick/xml-tree.h"
 #if defined(MAGICKCORE_LCMS_DELEGATE)
 #if defined(MAGICKCORE_HAVE_LCMS2_LCMS2_H)
-#include <lcms2/lcms2.h>
+#include <lcms2.h>
 #elif defined(MAGICKCORE_HAVE_LCMS2_H)
 #include "lcms2.h"
 #elif defined(MAGICKCORE_HAVE_LCMS_LCMS_H)
