diff -Naur orig/cmake/systemd.cmake patched/cmake/systemd.cmake
--- orig/cmake/systemd.cmake	2016-07-14 03:10:16.000000000 +0200
+++ patched/cmake/systemd.cmake	2016-08-07 00:39:12.072371020 +0200
@@ -63,8 +63,7 @@
                                ${INSTALL_SYSTEMD_UNITDIR}/mariadb@.service
                                ${INSTALL_SYSTEMD_UNITDIR}/mariadb@bootstrap.service.d/use_galera_new_cluster.conf")
         IF(DEB)
-          SET(SYSTEMD_EXECSTARTPRE "ExecStartPre=/usr/bin/install -m 755 -o mysql -g root -d /var/run/mysqld")
-          SET(SYSTEMD_EXECSTARTPOST "ExecStartPost=/etc/mysql/debian-start")
+          SET(SYSTEMD_EXECSTARTPOST "ExecStartPost=/usr/bin/mysqld-ping")
         ENDIF()
         MESSAGE(STATUS "Systemd features enabled")
       ELSE()
diff -Naur orig/support-files/mariadb.service.in patched/support-files/mariadb.service.in
--- orig/support-files/mariadb.service.in	2016-07-14 03:10:31.000000000 +0200
+++ patched/support-files/mariadb.service.in	2016-08-07 00:41:06.990276766 +0200
@@ -15,7 +15,6 @@
 [Unit]
 Description=MariaDB database server
 After=network.target
-After=syslog.target
 
 [Install]
 WantedBy=multi-user.target
@@ -74,7 +73,6 @@
 # Use the [service] section and Environment="MYSQLD_OPTS=...".
 # This isn't a replacement for my.cnf.
 # _WSREP_NEW_CLUSTER is for the exclusive use of the script galera_new_cluster
-@SYSTEMD_EXECSTARTPRE@
 ExecStart=/usr/sbin/mysqld $MYSQLD_OPTS $_WSREP_NEW_CLUSTER $_WSREP_START_POSITION
 @SYSTEMD_EXECSTARTPOST@
 
diff -Naur orig/support-files/mariadb@.service.in patched/support-files/mariadb@.service.in
--- orig/support-files/mariadb@.service.in	2016-07-14 03:10:31.000000000 +0200
+++ patched/support-files/mariadb@.service.in	2016-08-07 00:39:12.076371086 +0200
@@ -20,7 +20,6 @@
 [Unit]
 Description=MariaDB database server
 After=network.target
-After=syslog.target
 
 ConditionPathExists=@INSTALL_SYSCONF2DIR@/my%I.cnf
 
