diff -Naur orig/cmake/systemd.cmake patched/cmake/systemd.cmake
--- orig/cmake/systemd.cmake	2015-12-23 16:33:29.000000000 +0100
+++ patched/cmake/systemd.cmake	2015-12-25 04:11:04.834818850 +0100
@@ -62,8 +62,7 @@
                                ${INSTALL_SYSTEMD_UNITDIR}/mariadb@.service
                                ${INSTALL_SYSTEMD_UNITDIR}/mariadb@bootstrap.service.d/use_galera_new_cluster.conf")
         IF(DEB)
-          SET(SYSTEMD_EXECSTARTPRE "ExecStartPre=/usr/bin/install -m 755 -o mysql -g root -d /var/run/mysqld")
-          SET(SYSTEMD_EXECSTARTPOST "ExecStartPost=/etc/mysql/debian-start")
+          SET(SYSTEMD_EXECSTARTPOST "ExecStartPost=/usr/bin/mysqld-ping")
         ENDIF()
         MESSAGE(STATUS "Systemd features enabled")
       ELSE()
diff -Naur orig/support-files/mariadb.service.in patched/support-files/mariadb.service.in
--- orig/support-files/mariadb.service.in	2015-12-23 16:33:33.000000000 +0100
+++ patched/support-files/mariadb.service.in	2015-12-25 04:12:11.495149400 +0100
@@ -15,7 +15,6 @@
 [Unit]
 Description=MariaDB database server
 After=network.target
-After=syslog.target
 
 [Install]
 WantedBy=multi-user.target
@@ -56,7 +55,6 @@
 # Use the [service] section and Environment="MYSQLD_OPTS=...".
 # This isn't a replacement for my.cnf.
 # _WSREP_NEW_CLUSTER is for the exclusive use of the script galera_new_cluster
-@SYSTEMD_EXECSTARTPRE@
 ExecStart=/usr/sbin/mysqld $MYSQLD_OPTS $_WSREP_NEW_CLUSTER
 @SYSTEMD_EXECSTARTPOST@
 
diff -Naur orig/support-files/mariadb@.service.in patched/support-files/mariadb@.service.in
--- orig/support-files/mariadb@.service.in	2015-12-23 16:33:33.000000000 +0100
+++ patched/support-files/mariadb@.service.in	2015-12-25 04:12:59.751388690 +0100
@@ -20,7 +20,6 @@
 [Unit]
 Description=MariaDB database server
 After=network.target
-After=syslog.target
 
 ConditionPathExists=@INSTALL_SYSCONF2DIR@/my%I.cnf
 
