diff -urN mariadb-10.0.11.orig/packaging/rpm-oel/mysqld.service mariadb-10.0.11/packaging/rpm-oel/mysqld.service
--- mariadb-10.0.11.orig/packaging/rpm-oel/mysqld.service	2014-05-11 00:36:18.000000000 +0200
+++ mariadb-10.0.11/packaging/rpm-oel/mysqld.service	2014-05-27 10:46:42.000000000 +0200
@@ -30,19 +30,14 @@
 # Execute pre and post scripts as root
 PermissionsStartOnly=true
 
-# Needed to create system tables etc.
-ExecStartPre=/usr/bin/mysql-systemd-start pre
-
 # Start main service
 ExecStart=/usr/bin/mysqld_safe
 
 # Don't signal startup success before a ping works
-ExecStartPost=/usr/bin/mysql-systemd-start post
+ExecStartPost=/usr/bin/mysqld-ping
 
 # Give up if ping don't get an answer
 TimeoutSec=600
 
 Restart=always
 PrivateTmp=false
-
-
diff -urN mariadb-10.0.11.orig/packaging/rpm-oel/mysql-systemd-start mariadb-10.0.11/packaging/rpm-oel/mysql-systemd-start
--- mariadb-10.0.11.orig/packaging/rpm-oel/mysql-systemd-start	2014-05-11 00:36:18.000000000 +0200
+++ mariadb-10.0.11/packaging/rpm-oel/mysql-systemd-start	2014-05-27 10:48:08.484284520 +0200
@@ -8,29 +8,6 @@
 # post mode :  ping server until answer is received
 # 
 
-install_db () {    
-    # Note: something different than datadir=/var/lib/mysql requires SELinux policy changes (in enforcing mode)
-    datadir=$(/usr/bin/my_print_defaults server mysqld | grep '^--datadir=' | sed -n 's/--datadir=//p')
-    
-    # Restore log, dir, perms and SELinux contexts
-    [ -d "$datadir" ] || install -d -m 0755 -omysql -gmysql "$datadir" || exit 1
-    log=/var/log/mysqld.log
-    [ -e $log ] || touch $log
-    chmod 0640 $log
-    chown mysql:mysql $log || exit 1
-    if [ -x /usr/sbin/restorecon ]; then
-        /usr/sbin/restorecon "$datadir"
-        /usr/sbin/restorecon $log
-    fi
-
-    # If special mysql dir is in place, skip db install 
-    [ -d "$datadir/mysql" ] && exit 0
-
-    # Create initial db
-    /usr/bin/mysql_install_db --rpm --datadir="$datadir" --user=mysql
-    exit 0
-}
-
 pinger () {
     # Wait for ping to answer to signal startup completed,
     # might take a while in case of e.g. crash recovery
@@ -42,11 +19,5 @@
     exit 0
 }
 
-# main
-case $1 in
-    "pre") install_db ;;
-    "post") pinger ;;
-esac
-
+pinger
 exit 0
-
