diff -Naur mined-2011.19.2-orig/src/makefile.gcc mined-2011.19.2/src/makefile.gcc
--- mined-2011.19.2-orig/src/makefile.gcc	2011-09-13 00:00:00.000000000 +0200
+++ mined-2011.19.2/src/makefile.gcc	2011-11-18 16:30:35.000000000 +0100
@@ -23,7 +23,7 @@
 #docdir	= $(prefix)/share/doc/mined
 # SuSE:
 #docdir	= $(prefix)/share/doc/packages/mined
-
+desktopdir = $(prefix)/share/applications
 # directories for target "make localinstall"
 lprefix	= $(root)/usr/local
 lbindir	= $(lprefix)/bin
diff -Naur mined-2011.19.2-orig/src/mkinclud.mak mined-2011.19.2/src/mkinclud.mak
--- mined-2011.19.2-orig/src/mkinclud.mak	2011-11-15 00:00:00.000000000 +0100
+++ mined-2011.19.2/src/mkinclud.mak	2011-11-18 16:33:07.000000000 +0100
@@ -561,7 +561,9 @@
 	$(makelinks) linkdir=$(DESTDIR)$(bindir)
 	# ==== install runtime support library
 	mkdir -p $(DESTDIR)$(rundir)
-	/bin/cp -pr ../usrshare/* $(DESTDIR)$(rundir)
+	/bin/cp -pr ../usrshare/help $(rundir)
+	mkdir -p $(desktopdir)
+	$(INSTALL) --mode=+r,u+w ../usrshare/setup_install/mined.desktop $(desktopdir)
 	# ==== install manual page
 	mkdir -p $(DESTDIR)$(mandir)/man1
 	$(INSTALL) ../man/*.1 $(DESTDIR)$(mandir)/man1
