diff -Naur orig/gview_v4l2core/jpeg_decoder.c mod/gview_v4l2core/jpeg_decoder.c
--- orig/gview_v4l2core/jpeg_decoder.c	2016-04-25 17:58:06.000000000 +0200
+++ mod/gview_v4l2core/jpeg_decoder.c	2016-04-29 13:37:14.062844802 +0200
@@ -1436,7 +1436,7 @@
 		exit(-1);
 	}
 
-	codec_data->context->pix_fmt = PIX_FMT_YUV422P;
+	codec_data->context->pix_fmt = AV_PIX_FMT_YUV422P;
 	codec_data->context->width = width;
 	codec_data->context->height = height;
 	//jpeg_ctx->context->dsp_mask = (FF_MM_MMX | FF_MM_MMXEXT | FF_MM_SSE);
diff -Naur orig/gview_v4l2core/uvc_h264.c mod/gview_v4l2core/uvc_h264.c
--- orig/gview_v4l2core/uvc_h264.c	2016-04-25 17:58:06.000000000 +0200
+++ mod/gview_v4l2core/uvc_h264.c	2016-04-29 13:37:51.674800395 +0200
@@ -1036,7 +1036,7 @@
 	}
 	
 	h264_ctx->context->flags2 |= CODEC_FLAG2_FAST;
-	h264_ctx->context->pix_fmt = PIX_FMT_YUV420P;
+	h264_ctx->context->pix_fmt = AV_PIX_FMT_YUV420P;
 	h264_ctx->context->width = width;
 	h264_ctx->context->height = height;
 	//h264_ctx->context->dsp_mask = (FF_MM_MMX | FF_MM_MMXEXT | FF_MM_SSE);
