diff -Naurd a/src/decl.c b/src/decl.c
--- a/src/decl.c	1993-12-08 06:36:50.000000000 +0100
+++ b/src/decl.c	2013-08-06 06:41:03.000000000 +0200
@@ -789,8 +789,10 @@
 		mpp2 = &mp2->snext;
 	    if (mp2)
 		*mpp2 = mp2->snext;
+#if 0
 	    if (mp->kind == MK_CONST)
 		free_value(&mp->val);
+#endif
 	    freeexpr(mp->constdefn);
 	    if (mp->cbase)
 		forget_ctx(mp, 1);
diff -Naurd a/src/lex.c b/src/lex.c
--- a/src/lex.c	1993-12-08 06:36:49.000000000 +0100
+++ b/src/lex.c	2013-08-06 06:51:35.000000000 +0200
@@ -1044,7 +1044,7 @@
 
 
 
-void getline()
+void p2c_getline()
 {
     char *cp, *cp2;
 
@@ -1071,7 +1071,7 @@
 			infname = stralloc(cp);
 			infname[cp2 - cp] = 0;
 		    }
-		    getline();
+		    p2c_getline();
 		    return;
 		}
 		if (copysource && *inbuf) {
@@ -1088,7 +1088,7 @@
                     fprintf(stderr, "\n");
                 if (inputkind == INP_INCFILE) {
                     pop_input();
-                    getline();
+                    p2c_getline();
                 } else
                     strcpy(inbuf, "\001");
             }
@@ -2367,7 +2367,7 @@
 		else
 		    commentline(CMT_POST);
 		trailing = 0;
-                getline();
+                p2c_getline();
 		i = 0;
 		for (;;) {
 		    if (*inbufptr == ' ') {
@@ -2419,7 +2419,7 @@
         if (isspace(*inbufptr)) {
             inbufptr++;
         } else if (!*inbufptr) {
-            getline();
+            p2c_getline();
         } else if (*inbufptr == '{') {
             inbufptr++;
             comment(0);
@@ -2513,7 +2513,7 @@
         switch (*inbufptr++) {
 
             case 0:
-                getline();
+                p2c_getline();
                 break;
 
             case ' ':
@@ -2765,7 +2765,7 @@
             case 0:
 	        if (commenting_flag)
 		    saveinputcomment(inbufptr-1);
-                getline();
+                p2c_getline();
 		cp = curtokbuf;
 		for (;;) {
 		    inbufindent = 0;
@@ -2782,7 +2782,7 @@
 		    }
 		    if (!*inbufptr && !commenting_flag) {   /* blank line */
 			*cp++ = '\001';
-			getline();
+			p2c_getline();
 		    } else
 			break;
 		}
@@ -2797,10 +2797,10 @@
 		*cp++ = '\001';
 		*cp++ = '\014';
 		if (!*inbufptr && !commenting_flag) {
-		    getline();
+		    p2c_getline();
 		    while (!*inbufptr) {
 			*cp++ = '\001';
-			getline();
+			p2c_getline();
 		    }
 		}
 		*cp = 0;
diff -Naurd a/src/Makefile b/src/Makefile
--- a/src/Makefile	1993-12-08 06:36:48.000000000 +0100
+++ b/src/Makefile	2013-08-06 06:43:05.000000000 +0200
@@ -19,22 +19,16 @@
 SHELL = /bin/sh
 
 # Directories (private version)
-HOMEDIR = ../home
-INCDIR = ../home/p2c
-BINDIR = ..
-LIBDIR = ../home
-MANDIR = ../home
-MANFILE = p2c.cat          # human-readable manual (for cat.1)
 #MANFILE = p2c.man.inst    # uncompressed nroff source (for man.1)
 #MANFILE = p2c.man.Z       # compressed nroff source (for man.1.Z)
 
 # Directories (public version)
-#HOMEDIR = /usr/lib/p2c
-#INCDIR = /usr/include/p2c
-#BINDIR = /usr/bin
-#LIBDIR = /usr/lib
-#MANDIR = /usr/man/man1
-#MANFILE = p2c.man.inst
+HOMEDIR = $(DESTDIR)/usr/lib/p2c
+INCDIR = $(DESTDIR)/usr/include/p2c
+BINDIR = $(DESTDIR)/usr/bin
+LIBDIR = $(DESTDIR)/usr/lib
+MANDIR = $(DESTDIR)/usr/share/man/man1
+MANFILE = p2c.man.inst
 
 # Compiler options
 CC = cc                    # you may wish to use gcc here instead
diff -Naurd a/src/p2c.hdrs b/src/p2c.hdrs
--- a/src/p2c.hdrs	1993-12-08 06:36:55.000000000 +0100
+++ b/src/p2c.hdrs	2013-08-06 06:45:27.000000000 +0200
@@ -169,7 +169,7 @@
 extern void     clearprogress      PV();
 extern void     progress           PV();
 extern void     replacestrings     PP( (char *buf, Strlist *sl) );
-extern void     getline            PV();
+extern void     p2c_getline        PV();
 extern void     push_input_file    PP( (FILE *fp, char *fname, int isinclude) );
 extern void     include_as_import  PV();
 extern void     push_input_strlist PP( (Strlist *sp, char *fname) );
diff -Naurd a/src/parse.c b/src/parse.c
--- a/src/parse.c	1993-12-08 06:36:49.000000000 +0100
+++ b/src/parse.c	2013-08-06 06:52:58.000000000 +0200
@@ -4642,8 +4642,10 @@
 	}
     }
     if (conserve_mem) {
+#if 0
 	free_stmt((Stmt *)func->val.i);   /* is this safe? */
 	func->val.i = 0;
+#endif
 	forget_ctx(func, 0);
     }
     outsection(spacing);
@@ -5071,7 +5073,7 @@
 	out_include(fname, 1);
     outsection(majorspace);
     pop_input();
-    getline();
+    p2c_getline();
     gettok();
 }
 
diff -Naurd a/src/TAGS b/src/TAGS
--- a/src/TAGS	1993-12-08 06:36:55.000000000 +0100
+++ b/src/TAGS	2013-08-06 06:46:37.000000000 +0200
@@ -176,7 +176,7 @@
 char getchartok(2466,60243
 Static int getflag(2451,60064
 char *getinlinepart(2414,59241
-void getline(1047,22000
+void p2c_getline(1047,22000
 char *getparenstr(2481,60499
 void gettok(2728,65611
 void gettok(2752,66193
