diff -Naur orig/ext/vp8/gstvp8utils.h patched/ext/vp8/gstvp8utils.h
--- orig/ext/vp8/gstvp8utils.h	2011-12-11 19:49:11.000000000 +0100
+++ patched/ext/vp8/gstvp8utils.h	2015-09-26 23:42:58.498302743 +0200
@@ -25,23 +25,6 @@
 
 G_BEGIN_DECLS
 
-/* Some compatibility defines for older libvpx versions */
-#ifndef VPX_IMG_FMT_I420
-#define VPX_IMG_FMT_I420 IMG_FMT_I420
-#endif
-
-#ifndef VPX_PLANE_Y
-#define VPX_PLANE_Y PLANE_Y
-#endif
-
-#ifndef VPX_PLANE_U
-#define VPX_PLANE_U PLANE_U
-#endif
-
-#ifndef VPX_PLANE_V
-#define VPX_PLANE_V PLANE_V
-#endif
-
 const char * gst_vpx_error_name (vpx_codec_err_t status);
 
 G_END_DECLS
