diff -Naur microcode_ctl-1.29-orig/Makefile microcode_ctl-1.29/Makefile
--- microcode_ctl-1.29-orig/Makefile	2016-01-12 16:13:34.000000000 +0100
+++ microcode_ctl-1.29/Makefile	2016-03-13 16:52:47.102681428 +0100
@@ -19,11 +19,11 @@
 CFLAGS          = -g -Wall -O2 -I $(KERNELHEADER)
 
 DESTDIR         =
-PREFIX          = /usr/local
+PREFIX          = /
 
 INSDIR          = $(PREFIX)/sbin
-MANDIR          = $(PREFIX)/share/man/man8
-DOCDIR          = $(PREFIX)/share/doc/microcode_ctl
+MANDIR          = $(PREFIX)/usr/share/man/man8
+DOCDIR          = $(PREFIX)/usr/share/doc/microcode_ctl
 MICDIR          = /lib/firmware
 MICDIRAMD       = $(MICDIR)/amd-ucode
 
@@ -53,7 +53,6 @@
 	$(INS) -m 755 $(PROGRAM) $(DESTDIR)$(INSDIR)
 	$(INS) -m 644 $(MANPAGE) $(DESTDIR)$(MANDIR)
 	gzip -9f $(DESTDIR)$(MANDIR)/$(MANPAGE)
-	$(INS) -m 755 $(RCFILE) $(DESTDIR)$(RCFILETO)/$(RCFILEFINAL)
 	$(INS) -m 644 intel-ucode/microcode.dat $(DESTDIR)$(MICDIR)
 	$(INS) -m 644 amd-ucode/microcode_amd.bin $(DESTDIR)$(MICDIR)/amd-ucode/
 	$(INS) -m 644 amd-ucode/microcode_amd_fam15h.bin $(DESTDIR)$(MICDIR)/amd-ucode/
