diff -Naur orig/configure patched/configure
--- orig/configure	2015-07-09 11:50:42.000000000 +0200
+++ patched/configure	2015-07-09 22:51:28.466619625 +0200
@@ -2441,6 +2441,9 @@
     --target-arch=*)
       TARGET_MACHINE=`echo $option | cut -d'=' -f2`
       ;;
+	--target-cpu=*)
+      TARGET_CPU=`echo $option | cut -d'=' -f2`
+      ;;
     --disable-xpcom)
       [ $WITH_XPCOM -eq 1 ] && WITH_XPCOM=0
       ;;
