diff -Naur orig/configure patched/configure
--- orig/configure	2011-12-21 20:30:01.849762607 +0100
+++ patched/configure	2011-12-21 19:04:25.000000000 +0100
@@ -2304,6 +2304,9 @@
     --target-arch=*)
       TARGET_MACHINE=`echo $option | cut -d'=' -f2`
       ;;
+	--target-cpu=*)
+      TARGET_CPU=`echo $option | cut -d'=' -f2`
+      ;;
     --disable-xpcom)
       [ $WITH_XPCOM -eq 1 ] && WITH_XPCOM=0
       ;;
