diff -Naur orig/configure patched/configure
--- orig/configure	2016-07-22 14:32:25.000000000 +0200
+++ patched/configure	2016-07-22 14:39:08.839494282 +0200
@@ -1690,7 +1690,7 @@
         qt5_ver=`pkg-config Qt5Core --modversion 2>> $LOG`
         if [ $? -eq 0 ]; then
           echo "(Qt5 from pkg-config)" >> $LOG
-          FLGQT5=`pkg-config Qt5Core --cflags`
+          FLGQT5=`pkg-config Qt5Core --cflags -std=c++11`
           INCQT5=`strip_I "$FLGQT5"`
           LIBDIR5=`pkg-config Qt5Core --variable=libdir`
           LIBQT5=`pkg-config Qt5Core --libs`
@@ -1711,7 +1711,7 @@
       for q in $QT5DIR "$PWD/tools/linux.$TARGET_MACHINE"/qt/v5.*; do
         echo "(Qt5 from '$q')" >> $LOG
         INCQT5="$q/include $q/include/QtCore"
-        FLGQT5="-DQT_SHARED"
+        FLGQT5="-DQT_SHARED -std=c++11"
         I_INCQT5=`prefix_I "$INCQT5"`
         LIBQT5="-L$q/lib -lQt5CoreVBox"
         TOOLQT5="$q"
