diff -Naurdp a/init/main.c b/init/main.c
--- a/init/main.c	2013-10-06 23:00:20.000000000 +0200
+++ b/init/main.c	2013-10-10 18:04:29.558694978 +0200
@@ -204,6 +204,14 @@ static int __init debug_kernel(char *str
 	return 0;
 }
 
+static int __init verbose_kernel(char *str)
+{
+	if (*str)
+		return 0;
+	console_loglevel = 7;
+	return 1;
+}
+
 static int __init quiet_kernel(char *str)
 {
 	console_loglevel = 4;
@@ -211,6 +219,7 @@ static int __init quiet_kernel(char *str
 }
 
 early_param("debug", debug_kernel);
+early_param("verbose", verbose_kernel);
 early_param("quiet", quiet_kernel);
 
 static int __init loglevel(char *str)
diff -Naurdp a/kernel/printk/printk.c b/kernel/printk/printk.c
--- a/kernel/printk/printk.c	2013-10-06 23:00:20.000000000 +0200
+++ b/kernel/printk/printk.c	2013-10-10 18:06:25.976164769 +0200
@@ -59,7 +59,7 @@
 
 /* We show everything that is MORE important than this.. */
 #define MINIMUM_CONSOLE_LOGLEVEL 1 /* Minimum loglevel we let people use */
-#define DEFAULT_CONSOLE_LOGLEVEL 7 /* anything MORE serious than KERN_DEBUG */
+#define DEFAULT_CONSOLE_LOGLEVEL 1 /* anything MORE serious than KERN_DEBUG */
 
 int console_printk[4] = {
 	DEFAULT_CONSOLE_LOGLEVEL,	/* console_loglevel */
