diff -Naur orig/INSTALLDIR/lib/mozilla/python/komodo/Crypto/Util/RFC1751.py mod/INSTALLDIR/lib/mozilla/python/komodo/Crypto/Util/RFC1751.py
--- orig/INSTALLDIR/lib/mozilla/python/komodo/Crypto/Util/RFC1751.py	2016-04-07 23:13:44.000000000 +0200
+++ mod/INSTALLDIR/lib/mozilla/python/komodo/Crypto/Util/RFC1751.py	2016-05-19 14:17:25.982594997 +0200
@@ -1,4 +1,4 @@
-#!/usr/local/bin/python
+#!/usr/bin/python
 # rfc1751.py : Converts between 128-bit strings and a human-readable
 # sequence of words, as defined in RFC1751: "A Convention for
 # Human-Readable 128-bit Keys", by Daniel L. McDonald.
diff -Naur orig/INSTALLDIR/lib/mozilla/python/komodo/pathtools/patterns.py mod/INSTALLDIR/lib/mozilla/python/komodo/pathtools/patterns.py
--- orig/INSTALLDIR/lib/mozilla/python/komodo/pathtools/patterns.py	2016-04-07 23:13:44.000000000 +0200
+++ mod/INSTALLDIR/lib/mozilla/python/komodo/pathtools/patterns.py	2016-05-19 14:18:29.954481797 +0200
@@ -199,7 +199,7 @@
         through the ignored patterns.
 
     Doctests::
-        >>> pathnames = set(["/users/gorakhargosh/foobar.py", "/var/cache/pdnsd.status", "/etc/pdnsd.conf", "/usr/local/bin/python"])
+        >>> pathnames = set(["/users/gorakhargosh/foobar.py", "/var/cache/pdnsd.status", "/etc/pdnsd.conf", "/usr/bin/python"])
         >>> set(filter_paths(pathnames)) == pathnames
         True
         >>> set(filter_paths(pathnames, case_sensitive=False)) == pathnames
@@ -241,7 +241,7 @@
         ``True`` if any of the paths matches; ``False`` otherwise.
 
     Doctests::
-        >>> pathnames = set(["/users/gorakhargosh/foobar.py", "/var/cache/pdnsd.status", "/etc/pdnsd.conf", "/usr/local/bin/python"])
+        >>> pathnames = set(["/users/gorakhargosh/foobar.py", "/var/cache/pdnsd.status", "/etc/pdnsd.conf", "/usr/bin/python"])
         >>> match_any_paths(pathnames)
         True
         >>> match_any_paths(pathnames, case_sensitive=False)
diff -Naur orig/INSTALLDIR/lib/python/lib/python2.7/cgi.py mod/INSTALLDIR/lib/python/lib/python2.7/cgi.py
--- orig/INSTALLDIR/lib/python/lib/python2.7/cgi.py	2016-05-10 20:51:09.000000000 +0200
+++ mod/INSTALLDIR/lib/python/lib/python2.7/cgi.py	2016-05-19 14:17:00.938639470 +0200
@@ -1,4 +1,4 @@
-#! /usr/local/bin/python
+#! /usr/bin/python
 
 # NOTE: the above "/usr/local/bin/python" is NOT a mistake.  It is
 # intentionally NOT "/usr/bin/env python".  On many systems
