diff -Naur gqview-2.0.4.orig/gqview.desktop gqview-2.0.4/gqview.desktop
--- gqview-2.0.4.orig/gqview.desktop	2006-11-01 00:19:21.000000000 +0100
+++ gqview-2.0.4/gqview.desktop	2007-09-04 15:26:22.000000000 +0200
@@ -17,5 +17,5 @@
 #StartupNotify=false
 #StartupWMClass=gqview
 Encoding=UTF-8
-Categories=Application;Graphics;Viewer;
+Categories=Application;Graphics;Viewer;X-UHU-Menu/Graphics;
 MimeType=application/x-navi-animation;image/bmp;image/x-bmp;image/x-MS-bmp;image/gif;image/x-icon;image/jpeg;image/png;image/x-portable-anymap;image/x-portable-bitmap;image/x-portable-graymap;image/x-portable-pixmap;image/x-cmu-raster;image/x-sun-raster;image/x-tga;image/tiff;image/vnd.wap.wbmp;image/x-xbitmap;image/x-xpixmap;image/svg;image/svg+xml;image/x-png;image/xpm;image/x-ico;image/x-pcx;
