diff -urN gqview-2.0.4.orig/src/editors.c gqview-2.0.4/src/editors.c
--- gqview-2.0.4.orig/src/editors.c	2005-06-13 23:58:04.000000000 +0200
+++ gqview-2.0.4/src/editors.c	2007-10-26 12:45:30.000000000 +0200
@@ -47,7 +47,7 @@
 
 
 static gchar *editor_slot_defaults[] = {
-	N_("The Gimp"), "gimp-remote -n %f",
+	N_("The Gimp"), "gimp %f",
 	N_("XV"), "xv %f",
 	N_("Xpaint"), "xpaint %f",
 	NULL, NULL,
