diff -urN sk1libs-0.9.2pre.orig/src/imaging/libimagingft/_imagingft.c sk1libs-0.9.2pre/src/imaging/libimagingft/_imagingft.c
--- sk1libs-0.9.2pre.orig/src/imaging/libimagingft/_imagingft.c	2009-11-01 01:44:12.000000000 +0100
+++ sk1libs-0.9.2pre/src/imaging/libimagingft/_imagingft.c	2014-02-17 17:22:29.377307368 +0100
@@ -32,7 +32,7 @@
 #include FT_FREETYPE_H
 #else
 /* freetype 2.0 */
-#include <freetype/freetype.h>
+#include <freetype2/freetype.h>
 #endif
 
 #if PY_VERSION_HEX < 0x01060000
@@ -70,7 +70,7 @@
     const char* message;
 } ft_errors[] =
 
-#include <freetype/fterrors.h>
+#include <freetype2/fterrors.h>
 
 /* -------------------------------------------------------------------- */
 /* font objects */
