diff -Naur DirectFB-1.7.7-orig/configure DirectFB-1.7.7/configure
--- DirectFB-1.7.7-orig/configure	2015-02-11 15:44:22.000000000 +0100
+++ DirectFB-1.7.7/configure	2017-05-30 20:38:54.166341693 +0200
@@ -18568,10 +18568,10 @@
     if test "$enable_debug" = "no"; then
         CFLAGS="-g2 $CFLAGS"
     fi
-    CFLAGS="-O3 $CFLAGS"
+    CFLAGS="-O2 $CFLAGS"
     DIRECT_BUILD_DEBUGS=1
 else
-    CFLAGS="-O3 -g0 $CFLAGS"
+    CFLAGS="-O2 -g0 $CFLAGS"
     DIRECT_BUILD_DEBUGS=0
 fi
  if test "$enable_debug_support" = "yes"; then
diff -Naur DirectFB-1.7.7-orig/configure.in DirectFB-1.7.7/configure.in
--- DirectFB-1.7.7-orig/configure.in	2015-02-09 18:16:46.000000000 +0100
+++ DirectFB-1.7.7/configure.in	2017-05-30 20:37:39.489025529 +0200
@@ -486,10 +486,10 @@
     if test "$enable_debug" = "no"; then
         CFLAGS="-g2 $CFLAGS"
     fi
-    CFLAGS="-O3 $CFLAGS"
+    CFLAGS="-O2 $CFLAGS"
     DIRECT_BUILD_DEBUGS=1
 else
-    CFLAGS="-O3 -g0 $CFLAGS"
+    CFLAGS="-O2 -g0 $CFLAGS"
     DIRECT_BUILD_DEBUGS=0
 fi
 AM_CONDITIONAL(ENABLE_DEBUGS, test "$enable_debug_support" = "yes")
