diff -Nur orig/configure mod/configure
--- orig/configure	2017-05-12 16:46:43.000000000 +0200
+++ mod/configure	2017-05-12 16:48:03.409442553 +0200
@@ -16459,7 +16459,7 @@
 	OPT_CFLAGS=""
 	case "$CXX" in
     g++*|icc)
-		TRY_CFLAGS="-O3"
+		TRY_CFLAGS="-O2"
 		{ $as_echo "$as_me:${as_lineno-$LINENO}: checking if $CXX supports $TRY_CFLAGS $CXXFLAGS flags" >&5
 $as_echo_n "checking if $CXX supports $TRY_CFLAGS $CXXFLAGS flags... " >&6; }
 	SAVE_CXXFLAGS="$CXXFLAGS"
