diff -up gnomeradio-1.8/src/Makefile.am.v4l2 gnomeradio-1.8/src/Makefile.am
--- gnomeradio-1.8/src/Makefile.am.v4l2	2008-09-13 18:01:41.000000000 +0200
+++ gnomeradio-1.8/src/Makefile.am	2011-02-27 17:10:13.000000000 +0100
@@ -31,7 +31,6 @@ gnomeradio_SOURCES = \
             record.c \
             tech.c \
             trayicon.c \
-            v4l1.c \
             v4l2.c
 
 gnomeradio_LDADD = $(GNOME_LIBS) $(GSTREAMER_LIBS) $(LIRC)
diff -up gnomeradio-1.8/src/Makefile.in.v4l2 gnomeradio-1.8/src/Makefile.in
--- gnomeradio-1.8/src/Makefile.in.v4l2	2008-09-13 18:01:57.000000000 +0200
+++ gnomeradio-1.8/src/Makefile.in	2011-02-27 17:09:58.000000000 +0100
@@ -52,7 +52,7 @@ PROGRAMS = $(bin_PROGRAMS)
 am_gnomeradio_OBJECTS = bacon-volume.$(OBJEXT) gui.$(OBJEXT) \
 	lirc.$(OBJEXT) prefs.$(OBJEXT) radio.$(OBJEXT) \
 	rec_tech.$(OBJEXT) record.$(OBJEXT) tech.$(OBJEXT) \
-	trayicon.$(OBJEXT) v4l1.$(OBJEXT) v4l2.$(OBJEXT)
+	trayicon.$(OBJEXT) v4l2.$(OBJEXT)
 gnomeradio_OBJECTS = $(am_gnomeradio_OBJECTS)
 am__DEPENDENCIES_1 =
 gnomeradio_DEPENDENCIES = $(am__DEPENDENCIES_1) $(am__DEPENDENCIES_1) \
@@ -265,7 +265,6 @@ gnomeradio_SOURCES = \
             record.h \
             tech.h \
             trayicon.h \
-            v4l1.h \
             v4l2.h \
             bacon-volume.c \
             gui.c \
@@ -276,7 +275,6 @@ gnomeradio_SOURCES = \
             record.c \
             tech.c \
             trayicon.c \
-            v4l1.c \
             v4l2.c
 
 gnomeradio_LDADD = $(GNOME_LIBS) $(GSTREAMER_LIBS) $(LIRC)
@@ -360,7 +358,6 @@ distclean-compile:
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/record.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/tech.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/trayicon.Po@am__quote@
-@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/v4l1.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/v4l2.Po@am__quote@
 
 .c.o:
diff -up gnomeradio-1.8/src/prefs.c.v4l2 gnomeradio-1.8/src/prefs.c
--- gnomeradio-1.8/src/prefs.c.v4l2	2008-04-13 15:04:52.000000000 +0200
+++ gnomeradio-1.8/src/prefs.c	2011-02-27 17:09:58.000000000 +0100
@@ -112,7 +112,7 @@ gboolean load_settings(void)
 	/* Load general settings */
 	settings.device = gconf_client_get_string(client, "/apps/gnomeradio/device" , NULL);
 	if (!settings.device)
-		settings.device = g_strdup("/dev/radio");
+		settings.device = g_strdup("/dev/radio0");
 	settings.driver = gconf_client_get_string(client, "/apps/gnomeradio/driver" , NULL);
 	if (!settings.driver)
 		settings.driver = g_strdup("any");
diff -up gnomeradio-1.8/src/radio.c.v4l2 gnomeradio-1.8/src/radio.c
--- gnomeradio-1.8/src/radio.c.v4l2	2008-04-13 14:55:43.000000000 +0200
+++ gnomeradio-1.8/src/radio.c	2011-02-27 17:09:58.000000000 +0100
@@ -24,7 +24,6 @@
 #include <assert.h>
 
 #include "radio.h"
-#include "v4l1.h"
 #include "v4l2.h"
 
 static RadioDev *dev;
@@ -41,25 +40,10 @@ int radio_init(char *device, DriverType 
 	}
 
 	switch (driver) {
-		case DRIVER_V4L2:
-			goto try_v4l2;
 		case DRIVER_ANY:
-		case DRIVER_V4L1:
+		case DRIVER_V4L2:
 		default:
-			goto try_v4l1;
-	}
-
-try_v4l1:
-	dev = v4l1_radio_dev_new();
-	rv = dev->init (dev, device);
-	if (rv == 0) {
-        fprintf(stderr, "Initializing v4l1 failed\n");
-		dev->finalize (dev);
-		dev = NULL;
-		if (driver != DRIVER_ANY)
-			goto failure;
-	} else {
-		goto success;
+			goto try_v4l2;
 	}
 
 try_v4l2:
