diff -up gnomeradio-1.8/src/Makefile.am.ld gnomeradio-1.8/src/Makefile.am
--- gnomeradio-1.8/src/Makefile.am.ld	2011-02-27 17:11:12.000000000 +0100
+++ gnomeradio-1.8/src/Makefile.am	2011-02-27 17:11:34.000000000 +0100
@@ -33,4 +33,4 @@ gnomeradio_SOURCES = \
             trayicon.c \
             v4l2.c
 
-gnomeradio_LDADD = $(GNOME_LIBS) $(GSTREAMER_LIBS) $(LIRC)
+gnomeradio_LDADD = $(GNOME_LIBS) $(GSTREAMER_LIBS) $(LIRC) -lm
diff -up gnomeradio-1.8/src/Makefile.in.ld gnomeradio-1.8/src/Makefile.in
--- gnomeradio-1.8/src/Makefile.in.ld	2011-02-27 17:11:12.000000000 +0100
+++ gnomeradio-1.8/src/Makefile.in	2011-02-27 17:11:45.000000000 +0100
@@ -277,7 +277,7 @@ gnomeradio_SOURCES = \
             trayicon.c \
             v4l2.c
 
-gnomeradio_LDADD = $(GNOME_LIBS) $(GSTREAMER_LIBS) $(LIRC)
+gnomeradio_LDADD = $(GNOME_LIBS) $(GSTREAMER_LIBS) $(LIRC) -lm
 all: all-am
 
 .SUFFIXES:
