diff -up gnomeradio-1.8/configure.gnome-3.0 gnomeradio-1.8/configure
--- gnomeradio-1.8/configure.gnome-3.0	2008-09-13 17:57:42.000000000 +0200
+++ gnomeradio-1.8/configure	2010-12-07 23:15:00.177876740 +0100
@@ -22936,12 +22936,12 @@ if test -n "$PKG_CONFIG"; then
         pkg_cv_GNOME_CFLAGS="$GNOME_CFLAGS"
     else
         if test -n "$PKG_CONFIG" && \
-    { (echo "$as_me:$LINENO: \$PKG_CONFIG --exists --print-errors \"libgnomeui-2.0 >= 2.12 gconf-2.0 >= 2.12 gnome-vfs-2.0 >= 2.12 gnome-media-profiles >= 2.12\"") >&5
-  ($PKG_CONFIG --exists --print-errors "libgnomeui-2.0 >= 2.12 gconf-2.0 >= 2.12 gnome-vfs-2.0 >= 2.12 gnome-media-profiles >= 2.12") 2>&5
+    { (echo "$as_me:$LINENO: \$PKG_CONFIG --exists --print-errors \"libgnomeui-2.0 >= 2.12 gconf-2.0 >= 2.12 gnome-vfs-2.0 >= 2.12 libgnome-media-profiles-3.0 >= 2.12\"") >&5
+  ($PKG_CONFIG --exists --print-errors "libgnomeui-2.0 >= 2.12 gconf-2.0 >= 2.12 gnome-vfs-2.0 >= 2.12 libgnome-media-profiles-3.0 >= 2.12") 2>&5
   ac_status=$?
   echo "$as_me:$LINENO: \$? = $ac_status" >&5
   (exit $ac_status); }; then
-  pkg_cv_GNOME_CFLAGS=`$PKG_CONFIG --cflags "libgnomeui-2.0 >= 2.12 gconf-2.0 >= 2.12 gnome-vfs-2.0 >= 2.12 gnome-media-profiles >= 2.12" 2>/dev/null`
+  pkg_cv_GNOME_CFLAGS=`$PKG_CONFIG --cflags "libgnomeui-2.0 >= 2.12 gconf-2.0 >= 2.12 gnome-vfs-2.0 >= 2.12 libgnome-media-profiles-3.0 >= 2.12" 2>/dev/null`
 else
   pkg_failed=yes
 fi
@@ -22954,12 +22954,12 @@ if test -n "$PKG_CONFIG"; then
         pkg_cv_GNOME_LIBS="$GNOME_LIBS"
     else
         if test -n "$PKG_CONFIG" && \
-    { (echo "$as_me:$LINENO: \$PKG_CONFIG --exists --print-errors \"libgnomeui-2.0 >= 2.12 gconf-2.0 >= 2.12 gnome-vfs-2.0 >= 2.12 gnome-media-profiles >= 2.12\"") >&5
-  ($PKG_CONFIG --exists --print-errors "libgnomeui-2.0 >= 2.12 gconf-2.0 >= 2.12 gnome-vfs-2.0 >= 2.12 gnome-media-profiles >= 2.12") 2>&5
+    { (echo "$as_me:$LINENO: \$PKG_CONFIG --exists --print-errors \"libgnomeui-2.0 >= 2.12 gconf-2.0 >= 2.12 gnome-vfs-2.0 >= 2.12 libgnome-media-profiles-3.0 >= 2.12\"") >&5
+  ($PKG_CONFIG --exists --print-errors "libgnomeui-2.0 >= 2.12 gconf-2.0 >= 2.12 gnome-vfs-2.0 >= 2.12 libgnome-media-profiles-3.0 >= 2.12") 2>&5
   ac_status=$?
   echo "$as_me:$LINENO: \$? = $ac_status" >&5
   (exit $ac_status); }; then
-  pkg_cv_GNOME_LIBS=`$PKG_CONFIG --libs "libgnomeui-2.0 >= 2.12 gconf-2.0 >= 2.12 gnome-vfs-2.0 >= 2.12 gnome-media-profiles >= 2.12" 2>/dev/null`
+  pkg_cv_GNOME_LIBS=`$PKG_CONFIG --libs "libgnomeui-2.0 >= 2.12 gconf-2.0 >= 2.12 gnome-vfs-2.0 >= 2.12 libgnome-media-profiles-3.0 >= 2.12" 2>/dev/null`
 else
   pkg_failed=yes
 fi
@@ -22978,14 +22978,14 @@ else
         _pkg_short_errors_supported=no
 fi
         if test $_pkg_short_errors_supported = yes; then
-	        GNOME_PKG_ERRORS=`$PKG_CONFIG --short-errors --errors-to-stdout --print-errors "libgnomeui-2.0 >= 2.12 gconf-2.0 >= 2.12 gnome-vfs-2.0 >= 2.12 gnome-media-profiles >= 2.12"`
+	        GNOME_PKG_ERRORS=`$PKG_CONFIG --short-errors --errors-to-stdout --print-errors "libgnomeui-2.0 >= 2.12 gconf-2.0 >= 2.12 gnome-vfs-2.0 >= 2.12 libgnome-media-profiles-3.0 >= 2.12"`
         else
-	        GNOME_PKG_ERRORS=`$PKG_CONFIG --errors-to-stdout --print-errors "libgnomeui-2.0 >= 2.12 gconf-2.0 >= 2.12 gnome-vfs-2.0 >= 2.12 gnome-media-profiles >= 2.12"`
+	        GNOME_PKG_ERRORS=`$PKG_CONFIG --errors-to-stdout --print-errors "libgnomeui-2.0 >= 2.12 gconf-2.0 >= 2.12 gnome-vfs-2.0 >= 2.12 libgnome-media-profiles-3.0 >= 2.12"`
         fi
 	# Put the nasty error message in config.log where it belongs
 	echo "$GNOME_PKG_ERRORS" >&5
 
-	{ { echo "$as_me:$LINENO: error: Package requirements (libgnomeui-2.0 >= 2.12 gconf-2.0 >= 2.12 gnome-vfs-2.0 >= 2.12 gnome-media-profiles >= 2.12) were not met:
+	{ { echo "$as_me:$LINENO: error: Package requirements (libgnomeui-2.0 >= 2.12 gconf-2.0 >= 2.12 gnome-vfs-2.0 >= 2.12 libgnome-media-profiles-3.0 >= 2.12) were not met:
 
 $GNOME_PKG_ERRORS
 
@@ -22996,7 +22996,7 @@ Alternatively, you may set the environme
 and GNOME_LIBS to avoid the need to call pkg-config.
 See the pkg-config man page for more details.
 " >&5
-echo "$as_me: error: Package requirements (libgnomeui-2.0 >= 2.12 gconf-2.0 >= 2.12 gnome-vfs-2.0 >= 2.12 gnome-media-profiles >= 2.12) were not met:
+echo "$as_me: error: Package requirements (libgnomeui-2.0 >= 2.12 gconf-2.0 >= 2.12 gnome-vfs-2.0 >= 2.12 libgnome-media-profiles-3.0 >= 2.12) were not met:
 
 $GNOME_PKG_ERRORS
 
diff -up gnomeradio-1.8/src/prefs.c.gnome-3.0 gnomeradio-1.8/src/prefs.c
--- gnomeradio-1.8/src/prefs.c.gnome-3.0	2010-12-07 23:15:00.000000000 +0100
+++ gnomeradio-1.8/src/prefs.c	2010-12-07 23:41:48.576875625 +0100
@@ -15,7 +15,7 @@
  */
 
 #include <gconf/gconf-client.h>
-#include <profiles/audio-profile-choose.h>
+#include <libgnome-media-profiles/audio-profile-choose.h>
 #include <string.h>
 #include "config.h"
 #include "prefs.h"
diff -up gnomeradio-1.8/src/rec_tech.c.gnome-3.0 gnomeradio-1.8/src/rec_tech.c
--- gnomeradio-1.8/src/rec_tech.c.gnome-3.0	2007-02-25 00:33:36.000000000 +0100
+++ gnomeradio-1.8/src/rec_tech.c	2010-12-07 23:43:49.875123992 +0100
@@ -17,7 +17,7 @@
 
 #include "rec_tech.h"
 #include <gnome.h>
-#include <profiles/audio-profile.h>
+#include <libgnome-media-profiles/audio-profile.h>
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <unistd.h>
