--- gnomeradio-1.8.orig/data/gnomeradio.desktop.in	2007-02-25 00:33:38.000000000 +0100
+++ gnomeradio-1.8/data/gnomeradio.desktop.in	2009-05-08 12:54:10.000000000 +0200
@@ -1,9 +1,9 @@
 [Desktop Entry]
-_Name=FM-Radio Tuner
+_Name=GNOME Radio
 _Comment=Listen to FM-radio
 Exec=gnomeradio
 Icon=gnomeradio
-Categories=Application;AudioVideo;
+Categories=GNOME;GTK;AudioVideo;X-UHU-Menu/Multimedia/TVRadio;
 StartupNotify=true
 Terminal=false
 Type=Application
