diff -Naur orig/po/hu_HU.po patched/po/hu_HU.po
--- orig/po/hu_HU.po	2017-01-08 14:35:15.000000000 +0100
+++ patched/po/hu_HU.po	2017-01-08 23:26:15.866461655 +0100
@@ -22,10 +22,9 @@
 
 #: src/main.c:180
 msgid "translator-credits"
-msgstr ""
+msgstr "rezso <rezso@rezso.net>"
 
 #: src/main.c:182
-#, fuzzy
 msgid ""
 "An applet that shows the state of your CapsLock, NumLock, and ScrollLock "
 "keys."
@@ -35,7 +34,7 @@
 
 #: src/main.c:185
 msgid "Home page"
-msgstr ""
+msgstr "Weboldal"
 
 #: src/main.c:279
 msgid "Lock Keys Preferences"
diff -Naur orig/src/org.mate.applets.LockkeysApplet.mate-panel-applet.in patched/src/org.mate.applets.LockkeysApplet.mate-panel-applet.in
--- orig/src/org.mate.applets.LockkeysApplet.mate-panel-applet.in	2017-01-08 12:43:41.000000000 +0100
+++ patched/src/org.mate.applets.LockkeysApplet.mate-panel-applet.in	2017-01-08 23:28:50.296082552 +0100
@@ -4,20 +4,24 @@
 Name=Keyboard LED Indicator
 Name[en_GB]=Keyboard LED Indicator
 Name[bg_BG]=Индикатор от клавиатурата
+Name[hu_HU]=Billentyűzet LED jelző
 Name[ru_RU]=Индикатор клавиатуры
 Description=Display keyboard LED status
 Description[en_GB]=Display keyboard LED status
 Description[bg_BG]=Показва състоянието на клавиатурата
+Description[hu]=Billentyűzet LED állapotának kijelzése
 Description[ru_RU]=Показывает состояние клавиатуры
 
 [LockkeysApplet]
 Name=Keyboard LED Indicator
 Name[en_GB]=Keyboard LED Indicator
 Name[bg_BG]=Индикатор от клавиатурата
+Name[hu_HU]=Billentyűzet LED jelző
 Name[ru_RU]=Индикатор клавиатуры
 Description=Display keyboard LED status
 Description[en_GB]=Display keyboard LED status
 Description[bg_BG]=Показва състоянието на клавиатурата
+Description[hu]=Billentyűzet LED állapotának kijelzése
 Description[ru_RU]=Показывает состояние клавиатуры
 Icon=applet_lockkeys
 MateComponentId=OAFIID:MATE_LockkeysApplet
