diff -Naurd a/keycodes/Makefile b/keycodes/Makefile
--- a/keycodes/Makefile	2013-05-13 19:08:02.000000000 +0200
+++ b/keycodes/Makefile	2013-07-02 19:40:18.000000000 +0200
@@ -1,6 +1,6 @@
 CFLAGS = -Wall
-INCL_PKG_CFG = `pkg-config --cflags-only-I glib-2.0 ncurses`
-LIBS_PKG_CFG = `pkg-config --libs-only-l glib-2.0 ncurses`
+INCL_PKG_CFG = `pkg-config --cflags-only-I glib-2.0 `
+LIBS_PKG_CFG = `pkg-config --libs-only-l glib-2.0 `
 DEFS = -D_GNU_SOURCE 
 LIBS = $(LIBS_PKG_CFG) -lncurses 
 WLIBS = $(LIBS_PKG_CFG) -lncursesw
diff -Naurd a/Makefile b/Makefile
--- a/Makefile	2013-06-28 22:08:59.000000000 +0200
+++ b/Makefile	2013-07-02 19:10:40.000000000 +0200
@@ -4,7 +4,7 @@
 DEFS = -D_REENTRANT -D_POSIX_PTHREAD_SEMANTICS -D_GNU_SOURCE -D_VWM_SCREENSAVER_TIMEOUT=5 -D_DEBUG
 LIBS = $(LIBS_PKG_CFG) -lncurses -lviper -ldl -lmenu -lform -lgpm -lpseudo -lrt
 WLIBS = $(LIBS_PKG_CFG) -lncursesw -lviper -ldl -lmenu -lform -lgpm -lpseudo -lrt
-PREFIX = /usr/local/
+PREFIX = /usr
 libdir = ${prefix}/lib
 INCLUDES = -I./ $(INCL_PKG_CFG)
 MODDIR = /usr/lib/vwm/modules
@@ -43,15 +43,15 @@
 	rm -f vwm_wide.h
 
 install:
-	mkdir -p $(MODDIR)
+	mkdir -p $(DESTDIR)$(MODDIR)
 	chmod 644 vwm.h
-	cp -f vwm.h ${PREFIX}/include
+	cp -f vwm.h $(DESTDIR)${PREFIX}/include
 	chmod 755 vwm
-	cp -f vwm ${PREFIX}/bin
+	cp -f vwm $(DESTDIR)${PREFIX}/bin
 	cd modules/vwmterm3 && $(MAKE) install
 
 install_wide:
-	mkdir -p $(MODDIR)
+	mkdir -p $(DESTDIR)$(MODDIR)
 	chmod 755 vwm_wide
-	cp -f vwm_wide ${PREFIX}/bin
+	cp -f vwm_wide $(DESTDIR)${PREFIX}/bin
 	cd modules/vwmterm3 && $(MAKE) install_wide
diff -Naurd a/modules/sysmon/Makefile b/modules/sysmon/Makefile
--- a/modules/sysmon/Makefile	2013-05-13 19:08:02.000000000 +0200
+++ b/modules/sysmon/Makefile	2013-07-02 19:23:02.000000000 +0200
@@ -1,6 +1,6 @@
 CFLAGS = -Wall -O2
 DEFS = -D_REENTRANT -D_GNU_SOURCE
-prefix = /usr/local
+prefix = /usr
 libdir = ${prefix}/lib
 includedir = ${prefix}/include -I../..
 bindir = ${prefix}/bin
@@ -32,8 +32,8 @@
 
 install:
 	chmod 755 sysmon.so
-	cp -f sysmon.so $(MODDIR)
+	cp -f sysmon.so $(DESTDIR)$(MODDIR)
 
 install_wide:
 	chmod 755 sysmon_wide.so
-	cp -f sysmon_wide.so $(MODDIR)/sysmon.so
+	cp -f sysmon_wide.so $(DESTDIR)$(MODDIR)/sysmon.so
diff -Naurd a/modules/vwmterm3/Makefile b/modules/vwmterm3/Makefile
--- a/modules/vwmterm3/Makefile	2013-05-13 19:08:02.000000000 +0200
+++ b/modules/vwmterm3/Makefile	2013-07-02 19:12:27.000000000 +0200
@@ -1,7 +1,7 @@
 CFLAGS = -Wall -std=gnu99 -O2
 PKG_CFG = `pkg-config --cflags glib-2.0 gmodule-2.0`
 DEFS = -D_REENTRANT -D_GNU_SOURCE
-prefix = /usr/local
+prefix = /usr
 libdir = ${prefix}/lib
 includedir = -I${prefix}/include -I../..
 bindir = ${prefix}/bin
@@ -31,8 +31,8 @@
 
 install:
 	chmod 755 vwmterm3.so
-	cp -f vwmterm3.so $(MODDIR)
+	cp -f vwmterm3.so $(DESTDIR)$(MODDIR)
 
 install_wide:
 	chmod 755 vwmterm3_wide.so
-	cp -f vwmterm3_wide.so $(MODDIR)/vwmterm.so
+	cp -f vwmterm3_wide.so $(DESTDIR)$(MODDIR)/vwmterm.so
