diff -Naur orig/Makefile patched/Makefile
--- orig/Makefile	2009-10-20 21:57:55.000000000 +0200
+++ patched/Makefile	2016-07-20 00:31:27.400041398 +0200
@@ -3,7 +3,7 @@
 DEFS = -D_REENTRANT -D_POSIX_PTHREAD_SEMANTICS -D_GNU_SOURCE -D_VWM_SCREENSAVER_TIMEOUT=5 -D_DEBUG
 LIBS = -lncurses -lviper -lmenu -lform -lgpm -lpseudo
 WLIBS = -lncursesw -lviper -lmenu -lform -lgpm -lpseudo
-prefix = /usr/local
+prefix = /usr
 libdir = ${prefix}/lib
 headerdir = ${prefix}/include
 includedir = -I${headerdir} -I./
@@ -45,19 +45,19 @@
 	rm -f vwm_wide.h
 
 install:
-	mkdir -p $(moddir)
+	mkdir -p $(DESTDIR)$(moddir)
 	chmod 644 vwm.h
-	cp -f vwm.h $(headerdir)
+	cp -f vwm.h $(DESTDIR)$(headerdir)
 	chmod 755 vwm
-	cp -f vwm $(bindir)
+	cp -f vwm $(DESTDIR)$(bindir)
 	cd modules/vwmterm3 && $(MAKE) install
 
 install_wide:
-	mkdir -p $(moddir_wide)
+	mkdir -p$(DESTDIR) $(moddir_wide)
 	chmod 644 vwm.h
-	cp -f vwm.h $(headerdir)
+	cp -f vwm.h $(DESTDIR)$(headerdir)
 	chmod 644 vwm_wide.h
-	cp -f vwm_wide.h $(headerdir)
+	cp -f vwm_wide.h $(DESTDIR)$(headerdir)
 	chmod 755 vwm_wide
-	cp -f vwm_wide $(bindir)
+	cp -f vwm_wide $(DESTDIR)$(bindir)
 	cd modules/vwmterm3 && $(MAKE) install_wide
diff -Naur orig/modules/sysmon/Makefile patched/modules/sysmon/Makefile
--- orig/modules/sysmon/Makefile	2009-06-23 20:41:32.000000000 +0200
+++ patched/modules/sysmon/Makefile	2016-07-20 00:32:23.868966920 +0200
@@ -1,6 +1,6 @@
 CFLAGS = -Wall -O2
 DEFS = -D_REENTRANT -D_GNU_SOURCE
-prefix = /usr/local
+prefix = /usr
 libdir = ${prefix}/lib
 includedir = ${prefix}/include -I../..
 bindir = ${prefix}/bin
@@ -33,8 +33,8 @@
 
 install:
 	chmod 755 sysmon.so
-	cp -f sysmon.so $(moddir)
+	cp -f sysmon.so $(DESTDIR)$(moddir)
 
 install_wide:
 	chmod 755 sysmon_wide.so
-	cp -f sysmon_wide.so $(moddir_wide)/sysmon.so
+	cp -f sysmon_wide.so $(DESTDIR)$(moddir_wide)/sysmon.so
diff -Naur orig/modules/vwmterm3/Makefile patched/modules/vwmterm3/Makefile
--- orig/modules/vwmterm3/Makefile	2009-10-20 22:02:05.000000000 +0200
+++ patched/modules/vwmterm3/Makefile	2016-07-20 00:32:51.545421428 +0200
@@ -1,7 +1,7 @@
 CFLAGS = -Wall -std=gnu99 -O2
 PKG_CFG = `pkg-config --cflags glib-2.0 gmodule-2.0`
 DEFS = -D_REENTRANT -D_GNU_SOURCE
-prefix = /usr/local
+prefix = /usr
 libdir = ${prefix}/lib
 includedir = -I${prefix}/include -I../..
 bindir = ${prefix}/bin
@@ -32,8 +32,8 @@
 
 install:
 	chmod 755 vwmterm3.so
-	cp -f vwmterm3.so $(moddir)
+	cp -f vwmterm3.so $(DESTDIR)$(moddir)
 
 install_wide:
 	chmod 755 vwmterm3_wide.so
-	cp -f vwmterm3_wide.so $(moddir_wide)/vwmterm.so
+	cp -f vwmterm3_wide.so $(DESTDIR)$(moddir_wide)/vwmterm.so
