diff -Naur orig/plugins/decoder/ffmpeg/k3bffmpegwrapper.cpp patched/plugins/decoder/ffmpeg/k3bffmpegwrapper.cpp
--- orig/plugins/decoder/ffmpeg/k3bffmpegwrapper.cpp	2011-01-15 21:47:29.000000000 +0100
+++ patched/plugins/decoder/ffmpeg/k3bffmpegwrapper.cpp	2013-10-12 19:33:00.352222102 +0200
@@ -42,6 +42,7 @@
 #define FFMPEG_BUILD_PRE_4629
 #endif
 
+#define AVCODEC_MAX_AUDIO_FRAME_SIZE 192000
 
 K3bFFMpegWrapper* K3bFFMpegWrapper::s_instance = 0;
 
