diff -Naur orig/configure.ac mod/configure.ac
--- orig/configure.ac	2015-11-03 22:01:43.000000000 +0100
+++ mod/configure.ac	2015-11-25 21:18:31.453062032 +0100
@@ -223,10 +223,6 @@
     BDJAVA_CFLAGS='-I${abs_top_srcdir}/jni -I${abs_top_srcdir}/jni/'"${java_os}"
   ])
 
-  AC_CHECK_PROG(HAVE_ANT, [ant], yes, no)
-  if test "x$HAVE_ANT" = "xno"; then
-      AC_MSG_ERROR([BD-J requires ANT, but ant was not found. Please install it.])
-  fi
 
   AC_DEFINE([USING_BDJAVA], [1], ["Define to 1 if using BD-Java"])
   AC_DEFINE_UNQUOTED([JAVA_ARCH], ["$java_arch"], ["Defines the architecture of the java vm."])
diff -Naur orig/Makefile.am mod/Makefile.am
--- orig/Makefile.am	2015-11-03 22:01:43.000000000 +0100
+++ mod/Makefile.am	2015-11-25 21:06:54.138322058 +0100
@@ -215,7 +215,7 @@
 AM_CFLAGS += $(BDJAVA_CFLAGS)
 
 all-local:
-	ant -f $(top_srcdir)/src/libbluray/bdj/build.xml \
+	/usr/lib/apache-ant/bin/ant -f $(top_srcdir)/src/libbluray/bdj/build.xml \
 	    -Dbuild='$(abs_builddir)/src/libbluray/bdj/build' \
 	    -Ddist='$(abs_builddir)/.libs' \
 	    -Dsrc_awt=:java-$(BDJ_TYPE) \
@@ -223,7 +223,7 @@
 	    -Dversion='$(BDJ_TYPE)-$(VERSION)'
 
 clean-local:
-	ant -f $(top_srcdir)/src/libbluray/bdj/build.xml \
+	/usr/lib/apache-ant/bin/ant -f $(top_srcdir)/src/libbluray/bdj/build.xml \
 	    -Dbuild='$(abs_builddir)/src/libbluray/bdj/build' \
 	    -Ddist='$(abs_builddir)/.libs' \
 	    -Dversion='$(BDJ_TYPE)-$(VERSION)' \
