diff -Naur orig/joe/uformat.c mod/joe/uformat.c
--- orig/joe/uformat.c	2016-03-06 16:39:56.000000000 +0100
+++ mod/joe/uformat.c	2016-03-29 14:14:18.828934976 +0200
@@ -358,6 +358,7 @@
 	int c;
 	off_t to = p->byte;
 	int my_indents = 0;
+	off_t indents_size = 0;
 	
 	/* autoindent when called by utype */
 	if (!indents) {
@@ -375,7 +376,8 @@
 
 			indent = nindent(bw, q, 0);
 			pcol(tr, indent);
-			indents = brs(q, tr->byte - q->byte); /* Risky */
+			indents_size = tr->byte - q->byte;
+			indents = brs(q, indents_size);
 			prm(tr);
 		} else {
 			/* First line */
@@ -384,7 +386,8 @@
 
 			indent = nindent(bw, s, 1);
 			pcol(tr, indent);
-			indents = brs(s, tr->byte - s->byte); /* Risky */
+			indents_size = tr->byte - s->byte;
+			indents = brs(s, indents_size);
 			prm(tr);
 			if (!bw->o.autoindent) {
 				/* Don't indent second line of single-line paragraphs if autoindent is off */
@@ -392,7 +395,7 @@
 				ptrdiff_t orgx = tx;
 				while (tx && (indents[tx - 1] == ' ' || indents[tx - 1] == '\t'))
 					indents[--tx] = 0;
-				if (tx && orgx != tx) {
+				if (tx && orgx != tx && (tx < indents_size)) {
 					indents[tx++] = ' ';
 					indents[tx] = 0;
 				}
@@ -554,6 +557,7 @@
 	BW *bw;
 	int flag;
 	WIND_BW(bw, w);
+	off_t indents_size = 0;
 
 	p = pdup(bw->cursor, "uformat");
 	p_goto_bol(p);
@@ -584,21 +588,23 @@
 
 		indent = nindent(bw, q, 0);
 		pcol(r, indent);
-		indents = brs(q, r->byte - q->byte); /* Risky */
+		indents_size = r->byte - q->byte;
+		indents = brs(q, indents_size);
 		prm(r);
 	} else {
 		P *r = pdup(p, "uformat");
 		ptrdiff_t x, y;
 		indent = nindent(bw, p, 1); /* allowing * and - here */
 		pcol(r, indent);
-		indents = brs(p, r->byte - p->byte); /* Risky */
+		indents_size = r->byte - q->byte;
+		indents = brs(q, indents_size);
 		prm(r);
 		if (!bw->o.autoindent) {
 			/* Don't indent second line of single-line paragraphs if autoindent is off */
 			ptrdiff_t tx = zlen(indents);
 			while (tx && (indents[tx - 1] == ' ' || indents[tx - 1] == '\t'))
 				indents[--tx] = 0;
-			if (tx) {
+			if ((tx) && (tx < indents_size)) {
 				indents[tx++] = ' ';
 				indents[tx] = 0;
 			}
