diff -Naurdp xscreensaver-5.29/driver/XScreenSaver_ad.h xscreensaver-5.29-defaultconfig/driver/XScreenSaver_ad.h
--- xscreensaver-5.29/driver/XScreenSaver_ad.h	2014-06-08 23:43:53.000000000 +0200
+++ xscreensaver-5.29-defaultconfig/driver/XScreenSaver_ad.h	2014-06-19 14:28:48.517751877 +0200
@@ -6,12 +6,12 @@
 "*cycle:			0:10:00",
 "*lockTimeout:		0:00:00",
 "*passwdTimeout:		0:00:30",
-"*dpmsEnabled:		False",
+"*dpmsEnabled:		True",
 "*dpmsQuickoffEnabled:	False",
 "*dpmsStandby:		2:00:00",
 "*dpmsSuspend:		2:00:00",
 "*dpmsOff:		4:00:00",
-"*grabDesktopImages:	True",
+"*grabDesktopImages:	False",
 "*grabVideoFrames:	False",
 "*chooseRandomImages:	True",
 "*imageDirectory:	/Library/Desktop Pictures/",
@@ -28,7 +28,7 @@
 "*splashDuration:	0:00:05",
 "*visualID:		default",
 "*captureStderr: 	True",
-"*ignoreUninstalledPrograms: False",
+"*ignoreUninstalledPrograms: True",
 "*textMode:		file",
 "*textLiteral:		XScreenSaver",
 "*textFile:		",
diff -Naurdp xscreensaver-5.29/driver/XScreenSaver.ad.in xscreensaver-5.29-defaultconfig/driver/XScreenSaver.ad.in
--- xscreensaver-5.29/driver/XScreenSaver.ad.in	2014-06-08 23:43:53.000000000 +0200
+++ xscreensaver-5.29-defaultconfig/driver/XScreenSaver.ad.in	2014-06-19 14:29:22.821753098 +0200
@@ -33,12 +33,12 @@
 *cycle:			0:10:00
 *lockTimeout:		0:00:00
 *passwdTimeout:		0:00:30
-*dpmsEnabled:		False
+*dpmsEnabled:		True
 *dpmsQuickoffEnabled:	False
 *dpmsStandby:		2:00:00
 *dpmsSuspend:		2:00:00
 *dpmsOff:		4:00:00
-*grabDesktopImages:	True
+*grabDesktopImages:	False
 *grabVideoFrames:	False
 *chooseRandomImages:	@DEFAULT_IMAGES_P@
 ! This can be a local directory name, or the URL of an RSS or Atom feed.
@@ -56,7 +56,7 @@
 *splashDuration:	0:00:05
 *visualID:		default
 *captureStderr: 	True
-*ignoreUninstalledPrograms: False
+*ignoreUninstalledPrograms: True
 
 *textMode:		file
 *textLiteral:		XScreenSaver
