diff -Naurdp xscreensaver-5.29/driver/screensaver-properties.desktop.in xscreensaver-5.29-menu/driver/screensaver-properties.desktop.in
--- xscreensaver-5.29/driver/screensaver-properties.desktop.in	2009-10-07 04:26:41.000000000 +0200
+++ xscreensaver-5.29-menu/driver/screensaver-properties.desktop.in	2014-06-19 14:31:44.949758160 +0200
@@ -2,7 +2,7 @@
 Exec=xscreensaver-demo
 Icon=xscreensaver
 Terminal=false
-_Name=Screensaver
-_Comment=Change screensaver properties
+Name=XScreenSaver
+Comment=Change screensaver properties
 Type=Application
 Categories=Settings;DesktopSettings;Security;X-XFCE;
