diff -Naur wpa_supplicant-2.5-orig/wpa_supplicant/dbus/dbus_new_helpers.c wpa_supplicant-2.5/wpa_supplicant/dbus/dbus_new_helpers.c
--- wpa_supplicant-2.5-orig/wpa_supplicant/dbus/dbus_new_helpers.c	2015-09-27 21:02:05.000000000 +0200
+++ wpa_supplicant-2.5/wpa_supplicant/dbus/dbus_new_helpers.c	2016-05-12 11:51:35.001578527 +0200
@@ -847,7 +847,7 @@
 	const struct wpa_dbus_property_desc *dsc;
 	int i = 0;
 
-	if (iface == NULL)
+	if (iface == NULL || path == NULL)
 		return;
 
 	dbus_connection_get_object_path_data(iface->con, path,
diff -Naur wpa_supplicant-2.5-orig/wpa_supplicant/dbus/dbus_old.c wpa_supplicant-2.5/wpa_supplicant/dbus/dbus_old.c
--- wpa_supplicant-2.5-orig/wpa_supplicant/dbus/dbus_old.c	2015-09-27 21:02:05.000000000 +0200
+++ wpa_supplicant-2.5/wpa_supplicant/dbus/dbus_old.c	2016-05-12 11:54:30.653580852 +0200
@@ -423,7 +423,7 @@
 	if (wpa_s->global == NULL)
 		return;
 	iface = wpa_s->global->dbus;
-	if (iface == NULL)
+	if (iface == NULL || wpa_s->dbus_path == NULL)
 		return;
 
 	/* Only send signal if state really changed */
