diff -Naurd a/installer.sh b/installer.sh
--- a/installer.sh	2004-11-08 13:07:59.756379656 +0100
+++ b/installer.sh	2004-11-08 13:10:01.606310722 +0100
@@ -49,7 +49,7 @@
 esac
 
 # Default installation dir (you can change this with the --installdir option)
-INSTALLDIR="/usr/local/rkhunter"
+INSTALLDIR="$UB_INSTALLDIR/usr"
 INSTALLPREFIX="./files/"
 
 SHOWUSAGE=0
@@ -120,30 +120,24 @@
 overwrite:programs_good.dat:/db/programs_good.dat:Database%%Program%%versions
 overwrite:defaulthashes.dat:/db/defaulthashes.dat:Database%%Default%%file%%hashes
 overwrite:md5blacklist.dat:/db/md5blacklist.dat:Database%%MD5%%blacklisted%%files
-overwrite:CHANGELOG:/docs/CHANGELOG:Changelog
-overwrite:README:/docs/README:Readme%%and%%FAQ
-overwrite:WISHLIST:/docs/WISHLIST:Wishlist%%and%%TODO
 "
 
 # Prefix: INSTALLDIR
 INSTALLFILES2="
-nooverwrite:rkhunter.conf:/usr/local/etc/rkhunter.conf:RK%%Hunter%%configuration%%file
-overwrite:rkhunter:/usr/local/bin/rkhunter:RK%%Hunter%%binary
+nooverwrite:rkhunter.conf:$UB_INSTALLDIR/etc/rkhunter.conf:RK%%Hunter%%configuration%%file
+overwrite:rkhunter:$UB_INSTALLDIR/usr/bin/rkhunter:RK%%Hunter%%binary
 "
 
 # Create directories (only if they do not exist)
 CREATEDIRS="
 ${INSTALLDIR}
-${INSTALLDIR}/etc
 ${INSTALLDIR}/bin
 ${INSTALLDIR}/lib/rkhunter/db
-${INSTALLDIR}/lib/rkhunter/docs
 ${INSTALLDIR}/lib/rkhunter/scripts
 ${INSTALLDIR}/lib/rkhunter/tmp
-/usr/local/etc
 "
 
-CHECKDIR="/usr/local"
+CHECKDIR="/usr"
 
 
 # Functions
@@ -508,21 +502,15 @@
 done
 
 # Installation dir to configuration file
-INSTALLDIRCHECK=`cat /usr/local/etc/rkhunter.conf | grep "INSTALLDIR="`
-if [ "${INSTALLDIRCHECK}" = "" ]
-  then
-    echo "" >> /usr/local/etc/rkhunter.conf
-    echo "INSTALLDIR=${INSTALLDIR}" >> /usr/local/etc/rkhunter.conf
+    echo "" >> $UB_INSTALLDIR/etc/rkhunter.conf
+    echo "INSTALLDIR=${INSTALLDIR#$UB_INSTALLDIR}" >> $UB_INSTALLDIR/etc/rkhunter.conf
     echo "Configuration updated with installation path (${INSTALLDIR})"
-  else
-    echo "Configuration already updated."
-fi
 
 if [ ! ${INSTALL} = "nok" ]
 then
 	echo ""
 	echo $E "$t17"
-	echo "$t18 (/usr/local/bin/rkhunter)"
+	echo "$t18 (/usr/bin/rkhunter)"
 else
 	echo ""
 	echo $E "$t19"
