diff -Naurd b/files/programs_good.dat c/files/programs_good.dat
--- b/files/programs_good.dat	2004-09-12 15:00:00.000000000 +0200
+++ c/files/programs_good.dat	2004-11-08 13:23:54.083146861 +0100
@@ -1,12 +1,12 @@
 000:version:2004091000:
-httpd:%1.3.31%2.0.50%:
+httpd:%1.3.31%2.0.50%2.0.51%2.0.52%:
 sshd:%3.1p1%3.7.1p1%3.8p1%3.8.1p1%3.9p1%:
 exim:%4.30%4.32%4.34%4.41%:
-php:%4.3.8%5.0.0%5.0.1%:
-clamd:%0.70%0.72%0.75%0.75.1%:
+php:%4.3.8%5.0.0%5.0.1%5.0.2%:
+clamd:%0.70%0.72%0.75%0.75.1%0.80%:
 gpg:%1.2.4%1.2.5%1.2.6%1.3.4%:
 named:%8.4.2%8.4.4%:
 mc:%4.6.0%:
 procmail:%3.15.2%3.22%:
-proftpd:%1.2.10rc1%1.2.10rc2%1.2.10rc3%:
+proftpd:%1.2.10rc1%1.2.10rc2%1.2.10rc3%1.2.10%:
 openssl:%0.9.7d%0.9.6m%:
diff -Naurd b/files/rkhunter c/files/rkhunter
--- b/files/rkhunter	2004-09-12 15:00:20.000000000 +0200
+++ c/files/rkhunter	2004-11-08 13:18:51.467717542 +0100
@@ -1565,6 +1565,7 @@
 ${ROOTDIR}usr/lib/apache/1.3/mod_rootme.so
 ${ROOTDIR}usr/lib/apache2/modules/mod_rootme2.so
 ${ROOTDIR}usr/local/apache2/modules/mod_rootme2.so
+${ROOTDIR}usr/lib/apache/mod_rootme2.so
 "
 
 HTTPDCONFS="
@@ -4157,11 +4158,11 @@
 	displaytext -n "   Checking Apache2 modules ... "
 	jump=`expr ${defaultcolumn} - ${SIZE}`
 
-	if [ -d /etc/apache2/mods-enabled ]
+	if [ -d /etc/apache/mods/loaders ]
 	  then
-    	    for I in `ls /etc/apache2/mods-enabled/*`; do
+    	    for I in `ls /etc/apache/mods/loaders/*.load`; do
 	      SEARCHSTRING=`cat ${I} | egrep 'mod_rootme.so|mod_rootme2.so'`
-	      logtext -n "Checking Apache2 modules in /etc/apache2/mods-enabled ${I}... "
+	      logtext -n "Checking Apache2 modules in /etc/apache/mods/loaders ${I}... "
 	      if [ ! "${SEARCHSTRING}" = "" ];
 	        then
 	          logtext "Warning! Possible bad module found."
@@ -4267,7 +4268,7 @@
         VERSION=""
         case ${APPLICATION} in
           clamd)
-                VERSION=`${I}/clamd --version | grep 'ClamAV version' | awk '{ print $5 }'`
+                VERSION=`${I}/clamd --version 2>&1 | grep 'ClamAV version' | awk '{ print $5 }'`
                 ;;
 
           exim)
