diff -Naur ORIG/setup.py UHU/setup.py
--- ORIG/setup.py	2014-04-16 07:27:51.000000000 +0000
+++ UHU/setup.py	2014-07-09 14:43:07.346659318 +0000
@@ -15,6 +15,8 @@
 def _read(fname):
     try:
         return open(op.join(op.dirname(__file__), fname)).read()
+    except UnicodeDecodeError:
+        return open(op.join(op.dirname(__file__), fname), encoding="utf-8").read()
     except IOError:
         return ''
 
