diff -Naur xorg-server-1.15.0-orig/glx/glxdri2.c xorg-server-1.15.0/glx/glxdri2.c
--- xorg-server-1.15.0-orig/glx/glxdri2.c	2013-12-27 17:38:52.000000000 +0000
+++ xorg-server-1.15.0/glx/glxdri2.c	2014-03-07 07:24:08.134207206 +0000
@@ -972,7 +972,7 @@
                                           &screen->driConfigs, screen);
 
     if (screen->driScreen == NULL) {
-        LogMessage(X_ERROR, "AIGLX error: Calling driver entry point failed\n");
+        LogMessage(X_ERROR, "AIGLX error: Calling driver entry point failed (dri2)\n");
         goto handle_error;
     }
 
diff -Naur xorg-server-1.15.0-orig/glx/glxdriswrast.c xorg-server-1.15.0/glx/glxdriswrast.c
--- xorg-server-1.15.0-orig/glx/glxdriswrast.c	2013-12-15 07:48:08.000000000 +0000
+++ xorg-server-1.15.0/glx/glxdriswrast.c	2014-03-07 07:25:06.554205279 +0000
@@ -459,7 +459,7 @@
                                             &screen->driConfigs, screen);
 
     if (screen->driScreen == NULL) {
-        LogMessage(X_ERROR, "AIGLX error: Calling driver entry point failed\n");
+        LogMessage(X_ERROR, "AIGLX error: Calling driver entry point failed (swrast)\n");
         goto handle_error;
     }
 
