diff -Naur xorg-server-1.14.0-orig/hw/xfree86/common/xf86Config.c xorg-server-1.14.0/hw/xfree86/common/xf86Config.c
--- xorg-server-1.14.0-orig/hw/xfree86/common/xf86Config.c	2012-07-10 09:31:01.000000000 +0200
+++ xorg-server-1.14.0/hw/xfree86/common/xf86Config.c	2013-03-26 19:50:34.000000000 +0100
@@ -677,6 +677,7 @@
     FLAG_NOTRAPSIGNALS,
     FLAG_DONTVTSWITCH,
     FLAG_DONTZAP,
+    FLAG_ZAPWARNING,
     FLAG_DONTZOOM,
     FLAG_DISABLEVIDMODE,
     FLAG_ALLOWNONLOCAL,
@@ -714,6 +715,8 @@
      {0}, FALSE},
     {FLAG_DONTZAP, "DontZap", OPTV_BOOLEAN,
      {0}, FALSE},
+    { FLAG_ZAPWARNING,           "ZapWarning",                   OPTV_BOOLEAN,
+     {0}, FALSE}, 
     {FLAG_DONTZOOM, "DontZoom", OPTV_BOOLEAN,
      {0}, FALSE},
     {FLAG_DISABLEVIDMODE, "DisableVidModeExtension", OPTV_BOOLEAN,
@@ -802,6 +805,7 @@
     xf86GetOptValBool(FlagOptions, FLAG_NOTRAPSIGNALS, &xf86Info.notrapSignals);
     xf86GetOptValBool(FlagOptions, FLAG_DONTVTSWITCH, &xf86Info.dontVTSwitch);
     xf86GetOptValBool(FlagOptions, FLAG_DONTZAP, &xf86Info.dontZap);
+    xf86GetOptValBool(FlagOptions, FLAG_ZAPWARNING, &xf86Info.ZapWarning);
     xf86GetOptValBool(FlagOptions, FLAG_DONTZOOM, &xf86Info.dontZoom);
 
     xf86GetOptValBool(FlagOptions, FLAG_IGNORE_ABI, &xf86Info.ignoreABI);
diff -Naur xorg-server-1.14.0-orig/hw/xfree86/common/xf86Events.c xorg-server-1.14.0/hw/xfree86/common/xf86Events.c
--- xorg-server-1.14.0-orig/hw/xfree86/common/xf86Events.c	2013-02-12 05:25:25.000000000 +0100
+++ xorg-server-1.14.0/hw/xfree86/common/xf86Events.c	2013-03-26 19:42:39.000000000 +0100
@@ -179,13 +179,27 @@
     DebugF("ProcessActionEvent(%d,%x)\n", (int) action, arg);
     switch (action) {
     case ACTION_TERMINATE:
-        if (!xf86Info.dontZap) {
-            xf86Msg(X_INFO, "Server zapped. Shutting down.\n");
+        if (xf86Info.dontZap)
+            break;
+
+   if (xf86Info.ZapWarning) {
+       static struct timeval LastZap = { 0, 0};
+       struct timeval NewZap;
+
+       gettimeofday(&NewZap, NULL);
+
+       if ((NewZap.tv_sec - LastZap.tv_sec) >= 2) {
+           xf86OSRingBell(30, 1000, 50);
+           LastZap = NewZap;
+           break;
+       }
+   }
+           /* xf86Msg(X_INFO, "Server zapped. Shutting down.\n");*/
 #ifdef XFreeXDGA
             DGAShutdown();
 #endif
             GiveUp(0);
-        }
+
         break;
     case ACTION_NEXT_MODE:
         if (!xf86Info.dontZoom)
diff -Naur xorg-server-1.14.0-orig/hw/xfree86/common/xf86Globals.c xorg-server-1.14.0/hw/xfree86/common/xf86Globals.c
--- xorg-server-1.14.0-orig/hw/xfree86/common/xf86Globals.c	2012-07-06 21:17:17.000000000 +0200
+++ xorg-server-1.14.0/hw/xfree86/common/xf86Globals.c	2013-03-26 19:43:07.000000000 +0100
@@ -105,6 +105,7 @@
     .autoVTSwitch = TRUE,
     .ShareVTs = FALSE,
     .dontZap = FALSE,
+    .ZapWarning = TRUE,
     .dontZoom = FALSE,
     .notrapSignals = FALSE,
     .caughtSignal = FALSE,
diff -Naur xorg-server-1.14.0-orig/hw/xfree86/common/xf86Privstr.h xorg-server-1.14.0/hw/xfree86/common/xf86Privstr.h
--- xorg-server-1.14.0-orig/hw/xfree86/common/xf86Privstr.h	2012-07-06 21:17:17.000000000 +0200
+++ xorg-server-1.14.0/hw/xfree86/common/xf86Privstr.h	2013-03-26 19:43:33.000000000 +0100
@@ -68,6 +68,7 @@
     Bool autoVTSwitch;
     Bool ShareVTs;
     Bool dontZap;
+    Bool ZapWarning;
     Bool dontZoom;
     Bool notrapSignals;         /* don't exit cleanly - die at fault */
     Bool caughtSignal;
