diff -Naurdp mediastreamer-2.10.0/src/audiofilters/pulseaudio.c mediastreamer-2.10.0-pa/src/audiofilters/pulseaudio.c
--- mediastreamer-2.10.0/src/audiofilters/pulseaudio.c	2013-03-25 11:31:33.000000000 +0100
+++ mediastreamer-2.10.0-pa/src/audiofilters/pulseaudio.c	2014-12-03 19:20:47.742109363 +0100
@@ -22,7 +22,7 @@ Foundation, Inc., 59 Temple Place - Suit
 
 #include <pulse/pulseaudio.h>
 
-static const float latency_req=0.02;
+static const float latency_req=0.1;
 
 static void init_pulse_context();
 static void pulse_card_detect(MSSndCardManager *m);
