diff -Naur imap-2007f-orig/src/osdep/unix/Makefile imap-2007f/src/osdep/unix/Makefile
--- imap-2007f-orig/src/osdep/unix/Makefile	2011-07-23 02:20:10.000000000 +0200
+++ imap-2007f/src/osdep/unix/Makefile	2016-02-25 18:02:55.429274064 +0100
@@ -96,18 +96,18 @@
 LOGINPW=std
 SIGTYPE=bsd
 CRXTYPE=std
-ACTIVEFILE=/usr/lib/news/active
-SPOOLDIR=/usr/spool
+ACTIVEFILE=/var/lib/news/active
+SPOOLDIR=/var/spool
 MAILSPOOL=$(SPOOLDIR)/mail
 NEWSSPOOL=$(SPOOLDIR)/news
-RSHPATH=/usr/ucb/rsh
+RSHPATH=/usr/bin/rsh
 MD5PWD=/etc/cram-md5.pwd
 # Tries one of the test alternatives below if not specified.
-LOCKPGM=
+# LOCKPGM=
 # Test alternatives if LOCKPGM not specified
-LOCKPGM1=/usr/libexec/mlock
-LOCKPGM2=/usr/sbin/mlock
-LOCKPGM3=/etc/mlock
+# LOCKPGM1=/usr/libexec/mlock
+# LOCKPGM2=/usr/sbin/mlock
+# LOCKPGM3=/etc/mlock
 
 
 # Default formats for creating new mailboxes and for empty mailboxes in the
