From 894ca98a86fb56b5dd0fae2631a5ad1a5db64416 Mon Sep 17 00:00:00 2001
From: Jonathon Jongsma <jjongsma@redhat.com>
Date: Mon, 20 Jan 2014 10:27:36 -0600
Subject: [PATCH] Write vm name to config file as comment

---
 src/virt-viewer-app.c | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/src/virt-viewer-app.c b/src/virt-viewer-app.c
index 9182fe4..aa574be 100644
--- a/src/virt-viewer-app.c
+++ b/src/virt-viewer-app.c
@@ -242,6 +242,20 @@ virt_viewer_app_save_config(VirtViewerApp *self)
         g_warning("failed to create config directory");
     g_free(dir);
 
+    if (priv->uuid && priv->guest_name) {
+        // if there's no comment for this uuid settings group, add a comment
+        // with the vm name so user can make sense of it later.
+        gchar *comment = g_key_file_get_comment(priv->config, priv->uuid, NULL, &error);
+        if (error) {
+            g_warning("Unable to get comment from key file: %s", error->message);
+            g_clear_error(&error);
+        } else {
+            if (!comment || *comment == '\0')
+                g_key_file_set_comment(priv->config, priv->uuid, NULL, priv->guest_name, NULL);
+        }
+        g_free(comment);
+    }
+
     if ((data = g_key_file_to_data(priv->config, NULL, &error)) == NULL ||
         !g_file_set_contents(priv->config_file, data, -1, &error)) {
         g_warning("Couldn't save configuration: %s", error->message);
