diff -Naur orig/vapi/gdk-pixbuf-2.0.vapi patched/vapi/gdk-pixbuf-2.0.vapi
--- orig/vapi/gdk-pixbuf-2.0.vapi	2016-06-20 20:16:16.134739579 +0200
+++ patched/vapi/gdk-pixbuf-2.0.vapi	2017-10-15 01:53:33.878976816 +0200
@@ -104,6 +104,8 @@
 		public bool savev (string filename, string type, [CCode (array_length = false, array_null_terminated = true)] string[] option_keys, [CCode (array_length = false, array_null_terminated = true)] string[] option_values) throws GLib.Error;
 		public void scale (Gdk.Pixbuf dest, int dest_x, int dest_y, int dest_width, int dest_height, double offset_x, double offset_y, double scale_x, double scale_y, Gdk.InterpType interp_type);
 		public Gdk.Pixbuf scale_simple (int dest_width, int dest_height, Gdk.InterpType interp_type);
+		[Version (since = "2.2")]
+		public bool set_option (string key, string value);
 		[CCode (has_construct_function = false)]
 		public Pixbuf.subpixbuf (Gdk.Pixbuf src_pixbuf, int src_x, int src_y, int width, int height);
 		[CCode (cname = "gdk_pixbuf_new_from_data", has_construct_function = false)]
