diff -Naur blackbox-0.70.1-orig/lib/Util.hh blackbox-0.70.1/lib/Util.hh
--- blackbox-0.70.1-orig/lib/Util.hh	2005-04-06 16:01:18.000000000 +0200
+++ blackbox-0.70.1/lib/Util.hh	2011-11-08 10:48:00.000000000 +0100
@@ -25,6 +25,7 @@
 #ifndef __Util_hh
 #define __Util_hh
 
+#include <X11/Xutil.h>
 #include <limits.h>
 #include <string>
 
@@ -94,10 +95,8 @@
 
   std::string tolower(const std::string &string);
 
-#ifdef _XUTIL_H_
   std::string textPropertyToString(::Display *display,
                                    ::XTextProperty& text_prop);
-#endif
 
 } // namespace bt
 
