diff -Naur orig/CMakeLists.txt patched/CMakeLists.txt
--- orig/CMakeLists.txt	2016-10-03 01:25:32.000000000 +0200
+++ patched/CMakeLists.txt	2017-02-05 11:51:09.186712250 +0100
@@ -1,4 +1,4 @@
-project("DB Browser for SQLite")
+project("sqlitebrowser")
 cmake_minimum_required(VERSION 2.8.7)
 
 set(CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake" "${CMAKE_MODULE_PATH}")
@@ -376,28 +376,28 @@
 endif()
 
 #cpack
-set(CPACK_PACKAGE_DESCRIPTION_SUMMARY "DB Browser for SQLite")
-set(CPACK_PACKAGE_VENDOR "DB Browser for SQLite Team")
+set(CPACK_PACKAGE_DESCRIPTION_SUMMARY "sqlitebrowser")
+set(CPACK_PACKAGE_VENDOR "sqlitebrowser Team")
 set(CPACK_PACKAGE_DESCRIPTION_FILE "${CMAKE_CURRENT_SOURCE_DIR}/README.md")
 set(CPACK_RESOURCE_FILE_LICENSE "${CMAKE_CURRENT_SOURCE_DIR}/LICENSE")
 set(CPACK_PACKAGE_VERSION_MAJOR "3")
 set(CPACK_PACKAGE_VERSION_MINOR "9")
 set(CPACK_PACKAGE_VERSION_PATCH "1")
-set(CPACK_PACKAGE_INSTALL_DIRECTORY "DB Browser for SQLite")
+set(CPACK_PACKAGE_INSTALL_DIRECTORY "sqlitebrowser")
 if(WIN32 AND NOT UNIX)
 	# There is a bug in NSIS that does not handle full unix paths properly. Make
 	# sure there is at least one set of four (4) backlasshes.
 	set(CPACK_PACKAGE_ICON "${CMAKE_CURRENT_SOURCE_DIR}\\\\src\\\\iconwin.ico")
         set(CPACK_NSIS_MUI_ICON "${CMAKE_CURRENT_SOURCE_DIR}\\\\src\\\\iconwin.ico")
 	set(CPACK_NSIS_EXECUTABLES_DIRECTORY "/")
-	set(CPACK_NSIS_INSTALLED_ICON_NAME "DB Browser for SQLite.exe")
-	set(CPACK_NSIS_DISPLAY_NAME "DB Browser for SQLite")
+	set(CPACK_NSIS_INSTALLED_ICON_NAME "sqlitebrowser.exe")
+	set(CPACK_NSIS_DISPLAY_NAME "sqlitebrowser")
 	set(CPACK_NSIS_HELP_LINK "https:\\\\\\\\github.com\\\\sqlitebrowser\\\\sqlitebrowser")
 	set(CPACK_NSIS_URL_INFO_ABOUT "https:\\\\\\\\github.com\\\\sqlitebrowser\\\\sqlitebrowser")
 	set(CPACK_NSIS_CONTACT "justin@postgresql.org")
 	set(CPACK_NSIS_MODIFY_PATH OFF)
 	set(CPACK_NSIS_ENABLE_UNINSTALL_BEFORE_INSTALL ON)
-	set(CPACK_NSIS_MUI_FINISHPAGE_RUN "DB Browser for SQLite.exe")
+	set(CPACK_NSIS_MUI_FINISHPAGE_RUN "sqlitebrowser.exe")
 	set(CPACK_NSIS_COMPRESSOR "/SOLID lzma")
 
 	# VS redist
@@ -406,8 +406,8 @@
                Delete '\\\"$INSTDIR\\\\redist\\\\${VSREDIST}\\\"'
                ")
 else(WIN32 AND NOT UNIX)
-	set(CPACK_STRIP_FILES "bin/DB Browser for SQLite")
+	set(CPACK_STRIP_FILES "bin/sqlitebrowser")
 	set(CPACK_SOURCE_STRIP_FILES "")
 endif(WIN32 AND NOT UNIX)
-set(CPACK_PACKAGE_EXECUTABLES "DB Browser for SQLite" "DB Browser for SQLite")
+set(CPACK_PACKAGE_EXECUTABLES "sqlitebrowser" "sqlitebrowser")
 include(CPack)
