--- a/data/org.gnome.Polari.desktop.in.orig	2014-04-05 13:37:30.000000000 +0200
+++ a/data/org.gnome.Polari.desktop.in	2014-04-05 13:38:23.000000000 +0200
@@ -9,7 +9,7 @@
 DBusActivatable=true
 Actions=connections;
 X-GNOME-UsesNotifications=true
-Categories=GNOME;GTK;Network;IRCClient;
+Categories=GNOME;GTK;Network;IRCClient;X-UHU-Menu/Internet/Messengers;
 _Keywords=IRC;Internet;Relay;Chat;
 
 [Desktop Action connections]
