diff -Naur orig/dltest/dltest.c patched/dltest/dltest.c
--- orig/dltest/dltest.c	2014-03-26 22:20:55.000000000 +0100
+++ patched/dltest/dltest.c	2014-06-21 12:01:22.976182121 +0200
@@ -84,7 +84,7 @@
 
   if ( argc < 2 )
   {
-      printf( szSyntax );
+      printf( "%s",szSyntax );
       exit( 1 );
   }
 
diff -Naur orig/installer/myodbc-installer.c patched/installer/myodbc-installer.c
--- orig/installer/myodbc-installer.c	2014-03-26 22:20:55.000000000 +0100
+++ patched/installer/myodbc-installer.c	2014-06-21 12:02:25.124745441 +0200
@@ -148,7 +148,7 @@
 
 void main_usage()
 {
-  fprintf(stderr, usage);
+  fprintf(stderr, "%s",usage);
 }
 
 
