diff -Naur orig/cmake/FindMySQL.cmake patched/cmake/FindMySQL.cmake
--- orig/cmake/FindMySQL.cmake	2014-03-26 22:20:55.000000000 +0100
+++ patched/cmake/FindMySQL.cmake	2014-05-17 18:37:18.169649789 +0200
@@ -106,7 +106,7 @@
   STRING(REGEX REPLACE "[.]" "" MYSQL_NUM_VERSION "${MYSQL_NUM_VERSION}")
 
   IF(MYSQL_CXX_LINKAGE OR MYSQL_NUM_VERSION GREATER 50603)
-    EXECUTE_PROCESS(COMMAND ${MYSQL_CONFIG_EXECUTABLE} "--cxxflags"
+    EXECUTE_PROCESS(COMMAND ${MYSQL_CONFIG_EXECUTABLE} "--cflags"
             OUTPUT_VARIABLE _mysql_config_output
             )
     SET(MYSQL_CXX_LINKAGE 1)
