diff -Naur orig/driver/catalog_no_i_s.c patched/driver/catalog_no_i_s.c
--- orig/driver/catalog_no_i_s.c	2014-03-26 22:20:55.000000000 +0100
+++ patched/driver/catalog_no_i_s.c	2014-05-17 19:06:10.690448338 +0200
@@ -1093,7 +1093,7 @@
   unsigned long *lengths;
   SQLRETURN rc= SQL_SUCCESS;
 
-  my_init_dynamic_array(&records, sizeof(MY_FOREIGN_KEY_FIELD), 0, 0);
+  my_init_dynamic_array(&records, sizeof(MY_FOREIGN_KEY_FIELD), 0, 0, MYF(0));
 
   /* Get the list of tables that match szCatalog and szTable */
   pthread_mutex_lock(&stmt->dbc->lock);
diff -Naur orig/driver/desc.c patched/driver/desc.c
--- orig/driver/desc.c	2014-03-26 22:20:55.000000000 +0100
+++ patched/driver/desc.c	2014-05-17 19:07:00.690836598 +0200
@@ -63,13 +63,13 @@
      but in desc_get_rec we manually get a pointer to it. This avoids
      having to call set_dynamic after modifying the DESCREC.
   */
-  if (my_init_dynamic_array(&desc->records, sizeof(DESCREC), 0, 0))
+  if (my_init_dynamic_array(&desc->records, sizeof(DESCREC), 0, 0, MYF(0)))
   {
     x_free((char *)desc);
     return NULL;
   }
 
-  if (my_init_dynamic_array(&desc->bookmark, sizeof(DESCREC), 0, 0))
+  if (my_init_dynamic_array(&desc->bookmark, sizeof(DESCREC), 0, 0, MYF(0)))
   {
     delete_dynamic(&desc->records);
     x_free((char *)desc);
@@ -997,7 +997,7 @@
   delete_dynamic(&dest->records);
   if (my_init_dynamic_array(&dest->records, sizeof(DESCREC),
                             src->records.max_element,
-                            src->records.alloc_increment))
+                            src->records.alloc_increment, MYF(0)))
   {
     return set_desc_error(dest, "HY001",
               "Memory allocation error",
diff -Naur orig/driver/handle.c patched/driver/handle.c
--- orig/driver/handle.c	2014-03-26 22:20:55.000000000 +0100
+++ patched/driver/handle.c	2014-05-17 19:07:28.815055615 +0200
@@ -393,7 +393,7 @@
     }
   }
 
-  my_init_dynamic_array(*param_bind, sizeof(MYSQL_BIND), elements, 10);
+  my_init_dynamic_array(*param_bind, sizeof(MYSQL_BIND), elements, 10, MYF(0));
   memset((*param_bind)->buffer, 0, sizeof(MYSQL_BIND) *
 											(*param_bind)->max_element);
 
diff -Naur orig/driver/parse.c patched/driver/parse.c
--- orig/driver/parse.c	2014-03-26 22:20:55.000000000 +0100
+++ patched/driver/parse.c	2014-05-17 19:07:16.910963042 +0200
@@ -129,8 +129,8 @@
 
     /* TODO: Store offsets rather than ptrs. In this case we will be fine
        if work with copy of the originally parsed string */
-    my_init_dynamic_array(&pq->token,     sizeof(uint), 20, 10);
-    my_init_dynamic_array(&pq->param_pos, sizeof(uint), 10, 10);
+    my_init_dynamic_array(&pq->token,     sizeof(uint), 20, 10, MYF(0));
+    my_init_dynamic_array(&pq->param_pos, sizeof(uint), 10, 10, MYF(0));
   }
 
   return pq;
