diff -Naurdp a/lib/HTTP/Proxy.pm b/lib/HTTP/Proxy.pm
--- a/lib/HTTP/Proxy.pm	2013-04-07 12:54:12.000000000 +0200
+++ b/lib/HTTP/Proxy.pm	2014-02-27 11:10:09.263464994 +0100
@@ -380,20 +380,20 @@ sub serve_connections {
         # the header filters created a response,
         # we won't contact the origin server
         # FIXME should the response header and body be filtered?
+        my $headresponse = $self->response;
         goto SEND if defined $self->response;
 
         # FIXME - don't forward requests to ourselves!
 
         # pop a response
-        my ( $sent, $chunked ) = ( 0, 0 );
+        my ( $init, $sent, $chunked ) = ( 0, 0, 0 );
         $response = $self->agent->simple_request(
             $req,
             sub {
                 my ( $data, $response, $proto ) = @_;
 
-                # first time, filter the headers
-                if ( !$sent ) { 
-                    $sent++;
+                if ( !$init ) { 
+                    $init++;
                     $self->response( $response );
                     
                     # select the response filters
@@ -402,30 +402,38 @@ sub serve_connections {
 
                     $self->{headers}{response}
                          ->filter( $response->headers, $response );
-                    ( $last, $chunked ) =
-                      $self->_send_response_headers( $served );
+                    $headresponse = $self->response;
                 }
-
                 # filter and send the data
                 $self->log( DATA, "DATA",
                     "got " . length($data) . " bytes of body data" );
                 $self->{body}{response}->filter( \$data, $response, $proto );
+                if (length($data)) {
+                    # first time, filter the headers
+                    if ( !$sent ) { 
+                        $sent++;
+                        ( $last, $chunked ) = $self->_send_response_headers($served);
+                    }
+
                 if ($chunked) {
                     printf $conn "%x$CRLF%s$CRLF", length($data), $data
                       if length($data);    # the filter may leave nothing
                 }
                 else { print $conn $data; }
+                }
             },
             $self->chunk
         );
 
         # remove the header added by LWP::UA before it sends the response back
         $response->remove_header('Client-Date');
+        $headeresponse = $response;
 
         # the callback is not called by LWP::UA->request
         # in some cases (HEAD, redirect, error responses have no body)
-        if ( !$sent ) {
+        if ( !$init ) {
             $self->response($response);
+            $headeresponse = $response;
             $self->{$_}{response}->select_filters( $response )
               for qw( headers body );
             $self->{headers}{response}
@@ -434,7 +442,24 @@ sub serve_connections {
 
         # do a last pass, in case there was something left in the buffers
         my $data = "";    # FIXME $protocol is undef here too
+                if ( !$init ) { 
+                    $init++;
+                    $self->response( $response );
+                    
+                    # select the response filters
+                    $self->{$_}{response}->select_filters( $response )
+                      for qw( headers body );
+
+                    $self->{headers}{response}
+                         ->filter( $response->headers, $response );
+                }
         $self->{body}{response}->filter_last( \$data, $response, undef );
+					$headresponse ||= $self->response;
+   	             if ( !$sent ) { 
+				 	 $sent++;
+   	                 ( $last, $chunked ) =
+   	                   $self->_send_response_headers( $served);#, $headresponse );
+   	             }
         if ( length $data ) {
             if ($chunked) {
                 printf $conn "%x$CRLF%s$CRLF", length($data), $data;
@@ -454,6 +479,7 @@ sub serve_connections {
             $response->content_type( "text/plain" );
             $response->content($died);
             $self->response($response);
+            $headresponse = $response;
         }
 
       SEND:
@@ -495,10 +521,10 @@ sub serve_connections {
 # expects $served  (number of requests served)
 # returns $last and $chunked (last request served, chunked encoding)
 sub _send_response_headers {
-    my ( $self, $served ) = @_;
+    my ( $self, $served, $response ) = @_;
     my ( $last, $chunked ) = ( 0, 0 );
     my $conn = $self->client_socket;
-    my $response = $self->response;
+    $response ||= $self->response;
 
     # correct headers
     $response->remove_header("Content-Length")
