diff -Naur orig/lib/HTTP/Proxy.pm mod/lib/HTTP/Proxy.pm
--- orig/lib/HTTP/Proxy.pm	2015-06-16 02:47:39.000000000 +0200
+++ mod/lib/HTTP/Proxy.pm	2016-02-24 14:00:06.254816778 +0100
@@ -386,20 +386,20 @@
         # the header filters created a response,
         # we won't contact the origin server
         # FIXME should the response header and body be filtered?
+        my $headresponse = $self->response;
         goto SEND if defined $self->response;
 
         # FIXME - don't forward requests to ourselves!
 
         # pop a response
-        my ( $sent, $chunked ) = ( 0, 0 );
+        my ( $init, $sent, $chunked ) = ( 0, 0, 0 );
         $response = $self->agent->simple_request(
             $req,
             sub {
                 my ( $data, $response, $proto ) = @_;
 
-                # first time, filter the headers
-                if ( !$sent ) {
-                    $sent++;
+                if ( !$init ) { 
+                    $init++;
                     $self->response( $response );
 
                     # select the response filters
@@ -408,14 +408,20 @@
 
                     $self->{headers}{response}
                          ->filter( $response->headers, $response );
-                    ( $last, $chunked ) =
-                      $self->_send_response_headers( $served );
+                    $headresponse = $self->response;
                 }
 
                 # filter and send the data
                 $self->log( DATA, "DATA",
                     "got " . length($data) . " bytes of body data" );
                 $self->{body}{response}->filter( \$data, $response, $proto );
+                if (length($data)) {
+                    # first time, filter the headers
+                    if ( !$sent ) { 
+                        $sent++;
+                        ( $last, $chunked ) = $self->_send_response_headers($served);
+                    }
+
                 if ($chunked) {
                     printf $conn "%x$CRLF%s$CRLF", length($data), $data
                       if length($data);    # the filter may leave nothing
@@ -427,11 +433,13 @@
 
         # remove the header added by LWP::UA before it sends the response back
         $response->remove_header('Client-Date');
+        $headeresponse = $response;
 
         # the callback is not called by LWP::UA->request
         # in some cases (HEAD, redirect, error responses have no body)
-        if ( !$sent ) {
+        if ( !$init ) {
             $self->response($response);
+            $headeresponse = $response;
             $self->{$_}{response}->select_filters( $response )
               for qw( headers body );
             $self->{headers}{response}
@@ -440,7 +448,24 @@
 
         # do a last pass, in case there was something left in the buffers
         my $data = "";    # FIXME $protocol is undef here too
+                if ( !$init ) { 
+                    $init++;
+                    $self->response( $response );
+                    
+                    # select the response filters
+                    $self->{$_}{response}->select_filters( $response )
+                      for qw( headers body );
+
+                    $self->{headers}{response}
+                         ->filter( $response->headers, $response );
+                }
         $self->{body}{response}->filter_last( \$data, $response, undef );
+					$headresponse ||= $self->response;
+   	             if ( !$sent ) { 
+				 	 $sent++;
+   	                 ( $last, $chunked ) =
+   	                   $self->_send_response_headers( $served);#, $headresponse );
+   	             }
         if ( length $data ) {
             if ($chunked) {
                 printf $conn "%x$CRLF%s$CRLF", length($data), $data;
@@ -460,6 +485,7 @@
             $response->content_type( "text/plain" );
             $response->content($died);
             $self->response($response);
+            $headresponse = $response;
         }
 
       SEND:
@@ -501,10 +527,10 @@
 # expects $served  (number of requests served)
 # returns $last and $chunked (last request served, chunked encoding)
 sub _send_response_headers {
-    my ( $self, $served ) = @_;
+    my ( $self, $served, $response ) = @_;
     my ( $last, $chunked ) = ( 0, 0 );
     my $conn = $self->client_socket;
-    my $response = $self->response;
+    $response ||= $self->response;
 
     # correct headers
     $response->remove_header("Content-Length")
