diff -Naur orig/Makefile mod/Makefile
--- orig/Makefile	2016-01-10 21:41:46.000000000 +0100
+++ mod/Makefile	2016-04-24 22:19:52.820810011 +0200
@@ -75,7 +75,7 @@
 quick:
 	$(MAKE) target=UNIX \
 		CC=gcc CXX=g++ \
-		opt='-DTEST -DTRACE -DINLINEASM -DPOPCNT -DCPUS=4' \
+		opt='-DTEST -DTRACE -DPOPCNT -DCPUS=4' \
 		CFLAGS='-Wall -Wno-array-bounds -pipe -O3 -pthread' \
 		CXFLAGS='-Wall -Wno-array-bounds -pipe -O3 -pthread' \
 		LDFLAGS='$(LDFLAGS) -lstdc++' \
@@ -84,7 +84,7 @@
 unix-gcc:
 	$(MAKE) -j target=UNIX \
 		CC=gcc CXX=g++ \
-		opt='-DTEST -DINLINEASM -DPOPCNT -DCPUS=4' \
+		opt='-DTEST -DPOPCNT -DCPUS=4' \
 		CFLAGS='-Wall -Wno-array-bounds -pipe -O3 -fprofile-use \
 		-fprofile-correction -pthread' \
 		CXFLAGS='-Wall -Wno-array-bounds -pipe -O3 -fprofile-use \
@@ -95,7 +95,7 @@
 unix-gcc-profile:
 	$(MAKE) -j target=UNIX \
 		CC=gcc CXX=g++ \
-		opt='-DTEST -DINLINEASM -DPOPCNT -DCPUS=4' \
+		opt='-DTEST -DPOPCNT -DCPUS=4' \
 		CFLAGS='-Wall -Wno-array-bounds -pipe -O3 -fprofile-arcs \
 		-pthread' \
 		CXFLAGS='-Wall -Wno-array-bounds -pipe -O3 -fprofile-arcs \
diff -Naur orig/option.c mod/option.c
--- orig/option.c	2016-01-15 23:18:18.000000000 +0100
+++ mod/option.c	2016-04-24 22:05:07.840245875 +0200
@@ -1068,7 +1068,7 @@
     char *readstat = (char *) -1;
     int lines = 0;
 
-    helpfile = fopen("crafty.hlp", "r");
+    helpfile = fopen("BBOKDIR/crafty.hlp", "r");
     if (!helpfile) {
       printf("ERROR.  Unable to open \"crafty.hlp\" -- help unavailable\n");
       return 1;
