diff -Naurd orig/sed/execute.c mod/sed/execute.c
--- orig/sed/execute.c	2012-03-16 10:13:31.000000000 +0100
+++ mod/sed/execute.c	2013-12-05 22:17:00.000000000 +0100
@@ -1151,6 +1151,9 @@
   /* Exchange line and s_accum.  This can be much cheaper
      than copying s_accum.active into line.text (for huge lines). */
   line_exchange(&line, &s_accum, false);
+#ifdef HAVE_MBRTOWC
+  memcpy(&s_accum.mbstate, &line.mbstate, sizeof(line.mbstate));
+#endif
 
   /* Finish up. */
   if (count < sub->numb)
@@ -1364,6 +1367,9 @@
 			 (for huge lines).  See comment above for 'g' as
 			 to while the third argument is incorrect anyway.  */
 		      line_exchange(&line, &s_accum, true);
+#ifdef HAVE_MBRTOWC
+		      memcpy(&s_accum.mbstate, &line.mbstate, sizeof(line.mbstate));
+#endif
 		    }
                   else
                     flush_output(output_file.fp);
