diff -Naur orig/operations/external/ff-load.c patched/operations/external/ff-load.c
--- orig/operations/external/ff-load.c	2015-11-21 02:48:30.000000000 +0100
+++ patched/operations/external/ff-load.c	2015-11-22 22:19:20.595708555 +0100
@@ -56,6 +56,10 @@
 
 #include <libavformat/avformat.h>
 
+#ifndef AV_CODEC_FLAG2_FAST
+#define AV_CODEC_FLAG2_FAST CODEC_FLAG2_FAST
+#endif
+
 typedef struct
 {
   gint             width;
