diff -Naur orig/src/flDesktop.cpp mod/src/flDesktop.cpp
--- orig/src/flDesktop.cpp	2014-12-28 03:56:53.000000000 +0100
+++ mod/src/flDesktop.cpp	2015-04-08 19:53:39.820862113 +0200
@@ -433,7 +433,6 @@
     std::string inode ="inode/directory=";
     const char* defaults="/usr/share/applications/defaults.list";
     bool defaultsList = testFile(defaults);
-    if (!defaultsList){
         bool roxInstalled = testExec("rox-filer");
         bool pcmanfmInstalled = testExec("pcmanfm");
         bool nautilusInstalled = testExec("nautilus");
@@ -445,19 +444,4 @@
         else{result = "unknown ";}
         result += "INSTALLED";
         return result.c_str();
-    }
-    result = grep(inode.c_str(), defaults);
-    //std::cout<<result<<std::endl;
-    int len = inode.length();
-    std::string testResult = result;
-    if (len >= testResult.length()){return result.c_str();}
-    std::string STRresult = result;
-    STRresult = STRresult.substr(len, std::string::npos);
-    //std::cout<<STRresult<<std::endl;
-    int position = STRresult.find('.');
-    STRresult = STRresult.erase(position, std::string::npos);
-    std::cout<<STRresult<<std::endl;
-    result=STRresult;
-    //std::cout<<result<<"\n";
-    return result.c_str();
 }
