diff -Naur orig/data/rox.desktop mod/data/rox.desktop
--- orig/data/rox.desktop	2014-12-28 03:56:53.000000000 +0100
+++ mod/data/rox.desktop	2015-04-08 20:51:25.555755145 +0200
@@ -2,10 +2,10 @@
 Name=Rox
 GenericName=File Manager
 Icon=file-manager
-Exec=rox-filer -n
+Exec=rox -n
 Keywords=folder;manager;explore;disk;filesystem;
 Terminal=false
 Type=Application
-Categories=Utilities;Core;FileManager;
+Categories=Core;FileManager;
 Name[en_US]=Rox
 MimeType=inode/directory;
diff -Naur orig/fltk/jwm-desktop.cxx mod/fltk/jwm-desktop.cxx
--- orig/fltk/jwm-desktop.cxx	2014-12-28 03:56:53.000000000 +0100
+++ mod/fltk/jwm-desktop.cxx	2015-04-08 20:49:29.311752907 +0200
@@ -563,7 +563,7 @@
 }
 
 void DesktopUI::cb_Choose4_i(Fl_Button*, void*) {
-  int e =system("rox-filer -n /usr/share/applications &disown");
+  int e =system("rox -n /usr/share/applications &disown");
 if(e){/*do something*/};
 }
 void DesktopUI::cb_Choose4(Fl_Button* o, void* v) {
diff -Naur orig/fltk/jwm-desktop.fld mod/fltk/jwm-desktop.fld
--- orig/fltk/jwm-desktop.fld	2014-12-28 03:56:53.000000000 +0100
+++ mod/fltk/jwm-desktop.fld	2015-04-08 20:49:56.947753439 +0200
@@ -186,7 +186,7 @@
         }
         Fl_Button {} {
           label Choose
-          callback {int e =system("rox-filer -n /usr/share/applications &disown");
+          callback {int e =system("rox -n /usr/share/applications &disown");
 if(e){/*do something*/}}
           tooltip {Drag the files onto the desktop} xywh {405 320 85 25} box GTK_UP_BOX
         }
diff -Naur orig/src/flAutostart.cpp mod/src/flAutostart.cpp
--- orig/src/flAutostart.cpp	2014-12-28 03:56:53.000000000 +0100
+++ mod/src/flAutostart.cpp	2015-04-08 20:48:45.607752066 +0200
@@ -74,9 +74,9 @@
 */
 void flAutostart::useROX(bool rox){
     if(rox){
-        addAutostart("rox-filer -p pb_JWM");
+        addAutostart("roxr -p pb_JWM");
     }
     else{
-        removeAutostart("rox-filer -p pb_JWM");
+        removeAutostart("rox -p pb_JWM");
     }
 }
