diff -Naur orig/data/themes/Ambiance/Ambiance mod/data/themes/Ambiance/Ambiance
--- orig/data/themes/Ambiance/Ambiance	2014-12-28 03:56:53.000000000 +0100
+++ mod/data/themes/Ambiance/Ambiance	2015-03-31 11:26:42.000000000 +0200
@@ -1,30 +1,39 @@
 <?xml version="1.0"?>
 <JWM>
-
-<!-- MENUS -->
-
+    <!-- MENUS -->
     <!-- Main Menu -->
-    <Include>$HOME/.config/torimenu</Include>
+    <RootMenu height="0" onroot="5">
+        <Include>$HOME/.jwmrc-mjwm</Include>
+        <Program icon="uhu-64x64.png" label="Menügenerátor">jwmuhumenu</Program>
+    </RootMenu>
     <!--Shudown Menu-->
     <RootMenu height="0" onroot="9">
-        <Program icon="preferences-system-session.svg" label="Shutdown Menu">jwm-settings-manager --halt</Program>
-        <Program icon="system-restart.svg" label="Restart Computer">dbus-send --system --print-reply --dest=org.freedesktop.ConsoleKit /org/freedesktop/ConsoleKit/Manager org.freedesktop.ConsoleKit.Manager.Restart</Program>
-        <Restart label="Reload Desktop" icon="system-restart.svg"/>
-        <Exit label="Logout" confirm="true" icon="system-logout.svg"/>
+        <Restart label="JWM restart" icon="reload.png"/>
+        <Exit label="Kilépés" confirm="true" icon="system-log-out.png"/>
+        <Program icon="redo.png" label="Újraindítás">systemctl reboot</Program>
+        <Program icon="system-shutdown.png" label="Kikapcsolás">systemctl poweroff</Program>
+    </RootMenu>
+    <!--Places Menu-->
+    <RootMenu height="0" onroot="7">
+        <Program icon="application-x-gnome-saved-search.png" label="Applications">xdg-open /usr/share/applications</Program>
+        <Program icon="folder_home.png" label="Home">xdg-open ~/</Program>
+        <Program icon="folder-documents.png" label="Dokumentumok">xdg-open ~/Dokumentumok</Program>
+        <Program icon="folder-download.png" label="Letöltések">xdg-open ~/Letöltések</Program>
+        <Program icon="folder-music.png" label="Zenék">xdg-open -n ~/Zenék</Program>
+        <Program icon="folder-pictures.png" label="Képek">xdg-open ~/Képek</Program>
+        <Program icon="folder-videos.png" label="Videók">xdg-open ~/Videók</Program>
+        <Program icon="user-trash.png" label="Trash">xdg-open ~/.local/share/Trash/files</Program>
+        <Program icon="file-manager.png" label="Connected Devices">xdg-open /media</Program>
     </RootMenu>
-        <!--Places Menu-->
-    <Include>$HOME/.config/placesmenu</Include>
     <!-- Mouse Menu-->
     <RootMenu height="16" onroot="23">
-        <Program icon="application-default-icon.svg" label="Applications">xdg-open /usr/share/applications</Program>
-        <Program icon="folder-home.svg" label="Home">xdg-open ~/</Program>
+        <Include>$HOME/.jwmrc-mjwm</Include>
+        <Program icon="uhu-64x64.png" label="Menügenerátor">jwmuhumenu</Program>
+        <Program icon="folder_home.png" label="Home">xdg-open ~/</Program>
         <Program icon="jwm-settings-manager.svg" label="Settings">jwm-settings-manager</Program>
-        <Program icon="preferences-desktop-wallpaper.svg" label="Change Desktop Background">jwm-settings-manager --desktop</Program>
     </RootMenu>
     <!-- END MENUS -->
-
-
-<!-- GROUP -->
+    <!-- GROUP -->
     <Group>
         <Class>Pidgin</Class>
         <Option>sticky</Option>
@@ -43,22 +52,17 @@
         <Name>jwm-settings-manager</Name>
         <Option>icon:jwm-settings-manager.png</Option>
     </Group>
-
-
-<!-- PANEL -->
+    <!-- PANEL -->
     <Tray x="0" y="-1" height="34" autohide="false" valign="top" width="0" halign="fixed" layout="horizontal">
-        <TrayButton label="Apps" icon="/usr/share/pixmaps/distributor-logo.png" border="false">root:5</TrayButton>
-        <TrayButton label="Places" icon="folder.svg" border="false">root:7</TrayButton>
-        <TrayButton icon="browser.svg" border="false">exec:xdg-open /usr/share/doc/ToriOS/startpage.html</TrayButton>
+        <TrayButton label="Programok" icon="distributor-logo.png" border="false">root:5</TrayButton>
+        <TrayButton label="Helyek" icon="folder.png" border="false">root:7</TrayButton>
         <Pager labeled="true"/>
         <TaskList maxwidth="256"/>
         <Dock/>
         <Swallow name="xload" width="64">xload -nolabel -bg DimGrey -fg Grey -hl DarkGrey</Swallow>
         <Clock format="%a, %e %b %l:%M %p">xclock</Clock>
-        <TrayButton label="Shutdown" icon="system-shutdown.svg" border="false">root:9</TrayButton>
+        <TrayButton label="Kikapcsolás" icon="system-shutdown.png" border="false">root:9</TrayButton>
     </Tray>
-
-
     <!-- Panel Style -->
     <TaskListStyle>
         <ActiveBackground>#5A584F</ActiveBackground>
@@ -69,11 +73,11 @@
     <TrayStyle>
         <Font>ubuntu-12:antialias=true:encoding=utf8</Font>
         <Foreground>#ffffff</Foreground>
-        <Background>#484741</Background>
+        <Background>#5e5d55</Background>
         <Opacity>0.75</Opacity>
     </TrayStyle>
     <TrayButtonStyle>
-	    <Font>ubuntu-12:antialias=true:encoding=utf8</Font>
+        <Font>ubuntu-12:antialias=true:encoding=utf8</Font>
         <Foreground>#ffffff</Foreground>
         <Background>#484741</Background>
     </TrayButtonStyle>
@@ -91,8 +95,7 @@
         <ActiveBackground>#E45C2D</ActiveBackground>
         <Opacity>0.85</Opacity>
     </MenuStyle>
-
-<!-- WINDOW -->
+    <!-- WINDOW -->
     <WindowStyle>
         <Font>ubuntu-12:antialias=true:encoding=utf8</Font>
         <Width>4</Width>
@@ -118,8 +121,7 @@
     <SnapMode distance="10">border</SnapMode>
     <MoveMode>opaque</MoveMode>
     <ResizeMode>opaque</ResizeMode>
-
-<Key key="Up">up</Key>
+    <Key key="Up">up</Key>
     <Key key="Down">down</Key>
     <Key key="Right">right</Key>
     <Key key="Left">left</Key>
@@ -156,27 +158,23 @@
     <DoubleClickDelta>2</DoubleClickDelta>
     <!-- ICONS -->
     <IconPath>/usr/share/pixmaps</IconPath>
-    <IconPath>/usr/share/icons/Numix/48x48/actions/</IconPath>
-    <IconPath>/usr/share/icons/Numix/48x48/categories/</IconPath>
-    <IconPath>/usr/share/icons/Numix/48x48/places/</IconPath>
-    <IconPath>/usr/share/icons/Numix/48x48/status/</IconPath>
-    <IconPath>/usr/share/icons/Numix-Circle/48x48/apps</IconPath>
-<!-- DESKTOP -->
+    <IconPath>/usr/share/icons/gnome/48x48/actions</IconPath>
+    <IconPath>/usr/share/icons/gnome/48x48/places</IconPath>
+    <IconPath>/usr/share/icons/gnome/48x48/apps</IconPath>
+    <IconPath>/usr/share/icons/gnome/48x48/devices</IconPath>
+    <!-- DESKTOP -->
     <Desktops width="2" height="1">
         <Background type="image">/usr/share/backgrounds/default.png</Background>
     </Desktops>
-
-<!-- AUTOSTART -->
-   <StartupCommand>nm-applet</StartupCommand>
-   <StartupCommand>volumeicon</StartupCommand>
-   <StartupCommand>synclient MaxTapTime=0</StartupCommand>
-
-<!--POP UP -->
+    <!-- AUTOSTART -->
+    <StartupCommand>nm-applet</StartupCommand>
+    <StartupCommand>pasystray</StartupCommand>
+    <StartupCommand>synclient MaxTapTime=0</StartupCommand>
+    <!--POP UP -->
     <PopupStyle>
         <Font>ubuntu-12:antialias=true:encoding=utf8</Font>
         <Outline>#000000</Outline>
         <Foreground>#ffffff</Foreground>
         <Background>#484741</Background>
     </PopupStyle>
-
 </JWM>
diff -Naur orig/data/themes/Bassem/Bassem mod/data/themes/Bassem/Bassem
--- orig/data/themes/Bassem/Bassem	2014-12-28 03:56:53.000000000 +0100
+++ mod/data/themes/Bassem/Bassem	2015-04-22 21:20:25.000000000 +0200
@@ -4,32 +4,41 @@
 Israel <israel@torios.org>
 MIT license copyright 2014
 -->
-
 <JWM>
-
-<!-- MENUS -->
-<!-- Main Menu -->
-    <Include>$HOME/.config/torimenu</Include>
+    <!-- MENUS -->
+    <!-- Main Menu -->
+    <RootMenu height="0" onroot="5">
+        <Include>$HOME/.jwmrc-mjwm</Include>
+        <Program icon="uhu-64x64.png" label="Menügenerátor">jwmuhumenu</Program>
+    </RootMenu>
     <!--Shudown Menu-->
     <RootMenu height="0" onroot="9">
-        <Program icon="preferences-system-session.svg" label="Shutdown Menu">jwm-settings-manager --halt</Program>
-        <Program icon="system-restart.svg" label="Restart Computer">dbus-send --system --print-reply --dest=org.freedesktop.ConsoleKit /org/freedesktop/ConsoleKit/Manager org.freedesktop.ConsoleKit.Manager.Restart</Program>
-        <Restart label="Reload Desktop" icon="system-restart.svg"/>
-        <Exit label="Logout" confirm="true" icon="system-logout.svg"/>
+        <Restart label="JWM restart" icon="reload.png"/>
+        <Exit label="Kilépés" confirm="true" icon="system-log-out.png"/>
+        <Program icon="redo.png" label="Újraindítás">systemctl reboot</Program>
+        <Program icon="system-shutdown.png" label="Kikapcsolás">systemctl poweroff</Program>
     </RootMenu>
     <!--Places Menu-->
-    <Include>$HOME/.config/placesmenu</Include>
+    <RootMenu height="0" onroot="7">
+        <Program icon="application-x-gnome-saved-search.png" label="Applications">xdg-open /usr/share/applications</Program>
+        <Program icon="folder_home.png" label="Home">xdg-open ~/</Program>
+        <Program icon="folder-documents.png" label="Dokumentumok">xdg-open ~/Dokumentumok</Program>
+        <Program icon="folder-download.png" label="Letöltések">xdg-open ~/Letöltések</Program>
+        <Program icon="folder-music.png" label="Zenék">xdg-open -n ~/Zenék</Program>
+        <Program icon="folder-pictures.png" label="Képek">xdg-open ~/Képek</Program>
+        <Program icon="folder-videos.png" label="Videók">xdg-open ~/Videók</Program>
+        <Program icon="user-trash.png" label="Trash">xdg-open ~/.local/share/Trash/files</Program>
+        <Program icon="file-manager.png" label="Connected Devices">xdg-open /media</Program>
+    </RootMenu>
     <!-- Mouse Menu-->
     <RootMenu height="16" onroot="23">
-        <Program icon="application-default-icon.svg" label="Applications">xdg-open /usr/share/applications</Program>
-        <Program icon="folder-home.svg" label="Home">xdg-open ~/</Program>
+        <Include>$HOME/.jwmrc-mjwm</Include>
+        <Program icon="uhu-64x64.png" label="Menügenerátor">jwmuhumenu</Program>
+        <Program icon="folder_home.svg" label="Home">xdg-open ~/</Program>
         <Program icon="jwm-settings-manager.svg" label="Settings">jwm-settings-manager</Program>
-        <Program icon="preferences-desktop-wallpaper.svg" label="Change Desktop Background">jwm-settings-manager --desktop</Program>
     </RootMenu>
     <!-- END MENUS -->
-
-
-<!-- GROUP -->
+    <!-- GROUP -->
     <Group>
         <Class>Pidgin</Class>
         <Option>sticky</Option>
@@ -48,28 +57,25 @@
         <Name>jwm-settings-manager</Name>
         <Option>icon:jwm-settings-manager.png</Option>
     </Group>
-
-
-<!-- PANEL -->
+    <!-- PANEL -->
     <Tray x="0" y="-1" height="34" autohide="false" valign="top" width="0" halign="fixed" layout="horizontal">
         <TrayButton icon="distributor-logo.png" border="false">root:5</TrayButton>
         <Spacer/>
-        <Clock format="%a, %e %b %l:%M %p"></Clock>
+        <Clock format="%a, %e %b %l:%M %p"/>
         <Spacer/>
-        <TrayButton label="Shutdown" icon="system-shutdown.svg" border="false">root:9</TrayButton>
+        <TrayButton label="Kikapcsolás" icon="system-shutdown.svg" border="false">root:9</TrayButton>
     </Tray>
     <Tray x="0" y="-1" height="34" autohide="false" valign="bottom" width="0" halign="fixed" layout="horizontal">
         <Pager labeled="true"/>
         <TaskList maxwidth="256"/>
         <Swallow name="xload" width="64">xload -nolabel -bg DimGrey -fg Grey -hl DarkGrey</Swallow>
-        <TrayButton label="Places" icon="folder.svg" border="false">root:7</TrayButton>
+        <TrayButton label="Helyek" icon="folder.svg" border="false">root:7</TrayButton>
         <Dock/>
     </Tray>
-
-<!-- Panel Style -->
+    <!-- Panel Style -->
     <TaskListStyle>
         <ActiveBackground>#4A24AB</ActiveBackground>
-        <Background>#353535</Background>
+        <Background>#353535</Background>444444
         <Font>ubuntu-12:antialias=true:encoding=utf8</Font>
         <Foreground>#ffffff</Foreground>
     </TaskListStyle>
@@ -98,8 +104,7 @@
         <ActiveBackground>#4A24AB</ActiveBackground>
         <Opacity>0.85</Opacity>
     </MenuStyle>
-
-<!-- WINDOW -->
+    <!-- WINDOW -->
     <WindowStyle>
         <Font>ubuntu-12:antialias=true:encoding=utf8</Font>
         <Width>4</Width>
@@ -121,8 +126,7 @@
     <SnapMode distance="10">border</SnapMode>
     <MoveMode>opaque</MoveMode>
     <ResizeMode>opaque</ResizeMode>
-
-<Key key="Up">up</Key>
+    <Key key="Up">up</Key>
     <Key key="Down">down</Key>
     <Key key="Right">right</Key>
     <Key key="Left">left</Key>
@@ -159,28 +163,23 @@
     <DoubleClickDelta>2</DoubleClickDelta>
     <!-- ICONS -->
     <IconPath>/usr/share/pixmaps</IconPath>
-    <IconPath>/usr/share/icons/Numix/48x48/actions/</IconPath>
-    <IconPath>/usr/share/icons/Numix/48x48/categories/</IconPath>
-    <IconPath>/usr/share/icons/Numix/48x48/places/</IconPath>
-    <IconPath>/usr/share/icons/Numix/48x48/status/</IconPath>
-    <IconPath>/usr/share/icons/Numix-Circle/48x48/apps</IconPath>
-
-<!-- DESKTOP -->
+    <IconPath>/usr/share/icons/gnome/48x48/actions</IconPath>
+    <IconPath>/usr/share/icons/gnome/48x48/places</IconPath>
+    <IconPath>/usr/share/icons/gnome/48x48/apps</IconPath>
+    <IconPath>/usr/share/icons/gnome/48x48/devices</IconPath>
+    <!-- DESKTOP -->
     <Desktops width="2" height="1">
         <Background type="image">/usr/share/backgrounds/default.png</Background>
     </Desktops>
-
-<!-- AUTOSTART -->
-   <StartupCommand>nm-applet</StartupCommand>
-   <StartupCommand>volumeicon</StartupCommand>
-   <StartupCommand>synclient MaxTapTime=0</StartupCommand>
-
-<!--POP UP -->
+    <!-- AUTOSTART -->
+    <StartupCommand>nm-applet</StartupCommand>
+    <StartupCommand>retrovol -hide</StartupCommand>
+    <StartupCommand>synclient MaxTapTime=0</StartupCommand>
+    <!--POP UP -->
     <PopupStyle>
         <Font>ubuntu-12:antialias=true:encoding=utf8</Font>
         <Outline>#000000</Outline>
         <Foreground>#ffffff</Foreground>
         <Background>#353535</Background>
     </PopupStyle>
-
 </JWM>
diff -Naur orig/data/themes/DarkGnome2/DarkGnome2 mod/data/themes/DarkGnome2/DarkGnome2
--- orig/data/themes/DarkGnome2/DarkGnome2	2014-12-28 03:56:53.000000000 +0100
+++ mod/data/themes/DarkGnome2/DarkGnome2	2015-04-22 21:18:19.000000000 +0200
@@ -4,31 +4,41 @@
 Israel <israel@torios.org>
 MIT license copyright 2014
 -->
-
 <JWM>
-
-<!-- MENUS -->
-<!-- Main Menu -->
-    <Include>$HOME/.config/torimenu</Include>
+    <!-- MENUS -->
+    <!-- Main Menu -->
+    <RootMenu height="0" onroot="5">
+        <Include>$HOME/.jwmrc-mjwm</Include>
+        <Program icon="uhu-64x64.png" label="Menügenerátor">jwmuhumenu</Program>
+    </RootMenu>
     <!--Shudown Menu-->
     <RootMenu height="0" onroot="9">
-        <Program icon="preferences-system-session.svg" label="Shutdown Menu">jwm-settings-manager --halt</Program>
-        <Program icon="system-restart.svg" label="Restart Computer">dbus-send --system --print-reply --dest=org.freedesktop.ConsoleKit /org/freedesktop/ConsoleKit/Manager org.freedesktop.ConsoleKit.Manager.Restart</Program>
-        <Restart label="Reload Desktop" icon="system-restart.svg"/>
-        <Exit label="Logout" confirm="true" icon="system-logout.svg"/>
+        <Restart label="JWM restart" icon="reload.png"/>
+        <Exit label="Kilépés" confirm="true" icon="system-log-out.png"/>
+        <Program icon="redo.png" label="Újraindítás">systemctl reboot</Program>
+        <Program icon="system-shutdown.png" label="Kikapcsolás">systemctl poweroff</Program>
     </RootMenu>
     <!--Places Menu-->
-    <Include>$HOME/.config/placesmenu</Include>
+    <RootMenu height="0" onroot="7">
+        <Program icon="application-x-gnome-saved-search.png" label="Applications">xdg-open /usr/share/applications</Program>
+        <Program icon="folder_home.png" label="Home">xdg-open ~/</Program>
+        <Program icon="folder-documents.png" label="Dokumentumok">xdg-open ~/Dokumentumok</Program>
+        <Program icon="folder-download.png" label="Letöltések">xdg-open ~/Letöltések</Program>
+        <Program icon="folder-music.png" label="Zenék">xdg-open -n ~/Zenék</Program>
+        <Program icon="folder-pictures.png" label="Képek">xdg-open ~/Képek</Program>
+        <Program icon="folder-videos.png" label="Videók">xdg-open ~/Videók</Program>
+        <Program icon="user-trash.png" label="Trash">xdg-open ~/.local/share/Trash/files</Program>
+        <Program icon="file-manager.png" label="Connected Devices">xdg-open /media</Program>
+    </RootMenu>
     <!-- Mouse Menu-->
     <RootMenu height="16" onroot="23">
-        <Program icon="application-default-icon.svg" label="Applications">xdg-open /usr/share/applications</Program>
-        <Program icon="folder-home.svg" label="Home">xdg-open ~/</Program>
+        <Include>$HOME/.jwmrc-mjwm</Include>
+        <Program icon="uhu-64x64.png" label="Menügenerátor">jwmuhumenu</Program>
+        <Program icon="folder_home.svg" label="Home">xdg-open ~/</Program>
         <Program icon="jwm-settings-manager.svg" label="Settings">jwm-settings-manager</Program>
-        <Program icon="preferences-desktop-wallpaper.svg" label="Change Desktop Background">jwm-settings-manager --desktop</Program>
     </RootMenu>
     <!-- END MENUS -->
-    
-<!-- GROUP -->
+    <!-- GROUP -->
     <Group>
         <Class>Pidgin</Class>
         <Option>sticky</Option>
@@ -47,29 +57,25 @@
         <Name>jwm-settings-manager</Name>
         <Option>icon:jwm-settings-manager.png</Option>
     </Group>
-
-<!-- PANEL -->
+    <!-- PANEL -->
     <Tray height="34" autohide="false" valign="top" width="0" halign="fixed" layout="horizontal">
-        <TrayButton label="Apps" icon="/usr/share/pixmaps/distributor-logo.png" border="false">root:5</TrayButton>
-        <TrayButton label="Places" icon="folder.svg" border="false">root:7</TrayButton>
-        <TrayButton icon="browser.svg" border="false">exec:xdg-open /usr/share/doc/ToriOS/startpage.html</TrayButton>
-
+        <TrayButton label="Programok" icon="distributor-logo.png" border="false">root:5</TrayButton>
+        <TrayButton label="Helyek" icon="folder.png" border="false">root:7</TrayButton>
         <Spacer/>
         <Dock/>
         <Swallow name="xload" width="64">xload -nolabel -bg DimGrey -fg Grey -hl DarkGrey</Swallow>
         <Clock format="%a, %e %b %l:%M %p">xclock</Clock>
-        <TrayButton label="Shutdown" icon="system-shutdown.svg" border="false">root:9</TrayButton>
+        <TrayButton label="Kikapcsolás" icon="system-shutdown.svg" border="false">root:9</TrayButton>
     </Tray>
     <Tray x="0" y="-1" height="34" autohide="false" valign="bottom" width="0" halign="fixed" layout="horizontal">
-	<TrayButton icon="preferences-system-hotcorners.png" border="false">showdesktop</TrayButton>
+        <TrayButton icon="preferences-system-hotcorners.png" border="false">showdesktop</TrayButton>
         <TaskList maxwidth="256"/>
         <Pager labeled="false"/>
     </Tray>
-
     <!-- Panel Style -->
     <TaskListStyle>
-        <ActiveBackground>#A5A5A5</ActiveBackground>
-        <Background>#595959</Background>
+        <ActiveBackground>#BBBBBB:#000000</ActiveBackground>
+        <Background>#666666:#000000</Background>
         <Font>ubuntu-12:antialias=true:encoding=utf8</Font>
         <ActiveForeground>#ffffff</ActiveForeground>
         <Foreground>#BFBFBF</Foreground>
@@ -77,13 +83,14 @@
     <TrayStyle>
         <Font>ubuntu-12:antialias=true:encoding=utf8</Font>
         <Foreground>#ffffff</Foreground>
-        <Background>#464646</Background>
+        <Background>#58564D</Background>
         <Opacity>0.75</Opacity>
     </TrayStyle>
     <TrayButtonStyle>
-	    <Font>ubuntu-12:antialias=true:encoding=utf8</Font>
+        <Font>ubuntu-12:antialias=true:encoding=utf8</Font>
         <Foreground>#ffffff</Foreground>
-        <Background>#2F2F2F</Background>
+        <ActiveBackground>#AAAAAA:#000000</ActiveBackground>
+        <Background>#333333</Background>
     </TrayButtonStyle>
     <PagerStyle>
         <Outline>#000000</Outline>
@@ -99,21 +106,20 @@
         <ActiveBackground>#406477</ActiveBackground>
         <Opacity>0.85</Opacity>
     </MenuStyle>
-
-<!-- WINDOW -->
+    <!-- WINDOW -->
     <WindowStyle>
         <Font>ubuntu-12:antialias=true:encoding=utf8</Font>
         <Width>4</Width>
         <Height>20</Height>
         <Active>
             <Text>#ffffff</Text>
-            <Title>#58564D:#46443F</Title>
+            <Title>#BBBBBB:#000000</Title>
             <Outline>#353535</Outline>
             <Opacity>1.0</Opacity>
         </Active>
         <Inactive>
             <Text>#A1A1A1</Text>
-            <Title>#58564D:#46443F</Title>
+            <Title>#888888:#555555</Title>
             <Outline>#353535</Outline>
             <Opacity>0.7</Opacity>
         </Inactive>
@@ -122,8 +128,7 @@
     <SnapMode distance="10">border</SnapMode>
     <MoveMode>opaque</MoveMode>
     <ResizeMode>opaque</ResizeMode>
-
-<Key key="Up">up</Key>
+    <Key key="Up">up</Key>
     <Key key="Down">down</Key>
     <Key key="Right">right</Key>
     <Key key="Left">left</Key>
@@ -160,27 +165,23 @@
     <DoubleClickDelta>2</DoubleClickDelta>
     <!-- ICONS -->
     <IconPath>/usr/share/pixmaps</IconPath>
-    <IconPath>/usr/share/icons/Numix/48x48/actions/</IconPath>
-    <IconPath>/usr/share/icons/Numix/48x48/categories/</IconPath>
-    <IconPath>/usr/share/icons/Numix/48x48/places/</IconPath>
-    <IconPath>/usr/share/icons/Numix/48x48/status/</IconPath>
-    <IconPath>/usr/share/icons/Numix-Circle/48x48/apps</IconPath>
-<!-- DESKTOP -->
+    <IconPath>/usr/share/icons/gnome/48x48/actions</IconPath>
+    <IconPath>/usr/share/icons/gnome/48x48/places</IconPath>
+    <IconPath>/usr/share/icons/gnome/48x48/apps</IconPath>
+    <IconPath>/usr/share/icons/gnome/48x48/devices</IconPath>
+    <!-- DESKTOP -->
     <Desktops width="2" height="1">
         <Background type="image">/usr/share/backgrounds/default.png</Background>
     </Desktops>
-
-<!-- AUTOSTART -->
-   <StartupCommand>nm-applet</StartupCommand>
-   <StartupCommand>volumeicon</StartupCommand>
-   <StartupCommand>synclient MaxTapTime=0</StartupCommand>
-
-<!--POP UP -->
+    <!-- AUTOSTART -->
+    <StartupCommand>nm-applet</StartupCommand>
+    <StartupCommand>retrovol -hide</StartupCommand>
+    <StartupCommand>synclient MaxTapTime=0</StartupCommand>
+    <!--POP UP -->
     <PopupStyle>
         <Font>ubuntu-12:antialias=true:encoding=utf8</Font>
         <Outline>#000000</Outline>
         <Foreground>#ffffff</Foreground>
         <Background>#464646</Background>
     </PopupStyle>
-
 </JWM>
diff -Naur orig/data/themes/Default/Default mod/data/themes/Default/Default
--- orig/data/themes/Default/Default	2014-12-28 03:56:53.000000000 +0100
+++ mod/data/themes/Default/Default	2015-04-22 21:28:29.000000000 +0200
@@ -1,172 +1,186 @@
 <?xml version="1.0"?>
+
 <JWM>
-<!-- NOTE, if you hand edit this root:5 is the app menu and root:7 is the places menu  you have been duly warned :) -->
-    <!-- MENUS -->
-    <!-- Main Menu -->
-    <Include>$HOME/.config/torimenu</Include>
-    <!--Shudown Menu-->
-    <RootMenu height="0" onroot="9">
-        <Program icon="preferences-system-session.svg" label="Shutdown Menu">jwm-settings-manager --halt</Program>
-        <Program icon="system-restart.svg" label="Restart Computer NOW">dbus-send --system --print-reply --dest=org.freedesktop.ConsoleKit /org/freedesktop/ConsoleKit/Manager org.freedesktop.ConsoleKit.Manager.Restart</Program>
-        <Restart label="Reload Desktop" icon="system-restart.svg"/>
-        <Exit label="Logout" confirm="true" icon="system-logout.svg"/>
+
+    <!-- The root menu. -->
+    <RootMenu onroot="12">
+      <Include>$HOME/.jwmrc-mjwm</Include>
+        <Program icon="xterm-color_48x48.xpm" label="xterm">xterm</Program>
+        <Menu icon="folder.png" label="Alapok">
+            <Program icon="calc.png" label="Calculator">xcalc</Program>
+            <Program icon="gimp.png" label="Gimp">gimp</Program>
+            <Program icon="firefox.png" label="Firefox">firefox</Program>
+            <Program icon="MidnightCommander.png" label="mc">xterm -e mc</Program>
+        </Menu>
+        <Menu icon="folder.png" label="JWM eszközök">
+            <Program icon="uhu-64x64.png" label="Menügenerátor">jwmuhumenu</Program>
+            <Program icon="fonts.png" label="Fonts">xfontsel</Program>
+            <Program icon="preferences-system-windows.png" label="Window Properties">
+                xprop | xmessage -file -
+            </Program>
+        </Menu>
+        <Separator/>
+        <Program icon="lock.png" label="Lock">
+            xscreensaver-command -lock
+        </Program>
+        <Separator/>
+        <Restart label="JWM restart" icon="reload.png"/>
+        <Program icon="system-shutdown.png" label="Kikapcsolás">systemctl poweroff</Program>
+        <Program icon="redo.png" label="Újraindítás">systemctl reboot</Program>
+        <Exit label="Kilépés" confirm="true" icon="system-log-out.png"/>
     </RootMenu>
-    <!--Places Menu-->
-    <Include>$HOME/.config/placesmenu</Include>
-    <!-- Mouse Menu-->
-    <RootMenu height="16" onroot="23">
-        <Program icon="application-default-icon.svg" label="Applications">xdg-open /usr/share/applications</Program>
-        <Program icon="folder-home.svg" label="Home">xdg-open ~/</Program>
-        <Program icon="/usr/share/icons/jwm-settings-manager.svg" label="Settings">jwm-settings-manager</Program>
-        <Program icon="preferences-desktop-wallpaper.svg" label="Change Desktop Background">jwm-settings-manager --desktop</Program>
-    </RootMenu>
-    <!-- END MENUS -->
-    <!-- GROUP -->
+
+    <!-- Options for program groups. -->
     <Group>
         <Class>Pidgin</Class>
         <Option>sticky</Option>
     </Group>
     <Group>
-        <Name>gkrellm</Name>
-        <Option>nolist</Option>
-        <Option>sticky</Option>
+        <Name>xterm</Name>
+        <Option>vmax</Option>
+        <Option>icon:terminal.png</Option>
     </Group>
     <Group>
-        <Name>conky</Name>
-        <Option>nolist</Option>
-        <Option>sticky</Option>
+        <Name>xedit</Name>
+        <Option>icon:editor.png</Option>
     </Group>
     <Group>
-        <Name>jwm-settings-manager</Name>
-        <Option>icon:jwm-settings-manager.png</Option>
+        <Name>xcalc</Name>
+        <Option>icon:calculator.png</Option>
     </Group>
-    <!-- PANEL -->
-    <Tray x="0" y="-1" height="34" autohide="false" valign="top" width="0" halign="fixed" layout="horizontal">
-        <TrayButton label="Apps" icon="/usr/share/pixmaps/distributor-logo.png" border="false">root:5</TrayButton>
-        <TrayButton label="Places" icon="folder.svg" border="false">root:7</TrayButton>
-        <TrayButton icon="browser.svg" border="false">exec:xdg-open /usr/share/doc/ToriOS/startpage.html</TrayButton>
+
+    <!-- Tray at the bottom. -->
+    <Tray x="0" y="-1" height="32" autohide="false">
+
+        <TrayButton label="JWM">root:1</TrayButton>
+        <TrayButton label="_">showdesktop</TrayButton>
+
         <Pager labeled="true"/>
+
         <TaskList maxwidth="256"/>
+
         <Dock/>
-        <Swallow name="xload" width="64">xload -nolabel -bg DimGrey -fg Grey -hl DarkGrey</Swallow>
-        <Clock format="%a, %e %b %l:%M %p">xclock</Clock>
-        <TrayButton label="Shutdown" icon="system-shutdown.svg" border="false">root:9</TrayButton>
+        <Clock>xclock</Clock>
+
     </Tray>
-    <!-- Panel Style -->
-    <TaskListStyle>
-        <ActiveBackground>#940000</ActiveBackground>
-        <Background>#383838</Background>
-        <Font>ubuntu-12:antialias=true:encoding=utf8</Font>
-        <Foreground>#ffffff</Foreground>
-    </TaskListStyle>
-    <TrayStyle>
-        <Font>ubuntu-12:antialias=true:encoding=utf8</Font>
-        <Foreground>#ffffff</Foreground>
-        <Background>#383838</Background>
-        <Opacity>0.75</Opacity>
-    </TrayStyle>
-    <TrayButtonStyle>
-        <Font>ubuntu-12:antialias=true:encoding=utf8</Font>
-        <Foreground>#ffffff</Foreground>
-        <Background>#383838</Background>
-    </TrayButtonStyle>
-    <PagerStyle>
-        <Outline>#000000</Outline>
-        <Foreground>#808080</Foreground>
-        <Background>#383838</Background>
-        <ActiveForeground>#ffffff</ActiveForeground>
-        <ActiveBackground>#940000</ActiveBackground>
-    </PagerStyle>
-    <MenuStyle>
-        <Font>ubuntu-12:antialias=true:encoding=utf8</Font>
-        <Foreground>#ffffff</Foreground>
-        <Background>#383838</Background>
-        <ActiveBackground>#940000</ActiveBackground>
-        <Opacity>0.85</Opacity>
-    </MenuStyle>
-    <!-- WINDOW -->
+
+    <!-- Visual Styles -->
     <WindowStyle>
-        <Font>ubuntu-12:antialias=true:encoding=utf8</Font>
+        <Font>FreeSans-9:bold</Font>
         <Width>4</Width>
         <Height>20</Height>
         <Active>
-            <Text>#C7C7C7</Text>
-            <Title>#000000:#383838</Title>
+            <Text>#FFFFFF</Text>
+            <Title>#CC7700:#884400</Title>
             <Outline>#000000</Outline>
             <Opacity>1.0</Opacity>
         </Active>
         <Inactive>
-            <Text>#940000</Text>
-            <Title>#383838:#000000</Title>
+            <Text>#FFFFFF</Text>
+            <Title>#333333:#111111</Title>
             <Outline>#000000</Outline>
-            <Opacity>0.7</Opacity>
+            <Opacity>0.5:0.9:0.1</Opacity>
         </Inactive>
     </WindowStyle>
-    <ButtonClose>/usr/share/jwm-settings-manager/themes/Default/Buttons/close.png</ButtonClose>
-    <ButtonMax>/usr/share/jwm-settings-manager/themes/Default/Buttons/max.png</ButtonMax>
-    <ButtonMaxActive>/usr/share/jwm-settings-manager/themes/Default/Buttons/max-toggle.png</ButtonMaxActive>
-    <ButtonMin>/usr/share/jwm-settings-manager/themes/Default/Buttons/min.png</ButtonMin>
-    <FocusModel>click</FocusModel>
+    <TrayStyle>
+        <Font>FreeSans-12:bold</Font>
+        <Background>#111111</Background>
+        <Foreground>#FFFFFF</Foreground>
+        <Opacity>0.75</Opacity>
+    </TrayStyle>
+    <TaskListStyle>
+        <Font>FreeSans-12:bold</Font>
+        <ActiveForeground>#FFFFFF</ActiveForeground>
+        <ActiveBackground>#CC7700:#884400</ActiveBackground>
+        <Foreground>#FFFFFF</Foreground>
+        <Background>#777777:#111111</Background>
+    </TaskListStyle>
+    <PagerStyle>
+        <Outline>#000000</Outline>
+        <Text>#FFFFFF</Text>
+        <Foreground>#444444</Foreground>
+        <Background>#111111</Background>
+        <ActiveForeground>#CC7700</ActiveForeground>
+        <ActiveBackground>#884400</ActiveBackground>
+    </PagerStyle>
+    <MenuStyle>
+        <Font>FreeSans-11</Font>
+        <Foreground>#FFFFFF</Foreground>
+        <Background>#333333</Background>
+        <ActiveForeground>#FFFFFF</ActiveForeground>
+        <ActiveBackground>#CC7700:#884400</ActiveBackground>
+        <Opacity>0.85</Opacity>
+    </MenuStyle>
+    <PopupStyle>
+        <Font>FreeSans-10</Font>
+        <Outline>#000000</Outline>
+        <Foreground>#000000</Foreground>
+        <Background>#999999</Background>
+    </PopupStyle>
+
+    <!-- Path where icons can be found.
+         IconPath can be listed multiple times to allow searching
+         for icons in multiple paths.
+      -->
+    <IconPath>/usr/share/pixmaps</IconPath>
+    <IconPath>/usr/share/icons/gnome/48x48/actions</IconPath>
+    <IconPath>/usr/share/icons/gnome/48x48/places</IconPath>
+    <IconPath>/usr/share/icons/gnome/48x48/apps</IconPath>
+    <IconPath>/usr/share/icons/gnome/48x48/devices</IconPath>
+
+    <!-- Virtual Desktops -->
+    <!-- Desktop tags can be contained within Desktops for desktop names. -->
+    <Desktops width="2" height="1">
+        <!-- Default background. Note that a Background tag can be
+              contained within a Desktop tag to give a specific background
+              for that desktop.
+         -->
+        <Background type="image">/usr/share/wallpapers/UHU/blur.jpg</Background>
+    </Desktops>
+
+    <!-- Double click speed (in milliseconds) -->
+    <DoubleClickSpeed>400</DoubleClickSpeed>
+
+    <!-- Double click delta (in pixels) -->
+    <DoubleClickDelta>2</DoubleClickDelta>
+
+    <!-- The focus model (sloppy or click) -->
+    <FocusModel>sloppy</FocusModel>
+
+    <!-- The snap mode (none, screen, or border) -->
     <SnapMode distance="10">border</SnapMode>
+
+    <!-- The move mode (outline or opaque) -->
     <MoveMode>opaque</MoveMode>
+
+    <!-- The resize mode (outline or opaque) -->
     <ResizeMode>opaque</ResizeMode>
+
+    <StartupCommand>nm-applet</StartupCommand>
+    <StartupCommand>retrovol -hide</StartupCommand>
+    <StartupCommand>synclient MaxTapTime=0</StartupCommand>
+
+    <!-- Key bindings -->
     <Key key="Up">up</Key>
     <Key key="Down">down</Key>
     <Key key="Right">right</Key>
     <Key key="Left">left</Key>
-    <Key key="a">left</Key>
-    <Key key="s">down</Key>
-    <Key key="w">up</Key>
-    <Key key="d">right</Key>
-    <Key key="F12">fullscreen</Key>
+    <Key key="h">left</Key>
+    <Key key="j">down</Key>
+    <Key key="k">up</Key>
+    <Key key="l">right</Key>
     <Key key="Return">select</Key>
     <Key key="Escape">escape</Key>
-    <Key key="XF86AudioMute">exec:amixer -q sset Master toggle</Key>
-    <Key key="XF86AudioRaiseVolume">exec:amixer -q sset Master 3%+ unmute</Key>
-    <Key key="XF86AudioLowerVolume">exec:amixer -q sset Master 3%- unmute</Key>
-    <Key key="XF86WWW">exec:sensible-browser</Key>
-    <Key key="XF86Calculator">exec:xcalc</Key>
-    <Key key="Print">exec:screenie</Key>
-    <Key mask="CA" key="p">exec:screenie</Key>
-    <Key mask="CA" key="t">exec:x-terminal-emulator</Key>
-    <Key mask="CA" key="Delete">exec:xterm htop</Key>
-    <Key mask="A" key="Tab">next</Key>
-    <Key mask="CA" key="Tab">nextstacked</Key>
+
+    <Key mask="A" key="Tab">nextstacked</Key>
     <Key mask="A" key="F4">close</Key>
     <Key mask="A" key="#">desktop#</Key>
     <Key mask="A" key="F1">root:1</Key>
     <Key mask="A" key="F2">window</Key>
     <Key mask="A" key="F10">maximize</Key>
-    <Key mask="CA" key="Right">rdesktop</Key>
-    <Key mask="CA" key="Left">ldesktop</Key>
-    <Key mask="CA" key="Up">udesktop</Key>
-    <Key mask="CA" key="Down">ddesktop</Key>
-    <Key mask="CA" key="q">close</Key>
-    <!-- MOUSE -->
-    <DoubleClickSpeed>400</DoubleClickSpeed>
-    <DoubleClickDelta>2</DoubleClickDelta>
-    <!-- ICONS -->
-    <IconPath>/usr/share/pixmaps</IconPath>
-    <IconPath>/usr/share/icons</IconPath>
-    <IconPath>/usr/share/icons/Numix/48x48/actions/</IconPath>
-    <IconPath>/usr/share/icons/Numix/48x48/categories/</IconPath>
-    <IconPath>/usr/share/icons/Numix/48x48/places/</IconPath>
-    <IconPath>/usr/share/icons/Numix/48x48/status/</IconPath>
-    <IconPath>/usr/share/icons/Numix-Circle/48x48/apps</IconPath>
-    <!-- DESKTOP -->
-    <Desktops width="2" height="1">
-        <Background type="image">/usr/share/backgrounds/default.png</Background>
-    </Desktops>
-    <!-- AUTOSTART -->
-    <StartupCommand>nm-applet</StartupCommand>
-    <StartupCommand>volumeicon</StartupCommand>
-    <StartupCommand>synclient VertEdgeScroll=1</StartupCommand>
-    <StartupCommand>synclient MaxTapTime=0</StartupCommand>
-    <!--POP UP -->
-    <PopupStyle>
-        <Font>ubuntu-12:antialias=true:encoding=utf8</Font>
-        <Outline>#000000</Outline>
-        <Foreground>#ffffff</Foreground>
-        <Background>#383838</Background>
-    </PopupStyle>
+    <Key mask="A" key="Right">rdesktop</Key>
+    <Key mask="A" key="Left">ldesktop</Key>
+    <Key mask="A" key="Up">udesktop</Key>
+    <Key mask="A" key="Down">ddesktop</Key>
+
 </JWM>
diff -Naur orig/data/themes/LightGnome2/LightGnome2 mod/data/themes/LightGnome2/LightGnome2
--- orig/data/themes/LightGnome2/LightGnome2	2014-12-28 03:56:53.000000000 +0100
+++ mod/data/themes/LightGnome2/LightGnome2	2015-04-22 22:11:25.606188993 +0200
@@ -4,32 +4,41 @@
 Israel <israel@torios.org>
 MIT license copyright 2014
 -->
-
 <JWM>
-
-<!-- MENUS -->
+    <!-- MENUS -->
     <!-- Main Menu -->
-    <Include>$HOME/.config/torimenu</Include>
+    <RootMenu height="0" onroot="5">
+        <Include>$HOME/.jwmrc-mjwm</Include>
+        <Program icon="uhu-64x64.png" label="Menügenerátor">jwmuhumenu</Program>
+    </RootMenu>
     <!--Shudown Menu-->
     <RootMenu height="0" onroot="9">
-        <Program icon="preferences-system-session.svg" label="Shutdown Menu">jwm-settings-manager --halt</Program>
-        <Program icon="system-restart.svg" label="Restart Computer">dbus-send --system --print-reply --dest=org.freedesktop.ConsoleKit /org/freedesktop/ConsoleKit/Manager org.freedesktop.ConsoleKit.Manager.Restart</Program>
-        <Restart label="Reload Desktop" icon="system-restart.svg"/>
-        <Exit label="Logout" confirm="true" icon="system-logout.svg"/>
+        <Restart label="JWM restart" icon="reload.png"/>
+        <Exit label="Kilépés" confirm="true" icon="system-log-out.png"/>
+        <Program icon="redo.png" label="Újraindítás">systemctl reboot</Program>
+        <Program icon="system-shutdown.png" label="Kikapcsolás">systemctl poweroff</Program>
     </RootMenu>
     <!--Places Menu-->
-    <Include>$HOME/.config/placesmenu</Include>
+    <RootMenu height="0" onroot="7">
+        <Program icon="application-x-gnome-saved-search.png" label="Applications">xdg-open /usr/share/applications</Program>
+        <Program icon="folder_home.png" label="Home">xdg-open ~/</Program>
+        <Program icon="folder-documents.png" label="Dokumentumok">xdg-open ~/Dokumentumok</Program>
+        <Program icon="folder-download.png" label="Letöltések">xdg-open ~/Letöltések</Program>
+        <Program icon="folder-music.png" label="Zenék">xdg-open -n ~/Zenék</Program>
+        <Program icon="folder-pictures.png" label="Képek">xdg-open ~/Képek</Program>
+        <Program icon="folder-videos.png" label="Videók">xdg-open ~/Videók</Program>
+        <Program icon="user-trash.png" label="Trash">xdg-open ~/.local/share/Trash/files</Program>
+        <Program icon="file-manager.png" label="Connected Devices">xdg-open /media</Program>
+    </RootMenu>
     <!-- Mouse Menu-->
     <RootMenu height="16" onroot="23">
-        <Program icon="application-default-icon.svg" label="Applications">xdg-open /usr/share/applications</Program>
-        <Program icon="folder-home.svg" label="Home">xdg-open ~/</Program>
+        <Include>$HOME/.jwmrc-mjwm</Include>
+        <Program icon="uhu-64x64.png" label="Menügenerátor">jwmuhumenu</Program>
+        <Program icon="folder_home.png" label="Home">xdg-open ~/</Program>
         <Program icon="jwm-settings-manager.svg" label="Settings">jwm-settings-manager</Program>
-        <Program icon="preferences-desktop-wallpaper.svg" label="Change Desktop Background">jwm-settings-manager --desktop</Program>
     </RootMenu>
     <!-- END MENUS -->
-
-
-<!-- GROUP -->
+    <!-- GROUP -->
     <Group>
         <Class>Pidgin</Class>
         <Option>sticky</Option>
@@ -48,26 +57,21 @@
         <Name>jwm-settings-manager</Name>
         <Option>icon:jwm-settings-manager.png</Option>
     </Group>
-
-
-<!-- PANEL -->
+    <!-- PANEL -->
     <Tray height="34" autohide="false" valign="top" width="0" halign="fixed" layout="horizontal">
-        <TrayButton label="Apps" icon="/usr/share/pixmaps/distributor-logo.png" border="false">root:5</TrayButton>
-        <TrayButton label="Places" icon="folder.svg" border="false">root:7</TrayButton>
-        <TrayButton icon="browser.svg" border="false">exec:xdg-open /usr/share/doc/ToriOS/startpage.html</TrayButton>
+        <TrayButton label="Programok" icon="distributor-logo.png" border="false">root:5</TrayButton>
+        <TrayButton label="Helyek" icon="folder.png" border="false">root:7</TrayButton>
         <Spacer/>
         <Dock/>
         <Swallow name="xload" width="64">xload -nolabel -bg DimGrey -fg Grey -hl DarkGrey</Swallow>
         <Clock format="%a, %e %b %l:%M %p">xclock</Clock>
-        <TrayButton label="Shutdown" icon="system-shutdown.svg" border="false">root:9</TrayButton>
+        <TrayButton label="Kikapcsolás" icon="system-shutdown.png" border="false">root:9</TrayButton>
     </Tray>
     <Tray x="0" y="-1" height="34" autohide="false" valign="bottom" width="0" halign="fixed" layout="horizontal">
-	<TrayButton icon="preferences-system-hotcorners.png" border="false">showdesktop</TrayButton>
+        <TrayButton icon="preferences-system-hotcorners.png" border="false">showdesktop</TrayButton>
         <TaskList maxwidth="256"/>
         <Pager labeled="false"/>
     </Tray>
-
-
     <!-- Panel Style -->
     <TaskListStyle>
         <ActiveBackground>#C2C2BE</ActiveBackground>
@@ -82,7 +86,7 @@
         <Opacity>0.75</Opacity>
     </TrayStyle>
     <TrayButtonStyle>
-	    <Font>ubuntu-12:antialias=true:encoding=utf8</Font>
+        <Font>ubuntu-12:antialias=true:encoding=utf8</Font>
         <Foreground>#000000</Foreground>
         <Background>#E2E2DE</Background>
     </TrayButtonStyle>
@@ -100,9 +104,8 @@
         <ActiveBackground>#406477</ActiveBackground>
         <Opacity>0.85</Opacity>
     </MenuStyle>
-
-<!-- WINDOW -->
-<WindowStyle>
+    <!-- WINDOW -->
+    <WindowStyle>
         <Font>ubuntu-12:antialias=true:encoding=utf8</Font>
         <Width>4</Width>
         <Height>20</Height>
@@ -123,8 +126,7 @@
     <SnapMode distance="10">border</SnapMode>
     <MoveMode>opaque</MoveMode>
     <ResizeMode>opaque</ResizeMode>
-
-<Key key="Up">up</Key>
+    <Key key="Up">up</Key>
     <Key key="Down">down</Key>
     <Key key="Right">right</Key>
     <Key key="Left">left</Key>
@@ -161,28 +163,23 @@
     <DoubleClickDelta>2</DoubleClickDelta>
     <!-- ICONS -->
     <IconPath>/usr/share/pixmaps</IconPath>
-    <IconPath>/usr/share/icons/Numix/48x48/actions/</IconPath>
-    <IconPath>/usr/share/icons/Numix/48x48/categories/</IconPath>
-    <IconPath>/usr/share/icons/Numix/48x48/places/</IconPath>
-    <IconPath>/usr/share/icons/Numix/48x48/status/</IconPath>
-    <IconPath>/usr/share/icons/Numix-Circle/48x48/apps</IconPath>
-
-<!-- DESKTOP -->
+    <IconPath>/usr/share/icons/gnome/48x48/actions</IconPath>
+    <IconPath>/usr/share/icons/gnome/48x48/places</IconPath>
+    <IconPath>/usr/share/icons/gnome/48x48/apps</IconPath>
+    <IconPath>/usr/share/icons/gnome/48x48/devices</IconPath>
+    <!-- DESKTOP -->
     <Desktops width="2" height="1">
-        <Background type="image">/usr/share/backgrounds/default.png</Background>
+        <Background type="image">/usr/share/wallpapers/UHU/blur.jpg</Background>
     </Desktops>
-
-<!-- AUTOSTART -->
-   <StartupCommand>nm-applet</StartupCommand>
-   <StartupCommand>volumeicon</StartupCommand>
-   <StartupCommand>synclient MaxTapTime=0</StartupCommand>
-
-<!--POP UP -->
+    <!-- AUTOSTART -->
+    <StartupCommand>nm-applet</StartupCommand>
+    <StartupCommand>retrovol -hide</StartupCommand>
+    <StartupCommand>synclient MaxTapTime=0</StartupCommand>
+    <!--POP UP -->
     <PopupStyle>
         <Font>ubuntu-12:antialias=true:encoding=utf8</Font>
         <Outline>#000000</Outline>
         <Foreground>#000000</Foreground>
         <Background>#E2E2DE</Background>
     </PopupStyle>
-
 </JWM>
