diff -Naur clamz-0.5-orig/clamz.desktop clamz-0.5/clamz.desktop
--- clamz-0.5-orig/clamz.desktop	2010-04-30 04:52:29.000000000 +0200
+++ clamz-0.5/clamz.desktop	2012-12-14 09:31:10.000000000 +0100
@@ -2,7 +2,7 @@
 Encoding=UTF-8
 Name=Clamz MP3 Downloader (command-line)
 Comment=Download MP3 files from AmazonMP3.com
-Exec=clamz "--default-output-dir=\\${XDG_MUSIC_DIR:-\\$HOME/Music}/\\${album_artist}/\\${album}"
+Exec=clamz
 Terminal=true
 Type=Application
 Categories=Network;
