diff -Naur nagios-4.0.1-orig/daemon-init.in nagios-4.0.1/daemon-init.in
--- nagios-4.0.1-orig/daemon-init.in	2013-10-16 01:36:42.000000000 +0200
+++ nagios-4.0.1/daemon-init.in	2013-11-21 11:39:20.451979370 +0100
@@ -32,7 +32,7 @@
 
 test -e /etc/sysconfig/$prog && . /etc/sysconfig/$prog
 
-lockfile=/var/lock/subsys/$prog
+lockfile=/var/lock/$prog
 USE_RAMDISK=${USE_RAMDISK:-0}
 
 if test "$USE_RAMDISK" -ne 0 && test "$RAMDISK_SIZE"X != "X"; then
