--- a/Menu/gnome-rdp.desktop.orig	2011-11-10 19:06:34.000000000 +0100
+++ a/Menu/gnome-rdp.desktop	2011-11-10 19:07:16.000000000 +0100
@@ -7,7 +7,7 @@
 StartupNotify=true
 Terminal=false
 Type=Application
-Categories=RemoteAccess;Network;
+Categories=RemoteAccess;Network;X-UHU-Menu/Internet/Tools;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=gnome-rdp
 X-GNOME-Bugzilla-Component=General
