diff -Naur orig/samba3x-compat.h patched/samba3x-compat.h
--- orig/samba3x-compat.h	2009-04-11 12:08:17.000000000 +0200
+++ patched/samba3x-compat.h	2014-03-23 16:05:44.706510732 +0100
@@ -26,7 +26,7 @@
   #define SAMBA_30_COMPAT_H
 
   #include "config.h"
-  #include <libsmbclient.h>
+  #include <samba-4.0/libsmbclient.h>
   #include <sys/statvfs.h>
 
 
diff -Naur orig/usmb.c patched/usmb.c
--- orig/usmb.c	2009-04-11 12:08:17.000000000 +0200
+++ patched/usmb.c	2014-03-23 16:03:27.693246159 +0100
@@ -17,7 +17,7 @@
 #include "config.h"
 #include <sys/time.h>        // struct timeval needed by libsmbclient.h
 #include <unistd.h>
-#include <libsmbclient.h>
+#include <samba-4.0/libsmbclient.h>
 #include "samba3x-compat.h"
 #include <fuse.h>
 #include <assert.h>
diff -Naur orig/usmb_dir.c patched/usmb_dir.c
--- orig/usmb_dir.c	2009-04-11 12:08:17.000000000 +0200
+++ patched/usmb_dir.c	2014-03-23 16:07:15.151345465 +0100
@@ -16,7 +16,7 @@
 
 #include "config.h"
 #include <sys/time.h>        // struct timeval needed by libsmbclient.h
-#include <libsmbclient.h>
+#include <samba-4.0/libsmbclient.h>
 #include "samba3x-compat.h"
 #include <fuse.h>
 #include <dirent.h>
diff -Naur orig/usmb_file.c patched/usmb_file.c
--- orig/usmb_file.c	2009-04-11 12:08:17.000000000 +0200
+++ patched/usmb_file.c	2014-03-23 16:07:30.811489992 +0100
@@ -16,7 +16,7 @@
 
 #include "config.h"
 #include <sys/time.h>        // struct timeval needed by libsmbclient.h
-#include <libsmbclient.h>
+#include <samba-4.0/libsmbclient.h>
 #include "samba3x-compat.h"
 #include <limits.h>
 #include <assert.h>
diff -Naur orig/usmb.h patched/usmb.h
--- orig/usmb.h	2009-04-11 12:08:17.000000000 +0200
+++ patched/usmb.h	2014-03-23 16:09:22.452520408 +0100
@@ -19,7 +19,7 @@
 
   #include <stdbool.h>
   #include <stdint.h>
-  #include <libsmbclient.h>
+  #include <samba-4.0/libsmbclient.h>
 
   extern SMBCCTX *ctx;
 
diff -Naur orig/version.c patched/version.c
--- orig/version.c	2009-04-11 12:08:17.000000000 +0200
+++ patched/version.c	2014-03-23 16:07:53.547699842 +0100
@@ -17,7 +17,7 @@
 #include "config.h"
 #include <stdio.h>
 #include <sys/time.h>
-#include <libsmbclient.h>
+#include <samba-4.0/libsmbclient.h>
 #include <fuse.h>
 #include "version.h"
 
