diff -Naur orig/src/audacity.desktop.in mod/src/audacity.desktop.in
--- orig/src/audacity.desktop.in	2015-07-11 12:02:22.000000000 +0200
+++ mod/src/audacity.desktop.in	2015-08-26 15:09:01.347196763 +0200
@@ -5,10 +5,12 @@
 Name[uk]=Audacity
 GenericName=Sound Editor
 GenericName[de]=Audio-Editor
+GenericName[hu]=Hangszerkesztő
 GenericName[ru]=Редактор звуковых файлов
 GenericName[uk]=Редактор звукових файлів
 Comment=Record and edit audio files
 Comment[de]=Audio-Dateien aufnehmen und bearbeiten
+Comment[hu]=Hangfájlok felvétele és szerkesztése
 Comment[ru]=Запись и редактирование звуковых файлов
 Comment[uk]=Запис і редагування звукових файлів
 
