diff -Naur audacity-Audacity-2.1.2-orig/src/effects/vamp/LoadVamp.cpp audacity-Audacity-2.1.2/src/effects/vamp/LoadVamp.cpp
--- audacity-Audacity-2.1.2-orig/src/effects/vamp/LoadVamp.cpp	2016-01-08 23:05:48.000000000 +0100
+++ audacity-Audacity-2.1.2/src/effects/vamp/LoadVamp.cpp	2016-11-22 00:05:37.609640354 +0100
@@ -266,7 +266,7 @@
    Plugin *vp = PluginLoader::getInstance()->loadPlugin(key, 48000); // rate doesn't matter here
    if (!vp)
    {
-      return false;
+      return 0;
    }
 
    // We limit the listed plugin outputs to those whose results can
diff -Naur audacity-Audacity-2.1.2-orig/src/import/ImportFLAC.cpp audacity-Audacity-2.1.2/src/import/ImportFLAC.cpp
--- audacity-Audacity-2.1.2-orig/src/import/ImportFLAC.cpp	2016-01-08 23:05:48.000000000 +0100
+++ audacity-Audacity-2.1.2/src/import/ImportFLAC.cpp	2016-11-22 00:07:33.145641883 +0100
@@ -296,7 +296,7 @@
    int cnt;
    wxFile binaryFile;
    if (!binaryFile.Open(filename)) {
-      return false; // File not found
+      return NULL; // File not found
    }
 
 #ifdef USE_LIBID3TAG
@@ -313,7 +313,7 @@
 
    if (cnt == wxInvalidOffset || strncmp(buf, FLAC_HEADER, 4) != 0) {
       // File is not a FLAC file
-      return false;
+      return NULL;
    }
 
    // Open the file for import
