diff -Naurdp skype-4.3.0.37/skype.desktop skype-4.3.0.37-desktop/skype.desktop
--- skype-4.3.0.37/skype.desktop	2014-05-22 18:56:04.000000000 +0200
+++ skype-4.3.0.37-desktop/skype.desktop	2014-06-25 14:31:53.064113533 +0200
@@ -2,10 +2,9 @@
 Name=Skype
 Comment=Skype Internet Telephony
 Exec=skype %U
-Icon=skype.png
+Icon=skype
 Terminal=false
 Type=Application
-Encoding=UTF-8
-Categories=Network;Application;
+Categories=Network;Chat;VideoConference;
 MimeType=x-scheme-handler/skype;
 X-KDE-Protocols=skype
