diff -Naur orig/Nitrogen/default.theme mod/Nitrogen/default.theme
--- orig/Nitrogen/default.theme	2014-04-30 13:00:56.000000000 +0200
+++ mod/Nitrogen/default.theme	2014-12-21 23:09:40.197535868 +0100
@@ -31,6 +31,9 @@
 ################################################## Desktop
 
 DesktopBackgroundColor         = "#000000"
+DesktopBackgroundCenter=1
+DesktopBackgroundScaled=1
+DesktopBackgroundImage = 'ThrowingStones.jpg'
 
 ################################################## Taskbar
 
diff -Naur orig/Nitrogen/dframeAB.xpm mod/Nitrogen/dframeAB.xpm
--- orig/Nitrogen/dframeAB.xpm	2014-04-30 13:00:56.000000000 +0200
+++ mod/Nitrogen/dframeAB.xpm	2014-12-21 22:52:08.000000000 +0100
@@ -1,8 +1,8 @@
 /* XPM */
 static char * dframeAB_xpm[] = {
-"1 2 3 1",
+"2 2 3 1",
 " 	c None",
 ".	c #E0DFDE",
 "+	c #6A8EB8",
-".",
-"+"};
+"..",
+"++"};
diff -Naur orig/Nitrogen/dframeAT.xpm mod/Nitrogen/dframeAT.xpm
--- orig/Nitrogen/dframeAT.xpm	2014-04-30 13:00:56.000000000 +0200
+++ mod/Nitrogen/dframeAT.xpm	2014-12-21 22:52:45.000000000 +0100
@@ -1,8 +1,8 @@
 /* XPM */
 static char * dframeAT_xpm[] = {
-"1 2 3 1",
+"2 2 3 1",
 " 	c None",
 ".	c #6A8EB8",
 "+	c #E0DFDE",
-".",
-"+"};
+"..",
+"++"};
diff -Naur orig/Nitrogen/dframeIB.xpm mod/Nitrogen/dframeIB.xpm
--- orig/Nitrogen/dframeIB.xpm	2014-04-30 13:00:56.000000000 +0200
+++ mod/Nitrogen/dframeIB.xpm	2014-12-21 22:53:29.000000000 +0100
@@ -1,8 +1,8 @@
 /* XPM */
 static char * dframeIB_xpm[] = {
-"1 2 3 1",
+"2 2 3 1",
 " 	c None",
 ".	c #E0DFDE",
 "+	c #939393",
-".",
-"+"};
+"..",
+"++"};
diff -Naur orig/Nitrogen/dframeIT.xpm mod/Nitrogen/dframeIT.xpm
--- orig/Nitrogen/dframeIT.xpm	2014-04-30 13:00:56.000000000 +0200
+++ mod/Nitrogen/dframeIT.xpm	2014-12-21 22:54:10.000000000 +0100
@@ -1,8 +1,8 @@
 /* XPM */
 static char * dframeIT_xpm[] = {
-"1 2 3 1",
+"2 2 3 1",
 " 	c None",
 ".	c #CACACA",
 "+	c #ECECEA",
-".",
-"+"};
+"..",
+"++"};
diff -Naur orig/Nitrogen/frameAB.xpm mod/Nitrogen/frameAB.xpm
--- orig/Nitrogen/frameAB.xpm	2014-04-30 13:00:56.000000000 +0200
+++ mod/Nitrogen/frameAB.xpm	2014-12-21 22:54:42.000000000 +0100
@@ -1,8 +1,8 @@
 /* XPM */
 static char * frameAB_xpm[] = {
-"1 2 3 1",
+"2 2 3 1",
 " 	c None",
 ".	c #E0DFDE",
 "+	c #6A8EB8",
-".",
-"+"};
+"..",
+"++"};
diff -Naur orig/Nitrogen/frameAT.xpm mod/Nitrogen/frameAT.xpm
--- orig/Nitrogen/frameAT.xpm	2014-04-30 13:00:56.000000000 +0200
+++ mod/Nitrogen/frameAT.xpm	2014-12-21 22:55:16.000000000 +0100
@@ -1,8 +1,8 @@
 /* XPM */
 static char * frameAT_xpm[] = {
-"1 2 3 1",
+"2 2 3 1",
 " 	c None",
 ".	c #6A8EB8",
 "+	c #E0DFDE",
-".",
-"+"};
+"..",
+"++"};
diff -Naur orig/Nitrogen/frameIB.xpm mod/Nitrogen/frameIB.xpm
--- orig/Nitrogen/frameIB.xpm	2014-04-30 13:00:56.000000000 +0200
+++ mod/Nitrogen/frameIB.xpm	2014-12-21 22:55:47.000000000 +0100
@@ -1,8 +1,8 @@
 /* XPM */
-static char * dframeIB_xpm[] = {
-"1 2 3 1",
+static char * frameIB_xpm[] = {
+"2 2 3 1",
 " 	c None",
 ".	c #E0DFDE",
 "+	c #939393",
-".",
-"+"};
+"..",
+"++"};
diff -Naur orig/Nitrogen/frameIT.xpm mod/Nitrogen/frameIT.xpm
--- orig/Nitrogen/frameIT.xpm	2014-04-30 13:00:56.000000000 +0200
+++ mod/Nitrogen/frameIT.xpm	2014-12-21 22:56:14.000000000 +0100
@@ -1,8 +1,8 @@
 /* XPM */
-static char * dframeIT_xpm[] = {
-"1 2 3 1",
+static char * frameIT_xpm[] = {
+"2 2 3 1",
 " 	c None",
 ".	c #CACACA",
 "+	c #ECECEA",
-".",
-"+"};
+"..",
+"++"};
diff -Naur orig/Nitrogen/titleAB.xpm mod/Nitrogen/titleAB.xpm
--- orig/Nitrogen/titleAB.xpm	2014-04-30 13:00:56.000000000 +0200
+++ mod/Nitrogen/titleAB.xpm	2014-12-20 15:58:54.000000000 +0100
@@ -1,24 +1,24 @@
 /* XPM */
 static char * titleAB_xpm[] = {
-"1 19 2 1",
+"2 19 2 1",
 " 	c None",
 ".	c #E0DFDE",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-"."};
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+".."};
diff -Naur orig/Nitrogen/titleAL.xpm mod/Nitrogen/titleAL.xpm
--- orig/Nitrogen/titleAL.xpm	2014-04-30 13:00:56.000000000 +0200
+++ mod/Nitrogen/titleAL.xpm	2014-12-20 15:58:54.000000000 +0100
@@ -1,24 +1,24 @@
 /* XPM */
 static char * titleAL_xpm[] = {
-"1 19 2 1",
+"2 19 2 1",
 " 	c None",
 ".	c #E0DFDE",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-"."};
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+".."};
diff -Naur orig/Nitrogen/titleAM.xpm mod/Nitrogen/titleAM.xpm
--- orig/Nitrogen/titleAM.xpm	2014-04-30 13:00:56.000000000 +0200
+++ mod/Nitrogen/titleAM.xpm	2014-12-20 15:58:54.000000000 +0100
@@ -1,24 +1,24 @@
 /* XPM */
 static char * titleAM_xpm[] = {
-"1 19 2 1",
+"2 19 2 1",
 " 	c None",
 ".	c #E0DFDE",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-"."};
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+".."};
diff -Naur orig/Nitrogen/titleAP.xpm mod/Nitrogen/titleAP.xpm
--- orig/Nitrogen/titleAP.xpm	2014-04-30 13:00:56.000000000 +0200
+++ mod/Nitrogen/titleAP.xpm	2014-12-20 15:58:54.000000000 +0100
@@ -1,24 +1,24 @@
 /* XPM */
 static char * titleAP_xpm[] = {
-"1 19 2 1",
+"2 19 2 1",
 " 	c None",
 ".	c #E0DFDE",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-"."};
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+".."};
diff -Naur orig/Nitrogen/titleAR.xpm mod/Nitrogen/titleAR.xpm
--- orig/Nitrogen/titleAR.xpm	2014-04-30 13:00:56.000000000 +0200
+++ mod/Nitrogen/titleAR.xpm	2014-12-20 15:58:54.000000000 +0100
@@ -1,24 +1,24 @@
 /* XPM */
 static char * titleAR_xpm[] = {
-"1 19 2 1",
+"2 19 2 1",
 " 	c None",
 ".	c #E0DFDE",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-"."};
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+".."};
diff -Naur orig/Nitrogen/titleAS.xpm mod/Nitrogen/titleAS.xpm
--- orig/Nitrogen/titleAS.xpm	2014-04-30 13:00:56.000000000 +0200
+++ mod/Nitrogen/titleAS.xpm	2014-12-20 15:58:54.000000000 +0100
@@ -1,24 +1,24 @@
 /* XPM */
 static char * titleAS_xpm[] = {
-"1 19 2 1",
+"2 19 2 1",
 " 	c None",
 ".	c #E0DFDE",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-"."};
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+".."};
diff -Naur orig/Nitrogen/titleAT.xpm mod/Nitrogen/titleAT.xpm
--- orig/Nitrogen/titleAT.xpm	2014-04-30 13:00:56.000000000 +0200
+++ mod/Nitrogen/titleAT.xpm	2014-12-20 15:58:54.000000000 +0100
@@ -1,24 +1,24 @@
 /* XPM */
 static char * titleAT_xpm[] = {
-"1 19 2 1",
+"2 19 2 1",
 " 	c None",
 ".	c #E0DFDE",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-"."};
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+".."};
diff -Naur orig/Nitrogen/titleIB.xpm mod/Nitrogen/titleIB.xpm
--- orig/Nitrogen/titleIB.xpm	2014-04-30 13:00:56.000000000 +0200
+++ mod/Nitrogen/titleIB.xpm	2014-12-20 15:58:54.000000000 +0100
@@ -1,29 +1,29 @@
 /* XPM */
 static char * titleIB_xpm[] = {
-"1 24 2 1",
+"2 24 2 1",
 " 	c None",
 ".	c #E0DFDF",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-"."};
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+".."};
diff -Naur orig/Nitrogen/titleIL.xpm mod/Nitrogen/titleIL.xpm
--- orig/Nitrogen/titleIL.xpm	2014-04-30 13:00:56.000000000 +0200
+++ mod/Nitrogen/titleIL.xpm	2014-12-20 15:58:54.000000000 +0100
@@ -1,29 +1,29 @@
 /* XPM */
 static char * titleIL_xpm[] = {
-"1 24 2 1",
+"2 24 2 1",
 " 	c None",
 ".	c #E0DFDF",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-"."};
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+".."};
diff -Naur orig/Nitrogen/titleIM.xpm mod/Nitrogen/titleIM.xpm
--- orig/Nitrogen/titleIM.xpm	2014-04-30 13:00:56.000000000 +0200
+++ mod/Nitrogen/titleIM.xpm	2014-12-20 15:58:54.000000000 +0100
@@ -1,29 +1,29 @@
 /* XPM */
 static char * titleIM_xpm[] = {
-"1 24 2 1",
+"2 24 2 1",
 " 	c None",
 ".	c #E0DFDF",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-"."};
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+".."};
diff -Naur orig/Nitrogen/titleIP.xpm mod/Nitrogen/titleIP.xpm
--- orig/Nitrogen/titleIP.xpm	2014-04-30 13:00:56.000000000 +0200
+++ mod/Nitrogen/titleIP.xpm	2014-12-20 15:58:54.000000000 +0100
@@ -1,29 +1,29 @@
 /* XPM */
 static char * titleIP_xpm[] = {
-"1 24 2 1",
+"2 24 2 1",
 " 	c None",
 ".	c #E0DFDF",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-"."};
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+".."};
diff -Naur orig/Nitrogen/titleIR.xpm mod/Nitrogen/titleIR.xpm
--- orig/Nitrogen/titleIR.xpm	2014-04-30 13:00:56.000000000 +0200
+++ mod/Nitrogen/titleIR.xpm	2014-12-20 15:58:54.000000000 +0100
@@ -1,29 +1,29 @@
 /* XPM */
 static char * titleIR_xpm[] = {
-"1 24 2 1",
+"2 24 2 1",
 " 	c None",
 ".	c #E0DFDF",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-"."};
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+".."};
diff -Naur orig/Nitrogen/titleIS.xpm mod/Nitrogen/titleIS.xpm
--- orig/Nitrogen/titleIS.xpm	2014-04-30 13:00:56.000000000 +0200
+++ mod/Nitrogen/titleIS.xpm	2014-12-20 15:58:54.000000000 +0100
@@ -1,29 +1,29 @@
 /* XPM */
 static char * titleIS_xpm[] = {
-"1 24 2 1",
+"2 24 2 1",
 " 	c None",
 ".	c #E0DFDF",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-"."};
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+".."};
diff -Naur orig/Nitrogen/titleIT.xpm mod/Nitrogen/titleIT.xpm
--- orig/Nitrogen/titleIT.xpm	2014-04-30 13:00:56.000000000 +0200
+++ mod/Nitrogen/titleIT.xpm	2014-12-20 15:58:54.000000000 +0100
@@ -1,29 +1,29 @@
 /* XPM */
 static char * titleIT_xpm[] = {
-"1 24 2 1",
+"2 24 2 1",
 " 	c None",
 ".	c #E0DFDF",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-"."};
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+".."};
diff -Naur orig/Nitrogen/titleML.xpm mod/Nitrogen/titleML.xpm
--- orig/Nitrogen/titleML.xpm	2014-04-30 13:00:56.000000000 +0200
+++ mod/Nitrogen/titleML.xpm	2014-12-20 15:58:54.000000000 +0100
@@ -1,29 +1,29 @@
 /* XPM */
 static char * titleML_xpm[] = {
-"1 24 2 1",
+"2 24 2 1",
 " 	c None",
 ".	c #E0DFDF",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-"."};
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+".."};
