--- a/src/gnome-screenshot.desktop.in.orig	2013-03-07 20:03:39.000000000 +0100
+++ a/src/gnome-screenshot.desktop.in	2013-03-07 20:04:43.000000000 +0100
@@ -7,7 +7,7 @@
 Type=Application
 Icon=applets-screenshooter
 StartupNotify=true
-Categories=GTK;GNOME;Utility;
+Categories=GTK;GNOME;Utility;X-UHU-Menu/Graphics/Capture;
 NotShowIn=KDE;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=gnome-utils
