diff -Naur orig/supertux2.desktop mod/supertux2.desktop
--- orig/supertux2.desktop	2013-07-09 01:03:39.000000000 +0200
+++ mod/supertux2.desktop	2014-06-12 13:00:41.000000000 +0200
@@ -1,7 +1,5 @@
 [Desktop Entry]
 Type=Application
-Version=1.0
-Encoding=UTF-8
 Name=SuperTux 2
 Name[en]=SuperTux 2
 GenericName=Platform Game
@@ -36,7 +34,7 @@
 Comment[pt]=Joga este clássico de plataformas 2D
 Comment[pt_BR]=Encarne o pinguin Tux neste jogo inspirado em clássicos de Pular&Correr
 Comment[hu]=Egy klasszikus 2D-s oldalnézeti játék
-Icon=supertux.png
+Icon=supertux
 Exec=supertux2
 Terminal=false
 StartupNotify=false
