diff -Naur firefox-46.0-orig/browser/components/nsBrowserContentHandler.js firefox-46.0/browser/components/nsBrowserContentHandler.js
--- firefox-46.0-orig/browser/components/nsBrowserContentHandler.js	2016-04-22 02:37:20.000000000 +0200
+++ firefox-46.0/browser/components/nsBrowserContentHandler.js	2016-04-27 23:13:44.458416773 +0200
@@ -523,8 +523,6 @@
         switch (override) {
           case OVERRIDE_NEW_PROFILE:
             // New profile.
-            overridePage = Services.urlFormatter.formatURLPref("startup.homepage_welcome_url");
-            additionalPage = Services.urlFormatter.formatURLPref("startup.homepage_welcome_url.additional");
             // Turn on 'later run' pages for new profiles.
             LaterRun.enabled = true;
             break;
@@ -538,11 +536,6 @@
                                .getService(Components.interfaces.nsISessionStartup);
             willRestoreSession = ss.isAutomaticRestoreEnabled();
 
-            overridePage = Services.urlFormatter.formatURLPref("startup.homepage_override_url");
-            if (prefb.prefHasUserValue("app.update.postupdate"))
-              overridePage = getPostUpdateOverridePage(overridePage);
-
-            overridePage = overridePage.replace("%OLD_VERSION%", old_mstone);
             break;
         }
       }
