diff -Naur orig/config/baseconfig.mk patched/config/baseconfig.mk
--- orig/config/baseconfig.mk	2014-04-22 07:27:37.000000000 +0200
+++ patched/config/baseconfig.mk	2014-04-28 21:55:58.564212936 +0200
@@ -4,8 +4,8 @@
 # whether a normal build is happening or whether the check is running.
 includedir := $(includedir)/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
 idldir = $(datadir)/idl/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
-installdir = $(libdir)/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
-sdkdir = $(libdir)/$(MOZ_APP_NAME)-devel-$(MOZ_APP_VERSION)
+installdir = $(libdir)/$(MOZ_APP_NAME)
+sdkdir = $(libdir)/$(MOZ_APP_NAME)-devel
 ifndef TOP_DIST
 TOP_DIST = dist
 endif
