diff -Naur showimage-1.6.2-orig//showimage.desktop showimage-1.6.2//showimage.desktop
--- showimage-1.6.2-orig//showimage.desktop	2011-08-29 19:21:53.000000000 +0200
+++ showimage-1.6.2//showimage.desktop	2011-09-07 21:18:58.000000000 +0200
@@ -3,7 +3,7 @@
 Encoding=UTF-8
 Exec=showimage %F
 Icon=showimage
-Categories=Graphics;RasterGraphics;Viewer;
+Categories=Graphics;RasterGraphics;Viewer;X-UHU-Menu/Graphics;
 Terminal=false
 
 Name=ShowImage
