diff -Naur krusader-2.4.0-beta3-orig/krusader/krusader.desktop krusader-2.4.0-beta3/krusader/krusader.desktop
--- krusader-2.4.0-beta3-orig/krusader/krusader.desktop	2012-10-21 12:55:54.000000000 +0200
+++ krusader-2.4.0-beta3/krusader/krusader.desktop	2012-12-04 16:51:35.000000000 +0100
@@ -44,7 +44,7 @@
 Terminal=false
 Type=Application
 X-DocPath=krusader/index.html
-Categories=FileManager;Utility;Qt;KDE;
+Categories=FileManager;Utility;Qt;KDE;X-UHU-Menu/Tools/File_Managers;
 Comment=Twin-Panel File Manager
 Comment[bg]=Файлов мениджър с два панела
 Comment[bs]=Dvostrani menadžer datoteka
diff -Naur krusader-2.4.0-beta3-orig/krusader/krusader_root-mode.desktop krusader-2.4.0-beta3/krusader/krusader_root-mode.desktop
--- krusader-2.4.0-beta3-orig/krusader/krusader_root-mode.desktop	2012-10-21 12:55:54.000000000 +0200
+++ krusader-2.4.0-beta3/krusader/krusader_root-mode.desktop	2012-12-04 16:51:46.000000000 +0100
@@ -41,7 +41,7 @@
 Icon=krusader_root
 Terminal=false
 Type=Application
-Categories=Qt;KDE;System;
+Categories=Qt;KDE;System;X-UHU-Menu/Tools/File_Managers;
 OnlyShowIn=KDE;
 X-KDE-SubstituteUID=true
 X-KDE-Username=root
