diff -Nur orig/krusader/Search/krsearchdialog.cpp mod/krusader/Search/krsearchdialog.cpp
--- orig/krusader/Search/krsearchdialog.cpp	2016-10-22 15:30:11.000000000 +0200
+++ mod/krusader/Search/krsearchdialog.cpp	2016-11-05 13:46:32.000000000 +0100
@@ -477,14 +477,19 @@
 
 void KrSearchDialog::executed(const QString &name)
 {
+    // 'name' is (local) file path or complete URL
     QString path = name;
     QString fileName;
     if(!name.endsWith('/')) {
+        // not a directory, split filename and path
         int idx = name.lastIndexOf("/");
         fileName = name.mid(idx+1);
         path = name.left(idx);
     }
-    ACTIVE_FUNC->openUrl(QUrl::fromLocalFile(path), fileName);
+    QUrl url(path);
+    if (url.scheme().isEmpty())
+        url.setScheme("file");
+    ACTIVE_FUNC->openUrl(url, fileName);
     showMinimized();
 }
 
