diff -Naur orig/avidemux/common/ADM_commonUI/myOwnMenu.h mod/avidemux/common/ADM_commonUI/myOwnMenu.h
--- orig/avidemux/common/ADM_commonUI/myOwnMenu.h	2014-03-12 07:15:23.000000000 +0100
+++ mod/avidemux/common/ADM_commonUI/myOwnMenu.h	2014-03-18 15:44:50.091445074 +0100
@@ -83,7 +83,6 @@
 std::vector<MenuEntry> myMenuAudio(_myMenuAudio, _myMenuAudio + sizeof(_myMenuAudio) / sizeof(_myMenuAudio[0]));
 
 static const MenuEntry _myMenuHelp[] = {
-            {MENU_ACTION,QT_TRANSLATE_NOOP("adm","Build Option"),    NULL,ACT_BUILT_IN           ,NULL,NULL},
             {MENU_ACTION,QT_TRANSLATE_NOOP("adm","Plugins"),         NULL,ACT_PLUGIN_INFO        ,NULL,NULL},
 #ifdef _WIN32
 			{MENU_SEPARATOR,"-",NULL,ACT_DUMMY             ,NULL,NULL},
