diff -Nur orig/bootStrap.bash mod/bootStrap.bash
--- orig/bootStrap.bash	2016-11-19 08:58:39.000000000 +0100
+++ mod/bootStrap.bash	2016-12-09 16:57:46.154051890 +0100
@@ -54,7 +54,7 @@
         fi
         cd $BUILDDIR 
         cmake $COMPILER $PKG $FAKEROOT $QT_FLAVOR -DCMAKE_EDIT_COMMAND=vim $INSTALL_PREFIX $EXTRA $ASAN $DEBUG -G "$BUILDER" $SOURCEDIR || fail cmakeZ
-        make  $PARAL >& /tmp/log$BUILDDIR || fail "make, result in /tmp/log$BUILDDIR"
+        make  $PARAL
 	if  [ "x$PKG" != "x" ] ; then
           $FAKEROOT_COMMAND make package DESTDIR=$FAKEROOT_DIR/tmp || fail package
 	fi
@@ -238,7 +238,7 @@
 
 export TOP=$PWD
 export POSTFIX=""
-export FAKEROOT_DIR=$PWD/install
+export FAKEROOT_DIR=$UB_INSTALLDIR
 export PARAL="$O_PARAL"
 echo "Top dir : $TOP"
 echo "Fake installation directory=$FAKEROOT_DIR"
