diff -Naur orig/bootStrap.bash mod/bootStrap.bash
--- orig/bootStrap.bash	2015-06-10 07:42:47.000000000 +0200
+++ mod/bootStrap.bash	2015-07-22 16:16:22.639716388 +0200
@@ -35,7 +35,7 @@
         mkdir $BUILDDIR || fail mkdir
         cd $BUILDDIR 
         cmake $PKG $FAKEROOT $QT_FLAVOR -DCMAKE_EDIT_COMMAND=vim -DAVIDEMUX_SOURCE_DIR=$TOP -DCMAKE_INSTALL_PREFIX=/usr $EXTRA $DEBUG -G "$BUILDER" $SOURCEDIR || fail cmakeZ
-        make  $PARAL >& /tmp/log$BUILDDIR || fail "make, result in /tmp/log$BUILDDIR"
+        make  $PARAL
 	if  [ "x$PKG" != "x" ] ; then
           $FAKEROOT_COMMAND make package DESTDIR=$FAKEROOT_DIR/tmp || fail package
 	fi
@@ -171,7 +171,7 @@
 
 export TOP=$PWD
 export POSTFIX=""
-export FAKEROOT_DIR=$PWD/install
+export FAKEROOT_DIR=$UB_INSTALLDIR
 echo "Top dir : $TOP"
 echo "Fake installation directory=$FAKEROOT_DIR"
 if [ "x$debug" = "x1" ] ; then echo   
