diff -Naur orig/avidemux2.desktop patched/avidemux2.desktop
--- orig/avidemux2.desktop	2015-06-10 07:42:47.000000000 +0200
+++ patched/avidemux2.desktop	2015-09-26 21:51:23.885105949 +0200
@@ -1,8 +1,9 @@
 [Desktop Entry]
 Encoding=UTF-8
-Name=avidemux2
+Name=avidemux
 Comment=Video Editor
-Exec=avidemux2_gtk
+Comment[hu]=Videó szerkesztő
+Exec=avidemux
 MimeType=video/mpeg;video/quicktime;video/x-msvideo;video/x-anim;audio/x-mp3;audio/x-mp2;
 Icon=avidemux
 Terminal=false
