diff -Naur orig/avidemux/qt4/CMakeLists.txt patched/avidemux/qt4/CMakeLists.txt
--- orig/avidemux/qt4/CMakeLists.txt	2016-03-05 09:22:54.000000000 +0100
+++ patched/avidemux/qt4/CMakeLists.txt	2016-08-05 01:11:43.904618165 +0200
@@ -32,7 +32,7 @@
 ##########################################
 # Config
 ##########################################
-ADD_DEFINITIONS(-DADM_UI_TYPE_BUILD=ADM_UI_QT4)
+ADD_DEFINITIONS(-DADM_UI_TYPE_BUILD=ADM_UI_QT4 -std=c++11)
 SET(CONFIG_HEADER_TYPE ADM_BUILD_QT4)
 SET(UI_SUFFIX ${QT_EXTENSION})
 
