diff -Naur orig/xdao/gcdmaster.desktop mod/xdao/gcdmaster.desktop
--- orig/xdao/gcdmaster.desktop	2004-06-28 20:05:57.000000000 +0200
+++ mod/xdao/gcdmaster.desktop	2016-03-23 09:47:51.225966306 +0100
@@ -2,9 +2,10 @@
 Encoding=UTF-8
 Name=Gnome CD Master
 Comment=Gnome Audio CD editor and burner
+Comment[hu]=Gnome CD-író program
 Exec=gcdmaster %F
-Icon=gcdmaster.png
+Icon=gcdmaster
 Terminal=false
 Type=Application
 StartupNotify=true
-Categories=GNOME;AudioVideo;Application;X-Fedora;
+Categories=GNOME;AudioVideo;DiscBurning;
