diff -Naur orig/amule.desktop mod/amule.desktop
--- orig/amule.desktop	2011-10-08 18:48:02.000000000 +0200
+++ mod/amule.desktop	2014-06-12 14:38:50.000000000 +0200
@@ -1,10 +1,11 @@
 [Desktop Entry]
 Name=aMule
 Exec=amule
-Icon=amule
+Icon=/usr/share/pixmaps/amule.xpm
 Terminal=false
 Type=Application
 Categories=Network;P2P;
 Comment=A client for the eD2k network
 Comment[fr]=Un client pour le réseau eD2k
 Comment[tr]=eD2k ağı için istemci
+Comment[hu]=P2P Fájlcserélő
