diff -Naur orig/deluge/data/share/applications/deluge.desktop.in patched/deluge/data/share/applications/deluge.desktop.in
--- orig/deluge/data/share/applications/deluge.desktop.in	2011-07-22 20:12:47.000000000 +0200
+++ patched/deluge/data/share/applications/deluge.desktop.in	2011-08-27 16:48:11.000000000 +0200
@@ -9,6 +9,6 @@
 Icon=deluge
 Terminal=false
 Type=Application
-Categories=Network;FileTransfer;P2P;GTK;
+Categories=Network;FileTransfer;P2P;GTK;X-UHU-Menu/Internet/Downloaders;
 StartupNotify=true
 MimeType=application/x-bittorrent;x-scheme-handler/magnet;
