diff -Naur orig/data/gnome-subtitles.desktop.in patched/data/gnome-subtitles.desktop.in
--- orig/data/gnome-subtitles.desktop.in	2010-12-08 20:10:33.000000000 +0100
+++ patched/data/gnome-subtitles.desktop.in	2016-02-19 01:50:40.405553966 +0100
@@ -3,13 +3,14 @@
 Name=Gnome Subtitles
 GenericName=Subtitle Editor
 Comment=Video subtitling for the GNOME desktop
+Comment[hu]=Feliratok szerkesztése GNOME környezetben
 TryExec=gnome-subtitles
 Exec=gnome-subtitles %f
 Icon=gnome-subtitles
 StartupNotify=true
 Terminal=false
 Type=Application
-Categories=GNOME;GTK;Video;AudioVideo;AudioVideoEditing;
+Categories=GTK;GNOME;Video;AudioVideo;AudioVideoEditing;
 MimeType=application/x-subrip;text/x-mpsub;text/x-ssa;text/x-microdvd;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=gnome-subtitles
