From: Santiago Vila <sanvila@unex.es>
Subject: Correct default WAV sample rate
Bug-Debian: https://bugs.debian.org/744397
Applied-Upstream: http://sourceforge.net/p/fuse-emulator/code/5077/
Index: fuse-emulator-utils/tape2wav.c
===================================================================
--- fuse-emulator-utils.orig/tape2wav.c
+++ fuse-emulator-utils/tape2wav.c
@@ -41,7 +41,7 @@ static int read_tape( char *filename, li
 static int write_tape( char *filename, libspectrum_tape *tape );
 
 char *progname;
-int sample_rate = 41000;
+int sample_rate = 44100;
 
 int
 main( int argc, char **argv )
