diff -Naur orig/Makefile.in mod/Makefile.in
--- orig/Makefile.in	2016-02-06 21:46:03.000000000 +0100
+++ mod/Makefile.in	2016-02-08 19:19:08.924764056 +0100
@@ -330,7 +330,6 @@
 dist_doc_DATA = README
 doc_DATA = skeletonmm.tar.gz
 skeletonmm_data_files = \
-	skeletonmm/.gitignore						\
 	skeletonmm/AUTHORS						\
 	skeletonmm/COPYING						\
 	skeletonmm/ChangeLog						\
@@ -338,19 +337,15 @@
 	skeletonmm/README						\
 	skeletonmm/configure.ac						\
 	skeletonmm/skeletonmm.doap					\
-	skeletonmm/build/.gitignore					\
 	skeletonmm/codegen/Makefile.am					\
 	skeletonmm/codegen/extradefs/generate_extra_defs_skeleton.cc	\
 	skeletonmm/codegen/m4/convert.m4				\
 	skeletonmm/codegen/m4/convert_skeleton.m4			\
 	skeletonmm/codegen/m4/filelist.am				\
 	skeletonmm/doc/Makefile.am					\
-	skeletonmm/doc/reference/.gitignore				\
 	skeletonmm/doc/reference/Doxyfile.in				\
-	skeletonmm/examples/.gitignore					\
 	skeletonmm/examples/Makefile.am					\
 	skeletonmm/examples/example/example.cc				\
-	skeletonmm/skeleton/.gitignore					\
 	skeletonmm/skeleton/skeletonmm-uninstalled.pc.in		\
 	skeletonmm/skeleton/skeletonmm.h				\
 	skeletonmm/skeleton/skeletonmm.pc.in				\
@@ -358,7 +353,6 @@
 	skeletonmm/skeleton/skeletonmm/Makefile.am			\
 	skeletonmm/skeleton/skeletonmm/filelist.am			\
 	skeletonmm/skeleton/skeletonmm/wrap_init.h			\
-	skeletonmm/skeleton/skeletonmm/private/.gitignore		\
 	skeletonmm/skeleton/src/Makefile.am				\
 	skeletonmm/skeleton/src/filelist.am				\
 	skeletonmm/skeleton/src/skeleton.ccg				\
