diff -Naur telepathy-qt-0.9.3-orig/cmake/modules/FindQt.cmake telepathy-qt-0.9.3/cmake/modules/FindQt.cmake
--- telepathy-qt-0.9.3-orig/cmake/modules/FindQt.cmake	2012-07-13 19:40:23.000000000 +0200
+++ telepathy-qt-0.9.3/cmake/modules/FindQt.cmake	2014-02-18 10:17:38.000000000 +0100
@@ -74,7 +74,7 @@
 
 MACRO(QT_CREATE_MOC_COMMAND infile outfile moc_flags moc_options)
   IF(QT_VERSION_MAJOR MATCHES 4)
-    QT4_CREATE_MOC_COMMAND(${infile} ${outfile} "${moc_flags}" "${moc_options}")
+    QT4_CREATE_MOC_COMMAND(${infile} ${outfile} "${moc_flags}" "${moc_options}" "${moc_target}")
   ELSE(QT_VERSION_MAJOR MATCHES 4)
     IF(QT_VERSION_MAJOR MATCHES 5)
       QT5_CREATE_MOC_COMMAND(${infile} ${outfile} "${moc_flags}" "${moc_options}")
