diff -Naur orig/setup.py patched/setup.py
--- orig/setup.py	2016-02-01 00:17:36.000000000 +0100
+++ patched/setup.py	2016-02-01 04:13:41.126964974 +0100
@@ -18,6 +18,7 @@
 
 import os
 import sys
+import codecs
 
 try:
     import DistUtilsExtra.auto
@@ -34,8 +35,8 @@
     filename = os.path.join(libdir, 'catfish_lib', 'catfishconfig.py')
     oldvalues = {}
     try:
-        fin = open(filename, 'r', encoding='utf-8')
-        fout = open(filename + '.new', 'w', encoding='utf-8')
+        fin = codecs.open(filename, 'r', 'utf-8')
+        fout = codecs.open(filename + '.new', 'w', 'utf-8')
 
         for line in fin:
             fields = line.split(' = ')  # Separate variable from value
@@ -96,8 +97,8 @@
 def update_desktop_file(filename, script_path):
     """Update the desktop file with prefixed paths."""
     try:
-        fin = open(filename, 'r', encoding='utf-8')
-        fout = open(filename + '.new', 'w', encoding='utf-8')
+        fin = codecs.open(filename, 'r', 'utf-8')
+        fout = codecs.open(filename + '.new', 'w', 'utf-8')
 
         for line in fin:
             if 'Exec=' in line:
