diff --git a/data/ui/CatfishWindow.ui b/data/ui/CatfishWindow.ui
index e04fd10..53f9672 100644
--- a/data/ui/CatfishWindow.ui
+++ b/data/ui/CatfishWindow.ui
@@ -1218,7 +1218,7 @@
       </object>
     </child>
     <child>
-      <object class="GtkListBoxRow" id="listboxrow6">
+      <object class="GtkListBoxRow" id="application_menu_update">
         <property name="visible">True</property>
         <property name="can_focus">True</property>
         <property name="selectable">False</property>
