diff -Naurd a/res/converseen.desktop b/res/converseen.desktop
--- a/res/converseen.desktop	2014-05-05 19:53:42.000000000 +0200
+++ b/res/converseen.desktop	2014-05-06 19:09:59.000000000 +0200
@@ -6,6 +6,7 @@
 GenericName[ru]=Converseen
 GenericName[ro]=Converseen
 Comment=Batch image converter
+Comment[hu]=Kép átalakító
 Comment[it]=Convertitore batch di immagini
 Comment[ru]=Пакетное преобразование изображений
 Comment[ro]=Batch converter pentru imagini
