diff -Naur orig/CMakeLists.txt patched/CMakeLists.txt
--- orig/CMakeLists.txt	2015-03-22 21:27:46.000000000 +0100
+++ patched/CMakeLists.txt	2016-08-03 01:47:11.632803735 +0200
@@ -86,21 +86,30 @@
     loc/converseen_ja_JP.ts
 )
 
-FIND_PACKAGE(Qt4 REQUIRED)
+FIND_PACKAGE(Qt5Core REQUIRED)
+FIND_PACKAGE(Qt5Gui REQUIRED)
+FIND_PACKAGE(Qt5Network REQUIRED)
+FIND_PACKAGE(Qt5Widgets REQUIRED)
+FIND_PACKAGE(Qt5LinguistTools REQUIRED)
 FIND_PACKAGE(ImageMagick REQUIRED)
 FIND_PACKAGE(ImageMagick COMPONENTS Magick++)
 FIND_PACKAGE(ImageMagick COMPONENTS MagickWand)
 FIND_PACKAGE(ImageMagick COMPONENTS MagickCore)
 
-INCLUDE(${QT_USE_FILE})
-QT4_WRAP_UI(converseen_UIS_H ${converseen_UIS})  
-QT4_WRAP_CPP(converseen_MOC_SRCS ${converseen_MOC_HDRS})  
-QT4_ADD_RESOURCES(converseen_RESOURCES ${converseen_RESOURCES})
-QT4_ADD_TRANSLATION(converseen_RESOURCES ${converseen_LANG}) 
+INCLUDE_DIRECTORIES(${Qt5Core_INCLUDE_DIRS})
+INCLUDE_DIRECTORIES(${Qt5Gui_INCLUDE_DIRS})
+INCLUDE_DIRECTORIES(${Qt5Network_INCLUDE_DIRS})
+INCLUDE_DIRECTORIES(${Qt5Widgets_INCLUDE_DIRS})
+QT5_WRAP_UI(converseen_UIS_H ${converseen_UIS})
+QT5_WRAP_CPP(converseen_MOC_SRCS ${converseen_MOC_HDRS})
+QT5_ADD_RESOURCES(converseen_RESOURCES ${converseen_RESOURCES})
+QT5_ADD_TRANSLATION(converseen_RESOURCES ${converseen_LANG})
 
 ADD_DEFINITIONS(
 	-Wall
 	-W
+	-std=c++11
+	-fPIC
 	-DMAGICKCORE_HDRI_ENABLE=0
 	-DMAGICKCORE_QUANTUM_DEPTH=16
 	${QT_DEFINITIONS}
@@ -116,7 +125,7 @@
     ${converseen_RCS} 
 )  
   
-TARGET_LINK_LIBRARIES(${GUI} ${QT_LIBRARIES} ${ImageMagick_LIBRARIES} ${ImageMagick_Magick++_LIBRARIES})
+TARGET_LINK_LIBRARIES(${GUI} ${Qt5Network_LIBRARIES} ${Qt5Widgets_LIBRARIES} ${ImageMagick_LIBRARIES} ${ImageMagick_Magick++_LIBRARIES})
 
 install(PROGRAMS ${CMAKE_CURRENT_BINARY_DIR}/converseen DESTINATION ${CMAKE_INSTALL_PREFIX}/bin)
 install(FILES res/converseen.desktop DESTINATION ${CMAKE_INSTALL_PREFIX}/share/applications)
diff -Naur orig/converseen.pro patched/converseen.pro
--- orig/converseen.pro	2015-03-22 21:27:46.000000000 +0100
+++ patched/converseen.pro	2016-08-03 01:06:18.624396839 +0200
@@ -1,5 +1,5 @@
 TEMPLATE = app
-QT = gui core network
+QT = gui core network widgets
 CONFIG += qt \
  release \
  warn_on \
diff -Naur orig/src/inisettings.cpp patched/src/inisettings.cpp
--- orig/src/inisettings.cpp	2015-03-22 21:27:46.000000000 +0100
+++ patched/src/inisettings.cpp	2016-08-03 01:06:18.624396839 +0200
@@ -22,7 +22,7 @@
 */
 
 #include <QDir>
-#include <QDesktopServices>
+#include <QStandardPaths>
 #include "inisettings.h"
 
 #define INIFILENAME ".converseen.conf"
@@ -37,7 +37,7 @@
 
 void IniSettings::init()
 {
-	QString myPath =  QDesktopServices::storageLocation(QDesktopServices::DataLocation);
+	QString myPath =  QStandardPaths::writableLocation(QStandardPaths::DataLocation);
 	QString iniPath = QString("%1/%2").arg(myPath).arg(INIFILENAME);
 
     settings = new QSettings(iniPath, QSettings::IniFormat);
diff -Naur orig/src/main.cpp patched/src/main.cpp
--- orig/src/main.cpp	2015-03-22 21:27:46.000000000 +0100
+++ patched/src/main.cpp	2016-08-03 01:06:18.628396911 +0200
@@ -35,8 +35,6 @@
 
     QCoreApplication::setApplicationVersion(globals::VERSION);
 
-    QTextCodec::setCodecForTr(QTextCodec::codecForName("UTF-8"));
-    QTextCodec::setCodecForCStrings(QTextCodec::codecForName("UTF-8"));
     QTextCodec::setCodecForLocale(QTextCodec::codecForName("UTF-8"));
 
     QApplication app( argc, argv );
diff -Naur orig/src/mainwindowimpl.cpp patched/src/mainwindowimpl.cpp
--- orig/src/mainwindowimpl.cpp	2015-03-22 21:27:46.000000000 +0100
+++ patched/src/mainwindowimpl.cpp	2016-08-03 01:06:18.628396911 +0200
@@ -40,7 +40,7 @@
 
 // TODO: Inserire funzione per sostituire tutte le immagini
 
-MainWindowImpl::MainWindowImpl(QWidget * parent, Qt::WFlags f)
+MainWindowImpl::MainWindowImpl(QWidget * parent, Qt::WindowFlags f)
     : QMainWindow(parent, f)
 {
     setupUi(this);
@@ -93,7 +93,7 @@
     adjustSize();
     dockWidget->adjustSize();
 
-    treeWidget->header()->setResizeMode(0, QHeaderView::ResizeToContents);
+    treeWidget->header()->setSectionResizeMode(0, QHeaderView::ResizeToContents);
 
     loadOptions();
 
diff -Naur orig/src/mainwindowimpl.h patched/src/mainwindowimpl.h
--- orig/src/mainwindowimpl.h	2015-03-22 21:27:46.000000000 +0100
+++ patched/src/mainwindowimpl.h	2016-08-03 01:06:18.628396911 +0200
@@ -49,7 +49,7 @@
 {
     Q_OBJECT
 public:
-    MainWindowImpl( QWidget * parent = 0, Qt::WFlags f = 0 );
+    MainWindowImpl( QWidget * parent = 0, Qt::WindowFlags f = 0 );
     ~MainWindowImpl();
     void importListFromArgv(QString fileName);
 
diff -Naur orig/src/pixtreewidget.cpp patched/src/pixtreewidget.cpp
--- orig/src/pixtreewidget.cpp	2015-03-22 21:27:46.000000000 +0100
+++ patched/src/pixtreewidget.cpp	2016-08-03 01:06:18.628396911 +0200
@@ -27,6 +27,7 @@
 #include <QDragMoveEvent>
 #include <QDir>
 #include <QFileInfo>
+#include <QMimeData>
 #include <QUrl>
 #include "pixtreewidget.h"
 #include "formats.h"
