diff -Naur orig/src/applet.c patched/src/applet.c
--- orig/src/applet.c	2017-01-24 00:41:39.000000000 +0100
+++ patched/src/applet.c	2017-01-24 01:25:51.748141841 +0100
@@ -150,6 +150,11 @@
  */
 static void brisk_menu_applet_init(BriskMenuApplet *self)
 {
+        setlocale (LC_ALL, "");
+        bindtextdomain (GETTEXT_PACKAGE, MATELOCALEDIR);
+        bind_textdomain_codeset (GETTEXT_PACKAGE, "UTF-8");
+        textdomain (GETTEXT_PACKAGE);
+
         GtkWidget *toggle, *layout, *image, *label, *menu = NULL;
 
         self->binder = brisk_key_binder_new();
