diff -Naur partitionmanager-1.0.0-orig/src/partitionmanager.desktop partitionmanager-1.0.0/src/partitionmanager.desktop
--- partitionmanager-1.0.0-orig/src/partitionmanager.desktop	2009-08-17 11:43:51.000000000 +0200
+++ partitionmanager-1.0.0/src/partitionmanager.desktop	2009-08-22 20:39:39.000000000 +0200
@@ -96,7 +96,7 @@
 Comment[zh_TW]=管理磁碟、分割區與檔案系統
 Exec=partitionmanager
 Icon=partitionmanager
-Categories=System;Filesystem;KDE;Qt;
+Categories=System;Filesystem;KDE;Qt;X-UHU-Menu/Tools/Utilities;
 
 X-KDE-SubstituteUID=true
 X-KDE-StartupNotify=true
