--- anjuta.orig/anjuta.desktop.in.in	2009-03-18 09:24:10.000000000 +0100
+++ anjuta/anjuta.desktop.in.in	2009-04-02 20:16:15.000000000 +0200
@@ -7,6 +7,6 @@
 Terminal=false
 X-MultipleArgs=false
 Type=Application
-Categories=GNOME;GTK;Development;IDE;
+Categories=GTK;GNOME;Development;IDE;X-UHU-Menu/Development/Programming;
 StartupNotify=true
 MimeType=application/x-anjuta;
 X-GNOME-DocPath=anjuta/anjuta-manual.xml
