diff -Naur orig/CMakeLists.txt patched/CMakeLists.txt
--- orig/CMakeLists.txt	2016-09-13 13:14:17.000000000 +0200
+++ patched/CMakeLists.txt	2017-02-06 02:18:02.612915728 +0100
@@ -63,13 +63,13 @@
 
 # if plasma-workspace is installed in different prefix then pmc then it will not work
 
-set(STARTKDE_BINARY_DIR ${CMAKE_INSTALL_FULL_BINDIR})
+# set(STARTKDE_BINARY_DIR ${CMAKE_INSTALL_FULL_BINDIR})
 
-configure_file(plasma-mediacenter.desktop.cmake ${CMAKE_CURRENT_BINARY_DIR}/plasma-mediacenter.desktop)
-install(FILES
-        ${CMAKE_CURRENT_BINARY_DIR}/plasma-mediacenter.desktop
-        DESTINATION ${DATA_INSTALL_DIR}/xsessions
-)
+# configure_file(plasma-mediacenter.desktop.cmake ${CMAKE_CURRENT_BINARY_DIR}/plasma-mediacenter.desktop)
+# install(FILES
+#        ${CMAKE_CURRENT_BINARY_DIR}/plasma-mediacenter.desktop
+#        DESTINATION ${DATA_INSTALL_DIR}/xsessions
+# )
 set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${KDE_ENABLE_EXCEPTIONS}")
 set(CMAKE_AUTOMOC ON)
 
