diff -Naur orig/lxqt-config-brightness/translations/lxqt-config-brightness_es.desktop patched/lxqt-config-brightness/translations/lxqt-config-brightness_es.desktop
--- orig/lxqt-config-brightness/translations/lxqt-config-brightness_es.desktop	2016-09-24 00:44:36.000000000 +0200
+++ patched/lxqt-config-brightness/translations/lxqt-config-brightness_es.desktop	2016-09-25 18:40:30.400168346 +0200
@@ -1,4 +1,2 @@
 # Translations
-Comment[es]=Permite cambiar el brillo y el contraste del monitor
-GenericName[es]=Configuración de Brillo y contraste LXQt
 Name[es]=Configuración de Brillo y contraste LXQt
diff -Naur orig/lxqt-config-brightness/translations/lxqt-config-brightness_ru.desktop patched/lxqt-config-brightness/translations/lxqt-config-brightness_ru.desktop
--- orig/lxqt-config-brightness/translations/lxqt-config-brightness_ru.desktop	2016-09-24 00:44:36.000000000 +0200
+++ patched/lxqt-config-brightness/translations/lxqt-config-brightness_ru.desktop	2016-09-25 18:40:37.908258843 +0200
@@ -1,4 +1,2 @@
 # Translations
-Comment[ru]=Позволяет менять яркость и контраст монитора
-GenericName[ru]=Настройка яркости и контраста LXQt
 Name[ru]=Настройка яркости и контраста LXQt
