diff -Naur orig/Makefile.in patched/Makefile.in
--- orig/Makefile.in	2014-04-17 16:50:14.000000000 +0200
+++ patched/Makefile.in	2014-04-29 19:11:14.000000000 +0200
@@ -115,7 +115,7 @@
 	texture.$(OBJEXT) typeConvert.$(OBJEXT)
 am_xpaint_OBJECTS = $(am__objects_1) $(am__objects_2)
 xpaint_OBJECTS = $(am_xpaint_OBJECTS)
-xpaint_DEPENDENCIES = rw/.libs/librw.a
+xpaint_DEPENDENCIES = rw/.libs/librw.so
 AM_V_P = $(am__v_P_@AM_V@)
 am__v_P_ = $(am__v_P_@AM_DEFAULT_V@)
 am__v_P_0 = false
@@ -405,7 +405,7 @@
 
 SUBDIRS = rw
 xpaint_SOURCES = $(BASE_HDRS) $(BASE_SRCS)
-xpaint_LDADD = rw/.libs/librw.a @X_LIBS@ @X_LIBS_EXTRA@
+xpaint_LDADD = rw/.libs/librw.so @X_LIBS@ @X_LIBS_EXTRA@
 all: $(BUILT_SOURCES)
 	$(MAKE) $(AM_MAKEFLAGS) all-recursive
 
