diff -Naur orig/CBLAS/cblas.pc.in patched/CBLAS/cblas.pc.in
--- orig/CBLAS/cblas.pc.in	2016-06-19 00:15:11.000000000 +0200
+++ patched/CBLAS/cblas.pc.in	2016-07-07 06:50:13.738571886 +0200
@@ -1,9 +1,9 @@
-prefix=@prefix@
-libdir=@libdir@
+prefix=@CMAKE_INSTALL_PREFIX@
+libdir=@CMAKE_INSTALL_PREFIX@/@CMAKE_INSTALL_LIBDIR@
 
 Name: lapacke
 Description: C Standard Interface to BLAS Linear Algebra PACKage
 Version: @LAPACK_VERSION@
 URL: http://www.netlib.org/lapack/
-Libs: -L${libdir} -lcblas
+Libs: -L@CMAKE_INSTALL_PREFIX@/@CMAKE_INSTALL_LIBDIR@ -lcblas
 Requires: blas
diff -Naur orig/LAPACKE/lapacke.pc.in patched/LAPACKE/lapacke.pc.in
--- orig/LAPACKE/lapacke.pc.in	2016-06-19 00:15:11.000000000 +0200
+++ patched/LAPACKE/lapacke.pc.in	2016-07-07 06:50:30.154835096 +0200
@@ -1,9 +1,9 @@
-prefix=@prefix@
-libdir=@libdir@
+prefix=@CMAKE_INSTALL_PREFIX@
+libdir=@CMAKE_INSTALL_PREFIX@/@CMAKE_INSTALL_LIBDIR@
 
 Name: lapacke
 Description: C Standard Interface to LAPACK Linear Algebra PACKage
 Version: @LAPACK_VERSION@
 URL: http://www.netlib.org/lapack/
-Libs: -L${libdir} -llapacke
+Libs: -L@CMAKE_INSTALL_PREFIX@/@CMAKE_INSTALL_LIBDIR@ -llapacke
 Requires: lapack blas
