diff -Naur orig/configure patched/configure
--- orig/configure	2009-05-20 19:11:19.000000000 +0200
+++ patched/configure	2013-08-04 19:52:02.819662211 +0200
@@ -19541,7 +19541,7 @@
 done
 
 if test x$HAVE_GLES = xyes; then
-  GLES_LIBS="-lGLES_CM"
+  GLES_LIBS="-lGLESv1_CM"
   GLES_CFLAGS=""
 fi
 
diff -Naur orig/configure.ac patched/configure.ac
--- orig/configure.ac	2009-05-20 19:08:32.000000000 +0200
+++ patched/configure.ac	2013-08-04 19:52:14.143766727 +0200
@@ -339,7 +339,7 @@
 AC_CHECK_HEADERS([GLES/egl.h], [HAVE_EGL=yes], [HAVE_EGL=no])
 AC_CHECK_HEADERS([GLES/gl.h], [HAVE_GLES=yes], [HAVE_GLES=no])
 if test x$HAVE_GLES = xyes; then
-  GLES_LIBS="-lGLES_CM"
+  GLES_LIBS="-lGLESv1_CM"
   GLES_CFLAGS=""
 fi
 
