diff -Nur orig/cmake/GenerateExportHeader.cmake mod/cmake/GenerateExportHeader.cmake
--- orig/cmake/GenerateExportHeader.cmake	2016-11-19 09:33:06.000000000 +0100
+++ mod/cmake/GenerateExportHeader.cmake	2016-11-19 10:46:17.696374073 +0100
@@ -164,7 +164,7 @@
   if(CMAKE_COMPILER_IS_GNUCXX)
     exec_program(${CMAKE_C_COMPILER} ARGS --version
       OUTPUT_VARIABLE _gcc_version_info)
-    string(REGEX MATCH "[345]\\.[0-9]\\.[0-9]"
+    string(REGEX MATCH "[345678]\\.[0-9]\\.[0-9]"
       _gcc_version "${_gcc_version_info}")
     # gcc on mac just reports: "gcc (GCC) 3.3 20030304 ..." without the
     # patch level, handle this here:
