diff -Naur orig/share/extcalc.desktop mod/share/extcalc.desktop
--- orig/share/extcalc.desktop	2010-04-09 18:13:47.000000000 +0200
+++ mod/share/extcalc.desktop	2016-01-24 21:59:37.647786009 +0100
@@ -1,8 +1,9 @@
 [Desktop Entry]
-Encoding=UTF-8
 Name=Extcalc
+Comment=Scientific graphic calculator
+Comment[hu]=Többfunkciós grafikus számológép
 Exec=extcalc
 Terminal=false
 Type=Application
-Categories=Qt;Utility
-Icon=extcalc48.png
+Categories=Qt;Utility;
+Icon=extcalc48
diff -Naur orig/src/calclocale.h mod/src/calclocale.h
--- orig/src/calclocale.h	2010-04-09 18:13:47.000000000 +0200
+++ mod/src/calclocale.h	2016-01-24 22:17:54.336250687 +0100
@@ -21,7 +21,7 @@
 
 //This macros will be defined by the configure-script at a normal installation
 #ifndef INSTALLDIR
-#define INSTALLDIR "/usr/local/share/extcalc"
+#define INSTALLDIR "/usr/share/extcalc"
 #endif
 
 #define EXTCALCH_MSG1 qApp->translate("001","Error loading icon")
