diff -urN 0install-2.3.3.orig/setup.py 0install-2.3.3/setup.py
--- 0install-2.3.3.orig/setup.py	2013-08-12 16:44:36.000000000 +0200
+++ 0install-2.3.3/setup.py	2014-03-12 10:52:22.737312224 +0100
@@ -116,7 +116,7 @@
       author_email="zero-install-devel@lists.sourceforge.net",
       url="http://0install.net",
       scripts=['0launch', '0alias', '0store', '0store-secure-add', '0desktop'] + (['0install'] if pure_python else []),
-      data_files = [('man/man1', ['0launch.1', '0alias.1', '0store-secure-add.1', '0store.1', '0desktop.1', '0install.1']),
+      data_files = [('share/man/man1', ['0launch.1', '0alias.1', '0store-secure-add.1', '0store.1', '0desktop.1', '0install.1']),
 		    ('share/applications', ['share/applications/0install.desktop']),
 		    ('share/bash-completion/completions', ['share/bash-completion/completions/0install']),
 		    ('share/fish/completions', ['share/fish/completions/0install.fish']),
