diff -Naur libgnome-2.8.0.orig/schemas/desktop_gnome_interface.schemas.in libgnome-2.8.0/schemas/desktop_gnome_interface.schemas.in
--- libgnome-2.8.0.orig/schemas/desktop_gnome_interface.schemas.in.in	2004-08-18 09:21:38.000000000 +0200
+++ libgnome-2.8.0/schemas/desktop_gnome_interface.schemas.in.in	2004-12-10 22:32:16.627457123 +0100
@@ -53,7 +53,7 @@
       <applyto>/desktop/gnome/interface/toolbar_style</applyto>
       <owner>gnome</owner>
       <type>string</type>
-      <default>both-horiz</default>
+      <default>icons</default>
       <locale name="C">
         <short>Toolbar Style</short>
         <long>Toolbar Style. Valid values are "both", "both_horiz", "icon", and
