diff -Naur orig/libs/ui/KisResourceBundleManifest.cpp patched/libs/ui/KisResourceBundleManifest.cpp
--- orig/libs/ui/KisResourceBundleManifest.cpp	2017-05-26 08:47:55.134739579 +0200
+++ patched/libs/ui/KisResourceBundleManifest.cpp	2017-07-13 07:54:57.930371007 +0200
@@ -22,7 +22,6 @@
 #include <QDomElement>
 #include <QDomNode>
 #include <QDomNodeList>
-#include <QForeachContainer>
 
 #include <KoXmlNS.h>
 #include <KoXmlReader.h>
