--- a/data/org.gnome.baobab.desktop.in.in.orig	2013-03-07 20:00:40.000000000 +0100
+++ a/data/org.gnome.baobab.desktop.in.in	2013-03-07 20:00:59.000000000 +0100
@@ -9,7 +9,7 @@
 Type=Application
 StartupNotify=true
 MimeType=inode/directory;
-Categories=GTK;GNOME;System;Filesystem;
+Categories=GTK;GNOME;System;Filesystem;X-UHU-Menu/Tools/Utilities;
 NotShowIn=KDE;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=gnome-utils
