diff -Naur orig/libffmpegthumbnailer/jpegwriter.cpp patched/libffmpegthumbnailer/jpegwriter.cpp
--- orig/libffmpegthumbnailer/jpegwriter.cpp	2012-08-25 14:07:44.000000000 +0200
+++ patched/libffmpegthumbnailer/jpegwriter.cpp	2014-03-14 15:52:02.774406731 +0100
@@ -133,7 +133,7 @@
     bufWriter->m_DestMgr.next_output_byte = bufWriter->m_pDataBuffer;
     bufWriter->m_DestMgr.free_in_buffer = JPEG_WORK_BUFFER_SIZE;
 
-    return true;
+    return TRUE;
 }
 
 
