diff -Naur a/ffDiaporama.desktop b/ffDiaporama.desktop
--- a/ffDiaporama.desktop	2014-02-09 10:46:58.000000000 +0100
+++ b/ffDiaporama.desktop	2014-02-18 19:28:50.000000000 +0100
@@ -1,5 +1,5 @@
 [Desktop Entry]
-Version=1.0
+Version=1.1
 Name=Movie creator ffDiaporama
 Name[fr]=Créateur de film ffDiaporama
 Name[it]=Movie creator ffDiaporama
@@ -15,9 +15,10 @@
 Comment[it]=Creatore di pellicole cinematografiche da foto e filmati video
 Comment[es]=Creador de películas a partir de fotos y clips de vídeo
 Comment[el]=Δημιουργός ταινιών από φωτογραφίες και βίντεο κλιπ
+Comment[hu]=Mozgókép készítés
 Icon=ffdiaporama
 Terminal=false
 Type=Application
 Exec=ffDiaporama %F
 MimeType=application/ffDiaporama;
-Categories=GTK;GNOME;Qt;KDE;AudioVideo;
+Categories=GTK;GNOME;Qt;KDE;AudioVideo;X-UHU-Menu/Multimedia/VideoEdit;
