diff -Naur orig/res/flamerobin.desktop patched/res/flamerobin.desktop
--- orig/res/flamerobin.desktop	2008-07-21 12:00:42.000000000 +0200
+++ patched/res/flamerobin.desktop	2016-02-19 00:08:58.343295473 +0100
@@ -2,8 +2,9 @@
 Name=FlameRobin
 GenericName=Database administration tool
 Comment=Administration Tool for Firebird DBMS
+Comment[hu]=Adminisztrációs eszköz a Firebird DBMS-hez
 Exec=flamerobin
-Icon=/usr/share/pixmaps/flamerobin.png
+Icon=flamerobin
 Type=Application
 Terminal=false
 Categories=Development;Database;GTK
