diff -Naur orig/etc/useradd patched/etc/useradd
--- orig/etc/useradd	2014-03-01 18:50:05.000000000 +0100
+++ patched/etc/useradd	2016-05-14 02:26:18.016468032 +0200
@@ -1,5 +1,5 @@
 # useradd defaults file
-GROUP=1000
+GROUP=999
 HOME=/home
 INACTIVE=-1
 EXPIRE=
diff -Naur orig/src/useradd.c patched/src/useradd.c
--- orig/src/useradd.c	2014-05-09 12:20:28.000000000 +0200
+++ patched/src/useradd.c	2016-05-14 02:27:42.832888614 +0200
@@ -93,8 +93,8 @@
 /*
  * These defaults are used if there is no defaults file.
  */
-static gid_t def_group = 100;
-static const char *def_gname = "other";
+static gid_t def_group = 999;
+static const char *def_gname = "users";
 static const char *def_home = "/home";
 static const char *def_shell = "";
 static const char *def_template = SKEL_DIR;
