diff -Naur orig/src/eiciel.desktop.in patched/src/eiciel.desktop.in
--- orig/src/eiciel.desktop.in	2014-07-17 00:39:05.000000000 +0200
+++ patched/src/eiciel.desktop.in	2016-02-18 23:53:20.194643445 +0100
@@ -4,12 +4,13 @@
 GenericName=Access control list editor
 GenericName[ca]=Editor de llistes de control d'accés
 GenericName[es]=Editor de listas de control de acceso
+GenericName[hu]=Hozzáférési lista szerkesztő
 Comment=View or change ACL and User Extended Attributes on files and directories
 Comment[ca]=Visualitza o canvia la llista de control d'accés i els atributs d'usuari en fitxers i directoris
 Comment[es]=Visualiza o cambia la lista de control de acceso y los atributos de usuario en ficheros y directorios
-Exec=@DSK_bindir@/eiciel
+Comment[hu]=Kiterjesztett fájlattribútumok megjelenítése és módosítása
+Exec=eiciel
 Icon=eiciel
 Terminal=false
 Type=Application
 Categories=GNOME;System;FileTools;
-Version=@DSK_VERSION@
