diff -Naur orig/library/tulip-core/src/TlpTools.cpp mod/library/tulip-core/src/TlpTools.cpp
--- orig/library/tulip-core/src/TlpTools.cpp	2015-10-13 14:16:13.000000000 +0200
+++ mod/library/tulip-core/src/TlpTools.cpp	2016-01-26 14:21:58.971971056 +0100
@@ -56,7 +56,7 @@
 #ifdef _WIN32
 #define _TULIP_LIB_DIR "c:\\Tulip\\lib\\"
 #else
-#define _TULIP_LIB_DIR "/usr/local/lib/"
+#define _TULIP_LIB_DIR "/usr/lib/"
 #endif
 #endif
 
diff -Naur orig/tulip-config.in mod/tulip-config.in
--- orig/tulip-config.in	2015-10-13 14:16:45.000000000 +0200
+++ mod/tulip-config.in	2016-01-26 14:21:14.148081549 +0100
@@ -1,6 +1,6 @@
 #!/bin/bash
 
-thisdir=$(dirname "$0")
+thisdir=$( $(readlink -f dirname "$0"))
 if test "$thisdir" = "." ; then
   thisdir=$PWD
 fi
