diff -Naur orig/software/crash_handling/CMakeLists.txt mod/software/crash_handling/CMakeLists.txt
--- orig/software/crash_handling/CMakeLists.txt	2013-11-19 10:45:17.000000000 +0100
+++ mod/software/crash_handling/CMakeLists.txt	2013-11-20 19:45:51.000000000 +0100
@@ -8,7 +8,6 @@
 IF(LINUX OR MINGW)
 INCLUDE(CheckIncludeFile)
 # workaround for latest versions of GNU binutils
-SET(CMAKE_REQUIRED_DEFINITIONS "-DPACKAGE -DPACKAGE_VERSION")
 # workaround for mingw 4.8
 IF(MINGW)
 GET_FILENAME_COMPONENT(BFD_INCLUDE_PATH ${CMAKE_C_COMPILER} PATH)
@@ -39,7 +38,7 @@
 ADD_LIBRARY(crash_handling STATIC ${LIB_SRCS})
 
 IF(UNIX)
-	TARGET_LINK_LIBRARIES(crash_handling dl)
+	TARGET_LINK_LIBRARIES(crash_handling dl z)
 ENDIF(UNIX)
 
 IF(HAVE_BFD)
