diff -Naur orig/software/crash_handling/CMakeLists.txt mod/software/crash_handling/CMakeLists.txt
--- orig/software/crash_handling/CMakeLists.txt	2015-10-13 14:16:38.000000000 +0200
+++ mod/software/crash_handling/CMakeLists.txt	2016-01-26 14:25:58.339390743 +0100
@@ -8,7 +8,6 @@
 IF(LINUX OR MINGW)
 INCLUDE(CheckIncludeFile)
 # workaround for latest versions of GNU binutils
-SET(CMAKE_REQUIRED_DEFINITIONS "-DPACKAGE -DPACKAGE_VERSION")
 # workaround for mingw 4.8
 IF(MINGW)
 GET_FILENAME_COMPONENT(BFD_INCLUDE_PATH ${CMAKE_C_COMPILER} PATH)
