diff -Naur orig/pan.desktop mod/pan.desktop
--- orig/pan.desktop	2012-06-30 00:31:46.000000000 +0200
+++ mod/pan.desktop	2016-03-17 09:41:46.032344106 +0100
@@ -47,7 +47,7 @@
 Comment[fr]=Lire et poster des articles Usenet
 Comment[sl]=Berite in objavljajte prispevke Usenet
 Exec=pan
-Icon=pan.png
+Icon=pan
 Terminal=false
 Type=Application
 Categories=GTK;Application;Network;News
diff -Naur orig/pan.desktop.in mod/pan.desktop.in
--- orig/pan.desktop.in	2012-06-30 00:24:54.000000000 +0200
+++ mod/pan.desktop.in	2016-03-17 09:41:57.972292637 +0100
@@ -3,7 +3,7 @@
 _Name=Pan Newsreader
 _Comment=Read and Post Usenet Articles
 Exec=pan
-Icon=pan.png
+Icon=pan
 Terminal=false
 Type=Application
 Categories=GTK;Application;Network;News
