diff -Naur orig/src/textview.c patched/src/textview.c
--- orig/src/textview.c	2017-03-26 11:58:32.134741323 +0200
+++ patched/src/textview.c	2017-03-26 15:04:52.628569111 +0200
@@ -2277,7 +2277,8 @@
 				procheader_headername_equal(header->name,
 						 "X-Newsreader")) &&
 				(strstr(header->body, "Claws Mail") != NULL ||
-				strstr(header->body, "Sylpheed-Claws") != NULL)) {
+				strstr(header->body, "Sylpheed-Claws")) &&
+				strstr(header->body, "uhu-linux") != NULL) {
 			gtk_text_buffer_get_end_iter (buffer, &iter);
 			gtk_text_buffer_insert_with_tags_by_name
 				(buffer, &iter, header->body, -1,
