diff -Naur orig/src/prefs_common.c patched/src/prefs_common.c
--- orig/src/prefs_common.c	2017-03-26 15:06:18.094117103 +0200
+++ patched/src/prefs_common.c	2017-03-26 15:12:29.604105629 +0200
@@ -455,7 +455,7 @@
 	{"msgview_date_format", "FALSE", &prefs_common.msgview_date_format, P_BOOL,
 	 NULL, NULL, NULL},
 
-	{"next_on_delete", "FALSE", &prefs_common.next_on_delete, P_BOOL,
+	{"next_on_delete", "TRUE", &prefs_common.next_on_delete, P_BOOL,
 	 NULL, NULL, NULL},
 
 	 {"bold_unread", "TRUE", &prefs_common.bold_unread, P_BOOL,
