diff -Naurd a/utilities/texstudio.desktop b/utilities/texstudio.desktop
--- a/utilities/texstudio.desktop	2014-03-17 11:07:58.000000000 +0100
+++ b/utilities/texstudio.desktop	2014-05-12 20:00:16.000000000 +0200
@@ -6,6 +6,7 @@
 GenericName[fr]=Editeur LaTeX
 Comment=LaTeX development environment
 Comment[fr]=Environnement de développement LaTeX
+Comment[hu]=LaTeX szerkesztő
 Icon=/usr/share/texstudio/texstudio.svg
 MimeType=text/x-tex;
 Name=TeXstudio
