diff -Naur orig/wireshark.desktop mod/wireshark.desktop
--- orig/wireshark.desktop	2013-04-22 20:05:29.000000000 +0200
+++ mod/wireshark.desktop	2013-10-15 08:41:53.705432072 +0200
@@ -68,6 +68,7 @@
 Comment=Network traffic analyzer
 Comment[fi]=Verkkoliikenne analysaattori
 Comment[fr]=Analyseur de trafic réseau
+Comment[hu]=Profi hálózati forgalomfigyelő- és elemző
 Comment[sv]=Nätverkstrafikanalysator
 Icon=wireshark
 TryExec=wireshark
