diff -Naur xf86-video-chips-1.2.5-orig/src/ct_driver.c xf86-video-chips-1.2.5/src/ct_driver.c
--- xf86-video-chips-1.2.5-orig/src/ct_driver.c	2012-07-16 05:35:59.000000000 +0200
+++ xf86-video-chips-1.2.5/src/ct_driver.c	2013-03-31 11:14:26.000000000 +0200
@@ -96,9 +96,6 @@
 /* All drivers initialising the SW cursor need this */
 #include "mipointer.h"
 
-/* All drivers implementing backing store need this */
-#include "mibstore.h"
-
 /* All drivers using the mi banking wrapper need this */
 #ifdef HAVE_ISA
 #include "mibank.h"
