diff -Naur orig/Runtime/codelite.desktop.template mod/Runtime/codelite.desktop.template
--- orig/Runtime/codelite.desktop.template	2014-06-04 06:10:09.000000000 +0200
+++ mod/Runtime/codelite.desktop.template	2014-06-09 09:09:23.000000000 +0200
@@ -1,5 +1,7 @@
 [Desktop Entry]
 Name=CodeLite
+Comment=C/C++ IDE
+Comment[hu]=C/C++ fejlesztői környezet
 Exec=codelite %f
 Icon=%%PREFIX%%/share/codelite/images/cubes.png
 Terminal=false
