diff -Naur orig/linux/launchy.desktop mod/linux/launchy.desktop
--- orig/linux/launchy.desktop	2010-02-14 03:02:11.000000000 +0100
+++ mod/linux/launchy.desktop	2010-04-12 17:47:22.000000000 +0200
@@ -1,12 +1,11 @@
 [Desktop Entry]
-Version=1.0
+Version=1.1
 Name=Launchy
 Comment=Keystroke Launcher
+Comment[hu]=Program indító
 Exec=launchy %u
 Terminal=false
 X-MultipleArgs=false
 Type=Application
 Icon=launchy_icon
-Categories=Utility;
-
-
+Categories=Utility;X-UHU-Menu/Tools/Utilities;
