diff -ru launchy-2.5-orig/src/src.pro launchy-2.5/src/src.pro
--- launchy-2.5-orig/src/src.pro	2010-03-13 19:29:11.000000000 +0100
+++ launchy-2.5/src/src.pro	2010-11-21 22:30:13.150184890 +0100
@@ -70,6 +70,7 @@
                ../platforms/unix/platform_x11_hotkey.h \
                 platform_base_hotkey.h \
                 platform_base_hottrigger.h
+    LIBS = -lX11
 
     PREFIX = /usr
     DEFINES += SKINS_PATH=\\\"$$PREFIX/share/launchy/skins/\\\" \
