diff -Nauir orig/src/udev/udevd.c mod/src/udev/udevd.c
--- orig/src/udev/udevd.c	2015-06-17 23:29:40.000000000 +0200
+++ mod/src/udev/udevd.c	2016-06-26 09:36:38.910585602 +0200
@@ -1672,8 +1672,12 @@
                    we only do this on systemd systems, and only if we are directly spawned
                    by PID1. otherwise we are not guaranteed to have a dedicated cgroup */
                 r = cg_pid_get_path(SYSTEMD_CGROUP_CONTROLLER, 0, &cgroup);
-                if (r < 0)
-                        log_warning_errno(r, "failed to get cgroup: %m");
+                if (r < 0) {
+                        if (r == -ENOENT)
+                                log_debug_errno(r, "did not find dedicated cgroup: %m");
+                        else
+                                log_warning_errno(r, "failed to get cgroup: %m");
+                }
         }
 
         r = listen_fds(&fd_ctrl, &fd_uevent);
