diff -Naur orig/tmpfiles.d/legacy.conf patched/tmpfiles.d/legacy.conf
--- orig/tmpfiles.d/legacy.conf	2015-06-09 01:06:17.000000000 +0200
+++ patched/tmpfiles.d/legacy.conf	2016-01-16 17:17:38.526604945 +0100
@@ -24,7 +24,7 @@
 # On modern systems a BSD file lock is a better choice if
 # serialization is needed on those devices.
 
-d /run/lock/lockdev 0775 root lock -
+# d /run/lock/lockdev 0775 root lock -
 
 # /forcefsck, /fastboot and /forcequotacheck are deprecated in favor of the
 # kernel command line options 'fsck.mode=force', 'fsck.mode=skip' and
diff -Naur orig/tmpfiles.d/systemd.conf.m4 patched/tmpfiles.d/systemd.conf.m4
--- orig/tmpfiles.d/systemd.conf.m4	2015-01-22 15:14:36.000000000 +0100
+++ patched/tmpfiles.d/systemd.conf.m4	2016-01-16 17:19:01.039014102 +0100
@@ -8,7 +8,7 @@
 # See tmpfiles.d(5) for details
 
 d /run/user 0755 root root -
-F! /run/utmp 0664 root utmp -
+F! /run/utmp 0664 root root -
 
 d /run/systemd/ask-password 0755 root root -
 d /run/systemd/seats 0755 root root -
