diff -Naur orig/share/nt.desktop patched/share/nt.desktop
--- orig/share/nt.desktop	2005-08-31 15:32:13.000000000 +0200
+++ patched/share/nt.desktop	2016-02-13 18:51:19.151325400 +0100
@@ -13,9 +13,11 @@
 Comment[de]=Programm zum Downloaden von Dateien
 Comment[pl]=Program do pobierania plików z Internetu.
 Comment[ru]=программа для скачивания файлов
-Icon=nt.png
+Comment[hu]=Webszivattyú
+Icon=/usr/share/pixmaps/uhu/d4x.png
 Exec=d4x
 TryExec=d4x
 Terminal=false
 Type=Application
 Encoding=UTF-8
+Categories=Network;FileTransfer;
\ No newline at end of file
