--- a/main/face/list.cc
+++ a/main/face/list.cc
@@ -993,7 +993,7 @@
 		//FIXME: std::string and boost::lexical_cast!
 		char b[100];
 		d4x_percent_str(temp->Percent, b, sizeof(b));
-		sprintf(data, "%s%(%lli/%lli)",b,fsize_t(temp->Size),temp->finfo.size);
+		sprintf(data, "%s%% (%lli/%lli)",b,fsize_t(temp->Size),temp->finfo.size);
 		gtk_progress_bar_set_fraction(GTK_PROGRESS_BAR(ProgressOfDownload),(temp->Percent>100?100:temp->Percent)/100.0);
 		gtk_progress_bar_set_text(GTK_PROGRESS_BAR(ProgressOfDownload),data);
 	}else{
