diff -Naur orig/CMakeLists.txt patched/CMakeLists.txt
--- orig/CMakeLists.txt	2016-09-24 00:45:02.000000000 +0200
+++ patched/CMakeLists.txt	2016-09-25 20:16:13.051062122 +0200
@@ -23,6 +23,7 @@
 option(WITH_LXQT_SESSION "Build lxqt-session translations" ON)
 option(WITH_LXQT_SUDO "Build lxqt-sudo translations" ON)
 option(WITH_OBCONF_QT "Build obconf-qt translations" ON)
+option(WITH_PAVUCONTROL_QT "Build pavucontrol-qt translations" ON)
 option(WITH_PCMANFM_QT "Build pcmanfm-qt translations" ON)
 option(WITH_QTERMINAL "Build qterminal translations" ON)
 
@@ -86,5 +87,6 @@
 add_component(WITH_LXQT_SESSION lxqt-session)
 add_component(WITH_LXQT_SUDO lxqt-sudo)
 add_component(WITH_OBCONF_QT obconf-qt)
+add_component(WITH_PAVUCONTROL_QT pavucontrol-qt)
 add_component(WITH_PCMANFM_QT pcmanfm-qt)
 add_component(WITH_QTERMINAL qterminal)
diff -Naur orig/pavucontrol-qt/CMakeLists.txt patched/pavucontrol-qt/CMakeLists.txt
--- orig/pavucontrol-qt/CMakeLists.txt	1970-01-01 01:00:00.000000000 +0100
+++ patched/pavucontrol-qt/CMakeLists.txt	2016-09-25 20:16:46.023444730 +0200
@@ -0,0 +1,5 @@
+cmake_minimum_required(VERSION 3.0.2 FATAL_ERROR)
+
+project(pavucontrol-qt)
+
+build_component("." "${CMAKE_INSTALL_FULL_DATADIR}/pavucontrol-qt/translations")
