diff -Nur orig/qterminal/qterminal_hu.ts mod/qterminal/qterminal_hu.ts
--- orig/qterminal/qterminal_hu.ts	2016-12-11 01:27:50.000000000 +0100
+++ mod/qterminal/qterminal_hu.ts	2016-12-13 21:28:31.316691905 +0100
@@ -38,219 +38,219 @@
     <message>
         <location filename="../src/mainwindow.cpp" line="141"/>
         <source>&amp;Clear Current Tab</source>
-        <translation type="unfinished"></translation>
+        <translation>Lap &amp;törlése</translation>
     </message>
     <message>
         <location filename="../src/mainwindow.cpp" line="150"/>
         <source>&amp;Next Tab</source>
-        <translation type="unfinished"></translation>
+        <translation>&amp;Következő lap</translation>
     </message>
     <message>
         <location filename="../src/mainwindow.cpp" line="157"/>
         <source>&amp;Previous Tab</source>
-        <translation type="unfinished"></translation>
+        <translation>&amp;Előző lap</translation>
     </message>
     <message>
         <location filename="../src/mainwindow.cpp" line="164"/>
         <source>Move Tab &amp;Left</source>
-        <translation type="unfinished"></translation>
+        <translation>&amp;Lap balra</translation>
     </message>
     <message>
         <location filename="../src/mainwindow.cpp" line="171"/>
         <source>Move Tab &amp;Right</source>
-        <translation type="unfinished"></translation>
+        <translation>Lap jobb&amp;ra</translation>
     </message>
     <message>
         <location filename="../src/mainwindow.cpp" line="180"/>
         <source>Split Terminal &amp;Horizontally</source>
-        <translation type="unfinished"></translation>
+        <translation>Terminál &amp;vízszintes szétvágása</translation>
     </message>
     <message>
         <location filename="../src/mainwindow.cpp" line="187"/>
         <source>Split Terminal &amp;Vertically</source>
-        <translation type="unfinished"></translation>
+        <translation>Terminál &amp;függőleges szétvágása</translation>
     </message>
     <message>
         <location filename="../src/mainwindow.cpp" line="194"/>
         <source>&amp;Collapse Subterminal</source>
-        <translation type="unfinished"></translation>
+        <translation>Alter&amp;minál összevonása</translation>
     </message>
     <message>
         <location filename="../src/mainwindow.cpp" line="201"/>
         <source>N&amp;ext Subterminal</source>
-        <translation type="unfinished"></translation>
+        <translation>Következő al&amp;terminál</translation>
     </message>
     <message>
         <location filename="../src/mainwindow.cpp" line="208"/>
         <source>P&amp;revious Subterminal</source>
-        <translation type="unfinished"></translation>
+        <translation>Előző alte&amp;rminál</translation>
     </message>
     <message>
         <location filename="../src/mainwindow.cpp" line="218"/>
         <source>Copy &amp;Selection</source>
-        <translation type="unfinished"></translation>
+        <translation>Kijelölés má&amp;solása</translation>
     </message>
     <message>
         <location filename="../src/mainwindow.cpp" line="225"/>
         <source>Paste Clip&amp;board</source>
-        <translation type="unfinished"></translation>
+        <translation>Vágólap &amp;beillesztése</translation>
     </message>
     <message>
         <location filename="../src/mainwindow.cpp" line="232"/>
         <source>Paste S&amp;election</source>
-        <translation type="unfinished"></translation>
+        <translation>Kijelölés b&amp;eillesztése</translation>
     </message>
     <message>
         <location filename="../src/mainwindow.cpp" line="239"/>
         <source>Zoom &amp;in</source>
-        <translation type="unfinished"></translation>
+        <translation>Nagy&amp;ít</translation>
     </message>
     <message>
         <location filename="../src/mainwindow.cpp" line="246"/>
         <source>Zoom &amp;out</source>
-        <translation type="unfinished"></translation>
+        <translation>Ki&amp;csinyít</translation>
     </message>
     <message>
         <location filename="../src/mainwindow.cpp" line="253"/>
         <source>Zoom rese&amp;t</source>
-        <translation type="unfinished"></translation>
+        <translation>Nagyí&amp;tás vissza</translation>
     </message>
     <message>
         <location filename="../src/mainwindow.cpp" line="262"/>
         <source>&amp;Find...</source>
-        <translation type="unfinished"></translation>
+        <translation>&amp;Keres...</translation>
     </message>
     <message>
         <location filename="../src/mainwindow.cpp" line="274"/>
         <source>&amp;Save Session</source>
-        <translation type="unfinished"></translation>
+        <translation>Menet menté&amp;s</translation>
     </message>
     <message>
         <location filename="../src/mainwindow.cpp" line="280"/>
         <source>&amp;Load Session</source>
-        <translation type="unfinished"></translation>
+        <translation>Menet tö&amp;ltés</translation>
     </message>
     <message>
         <location filename="../src/mainwindow.cpp" line="287"/>
         <source>&amp;Toggle Menu</source>
-        <translation type="unfinished"></translation>
+        <translation>Me&amp;nü ki-be</translation>
     </message>
     <message>
         <location filename="../src/mainwindow.cpp" line="295"/>
         <source>Rename session</source>
-        <translation type="unfinished">Menet átnevezése</translation>
+        <translation>Menet átnevezése</translation>
     </message>
     <message>
         <location filename="../src/mainwindow.cpp" line="314"/>
         <source>&amp;New Tab</source>
-        <translation type="unfinished"></translation>
+        <translation>Új la&amp;p</translation>
     </message>
     <message>
         <location filename="../src/mainwindow.cpp" line="321"/>
         <source>New Tab From &amp;Preset</source>
-        <translation type="unfinished"></translation>
+        <translation>Új lap a kés&amp;zletből</translation>
     </message>
     <message>
         <location filename="../src/mainwindow.cpp" line="322"/>
         <source>1 &amp;Terminal</source>
-        <translation type="unfinished"></translation>
+        <translation>1 &amp;terminál</translation>
     </message>
     <message>
         <location filename="../src/mainwindow.cpp" line="324"/>
         <source>2 &amp;Horizontal Terminals</source>
-        <translation type="unfinished"></translation>
+        <translation>2 &amp;vízszintes terminál</translation>
     </message>
     <message>
         <location filename="../src/mainwindow.cpp" line="326"/>
         <source>2 &amp;Vertical Terminals</source>
-        <translation type="unfinished"></translation>
+        <translation>2 &amp;függőleges terminál</translation>
     </message>
     <message>
         <location filename="../src/mainwindow.cpp" line="328"/>
         <source>4 Terminal&amp;s</source>
-        <translation type="unfinished"></translation>
+        <translation>4 te&amp;rminál</translation>
     </message>
     <message>
         <location filename="../src/mainwindow.cpp" line="332"/>
         <source>&amp;Close Tab</source>
-        <translation type="unfinished"></translation>
+        <translation>Lap be&amp;csukás</translation>
     </message>
     <message>
         <location filename="../src/mainwindow.cpp" line="339"/>
         <source>&amp;New Window</source>
-        <translation type="unfinished"></translation>
+        <translation>Új abla&amp;k</translation>
     </message>
     <message>
         <location filename="../src/mainwindow.cpp" line="373"/>
         <source>&amp;Hide Window Borders</source>
-        <translation type="unfinished"></translation>
+        <translation>Ablakkeret &amp;kikapcsolása</translation>
     </message>
     <message>
         <location filename="../src/mainwindow.cpp" line="388"/>
         <source>&amp;Show Tab Bar</source>
-        <translation type="unfinished"></translation>
+        <translation>Lapfül&amp;sáv látható</translation>
     </message>
     <message>
         <location filename="../src/mainwindow.cpp" line="400"/>
         <source>Fullscreen</source>
-        <translation type="unfinished"></translation>
+        <translation>Telikép</translation>
     </message>
     <message>
         <location filename="../src/mainwindow.cpp" line="420"/>
         <source>&amp;Bottom</source>
-        <translation type="unfinished"></translation>
+        <translation>Al&amp;ul</translation>
     </message>
     <message>
         <location filename="../src/mainwindow.cpp" line="421"/>
         <source>&amp;Top</source>
-        <translation type="unfinished"></translation>
+        <translation>&amp;Felül</translation>
     </message>
     <message>
         <location filename="../src/mainwindow.cpp" line="422"/>
         <location filename="../src/mainwindow.cpp" line="453"/>
         <source>&amp;Right</source>
-        <translation type="unfinished"></translation>
+        <translation>&amp;Jobbra</translation>
     </message>
     <message>
         <location filename="../src/mainwindow.cpp" line="423"/>
         <location filename="../src/mainwindow.cpp" line="454"/>
         <source>&amp;Left</source>
-        <translation type="unfinished"></translation>
+        <translation>Ba&amp;lra</translation>
     </message>
     <message>
         <location filename="../src/mainwindow.cpp" line="438"/>
         <source>&amp;Tabs Layout</source>
-        <translation type="unfinished"></translation>
+        <translation>La&amp;pfülek helye</translation>
     </message>
     <message>
         <location filename="../src/mainwindow.cpp" line="452"/>
         <source>&amp;None</source>
-        <translation type="unfinished"></translation>
+        <translation>&amp;Nincs</translation>
     </message>
     <message>
         <location filename="../src/mainwindow.cpp" line="470"/>
         <source>S&amp;crollbar Layout</source>
-        <translation type="unfinished"></translation>
+        <translation>&amp;Gördítősáv helye</translation>
     </message>
     <message>
         <location filename="../src/mainwindow.cpp" line="481"/>
         <source>&amp;BlockCursor</source>
-        <translation type="unfinished"></translation>
+        <translation>&amp;Tömb alak</translation>
     </message>
     <message>
         <location filename="../src/mainwindow.cpp" line="482"/>
         <source>&amp;UnderlineCursor</source>
-        <translation type="unfinished"></translation>
+        <translation>Aláh&amp;úzásjel</translation>
     </message>
     <message>
         <location filename="../src/mainwindow.cpp" line="483"/>
         <source>&amp;IBeamCursor</source>
-        <translation type="unfinished"></translation>
+        <translation>&amp;I alak</translation>
     </message>
     <message>
         <location filename="../src/mainwindow.cpp" line="499"/>
         <source>&amp;Keyboard Cursor Shape</source>
-        <translation type="unfinished"></translation>
+        <translation>Billentyű&amp;kurzor</translation>
     </message>
     <message>
         <location filename="../src/mainwindow.cpp" line="576"/>
@@ -389,17 +389,17 @@
     <message>
         <location filename="../src/forms/propertiesdialog.ui" line="305"/>
         <source>Keyboard cursor shape</source>
-        <translation type="unfinished"></translation>
+        <translation>Billentyűkurzor</translation>
     </message>
     <message>
         <location filename="../src/forms/propertiesdialog.ui" line="312"/>
         <source>Change window title based on current terminal</source>
-        <translation type="unfinished"></translation>
+        <translation>Ablakfelirat az aktuális alapján</translation>
     </message>
     <message>
         <location filename="../src/forms/propertiesdialog.ui" line="319"/>
         <source>Change window icon based on current terminal</source>
-        <translation type="unfinished"></translation>
+        <translation>Ablakikon az aktuális alapján</translation>
     </message>
     <message>
         <location filename="../src/forms/propertiesdialog.ui" line="352"/>
@@ -419,12 +419,12 @@
     <message>
         <location filename="../src/forms/propertiesdialog.ui" line="414"/>
         <source>Save Size when closing</source>
-        <translation type="unfinished"></translation>
+        <translation>Bezáráskor méretmentés</translation>
     </message>
     <message>
         <location filename="../src/forms/propertiesdialog.ui" line="421"/>
         <source>Save Position when closing</source>
-        <translation type="unfinished"></translation>
+        <translation>Bezáráskor helyzetmentés</translation>
     </message>
     <message>
         <location filename="../src/forms/propertiesdialog.ui" line="428"/>
@@ -558,17 +558,17 @@
     <message>
         <location filename="../src/propertiesdialog.cpp" line="73"/>
         <source>BlockCursor</source>
-        <translation type="unfinished"></translation>
+        <translation>Tömb alak</translation>
     </message>
     <message>
         <location filename="../src/propertiesdialog.cpp" line="73"/>
         <source>UnderlineCursor</source>
-        <translation type="unfinished"></translation>
+        <translation>Aláhúzásjel</translation>
     </message>
     <message>
         <location filename="../src/propertiesdialog.cpp" line="73"/>
         <source>IBeamCursor</source>
-        <translation type="unfinished"></translation>
+        <translation>I alak</translation>
     </message>
     <message>
         <location filename="../src/propertiesdialog.cpp" line="85"/>
@@ -652,37 +652,37 @@
     <message>
         <location filename="../src/forms/qterminal.ui" line="71"/>
         <source>&amp;File</source>
-        <translation type="unfinished"></translation>
+        <translation>&amp;Fájl</translation>
     </message>
     <message>
         <location filename="../src/forms/qterminal.ui" line="79"/>
         <source>&amp;Actions</source>
-        <translation type="unfinished"></translation>
+        <translation>&amp;Tevékenység</translation>
     </message>
     <message>
         <location filename="../src/forms/qterminal.ui" line="84"/>
         <source>&amp;Help</source>
-        <translation type="unfinished"></translation>
+        <translation>&amp;Súgó</translation>
     </message>
     <message>
         <location filename="../src/forms/qterminal.ui" line="91"/>
         <source>&amp;View</source>
-        <translation type="unfinished"></translation>
+        <translation>&amp;Nézet</translation>
     </message>
     <message>
         <location filename="../src/forms/qterminal.ui" line="96"/>
         <source>&amp;Edit</source>
-        <translation type="unfinished"></translation>
+        <translation>Sz&amp;erkeszt</translation>
     </message>
     <message>
         <location filename="../src/forms/qterminal.ui" line="112"/>
         <source>&amp;About...</source>
-        <translation type="unfinished"></translation>
+        <translation>&amp;Rólunk</translation>
     </message>
     <message>
         <location filename="../src/forms/qterminal.ui" line="117"/>
         <source>About &amp;Qt...</source>
-        <translation type="unfinished"></translation>
+        <translation>A QT-ról...</translation>
     </message>
     <message>
         <location filename="../src/forms/qterminal.ui" line="122"/>
