--- a/data/Makefile.in.orig	2014-03-30 21:23:23.000000000 +0200
+++ a/data/Makefile.in	2014-03-30 21:23:36.000000000 +0200
@@ -1201,12 +1201,6 @@
 		chown root:gdm $(DESTDIR)$(workingdir) || : ; \
 	fi
 
-	if test '!' -d $(DESTDIR)$(workingdir)/.local/share/applications; then \
-		$(mkinstalldirs) $(DESTDIR)$(workingdir)/.local/share/applications; \
-		chmod 0755 $(DESTDIR)$(workingdir)/.local/share/applications; \
-		chown gdm:gdm $(DESTDIR)$(workingdir)/.local/share/applications || : ; \
-	fi
-
 	if test '!' -d $(DESTDIR)$(cachedir); then \
 		$(mkinstalldirs) $(DESTDIR)$(cachedir); \
 		chmod 1755 $(DESTDIR)$(cachedir); \
