diff -Naur orig/main/configure.ac patched/main/configure.ac
--- orig/main/configure.ac	2016-12-25 21:30:44.000000000 +0100
+++ patched/main/configure.ac	2016-12-28 05:02:53.950188664 +0100
@@ -3931,6 +3931,7 @@
 dnl ===================================================================
 dnl Check for system lucene
 dnl ===================================================================
+if test "$SOLAR_JAVA" != ""; then
 AC_MSG_CHECKING([which lucene to use])
 if test -n "$with_system_lucene" -o -n "$with_system_libs" && \
        test "$with_system_lucene" != "no" && test "$with_system_jars" != "no"; then
@@ -4008,6 +4009,7 @@
 AC_SUBST(SYSTEM_LUCENE)
 AC_SUBST(LUCENE_CORE_JAR)
 AC_SUBST(LUCENE_ANALYZERS_JAR)
+fi
 
 AC_MSG_CHECKING([whether to build the MySQL Connector extension])
 if test -n "$enable_mysql_connector" -a "$enable_mysql_connector" != "no"; then
@@ -4123,6 +4125,7 @@
 dnl ===================================================================
 dnl Check for system hsqldb
 dnl ===================================================================
+if test "$SOLAR_JAVA" != ""; then
 AC_MSG_CHECKING([which hsqldb to use])
 if test -n "$with_system_hsqldb" -o -n "$with_system_libs" && \
        test "$with_system_hsqldb" != "no" && test "$with_system_jars" != "no"; then
@@ -4167,6 +4170,7 @@
 fi
 AC_SUBST(SYSTEM_HSQLDB)
 AC_SUBST(HSQLDB_JAR)
+fi
 
 dnl ===================================================================
 dnl Check for system beanshell
@@ -6448,8 +6452,8 @@
 KDE4_LIBS=""
 MOC4="moc"
 if test "$test_kde4" = "yes" -a "$ENABLE_KDE4" = "TRUE" ; then
-   qt_incdirs="$QT4INC $QT4DIR /usr/include/qt4 /usr/include $x_includes"
-   qt_libdirs="$QT4LIB /usr/lib $x_libraries"
+   qt_incdirs="$QT4INC $QT4DIR/include /usr/include/qt4 /usr/include $x_includes"
+   qt_libdirs="$QT4LIB $QT4DIR/lib /usr/lib $x_libraries"
    
    kde_incdirs="/usr/include /usr/include/kde4 $x_includes"
    kde_libdirs="/usr/lib /usr/lib/kde4 /usr/lib/kde4/devel $x_libraries"
@@ -6506,7 +6510,7 @@
    AC_PATH_PROG( MOCQT4, moc-qt4, no, [$QT4DIR/bin:$PATH] )
    MOC4="$MOCQT4"
    if test "$MOC4" = "no" ; then
-      AC_PATH_PROG( MOC4, moc, no, [$PATH:$QT4DIR/bin] )
+      AC_PATH_PROG( MOC4, moc, no, [$QT4DIR/bin:$PATH] )
       if test "$MOC4" = "no" ; then
          AC_MSG_ERROR([Qt Meta Object Compiler not found.  Please specify the root of your Qt installation by exporting QT4DIR before running "configure".])
       fi
@@ -6998,7 +7002,6 @@
    AC_MSG_RESULT([Ant lib directory found.])
 fi
 AC_SUBST(ANT_LIB)
-fi
 
 ant_minver=1.6.0
 # update for more extensions...
@@ -7068,6 +7071,7 @@
   fi
 fi
 rm -f conftest* core core.* *.core
+fi
 
 OOO_JUNIT_JAR=
 if test "$SOLAR_JAVA" != "" && test "$with_junit" != "no"; then
