diff -Naur orig/main/set_soenv.in patched/main/set_soenv.in
--- orig/main/set_soenv.in	2014-02-25 09:24:52.000000000 +0100
+++ patched/main/set_soenv.in	2014-07-15 14:08:05.135908913 +0200
@@ -2005,9 +2005,6 @@
 #
 # Writing the aliases to file.
 ToFile( "Aliases.",          $empty,             "c" );
-ToFile( "Don't set aliases when bootstrapping", $empty, "c" );
-ToFile( 'if ( ! ${?_ISBOOTSTRAP_} ) then', $empty, "y");
-ToFile( "if test -z \"\$_ISBOOTSTRAP_\"; then", $empty, "z");
 
 ToFile( "alias mkout",       $mkout,             "a" );
 ToFile( "alias deliver",     $deliver,           "a" );
@@ -2022,7 +2019,6 @@
 }
 
 ToFile( "endif",             $empty, "y");
-ToFile( "fi",                $empty, "z");
 
 # Check if the right dmake version is used. (W32 only)
 if ( $platform =~ m/cygwin/ ) {
@@ -2144,8 +2140,7 @@
 open( OUT, ">$bootfile" ) ||
 die "Cannot open $bootfile: $!\n";
 print OUT "#!/bin/sh\n";
-print OUT "_ISBOOTSTRAP_=TRUE ; export _ISBOOTSTRAP_ ; . $SRC_ROOT/$outfile.sh\n"; 
-print OUT "unset _ISBOOTSTRAP_; export _ISBOOTSTRAP_\n"; 
+print OUT " . $SRC_ROOT/$outfile.sh\n";
 if ( $MINGW eq "yes" )
 {  print OUT 'PATH="/usr/bin:$PATH" : export PATH'."\n"; }
 close( OUT ) || print "Can't close $bootfile: $!";
