diff -Naur libindicate-qt-0.2.5.91-orig/CMakeLists.txt libindicate-qt-0.2.5.91/CMakeLists.txt
--- libindicate-qt-0.2.5.91-orig/CMakeLists.txt	2011-02-03 17:35:18.000000000 +0100
+++ libindicate-qt-0.2.5.91/CMakeLists.txt	2012-04-07 10:07:16.000000000 +0200
@@ -17,7 +17,7 @@
 
 add_definitions(-Wall)
 
-pkg_check_modules(INDICATE REQUIRED indicate-0.5)
+pkg_check_modules(INDICATE REQUIRED indicate-0.6)
 
 include_directories(
     ${QT_INCLUDE_DIR}
