diff -Naur mesa-demos-8.1.0-orig/src/egl/Makefile.in mesa-demos-8.1.0/src/egl/Makefile.in
--- mesa-demos-8.1.0-orig/src/egl/Makefile.in	2013-02-24 12:12:04.000000000 +0100
+++ mesa-demos-8.1.0/src/egl/Makefile.in	2013-03-05 19:04:37.000000000 +0100
@@ -304,7 +304,6 @@
 SUBDIRS = \
 	eglut \
 	opengl \
-	openvg \
 	opengles1 \
 	opengles2 \
 	oes_vg
diff -Naur mesa-demos-8.1.0-orig/src/egl/openvg/Makefile.in mesa-demos-8.1.0/src/egl/openvg/Makefile.in
--- mesa-demos-8.1.0-orig/src/egl/openvg/Makefile.in	2013-02-24 12:12:04.000000000 +0100
+++ mesa-demos-8.1.0/src/egl/openvg/Makefile.in	2013-03-05 19:05:15.000000000 +0100
@@ -386,7 +386,6 @@
 text_CFLAGS = $(AM_CFLAGS) @FREETYPE2_CFLAGS@
 text_LDADD = @FREETYPE2_LIBS@ ../eglut/libeglut_x11.la
 SUBDIRS = \
-	trivial
 
 all: all-recursive
 
diff -Naur mesa-demos-8.1.0-orig/src/Makefile.in mesa-demos-8.1.0/src/Makefile.in
--- mesa-demos-8.1.0-orig/src/Makefile.in	2013-02-24 12:12:03.000000000 +0100
+++ mesa-demos-8.1.0/src/Makefile.in	2013-03-05 19:05:56.000000000 +0100
@@ -317,7 +317,6 @@
 	redbook \
 	samples \
 	slang \
-	tests \
 	tools \
 	trivial \
 	vp \
diff -Naur mesa-demos-8.1.0-orig/src/tests/Makefile.in mesa-demos-8.1.0/src/tests/Makefile.in
--- mesa-demos-8.1.0-orig/src/tests/Makefile.in	2013-02-24 12:12:05.000000000 +0100
+++ mesa-demos-8.1.0/src/tests/Makefile.in	2013-03-05 19:07:30.000000000 +0100
@@ -551,7 +551,7 @@
 	mipgen.c mipmap_comp.c mipmap_comp_tests.c mipmap_limits.c \
 	mipmap_tunnel.c mipmap_view.c multipal.c multitexarray.c \
 	multiwindow.c no_s3tc.c packedpixels.c pbo.c persp_hint.c \
-	prim.c prog_parameter.c quads.c random.c readrate.c \
+	prog_parameter.c quads.c random.c readrate.c \
 	rubberband.c scissor.c scissor-viewport.c seccolor.c \
 	shader-interp.c shader_api.c shadow-sample.c sharedtex.c \
 	stencilreaddraw.c stencilwrap.c step.c streaming_rect.c \
@@ -576,7 +576,7 @@
 	manytex.c mapbufrange.c minmag.c mipgen.c mipmap_comp.c \
 	mipmap_comp_tests.c mipmap_limits.c mipmap_tunnel.c \
 	mipmap_view.c multipal.c multitexarray.c multiwindow.c \
-	no_s3tc.c packedpixels.c pbo.c persp_hint.c prim.c \
+	no_s3tc.c packedpixels.c pbo.c persp_hint.c \
 	prog_parameter.c quads.c random.c readrate.c rubberband.c \
 	scissor.c scissor-viewport.c seccolor.c shader-interp.c \
 	shader_api.c shadow-sample.c sharedtex.c stencilreaddraw.c \
