diff -Naur mesa-demos-8.1.0-orig/src/egl/opengles1/drawtex.c mesa-demos-8.1.0/src/egl/opengles1/drawtex.c
--- mesa-demos-8.1.0-orig/src/egl/opengles1/drawtex.c	2012-12-15 07:17:06.000000000 +0000
+++ mesa-demos-8.1.0/src/egl/opengles1/drawtex.c	2014-06-29 06:30:19.426585741 +0000
@@ -16,6 +16,7 @@
 #include <string.h>
 #include <GLES/gl.h>
 #include <GLES/glext.h>
+#include <EGL/egl.h>
 
 #include "eglut.h"
 
@@ -25,13 +26,14 @@
 static GLboolean animate = GL_FALSE;
 static int win;
 
+static PFNGLDRAWTEXFOESPROC glDrawTexfOES_func = NULL;
 
 static void
 draw(void)
 {
    glClear(GL_COLOR_BUFFER_BIT);
 
-   glDrawTexfOES(view_posx, view_posy, 0.0, width, height);
+   glDrawTexfOES_func(view_posx, view_posy, 0.0, width, height);
 }
 
 
@@ -128,6 +130,14 @@
       exit(1);
    }
 
+ glDrawTexfOES_func = (PFNGLDRAWTEXFOESPROC) eglGetProcAddress("glDrawTexfOES");
+
+ if (!glDrawTexfOES_func) {
+ fprintf(stderr, "Sorry, failed to resolve glDrawTexfOES function\n");
+ exit(1);
+ }
+
+
    glClearColor(0.4, 0.4, 0.4, 0.0);
 
    make_smile_texture();
