diff -Naur orig/src/get patched/src/get
--- orig/src/get	2014-07-03 12:33:14.000000000 +0200
+++ patched/src/get	2015-05-31 00:44:16.078774653 +0200
@@ -205,6 +205,19 @@
 			
 			cd ..
 			tar -zcv --force-local -f "$filename" "$tmpsub" || return
+		elif [[ "${revarray[0]}" = *[[:alnum:]]* ]]; then  ## commit
+			tmpsub="$(basename "$url")"
+			tmpsub=${tmpsub/.git/}
+			tmpsub=${tmpsub}"-"${revarray[0]}
+			mkdir -p "$tmpdir/$tmpsub"
+			cd "$tmpdir/$tmpsub"
+
+			git clone "git://$url" "$tmpsub"
+			cd "$tmpsub"
+			git checkout "${revarray[0]}"
+			
+			cd ..
+			tar -zcv --force-local -f "$filename" "$tmpsub" || return
 		else  ## YYYYMMDD
 			tmpsub="$(basename "$url")"
 			tmpsub=${tmpsub/.git/}
