diff -Naur orig/sites/fedora patched/sites/fedora
--- orig/sites/fedora	2006-04-06 12:52:33.000000000 +0200
+++ patched/sites/fedora	2015-06-13 14:35:59.879895082 +0200
@@ -1,7 +1,7 @@
-http://download.fedora.redhat.com/pub/fedora/linux/core/5/source/SRPMS/
-ftp://download.fedora.redhat.com/pub/fedora/linux/core/5/source/SRPMS/
-ftp://gd.tuwien.ac.at/opsys/linux/fedora/core/5/source/SRPMS/
-ftp://ftp.funet.fi/pub/mirrors/ftp.redhat.com/pub/fedora/linux/core/5/source/SRPMS/
-ftp://ftp.tu-chemnitz.de/pub/linux/fedora-core/5/source/SRPMS/
-ftp://mirrors.kernel.org/fedora/core/5/source/SRPMS/
-rsync://ftp.uninett.no/fedora-linux-core/5/source/SRPMS/
+http://mirror.datacenter.by/pub/fedoraproject.org/linux//releases/22/Workstation/source/SRPMS/
+ftp://mirror.datacenter.by/pub/fedoraproject.org/linux//releases/22/Workstation/source/SRPMS/
+ftp://gd.tuwien.ac.at/opsys/linux/fedora/linux/releases/22/Workstation/source/SRPMS/
+ftp://ftp.funet.fi/pub/mirrors/ftp.redhat.com/pub/fedora/linux/releases/22/Workstation/source/SRPMS/
+ftp://ftp.tu-chemnitz.de/pub/linux/fedora/linux/releases/22/Workstation/source/SRPMS/
+ftp://mirrors.kernel.org/fedora/releases/22/Workstation/source/SRPMS/
+ftp://ftp.uninett.no/fedora/releases/22/Workstation/source/SRPMS/
diff -Naur orig/sites/gnome patched/sites/gnome
--- orig/sites/gnome	2004-07-29 13:56:32.000000000 +0200
+++ patched/sites/gnome	2015-06-13 14:23:13.524094929 +0200
@@ -2,20 +2,8 @@
 ftp://ftp.gnome.org/pub/gnome/
 http://ftp.gnome.org/pub/GNOME/
 http://ftp.gnome.org/pub/gnome/
-http://www.download-ftp-center.com/mirrors/gnome.org/
-http://ftp.unina.it/pub/linux/GNOME/
 http://fr.rpmfind.net/linux/gnome.org/
 http://fr2.rpmfind.net/linux/gnome.org/
 http://ftp.acc.umu.se/pub/GNOME/
-http://ftp.belnet.be/mirror/ftp.gnome.org/
-ftp://ftp.dataplus.se/pub/GNOME/
-ftp://ftp.dit.upm.es/pub/GNOME/
-http://ftp.linux.org.uk/mirrors/ftp.gnome.org/
-ftp://ftp.no.gnome.org/pub/GNOME/
-http://www.mirror.ac.uk/sites/ftp.gnome.org/pub/GNOME/
+http://ftp.belnet.be/ftp.gnome.org/
 http://ftp.rpmfind.net/linux/gnome.org/
-ftp://ftp.ussg.iu.edu/pub/gnome/
-http://archive.progeny.com/GNOME/
-ftp://mirror.pacific.net.au/linux/GNOME/
-ftp://ftp.isu.edu.tw/pub/Unix/Desktop/GNOME/
-http://www.zentek-international.com/mirrors/gnome/
diff -Naur orig/sites/gnu patched/sites/gnu
--- orig/sites/gnu	2003-08-08 11:00:33.000000000 +0200
+++ patched/sites/gnu	2015-06-13 14:06:16.811053321 +0200
@@ -2,8 +2,6 @@
 ftp://ftp.gnu.org/pub/gnu/
 ftp://mirrors.kernel.org/gnu/
 ftp://gd.tuwien.ac.at/gnu/gnusrc/
-ftp://ftp.fi.muni.cz/pub/gnu/
+ftp://ftp.fi.muni.cz/pub/gnu/gnu/
 ftp://ftp.funet.fi/pub/gnu/prep/
 ftp://ftp.cs.tu-berlin.de/pub/gnu/
-ftp://sunsite.cnlab-switch.ch/mirror/gnu/
-ftp://ftp.mirror.ac.uk/sites/ftp.gnu.org/gnu/
diff -Naur orig/sites/kde patched/sites/kde
--- orig/sites/kde	2003-08-04 10:19:36.000000000 +0200
+++ patched/sites/kde	2015-06-13 14:21:38.507623768 +0200
@@ -1,19 +1,6 @@
-ftp://ftp.kde.org/pub/kde/
-http://kde.oregonstate.edu/pub/kde/
-ftp://ftp.oregonstate.edu/pub/kde/
-ftp://kde.us.themoes.org/pub/kde/
-ftp://ftp.gtlib.cc.gatech.edu/pub/kde/
-ftp://ftp.rutgers.edu/pub/kde/
-ftp://ibiblio.org/pub/packages/desktops/kde/
-ftp://mirrors.midco.net/pub/kde/
-ftp://download.at.kde.org/pub/kde/
-ftp://ftp.solnet.ch/mirror/KDE/
-ftp://ftp.de.kde.org/pub/kde/
-ftp://ftp.gwdg.de/pub/x11/kde/
+http://download.kde.org/
+ftp://ftp.gwdg.de/pub/linux/kde/
 ftp://ftp.uni-kl.de/pub/linux/kde/
 ftp://ftp-stud.fht-esslingen.de/pub/Mirrors/ftp.kde.org/pub/kde/
-ftp://ftp.du.se/pub/mirrors/kde/
-ftp://ftp.eu.uu.net/pub/kde/
-ftp://ftp.tiscali.nl/pub/mirrors/kde/
-ftp://kde.uk.themoes.org/pub/kde/
-http://download.uk.kde.org/pub/kde/
+ftp://ftp.ussg.iu.edu/pub/kde/
+ftp://ftp.isu.edu.tw/pub/Unix/Desktop/KDE/
diff -Naur orig/sites/uhu patched/sites/uhu
--- orig/sites/uhu	2004-07-29 13:56:32.000000000 +0200
+++ patched/sites/uhu	2015-06-13 13:41:42.319741714 +0200
@@ -1,5 +1,4 @@
 rsync://rsync.uhulinux.hu/ftp/
 ftp://ftp.uhulinux.hu/
 ftp://ftp1.uhulinux.hu/
-ftp://ftp2.uhulinux.hu/
-ftp://ftp3.uhulinux.hu/
+http://download.uhulinux.hu/
