diff -Naur orig/CMakeLists.txt mod/CMakeLists.txt
--- orig/CMakeLists.txt	2016-05-02 10:29:11.000000000 +0200
+++ mod/CMakeLists.txt	2016-05-02 11:25:56.651630901 +0200
@@ -40,7 +40,7 @@
 set(pixmapdir ${datadir}/pixmaps)
 set(pkgdatadir ${datadir}/${name})
 set(localedir ${datadir}/locale)
-set(appsdir $ENV{HOME}/.local/share/applications)
+set(appsdir ${datadir}/applications)
 set(mandir ${datadir}/man)
 
 set(CMAKE_CXX_FLAGS "\
diff -Naur orig/data/me-tv.desktop mod/data/me-tv.desktop
--- orig/data/me-tv.desktop	2016-05-02 10:29:11.000000000 +0200
+++ mod/data/me-tv.desktop	2016-05-02 11:02:00.000000000 +0200
@@ -1,10 +1,10 @@
 [Desktop Entry]
 Type=Application
-Encoding=UTF-8
 Name=Me TV
 Comment=A digital television (DVB) viewer for GTK+3
-Exec=/home/russel/Built/bin/me-tv %U
-Icon=/home/russel/Built/share/pixmaps/me-tv.png
+Comment[hu]=Digitális televízó néző
+Exec=me-tv %U
+Icon=me-tv
 Terminal=false
 StartupNotify=true
 Categories=AudioVideo;GTK;TV;Player;
