diff -Naur xf86-video-sisusb-0.9.6-orig/src/sisusb_driver.c xf86-video-sisusb-0.9.6/src/sisusb_driver.c
--- xf86-video-sisusb-0.9.6-orig/src/sisusb_driver.c	2012-07-17 07:21:16.000000000 +0200
+++ xf86-video-sisusb-0.9.6/src/sisusb_driver.c	2013-03-31 18:04:06.000000000 +0200
@@ -39,7 +39,6 @@
 #include "shadowfb.h"
 #include "micmap.h"
 #include "mipointer.h"
-#include "mibstore.h"
 
 #include "sisusb_regs.h"
 #include "sisusb_dac.h"
