diff -Naur orig/src/frontends/qt4/GuiViewSource.cpp patched/src/frontends/qt4/GuiViewSource.cpp
--- orig/src/frontends/qt4/GuiViewSource.cpp	2016-10-10 17:42:40.000000000 +0200
+++ patched/src/frontends/qt4/GuiViewSource.cpp	2017-02-05 12:06:03.107463633 +0100
@@ -222,7 +222,7 @@
 		const QChar * oc = old.constData();
 		const QChar * nc = qcontent.constData();
 		int pos = 0;
-		while (*oc != '\0' && *nc != '\0' && *oc == *nc) {
+		while (*oc != QChar(0) && *nc != QChar(0) && *oc == *nc) {
 			++oc;
 			++nc;
 			++pos;
