diff -Naur orig/lib/irrlicht/source/Irrlicht/CImageLoaderJPG.cpp patched/lib/irrlicht/source/Irrlicht/CImageLoaderJPG.cpp
--- orig/lib/irrlicht/source/Irrlicht/CImageLoaderJPG.cpp	2016-05-12 02:31:37.000000000 +0200
+++ patched/lib/irrlicht/source/Irrlicht/CImageLoaderJPG.cpp	2016-07-11 01:22:25.588774363 +0200
@@ -66,7 +66,7 @@
 boolean CImageLoaderJPG::fill_input_buffer (j_decompress_ptr cinfo)
 {
 	// DO NOTHING
-	return 1;
+	return TRUE;
 }
 
 
