diff -Naurd a/curses.h b/curses.h
--- a/curses.h	2008-07-15 19:13:25.000000000 +0200
+++ b/curses.h	2012-05-19 07:19:43.000000000 +0200
@@ -1,6 +1,6 @@
 /* Public Domain Curses */
 
-/* $Id: curses.h,v 1.295 2008/07/15 17:13:25 wmcbrine Exp $ */
+/* $Id: curses.h,v 1.298 2012/05/19 05:19:43 wmcbrine Exp $ */
 
 /*----------------------------------------------------------------------*
  *                              PDCurses                                *
@@ -142,12 +142,16 @@
  *                             10000 <- mouse position report
  *                            100000 <- mouse wheel up
  *                           1000000 <- mouse wheel down
+ *                          10000000 <- mouse wheel left
+ *                         100000000 <- mouse wheel right
  */
 
 #define PDC_MOUSE_MOVED         0x0008
 #define PDC_MOUSE_POSITION      0x0010
 #define PDC_MOUSE_WHEEL_UP      0x0020
 #define PDC_MOUSE_WHEEL_DOWN    0x0040
+#define PDC_MOUSE_WHEEL_LEFT    0x0080
+#define PDC_MOUSE_WHEEL_RIGHT   0x0100
 
 #define A_BUTTON_CHANGED        (Mouse_status.changes & 7)
 #define MOUSE_MOVED             (Mouse_status.changes & PDC_MOUSE_MOVED)
@@ -156,6 +160,8 @@
 #define BUTTON_STATUS(x)        (Mouse_status.button[(x) - 1])
 #define MOUSE_WHEEL_UP          (Mouse_status.changes & PDC_MOUSE_WHEEL_UP)
 #define MOUSE_WHEEL_DOWN        (Mouse_status.changes & PDC_MOUSE_WHEEL_DOWN)
+#define MOUSE_WHEEL_LEFT        (Mouse_status.changes & PDC_MOUSE_WHEEL_LEFT)
+#define MOUSE_WHEEL_RIGHT       (Mouse_status.changes & PDC_MOUSE_WHEEL_RIGHT)
 
 /* mouse bit-masks */
 
@@ -409,7 +415,6 @@
 # define A_ITALIC     A_INVIS
 # define A_PROTECT    (A_UNDERLINE | A_LEFTLINE | A_RIGHTLINE)
 
-# define PDC_ATTR_SHIFT  19
 # define PDC_COLOR_SHIFT 24
 #else
 # define A_BOLD       (chtype)0x0100  /* X/Open */
@@ -429,7 +434,6 @@
 # define A_ITALIC     A_NORMAL
 # define A_INVIS      A_NORMAL
 
-# define PDC_ATTR_SHIFT   8
 # define PDC_COLOR_SHIFT 11
 #endif
 
@@ -443,6 +447,8 @@
 /* For use with attr_t -- X/Open says, "these shall be distinct", so 
    this is a non-conforming implementation. */
 
+#define WA_NORMAL     A_NORMAL
+
 #define WA_ALTCHARSET A_ALTCHARSET
 #define WA_BLINK      A_BLINK
 #define WA_BOLD       A_BOLD
@@ -460,6 +466,8 @@
 #define WA_TOP        A_NORMAL
 #define WA_VERTICAL   A_NORMAL
 
+#define WA_ATTRIBUTES A_ATTRIBUTES
+
 /*** Alternate character set macros ***/
 
 /* 'w' = 32-bit chtype; acs_map[] index | A_ALTCHARSET
diff -Naurd a/pdcurses/window.c b/pdcurses/window.c
--- a/pdcurses/window.c	2008-07-13 18:08:18.000000000 +0200
+++ b/pdcurses/window.c	2012-05-11 18:48:28.000000000 +0200
@@ -2,7 +2,7 @@
 
 #include <curspriv.h>
 
-RCSID("$Id: window.c,v 1.62 2008/07/13 16:08:18 wmcbrine Exp $")
+RCSID("$Id: window.c,v 1.63 2012/05/11 16:48:28 wmcbrine Exp $")
 
 /*man-start**************************************************************
 
@@ -323,6 +323,7 @@
     win->_leaveit = orig->_leaveit;
     win->_scroll = orig->_scroll;
     win->_nodelay = orig->_nodelay;
+    win->_delayms = orig->_delayms;
     win->_use_keypad = orig->_use_keypad;
     win->_immed = orig->_immed;
     win->_sync = orig->_sync;
@@ -410,6 +411,7 @@
     new->_leaveit = win->_leaveit;
     new->_scroll = win->_scroll;
     new->_nodelay = win->_nodelay;
+    new->_delayms = win->_delayms;
     new->_use_keypad = win->_use_keypad;
     new->_tmarg = win->_tmarg;
     new->_bmarg = win->_bmarg;
@@ -486,6 +488,7 @@
     new->_leaveit = win->_leaveit;
     new->_scroll = win->_scroll;
     new->_nodelay = win->_nodelay;
+    new->_delayms = win->_delayms;
     new->_use_keypad = win->_use_keypad;
     new->_tmarg = (win->_tmarg > new->_maxy - 1) ? 0 : win->_tmarg;
     new->_bmarg = (win->_bmarg == win->_maxy - 1) ?
diff -Naurd a/sdl1/pdckbd.c b/sdl1/pdckbd.c
--- a/sdl1/pdckbd.c	2008-07-14 06:24:52.000000000 +0200
+++ b/sdl1/pdckbd.c	2011-04-27 18:38:11.000000000 +0200
@@ -2,7 +2,7 @@
 
 #include "pdcsdl.h"
 
-RCSID("$Id: pdckbd.c,v 1.20 2008/07/14 04:24:52 wmcbrine Exp $")
+RCSID("$Id: pdckbd.c,v 1.21 2011/04/23 06:50:02 wmcbrine Exp $")
 
 /*man-start**************************************************************
 
@@ -282,12 +282,24 @@
 
         /* handle scroll wheel */
 
-        if ((btn == 4 || btn == 5) && action == BUTTON_RELEASED)
+        if ((btn >= 4 && btn <= 7) && action == BUTTON_RELEASED)
         {
             pdc_mouse_status.x = pdc_mouse_status.y = -1;
 
-            pdc_mouse_status.changes = (btn == 5) ?
-                PDC_MOUSE_WHEEL_DOWN : PDC_MOUSE_WHEEL_UP;
+            switch (btn)
+            {
+            case 4:
+                pdc_mouse_status.changes = PDC_MOUSE_WHEEL_UP;
+                break;
+            case 5:
+                pdc_mouse_status.changes = PDC_MOUSE_WHEEL_DOWN;
+                break;
+            case 6:
+                pdc_mouse_status.changes = PDC_MOUSE_WHEEL_LEFT;
+                break;
+            case 7:
+                pdc_mouse_status.changes = PDC_MOUSE_WHEEL_RIGHT;
+            }
 
             return KEY_MOUSE;
         }
diff -Naurd a/x11/x11.c b/x11/x11.c
--- a/x11/x11.c	2008-07-14 06:33:26.000000000 +0200
+++ b/x11/x11.c	2012-11-26 09:22:50.000000000 +0100
@@ -2,7 +2,7 @@
 
 #include "pdcx11.h"
 
-RCSID("$Id: x11.c,v 1.94 2008/07/14 04:33:26 wmcbrine Exp $")
+RCSID("$Id: x11.c,v 1.96 2012/11/26 08:22:50 wmcbrine Exp $")
 
 #ifdef HAVE_DECKEYSYM_H
 # include <DECkeysym.h>
@@ -1964,16 +1964,29 @@
     {
     case ButtonPress:
         /* Handle button 4 and 5, which are normally mapped to the wheel
-           mouse scroll up and down */
+           mouse scroll up and down, and button 6 and 7, which are
+           normally mapped to the wheel mouse scroll left and right */
 
-        if (button_no == 4 || button_no == 5)
+        if (button_no >= 4 && button_no <= 7)
         {
             /* Send the KEY_MOUSE to curses program */
 
             memset(&Mouse_status, 0, sizeof(Mouse_status));
 
-            Mouse_status.changes = (button_no == 5) ? 
-                PDC_MOUSE_WHEEL_DOWN : PDC_MOUSE_WHEEL_UP;
+            switch(button_no)
+            {
+               case 4:
+                  Mouse_status.changes = PDC_MOUSE_WHEEL_UP;
+                  break;
+               case 5:
+                  Mouse_status.changes = PDC_MOUSE_WHEEL_DOWN;
+                  break;
+               case 6:
+                  Mouse_status.changes = PDC_MOUSE_WHEEL_LEFT;
+                  break;
+               case 7:
+                  Mouse_status.changes = PDC_MOUSE_WHEEL_RIGHT;
+            }
 
             MOUSE_X_POS = MOUSE_Y_POS = -1;
             _send_key_to_curses(KEY_MOUSE, &Mouse_status, TRUE);
@@ -3223,7 +3236,15 @@
 #endif
 
     /* Wait for events */
+    {
+        XEvent event;
+
+        for (;;) /* forever */
+        {
+             XtAppNextEvent(app_context, &event);
+             XtDispatchEvent(&event);
+        }
+    }
 
-    XtAppMainLoop(app_context);
     return OK;          /* won't get here */
 }
