diff -Naur orig/configure.ac patched/configure.ac
--- orig/configure.ac	2014-06-19 19:58:36.000000000 +0200
+++ patched/configure.ac	2015-04-18 21:21:49.761070391 +0200
@@ -658,18 +658,9 @@
     mocversion=`$MOCQT52 -v 2>&1`
     mocversiongrep=`echo $mocversion | grep "Qt 5\|moc 5"`
     if test x"$mocversiongrep" != x"$mocversion"; then
-      AC_CHECK_TOOL(QTCHOOSER, qtchooser)
-      AC_MSG_CHECKING([for qtchooser])
-      qt5tooldir=`QT_SELECT=qt5 qtchooser -print-env | grep QTTOOLDIR | cut -d '=' -f 2 | cut -d \" -f 2`
-      mocversion=`$qt5tooldir/moc -v 2>&1`
-      mocversiongrep=`echo $mocversion | grep "Qt 5\|moc 5"`
-      if test x"$mocversiongrep" != x"$mocversion"; then
         # no valid moc found
-        enable_poppler_qt5=no;
-        MOCQT5="not found"
-      else
-        MOCQT5=$qt5tooldir/moc
-      fi
+        enable_poppler_qt5=yes;
+        MOCQT5=$MOCQT52
     else
       MOCQT5=$MOCQT52
     fi
