diff -urN lilypond-2.16.2.orig/lily/freetype-error.cc lilypond-2.16.2/lily/freetype-error.cc
--- lilypond-2.16.2.orig/lily/freetype-error.cc	2013-01-04 14:41:56.000000000 +0100
+++ lilypond-2.16.2/lily/freetype-error.cc	2014-02-17 14:39:19.179681052 +0100
@@ -31,7 +31,7 @@
   const char *err_msg;
 } ft_errors[] =
 
-#include <freetype/fterrors.h>
+#include <freetype2/fterrors.h>
 
   ;
 
diff -urN lilypond-2.16.2.orig/lily/open-type-font.cc lilypond-2.16.2/lily/open-type-font.cc
--- lilypond-2.16.2.orig/lily/open-type-font.cc	2013-01-04 14:41:56.000000000 +0100
+++ lilypond-2.16.2/lily/open-type-font.cc	2014-02-17 14:39:33.843244695 +0100
@@ -23,7 +23,7 @@
 
 using namespace std;
 
-#include <freetype/tttables.h>
+#include <freetype2/tttables.h>
 
 #include "dimensions.hh"
 #include "international.hh"
diff -urN lilypond-2.16.2.orig/lily/pango-font.cc lilypond-2.16.2/lily/pango-font.cc
--- lilypond-2.16.2.orig/lily/pango-font.cc	2013-01-04 14:41:57.000000000 +0100
+++ lilypond-2.16.2/lily/pango-font.cc	2014-02-17 14:39:10.341533137 +0100
@@ -22,7 +22,7 @@
 #define PANGO_ENABLE_BACKEND
 
 #include <pango/pangoft2.h>
-#include <freetype/ftxf86.h>
+#include <freetype2/ftxf86.h>
 
 #include <map>
 #include <cstdio>
diff -urN lilypond-2.16.2.orig/lily/ttf.cc lilypond-2.16.2/lily/ttf.cc
--- lilypond-2.16.2.orig/lily/ttf.cc	2013-01-04 14:41:56.000000000 +0100
+++ lilypond-2.16.2/lily/ttf.cc	2014-02-17 14:39:42.269292446 +0100
@@ -20,7 +20,7 @@
 #include <cstdio>
 #include "freetype.hh"
 
-#include <freetype/tttables.h>
+#include <freetype2/tttables.h>
 
 #include "international.hh"
 #include "memory-stream.hh"
