diff -Naurdp a/include/linux/aio.h b/include/linux/aio.h
--- a/include/linux/aio.h	2013-10-06 23:00:20.000000000 +0200
+++ b/include/linux/aio.h	2013-10-10 18:04:25.438926067 +0200
@@ -9,6 +9,8 @@
 
 #include <linux/atomic.h>
 
+#include <linux/compiler.h>
+
 struct kioctx;
 struct kiocb;
 
diff -Naurdp a/include/linux/blkdev.h b/include/linux/blkdev.h
--- a/include/linux/blkdev.h	2013-10-06 23:00:20.000000000 +0200
+++ b/include/linux/blkdev.h	2013-10-10 18:03:28.286131934 +0200
@@ -22,6 +22,7 @@
 #include <linux/rcupdate.h>
 
 #include <asm/scatterlist.h>
+#include <linux/compiler.h>
 
 struct module;
 struct scsi_ioctl_command;
diff -Naurdp a/include/linux/compat.h b/include/linux/compat.h
--- a/include/linux/compat.h	2013-10-06 23:00:20.000000000 +0200
+++ b/include/linux/compat.h	2013-10-10 18:03:28.362127671 +0200
@@ -7,6 +7,7 @@
 
 #ifdef CONFIG_COMPAT
 
+#include <linux/compiler.h>
 #include <linux/stat.h>
 #include <linux/param.h>	/* for HZ */
 #include <linux/sem.h>
diff -Naurdp a/include/linux/hugetlb.h b/include/linux/hugetlb.h
--- a/include/linux/hugetlb.h	2013-10-06 23:00:20.000000000 +0200
+++ b/include/linux/hugetlb.h	2013-10-10 18:03:28.362127671 +0200
@@ -15,6 +15,7 @@ struct mmu_gather;
 #include <linux/mempolicy.h>
 #include <linux/shm.h>
 #include <asm/tlbflush.h>
+#include <linux/compiler.h>
 
 struct hugepage_subpool {
 	spinlock_t lock;
diff -Naurdp a/include/linux/inetdevice.h b/include/linux/inetdevice.h
--- a/include/linux/inetdevice.h	2013-10-06 23:00:20.000000000 +0200
+++ b/include/linux/inetdevice.h	2013-10-10 18:03:28.362127671 +0200
@@ -1,6 +1,8 @@
 #ifndef _LINUX_INETDEVICE_H
 #define _LINUX_INETDEVICE_H
 
+#include <linux/compiler.h>
+
 #ifdef __KERNEL__
 
 #include <linux/bitmap.h>
diff -Naurdp a/include/linux/mmzone.h b/include/linux/mmzone.h
--- a/include/linux/mmzone.h	2013-10-06 23:00:20.000000000 +0200
+++ b/include/linux/mmzone.h	2013-10-10 18:03:28.362127671 +0200
@@ -18,6 +18,7 @@
 #include <linux/page-flags-layout.h>
 #include <linux/atomic.h>
 #include <asm/page.h>
+#include <linux/compiler.h>
 
 /* Free memory management - zoned buddy allocator.  */
 #ifndef CONFIG_FORCE_MAX_ZONEORDER
diff -Naurdp a/include/linux/namei.h b/include/linux/namei.h
--- a/include/linux/namei.h	2013-10-06 23:00:20.000000000 +0200
+++ b/include/linux/namei.h	2013-10-10 18:03:28.362127671 +0200
@@ -5,6 +5,7 @@
 #include <linux/errno.h>
 #include <linux/linkage.h>
 #include <linux/path.h>
+#include <linux/compiler.h>
 
 struct vfsmount;
 
diff -Naurdp a/include/linux/nfs_fs.h b/include/linux/nfs_fs.h
--- a/include/linux/nfs_fs.h	2013-10-06 23:00:20.000000000 +0200
+++ b/include/linux/nfs_fs.h	2013-10-10 18:03:28.366127447 +0200
@@ -38,6 +38,8 @@
 
 #include <linux/mempool.h>
 
+#include <linux/compiler.h>
+
 /*
  * These are the default flags for swap requests
  */
diff -Naurdp a/include/linux/oprofile.h b/include/linux/oprofile.h
--- a/include/linux/oprofile.h	2013-10-06 23:00:20.000000000 +0200
+++ b/include/linux/oprofile.h	2013-10-10 18:03:28.366127447 +0200
@@ -19,6 +19,7 @@
 #include <linux/errno.h>
 #include <linux/printk.h>
 #include <linux/atomic.h>
+#include <linux/compiler.h>
  
 /* Each escaped entry is prefixed by ESCAPE_CODE
  * then one of the following codes, then the
diff -Naurdp a/include/linux/proc_fs.h b/include/linux/proc_fs.h
--- a/include/linux/proc_fs.h	2013-10-06 23:00:20.000000000 +0200
+++ b/include/linux/proc_fs.h	2013-10-10 18:04:10.639756197 +0200
@@ -6,6 +6,7 @@
 
 #include <linux/types.h>
 #include <linux/fs.h>
+#include <linux/compiler.h>
 
 struct proc_dir_entry;
 
diff -Naurdp a/include/linux/security.h b/include/linux/security.h
--- a/include/linux/security.h	2013-10-06 23:00:20.000000000 +0200
+++ b/include/linux/security.h	2013-10-10 18:03:55.284617512 +0200
@@ -27,6 +27,7 @@
 #include <linux/slab.h>
 #include <linux/err.h>
 #include <linux/string.h>
+#include <linux/compiler.h>
 
 struct linux_binprm;
 struct cred;
diff -Naurdp a/include/linux/selection.h b/include/linux/selection.h
--- a/include/linux/selection.h	2013-10-06 23:00:20.000000000 +0200
+++ b/include/linux/selection.h	2013-10-10 18:03:28.374126998 +0200
@@ -9,6 +9,7 @@
 
 #include <linux/tiocl.h>
 #include <linux/vt_buffer.h>
+#include <linux/compiler.h>
 
 struct tty_struct;
 
diff -Naurdp a/include/linux/syscalls.h b/include/linux/syscalls.h
--- a/include/linux/syscalls.h	2013-10-06 23:00:20.000000000 +0200
+++ b/include/linux/syscalls.h	2013-10-10 18:03:28.374126998 +0200
@@ -11,6 +11,8 @@
 #ifndef _LINUX_SYSCALLS_H
 #define _LINUX_SYSCALLS_H
 
+#include <linux/compiler.h>
+
 struct epoll_event;
 struct iattr;
 struct inode;
diff -Naurdp a/include/linux/tty_driver.h b/include/linux/tty_driver.h
--- a/include/linux/tty_driver.h	2013-10-06 23:00:20.000000000 +0200
+++ b/include/linux/tty_driver.h	2013-10-10 18:03:28.518118921 +0200
@@ -237,6 +237,7 @@
 #include <linux/list.h>
 #include <linux/cdev.h>
 #include <linux/termios.h>
+#include <linux/compiler.h>
 
 struct tty_struct;
 struct tty_driver;
diff -Naurdp a/include/linux/tty_ldisc.h b/include/linux/tty_ldisc.h
--- a/include/linux/tty_ldisc.h	2013-10-06 23:00:20.000000000 +0200
+++ b/include/linux/tty_ldisc.h	2013-10-10 18:03:28.558116678 +0200
@@ -125,6 +125,7 @@
 #include <linux/fs.h>
 #include <linux/wait.h>
 #include <linux/wait.h>
+#include <linux/compiler.h>
 
 
 /*
diff -Naurdp a/include/linux/vt_kern.h b/include/linux/vt_kern.h
--- a/include/linux/vt_kern.h	2013-10-06 23:00:20.000000000 +0200
+++ b/include/linux/vt_kern.h	2013-10-10 18:03:28.558116678 +0200
@@ -14,6 +14,7 @@
 #include <linux/mm.h>
 #include <linux/consolemap.h>
 #include <linux/notifier.h>
+#include <linux/compiler.h>
 
 /*
  * Presently, a lot of graphics programs do not restore the contents of
diff -Naurdp a/include/linux/writeback.h b/include/linux/writeback.h
--- a/include/linux/writeback.h	2013-10-06 23:00:20.000000000 +0200
+++ b/include/linux/writeback.h	2013-10-10 18:03:28.578115556 +0200
@@ -26,6 +26,8 @@ DECLARE_PER_CPU(int, dirty_throttle_leak
 #define DIRTY_SCOPE		8
 #define DIRTY_FULL_SCOPE	(DIRTY_SCOPE / 2)
 
+#include <linux/compiler.h>
+
 struct backing_dev_info;
 
 /*
diff -Naurdp a/include/scsi/sg.h b/include/scsi/sg.h
--- a/include/scsi/sg.h	2013-10-06 23:00:20.000000000 +0200
+++ b/include/scsi/sg.h	2013-10-10 18:03:28.582115332 +0200
@@ -82,6 +82,7 @@ typedef struct sg_iovec /* same structur
     size_t iov_len;             /* Length in bytes  */
 } sg_iovec_t;
 
+#include <linux/compiler.h>
 
 typedef struct sg_io_hdr
 {
