diff -Naur orig/etc/mplayer.desktop patched/etc/mplayer.desktop
--- orig/etc/mplayer.desktop	2015-09-04 11:43:54.000000000 +0200
+++ patched/etc/mplayer.desktop	2015-10-04 15:21:02.948070033 +0200
@@ -6,6 +6,7 @@
 GenericName[de]=Medienwiedergabe
 GenericName[es]=Reproductor multimedia
 GenericName[fr]=Lecteur multimédia
+GenericName[hu]=Médialejátszó
 GenericName[it]=Lettore multimediale
 GenericName[ja]=メディアプレーヤー
 X-GNOME-FullName=MPlayer Media Player
@@ -28,5 +29,5 @@
 TryExec=gmplayer
 Exec=gmplayer %F
 Terminal=false
-Categories=GTK;AudioVideo;Audio;Video;Player;TV;
+Categories=GTK;AudioVideo;Audio;Video;Player;
 MimeType=application/x-cd-image;application/x-cue;application/mxf;application/x-netshow-channel;application/ogg;application/ram;application/vnd.adobe.flash.movie;application/vnd.rn-realmedia;application/x-shockwave-flash;application/smil;audio/aac;audio/ac3;audio/x-adpcm;audio/x-aiff;audio/AMR;audio/AMR-WB;audio/x-ape;audio/basic;audio/flac;audio/x-flac;audio/x-flac+ogg;audio/x-m4b;audio/x-matroska;audio/mp2;audio/mp4;audio/mpeg;audio/x-mpegurl;audio/x-ms-asx;audio/x-ms-wma;audio/x-musepack;audio/ogg;audio/vnd.rn-realaudio;audio/x-scpls;audio/x-voc;audio/x-vorbis+ogg;audio/x-wav;audio/x-wavpack;video/3gpp;video/3gpp2;video/dv;video/x-flic;video/x-flv;video/x-matroska;video/mp2t;video/mp4;video/mpeg;video/vnd.mpegurl;video/x-ms-asf;video/x-ms-wmv;video/x-msvideo;video/x-nsv;video/ogg;video/x-ogm+ogg;video/quicktime;video/vnd.rn-realvideo;video/x-theora+ogg;video/webm;x-content/audio-cdda;x-content/audio-dvd;x-content/video-dvd;x-content/video-svcd;x-content/video-vcd;
