diff -Naur orig/etc/mplayer.desktop patched/etc/mplayer.desktop
--- orig/etc/mplayer.desktop	2012-05-28 18:40:56.000000000 +0200
+++ patched/etc/mplayer.desktop	2012-05-28 18:48:35.919376778 +0200
@@ -6,6 +6,7 @@
 GenericName[de]=Medienwiedergabe
 GenericName[es]=Reproductor multimedia
 GenericName[fr]=Lecteur multimédia
+GenericName[hu]=Médialejátszó
 GenericName[it]=Lettore multimediale
 GenericName[ja]=メディアプレーヤー
 X-GNOME-FullName=MPlayer Media Player
@@ -26,5 +27,5 @@
 TryExec=gmplayer
 Exec=gmplayer %F
 Terminal=false
-Categories=GTK;AudioVideo;Audio;Video;Player;TV;
+Categories=GTK;AudioVideo;Audio;Video;Player;TV;X-UHU-Menu/Multimedia/VideoPlayers;
 MimeType=application/mxf;application/x-netshow-channel;application/ogg;application/ram;application/vnd.rn-realmedia;application/x-shockwave-flash;application/smil;audio/ac3;audio/x-adpcm;audio/x-aiff;audio/AMR;audio/AMR-WB;audio/x-ape;audio/basic;audio/flac;audio/x-flac;audio/x-flac+ogg;audio/x-m4b;audio/x-matroska;audio/mp2;audio/mp4;audio/mpeg;audio/x-mpegurl;audio/x-ms-asx;audio/x-ms-wma;audio/x-musepack;audio/ogg;audio/vnd.rn-realaudio;audio/x-scpls;audio/x-voc;audio/x-vorbis+ogg;audio/x-wav;audio/x-wavpack;video/3gpp;video/3gpp2;video/dv;video/x-flic;video/x-flv;video/x-matroska;video/mp2t;video/mp4;video/mpeg;video/x-ms-asf;video/x-ms-wmv;video/x-msvideo;video/x-nsv;video/ogg;video/x-ogm+ogg;video/quicktime;video/vnd.rn-realvideo;video/x-theora+ogg;video/webm;
