diff -Naur orig/src/assembler_opt.S mod/src/assembler_opt.S
--- orig/src/assembler_opt.S	2007-07-09 22:21:36.000000000 +0200
+++ mod/src/assembler_opt.S	2014-04-03 19:40:41.000000000 +0200
@@ -407,3 +407,6 @@
     leave
     ret
 
+#if defined(__linux__) && defined(__ELF__)
+.section .note.GNU-stack,"",%progbits
+#endif
diff -Naur orig/src/assembler.S mod/src/assembler.S
--- orig/src/assembler.S	2009-01-08 21:54:08.000000000 +0100
+++ mod/src/assembler.S	2014-04-03 19:40:41.000000000 +0200
@@ -2076,3 +2076,6 @@
 _et_col2:       .int 0    #13
 #_et_col1:       .byte 11
 #_et_col2:       .byte 13
+#if defined(__linux__) && defined(__ELF__)
+.section .note.GNU-stack,"",%progbits
+#endif
