diff -Naur orig/src/divx.h patched/src/divx.h
--- orig/src/divx.h	2008-12-21 10:35:57.000000000 +0100
+++ patched/src/divx.h	2016-05-03 07:40:39.501359088 +0200
@@ -61,7 +61,7 @@
 int is_divx_in_progress(void);
 int is_http_in_progress(void);
 int divx_display_frames(void);
-inline void write_audio_to_avi(short int *buffer, int size);
+void write_audio_to_avi(short int *buffer, int size);
 void write_sub(char *s);
 void divx_audio_start(void);
 void divx_audio_stop(void);
diff -Naur orig/src/dvbtools.c patched/src/dvbtools.c
--- orig/src/dvbtools.c	2008-12-07 12:43:24.000000000 +0100
+++ patched/src/dvbtools.c	2016-05-03 07:25:09.068745322 +0200
@@ -16,7 +16,6 @@
 
 #include "translation.h"
 
-extern char *strndup (__const char *__string, size_t __n); //dont know why function not detected ?!?
 #define MAX_NB_CITY 1000 // FIXME ?
 
 const struct STRTAB inversion_list [] = {
