diff -Naur xdtv-2.4.1-orig/libavcodec/libx264.c xdtv-2.4.1/libavcodec/libx264.c
--- xdtv-2.4.1-orig/libavcodec/libx264.c	2009-03-22 16:02:55.000000000 +0100
+++ xdtv-2.4.1/libavcodec/libx264.c	2010-02-04 20:51:15.000000000 +0100
@@ -166,7 +166,7 @@
     x4->params.b_cabac = avctx->coder_type == FF_CODER_TYPE_AC;
     x4->params.i_bframe_adaptive = avctx->b_frame_strategy;
     x4->params.i_bframe_bias = avctx->bframebias;
-    x4->params.b_bframe_pyramid = avctx->flags2 & CODEC_FLAG2_BPYRAMID;
+    x4->params.i_bframe_pyramid = avctx->flags2 & CODEC_FLAG2_BPYRAMID;
     avctx->has_b_frames= avctx->flags2 & CODEC_FLAG2_BPYRAMID ? 2 : !!avctx->max_b_frames;
 
     x4->params.i_keyint_min = avctx->keyint_min;
