diff -Naur smplayer-0.8.1-orig/smplayer.desktop smplayer-0.8.1/smplayer.desktop
--- smplayer-0.8.1-orig/smplayer.desktop	2012-03-23 08:57:49.000000000 +0100
+++ smplayer-0.8.1/smplayer.desktop	2012-09-20 15:44:16.000000000 +0200
@@ -1,5 +1,5 @@
 [Desktop Entry]
-Categories=Qt;KDE;AudioVideo;Player;Video;
+Categories=Qt;KDE;AudioVideo;Player;Video;X-UHU-Menu/Multimedia/VideoPlayers;
 Comment=A great MPlayer front-end
 Comment[cs]=Výborný frontend pro MPlayer
 Comment[es]=Un magnífico interfaz gráfico para el MPlayer
