diff -urN apt-0.5.14.orig/RSYNC apt-0.5.14/RSYNC
--- apt-0.5.14.orig/RSYNC	1970-01-01 01:00:00.000000000 +0100
+++ apt-0.5.14/RSYNC	2003-11-26 22:36:50.000000000 +0100
@@ -0,0 +1,22 @@
+rsync method in apt
+-------------------
+
+Using the rsync method of apt can save you huge network traffic by only
+downloading the diff against a previous version of the packages.
+
+As rsync requires much more resources (especially CPU) on server side, we
+kindly ask you not use to this feature if you have a network bandwidth of
+10Mbit/s or more. Use ftp or http in this case instead.
+
+To use rsync, you have to:
+
+- install the rsync UHU package
+
+- put the required rsync:// URL into /etc/apt/sources.list
+
+- if you have a firewall, enable outgoing requests to port TCP 873
+
+- whenever upgrading the system: "apt-get upgrade" lists the packages to be
+downloaded. Before confirming apt-get doing so, put the old version of these
+packages (at least the bigger ones) to /var/cache/apt/archives (copy them
+from your UHU-Linux CD or wherever you have them). Then tell apt to continue.
diff -urN apt-0.5.14.orig/RSYNC_HU apt-0.5.14/RSYNC_HU
--- apt-0.5.14.orig/RSYNC_HU	1970-01-01 01:00:00.000000000 +0100
+++ apt-0.5.14/RSYNC_HU	2003-11-26 22:38:22.000000000 +0100
@@ -0,0 +1,25 @@
+rsync letöltési mód az apt-ben
+------------------------------
+
+Az apt rsync letöltési módjával jelentősen csökkenthető a hálózati forgalom,
+ilyenkor csak a csomagok előző változatához képesti különbségeket tölti le
+az apt.
+
+Mivel az rsync lényegesen több erőforrást (elsősorban processzoridőt)
+igényel szerver oldalon, ezért kérjük, hogy csak akkor használja, ha hálózati
+sávszélessége nem éri el a 10Mbit/s-ot. Ha ilyen, vagy ennél is nagyobb
+sebességgel tud letölteni, használja az ftp vagy http protokollt.
+
+Az rsync használatához
+
+- telepítse az rsync UHU csomagot
+
+- a kívánt rsync:// kezdetű URL-t helyezze el a /etc/apt/sources.list fájlban
+
+- ha van tűzfala, engedélyezze a 873-as TCP portra irányuló kimenő kéréseket
+
+- a rendszer frissítése során: az "apt-get upgrade" parancs megjeleníti a
+frissítendő csomagok listáját. Mielőtt megerősítést ad az apt-get
+parancsnak a csomagok letöltésére, keresse elő (legalábbis a nagyobb
+csomagokra) azok régi verzióját (például az UHU-Linux CD-ről) és másolja be
+/var/cache/apt/archives alá. Ez után engedélyezze az apt-nek a folytatást.
