diff -Naur orig/apt-pkg/contrib/strutl.cc patched/apt-pkg/contrib/strutl.cc
--- orig/apt-pkg/contrib/strutl.cc	2014-10-15 19:55:49.000000000 +0200
+++ patched/apt-pkg/contrib/strutl.cc	2014-10-18 22:21:37.624772425 +0200
@@ -381,7 +381,7 @@
    
    /* bytes, KiloBytes, MegaBytes, GigaBytes, TeraBytes, PetaBytes, 
       ExaBytes, ZettaBytes, YottaBytes */
-   char Ext[] = {'\0','k','M','G','T','P','E','Z','Y'};
+   char Ext[] = {'\0','K','M','G','T','P','E','Z','Y'};
    int I = 0;
    while (I <= 8)
    {
@@ -396,7 +396,7 @@
          sprintf(S,"%'.0f %c",ASize,Ext[I]);
 	 break;
       }
-      ASize /= 1000.0;
+      ASize /= 1024.0;
       I++;
    }
    
