diff -Naurd apt-0.5.14.orig/apt-pkg/contrib/strutl.cc apt-0.5.14/apt-pkg/contrib/strutl.cc
--- apt-0.5.14.orig/apt-pkg/contrib/strutl.cc	2003-07-18 16:15:11.000000000 +0200
+++ apt-0.5.14/apt-pkg/contrib/strutl.cc	2003-11-23 17:57:09.000000000 +0100
@@ -258,7 +258,7 @@
    
    /* bytes, KiloBytes, MegaBytes, GigaBytes, TeraBytes, PetaBytes, 
       ExaBytes, ZettaBytes, YottaBytes */
-   char Ext[] = {'\0','k','M','G','T','P','E','Z','Y'};
+   char Ext[] = {'\0','K','M','G','T','P','E','Z','Y'};
    int I = 0;
    while (I <= 8)
    {
@@ -273,7 +273,7 @@
          sprintf(S,"%.0f%c",ASize,Ext[I]);
 	 break;
       }
-      ASize /= 1000.0;
+      ASize /= 1024.0;
       I++;
    }
    
