diff -Naur orig/Makefile patched/Makefile
--- orig/Makefile	2016-06-22 15:01:11.000000000 +0200
+++ patched/Makefile	2016-07-21 23:49:48.534141803 +0200
@@ -9,8 +9,8 @@
 .PHONY: default
 default: startup all
 
-.PHONY: fast headers library clean veryclean all binary program doc test update-po
-all headers library clean veryclean binary program doc manpages docbook test update-po startup dirs:
+.PHONY: fast headers library clean veryclean all binary program doc update-po
+all headers library clean veryclean binary program doc manpages docbook update-po startup dirs:
 	$(MAKE) -C vendor $@
 	$(MAKE) -C apt-pkg $@
 	$(MAKE) -C apt-inst $@
@@ -21,8 +21,6 @@
 	$(MAKE) -C dselect $@
 	$(MAKE) -C doc $@
 	$(MAKE) -C po $@
-	# FIXME: -C test has issue swith parallel builds, investigate!
-	-$(MAKE) -C test $@
 
 fast:
 	$(MAKE) -C vendor all
@@ -32,9 +30,8 @@
 	$(MAKE) -C methods all
 	$(MAKE) -C cmdline all
 	$(MAKE) -C ftparchive all
-	$(MAKE) -C test all
 
-all headers library clean veryclean binary program doc manpages docbook test update-po: startup dirs
+all headers library clean veryclean binary program doc manpages docbook update-po: startup dirs
 
 dirs: startup
 
