diff -Naur orig/buildlib/docbook.mak patched/buildlib/docbook.mak
--- orig/buildlib/docbook.mak	2014-10-15 19:55:49.000000000 +0200
+++ patched/buildlib/docbook.mak	2014-10-18 22:28:27.926807003 +0200
@@ -52,24 +52,24 @@
 # Rules to build Text documentations
 ifdef XSLTPROC
 
-DOCBOOK_TEXT_STYLESHEET := docbook-text-style.xsl
+DOCBOOK_HTML_STYLESHEET := docbook-html-style.xsl
 
 # Install generation hooks
-docbook: $($(LOCAL)-TEXT)
+docbook: $($(LOCAL)-HTML)
 veryclean: veryclean/text/$(LOCAL)
 
 vpath %.dbk $(SUBDIRS)
-vpath $(DOCBOOK_TEXT_STYLESHEET) $(SUBDIRS)
-$(DOC)/%.text: %.dbk $(DOCBOOK_TEXT_STYLESHEET) $(INCLUDES)
+vpath $(DOCBOOK_HTML_STYLESHEET) $(SUBDIRS)
+$(DOC)/%.text: %.dbk $(DOCBOOK_HTML_STYLESHEET) $(INCLUDES)
 	echo Creating text for $< to $@
 	$(DOCBOOK) \
 		--stringparam l10n.gentext.default.language $(LC) \
-		$(<D)/$(DOCBOOK_TEXT_STYLESHEET) $< | \
+		$(<D)/$(DOCBOOK_HTML_STYLESHEET) $< | \
 		LC_ALL=C.UTF-8 $(DOCBOOK2TEXT) > $@ || exit 198
 
 # Clean rule
 .PHONY: veryclean/text/$(LOCAL)
 veryclean/text/$(LOCAL):
-	-rm -rf $($(@F)-TEXT)
+	-rm -rf $($(@F)-HTML)
 
 endif
diff -Naur orig/doc/docbook-html-style.xsl patched/doc/docbook-html-style.xsl
--- orig/doc/docbook-html-style.xsl	2014-10-15 19:55:49.000000000 +0200
+++ patched/doc/docbook-html-style.xsl	2014-10-18 22:24:50.353728116 +0200
@@ -2,7 +2,7 @@
 <xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform" version="1.0">
 
   <!-- Import our base stylesheet -->
-  <xsl:import href="/usr/share/xml/docbook/stylesheet/docbook-xsl/xhtml-1_1/chunk.xsl" />
+  <xsl:import href="/usr/share/sgml/docbook/xsl-stylesheets/manpages/docbook.xsl" />
 
   <!-- Since we use xsltproc (not saxon), add a workaround to ensure UTF-8 -->
   <xsl:template xmlns="http://www.w3.org/1999/xhtml" name="head.content.generator">
diff -Naur orig/doc/docbook-text-style.xsl patched/doc/docbook-text-style.xsl
--- orig/doc/docbook-text-style.xsl	2014-10-15 19:55:49.000000000 +0200
+++ patched/doc/docbook-text-style.xsl	2014-10-18 22:24:50.353728116 +0200
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform" version="1.0">
 
-  <xsl:import href="/usr/share/xml/docbook/stylesheet/docbook-xsl/xhtml-1_1/docbook.xsl" />
+  <xsl:import href="/usr/share/sgml/docbook/xsl-stylesheets/manpages/docbook.xsl" />
 
   <!-- Parameters for optimal text output. -->
   <xsl:param name="callout.graphics" select="0"/>
diff -Naur orig/doc/manpage-style.xsl patched/doc/manpage-style.xsl
--- orig/doc/manpage-style.xsl	2014-10-15 19:55:49.000000000 +0200
+++ patched/doc/manpage-style.xsl	2014-10-18 22:24:50.353728116 +0200
@@ -2,7 +2,7 @@
  xmlns:xsl="http://www.w3.org/1999/XSL/Transform"
  version="1.0">
 
-<xsl:import href="/usr/share/xml/docbook/stylesheet/nwalsh/manpages/docbook.xsl" />
+<xsl:import href="/usr/share/sgml/docbook/xsl-stylesheets/manpages/docbook.xsl" />
 
 <xsl:param name="man.output.encoding" select="'UTF-8'" />
 
diff -Naur orig/doc/po4a.conf patched/doc/po4a.conf
--- orig/doc/po4a.conf	2014-10-15 19:55:49.000000000 +0200
+++ patched/doc/po4a.conf	2014-10-18 22:24:50.353728116 +0200
@@ -3,7 +3,7 @@
 
 # Entities need to be present, even if not translated
 [po4a_alias:entity] text opt:"-k 0"
-[po4a_alias:manpage] docbook opt:"\
+[po4a_alias:manpage] docbook opt:" \
 	-o untranslated='<term><option> <refname> <refentrytitle> <date> <refentryinfo> <phrase> <variablelist> <term><command>' \
 	-o break='<term><option> <term><command>' \
 	-o nodefault='<option> <refname> <refentrytitle> <date> <refentryinfo> <phrase> <variablelist> <command>' \
