diff -Naur orig/menu_data/aqemu.desktop mod/menu_data/aqemu.desktop
--- orig/menu_data/aqemu.desktop	2009-06-04 09:29:38.000000000 +0200
+++ mod/menu_data/aqemu.desktop	2016-04-25 17:29:34.074615240 +0200
@@ -1,14 +1,8 @@
 [Desktop Entry]
-Encoding=UTF-8
 Name=AQEMU
 Exec=aqemu
 Terminal=false
 Type=Application
-Icon=aqemu_48x48.png
+Icon=aqemu_48x48
 StartupNotify=true
 Categories=System;Emulator;
-
-Name[ru]=AQEMU
-Name[en]=AQEMU
-Comment[ru]=Управление виртуальными машинами
-Comment[en]=Virtual machine manager
