diff -Naur orig/gcc/c-family/c.opt patched/gcc/c-family/c.opt
--- orig/gcc/c-family/c.opt	2014-04-03 15:41:55.000000000 +0200
+++ patched/gcc/c-family/c.opt	2014-10-24 16:32:14.668964846 +0200
@@ -773,7 +773,7 @@
 ; documented in common.opt
 
 Wunused-local-typedefs
-C ObjC C++ ObjC++ Var(warn_unused_local_typedefs) Warning EnabledBy(Wunused)
+C ObjC C++ ObjC++ Var(warn_unused_local_typedefs) Warning
 Warn when typedefs locally defined in a function are not used
 
 Wunused-macros
