diff -Naur orig/unix/chessx.desktop mod/unix/chessx.desktop
--- orig/unix/chessx.desktop	2013-03-12 08:50:45.000000000 +0100
+++ mod/unix/chessx.desktop	2014-04-14 23:19:46.000000000 +0200
@@ -2,9 +2,11 @@
 Type=Application
 Name=ChessX
 Comment=Free chess database
+Comment[hu]=Sakkadatbázis
 GenericName=Chess Game
+GenericName[hu]=Sakkjáték
 Icon=chessx
 Exec=chessx
 Terminal=false
-Categories=Game;BoardGame;
+Categories=Game;BoardGame;X-UHU-Menu/Games/Board;
 MimeType=application/x-chess-pgn;
