diff -Naur faac-1.28-orig/configure.in faac-1.28/configure.in
--- faac-1.28-orig/configure.in	2009-02-05 01:55:38.000000000 +0100
+++ faac-1.28/configure.in	2013-03-06 19:52:05.000000000 +0100
@@ -3,7 +3,7 @@
 AC_CONFIG_AUX_DIR(.)
 AM_INIT_AUTOMAKE
 
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS(config.h)
 
 AC_ARG_WITH( mp4v2,  [  --with-mp4v2            compile libmp4v2],WITHMP4V2=$withval, WITHMP4V2=yes)
 AC_ARG_ENABLE( drm,  [  --enable-drm            Digital Radio Mondiale support], enable_drm=$enableval, enable_drm=no)
