diff -Naur Cinnamon-fbb12f3/configure.ac Cinnamon-4.0.1/configure.ac
--- Cinnamon-fbb12f3/configure.ac	2018-12-02 09:29:17.000000000 +0100
+++ Cinnamon-4.0.1/configure.ac	2018-11-12 16:01:43.000000000 +0100
@@ -81,7 +81,7 @@
                                libstartup-notification-1.0 >= $STARTUP_NOTIFICATION_MIN_VERSION
                                gobject-introspection-1.0 >= $GOBJECT_INTROSPECTION_MIN_VERSION
                                polkit-agent-1 >= $POLKIT_MIN_VERSION xfixes
-                               libnm atk-bridge-2.0)
+                               libnm-glib libnm-util atk-bridge-2.0)
 
 PKG_CHECK_MODULES(CINNAMON_PERF_HELPER, gtk+-3.0 gio-2.0)
 
diff -Naur Cinnamon-fbb12f3/files/usr/share/cinnamon/applets/network@cinnamon.org/applet.js Cinnamon-4.0.1/files/usr/share/cinnamon/applets/network@cinnamon.org/applet.js
--- Cinnamon-fbb12f3/files/usr/share/cinnamon/applets/network@cinnamon.org/applet.js	2018-12-02 09:29:17.000000000 +0100
+++ Cinnamon-4.0.1/files/usr/share/cinnamon/applets/network@cinnamon.org/applet.js	2018-11-12 16:01:43.000000000 +0100
@@ -2,7 +2,8 @@
 const GLib = imports.gi.GLib;
 const GObject = imports.gi.GObject;
 const Lang = imports.lang;
-const NM = imports.gi.NM;
+const NetworkManager = imports.gi.NetworkManager;
+const NMClient = imports.gi.NMClient;
 const Signals = imports.signals;
 const St = imports.gi.St;
 const Mainloop = imports.mainloop;
@@ -34,14 +35,26 @@
 };
 
 // small optimization, to avoid using [] all the time
-const NM80211Mode = NM['80211Mode'];
-const NM80211ApFlags = NM['80211ApFlags'];
-const NM80211ApSecurityFlags = NM['80211ApSecurityFlags'];
+const NM80211Mode = NetworkManager['80211Mode'];
+const NM80211ApFlags = NetworkManager['80211ApFlags'];
+const NM80211ApSecurityFlags = NetworkManager['80211ApSecurityFlags'];
 
 // number of wireless networks that should be visible
 // (the remaining are placed into More...)
 const NUM_VISIBLE_NETWORKS = 5;
 
+function ssidCompare(one, two) {
+    if (!one || !two)
+        return false;
+    if (one.length != two.length)
+        return false;
+    for (let i = 0; i < one.length; i++) {
+        if (one[i] != two[i])
+            return false;
+    }
+    return true;
+}
+
 // shared between NMNetworkMenuItem and NMDeviceWWAN
 function signalToIcon(value) {
     if (value > 80)
@@ -63,7 +76,7 @@
 }
 
 function ssidToLabel(ssid) {
-    let label = NM.utils_ssid_to_utf8(ssid.get_data());
+    let label = NetworkManager.utils_ssid_to_utf8(ssid);
     if (!label)
         label = _("<unknown>");
     return label;
@@ -73,11 +86,6 @@
     this._init.apply(this, arguments);
 }
 
-function default_to_private_connections(client) {
-    let perms = client.get_permission_result (NM.ClientPermission.SETTINGS_MODIFY_SYSTEM);
-    return (perms != NM.ClientPermissionResult.YES);
-}
-
 NMNetworkMenuItem.prototype = {
     __proto__: PopupMenu.PopupBaseMenuItem.prototype,
 
@@ -192,10 +200,8 @@
             this.setToggleState(device.connected);
             // if this device is not controllable, hide the switch
             this._switch.actor.visible = device.controllable;
-        } else {
+        } else
             this.setStatus('');
-            this._switch.actor.visible = false;
-        }
     },
 
     activate: function(event) {
@@ -380,7 +386,7 @@
     },
 
     deactivate: function() {
-        this.device.disconnect_async(null, null);
+        this.device.disconnect(function() {});
     },
 
     activate: function() {
@@ -391,16 +397,16 @@
         // pick the most recently used connection and connect to that
         // or if no connections ever set, create an automatic one
         if (this._connections.length > 0) {
-            this._client.activate_connection_async(this._connections[0].connection, this.device, null, null, null);
+            this._client.activate_connection(this._connections[0].connection, this.device, null, null);
         } else if (this._autoConnectionName) {
             let connection = this._createAutomaticConnection();
             if (connection)
-                this._client.add_and_activate_connection_async(connection, this.device, null, null, null);
+                this._client.add_and_activate_connection(connection, this.device, null, null);
         }
     },
 
     get connected() {
-        return this.device.state == NM.DeviceState.ACTIVATED;
+        return this.device.state == NetworkManager.DeviceState.ACTIVATED;
     },
 
     setActiveConnection: function(activeConnection) {
@@ -487,25 +493,25 @@
 
     get statusLabel(){
         switch(this.device.state) {
-        case NM.DeviceState.DISCONNECTED:
-        case NM.DeviceState.ACTIVATED:
+        case NetworkManager.DeviceState.DISCONNECTED:
+        case NetworkManager.DeviceState.ACTIVATED:
             return null;
-        case NM.DeviceState.UNMANAGED:
+        case NetworkManager.DeviceState.UNMANAGED:
             /* Translators: this is for network devices that are physically present but are not
                under NetworkManager's control (and thus cannot be used in the menu) */
             return _("unmanaged");
-        case NM.DeviceState.DEACTIVATING:
+        case NetworkManager.DeviceState.DEACTIVATING:
             return _("disconnecting...");
-        case NM.DeviceState.PREPARE:
-        case NM.DeviceState.CONFIG:
-        case NM.DeviceState.IP_CONFIG:
-        case NM.DeviceState.IP_CHECK:
-        case NM.DeviceState.SECONDARIES:
+        case NetworkManager.DeviceState.PREPARE:
+        case NetworkManager.DeviceState.CONFIG:
+        case NetworkManager.DeviceState.IP_CONFIG:
+        case NetworkManager.DeviceState.IP_CHECK:
+        case NetworkManager.DeviceState.SECONDARIES:
             return _("connecting...");
-        case NM.DeviceState.NEED_AUTH:
+        case NetworkManager.DeviceState.NEED_AUTH:
             /* Translators: this is for network connections that require some kind of key or password */
             return _("authentication required");
-        case NM.DeviceState.UNAVAILABLE:
+        case NetworkManager.DeviceState.UNAVAILABLE:
             // This state is actually a compound of various states (generically unavailable,
             // firmware missing, carrier not available), that are exposed by different properties
             // (whose state may or may not updated when we receive state-changed).
@@ -516,7 +522,7 @@
                    module, which is missing */
                 return _("firmware missing");
             }
-            if (this.device.capabilities & NM.DeviceCapabilities.CARRIER_DETECT) {
+            if (this.device.capabilities & NetworkManager.DeviceCapabilities.CARRIER_DETECT) {
                 if (!this._carrierChangedId)
                     this._carrierChangedId = this.device.connect('notify::carrier', Lang.bind(this, this._substateChanged));
                 if (!this.carrier) {
@@ -527,7 +533,7 @@
             /* Translators: this is for a network device that cannot be activated (for example it
                is disabled by rfkill, or it has no coverage */
             return _("unavailable");
-        case NM.DeviceState.FAILED:
+        case NetworkManager.DeviceState.FAILED:
             return _("connection failed");
         default:
             log('Device state invalid, is %d'.format(this.device.state));
@@ -537,7 +543,7 @@
 
     get controllable(){
         // controllable for every state except unavailable or unmanaged
-        if(this.device.state === NM.DeviceState.UNAVAILABLE || this.device.state === NM.DeviceState.UNMANAGED)
+        if(this.device.state === NetworkManager.DeviceState.UNAVAILABLE || this.device.state === NetworkManager.DeviceState.UNMANAGED)
             return false;
 
         return true;
@@ -569,7 +575,7 @@
     },
 
     _shouldShowConnectionList: function() {
-        return (this.device.state >= NM.DeviceState.DISCONNECTED);
+        return (this.device.state >= NetworkManager.DeviceState.DISCONNECTED);
     },
 
     _createSection: function() {
@@ -586,7 +592,7 @@
             for(let j = 0; j < this._connections.length; ++j) {
                 let obj = this._connections[j];
                 if (this._activeConnection &&
-                    obj.connection == this._activeConnection.connection)
+                    obj.connection == this._activeConnection._connection)
                     continue;
                 obj.item = this._createConnectionItem(obj);
 
@@ -604,7 +610,7 @@
             this._autoConnectionItem.connect('activate', Lang.bind(this, function() {
                 let connection = this._createAutomaticConnection();
                 if (connection)
-                    this._client.add_and_activate_connection_async(connection, this.device, null, null, null);
+                    this._client.add_and_activate_connection(connection, this.device, null, null);
             }));
             this.section.addMenuItem(this._autoConnectionItem);
         }
@@ -615,14 +621,14 @@
         let item = new PopupMenu.PopupMenuItem(obj.name);
 
         item.connect('activate', Lang.bind(this, function() {
-            this._client.activate_connection_async(connection, this.device, null, null, null);
+            this._client.activate_connection(connection, this.device, null, null);
         }));
         return item;
     },
 
     _createActiveConnectionItem: function() {
         let title;
-        let active = this._activeConnection.connection;
+        let active = this._activeConnection._connection;
         if (active) {
             title = active._name;
         } else {
@@ -640,11 +646,11 @@
             return;
         }
 
-        if (oldstate == NM.DeviceState.ACTIVATED) {
+        if (oldstate == NetworkManager.DeviceState.ACTIVATED) {
             this.emit('network-lost');
         }
 
-        if (newstate == NM.DeviceState.FAILED) {
+        if (newstate == NetworkManager.DeviceState.FAILED) {
             this.emit('activation-failed', reason);
         }
 
@@ -688,10 +694,10 @@
             }
             else {
                 switch (this.device.get_device_type()) {
-                    case NM.DeviceType.ETHERNET: return _("Ethernet");
-                    case NM.DeviceType.WIFI: return _("Wifi");
-                    case NM.DeviceType.MODEM: return _("Modem");
-                    case NM.DeviceType.BT: return _("Bluetooth");
+                    case NetworkManager.DeviceType.ETHERNET: return _("Ethernet");
+                    case NetworkManager.DeviceType.WIFI: return _("Wifi");
+                    case NetworkManager.DeviceType.MODEM: return _("Modem");
+                    case NetworkManager.DeviceType.BT: return _("Bluetooth");
                     default: return "";
                 }
             }
@@ -742,17 +748,15 @@
     },
 
     _createAutomaticConnection: function() {
-        let connection = new NM.SimpleConnection();
-        connection._uuid = NM.utils_uuid_generate();
-        connection.add_setting(new NM.SettingWired());
-        let setting_conn = new NM.SettingConnection({
+        let connection = new NetworkManager.Connection();
+        connection._uuid = NetworkManager.utils_uuid_generate();
+        connection.add_setting(new NetworkManager.SettingWired());
+        connection.add_setting(new NetworkManager.SettingConnection({
             uuid: connection._uuid,
             id: this._autoConnectionName,
-            type: NM.SETTING_WIRED_SETTING_NAME,
+            type: NetworkManager.SETTING_WIRED_SETTING_NAME,
             autoconnect: true
-        });
-        setting_conn.add_permission('user', GLib.get_user_name(), null);
-        connection.add_setting(setting_conn);
+        }));
         return connection;
     }
 };
@@ -777,26 +781,26 @@
             try {
                 is_wwan = true;
                 this.mobileDevice = new ModemManager.BroadbandModem(device.udi, device.current_capabilities);
-                if (this._capabilities & NM.DeviceModemCapabilities.GSM_UMTS) {
-                    this._connectionType = NM.SETTING_GSM_SETTING_NAME;
-                } else if (this._capabilities & NM.DeviceModemCapabilities.LTE) {
-                    this._connectionType = NM.SETTING_GSM_SETTING_NAME;
-                } else if (this._capabilities & NM.DeviceModemCapabilities.CDMA_EVDO) {
-                    this._connectionType = NM.SETTING_CDMA_SETTING_NAME;
+                if (this._capabilities & NetworkManager.DeviceModemCapabilities.GSM_UMTS) {
+                    this._connectionType = NetworkManager.SETTING_GSM_SETTING_NAME;
+                } else if (this._capabilities & NetworkManager.DeviceModemCapabilities.LTE) {
+                    this._connectionType = NetworkManager.SETTING_GSM_SETTING_NAME;
+                } else if (this._capabilities & NetworkManager.DeviceModemCapabilities.CDMA_EVDO) {
+                    this._connectionType = NetworkManager.SETTING_CDMA_SETTING_NAME;
                 }
             }
             catch (e){
                 global.logError(e);
             }
-        } else if (this._capabilities & NM.DeviceModemCapabilities.GSM_UMTS) {
+        } else if (this._capabilities & NetworkManager.DeviceModemCapabilities.GSM_UMTS) {
             is_wwan = true;
             this.mobileDevice = new ModemManager.ModemGsm(device.udi);
-            this._connectionType = NM.SETTING_GSM_SETTING_NAME;
-        } else if (this._capabilities & NM.DeviceModemCapabilities.CDMA_EVDO) {
+            this._connectionType = NetworkManager.SETTING_GSM_SETTING_NAME;
+        } else if (this._capabilities & NetworkManager.DeviceModemCapabilities.CDMA_EVDO) {
             is_wwan = true;
             this.mobileDevice = new ModemManager.ModemCdma(device.udi);
-            this._connectionType = NM.SETTING_CDMA_SETTING_NAME;
-        } else if (this._capabilities & NM.DeviceModemCapabilities.LTE) {
+            this._connectionType = NetworkManager.SETTING_CDMA_SETTING_NAME;
+        } else if (this._capabilities & NetworkManager.DeviceModemCapabilities.LTE) {
             is_wwan = true;
             // FIXME: support signal quality
         }
@@ -844,7 +848,7 @@
     },
 
     get connected() {
-        return this._enabled && this.device.state == NM.DeviceState.ACTIVATED;
+        return this._enabled && this.device.state == NetworkManager.DeviceState.ACTIVATED;
     },
 
     destroy: function() {
@@ -913,13 +917,13 @@
     },
 
     _createAutomaticConnection: function() {
-        let connection = new NM.SimpleConnection();
-        connection._uuid = NM.utils_uuid_generate();
-        connection.add_setting(new NM.SettingBluetooth());
-        connection.add_setting(new NM.SettingConnection({
+        let connection = new NetworkManager.Connection();
+        connection._uuid = NetworkManager.utils_uuid_generate();
+        connection.add_setting(new NetworkManager.SettingBluetooth());
+        connection.add_setting(new NetworkManager.SettingConnection({
             uuid: connection._uuid,
             id: this._autoConnectionName,
-            type: NM.SETTING_BLUETOOTH_SETTING_NAME,
+            type: NetworkManager.SETTING_BLUETOOTH_SETTING_NAME,
             autoconnect: false
         }));
         return connection;
@@ -959,7 +963,7 @@
     },
 
     connectionValid: function(connection) {
-        return connection._type == NM.SETTING_VPN_SETTING_NAME;
+        return connection._type == NetworkManager.SETTING_VPN_SETTING_NAME;
     },
 
     get empty() {
@@ -982,7 +986,7 @@
 
     deactivate: function() {
         if (this._activeConnection)
-            this._client.deactivate_connection(this._activeConnection, null);
+            this._client.deactivate_connection(this._activeConnection);
     },
 
     statusLabel: null,
@@ -1002,8 +1006,6 @@
         this._overflowItem = null;
         this._networks = [ ];
 
-        this._client = client;
-
         // breaking the layers with this, but cannot call
         // this.connectionValid until I have a device
         this.device = device;
@@ -1126,7 +1128,7 @@
             for (let i = 0; i < bestApObj.accessPoints.length; i++) {
                 let ap = bestApObj.accessPoints[i];
                 if (ap.connection_valid(best)) {
-                    this._client.activate_connection_async(best, this.device, ap.path, null, null);
+                    this._client.activate_connection(best, this.device, ap.dbus_path, null);
                     break;
                 }
             }
@@ -1140,7 +1142,7 @@
         if (this._networks.length > 0) {
             let connection = this._createAutomaticConnection(this._networks[0]);
             let accessPoints = sortAccessPoints(this._networks[0].accessPoints);
-            this._client.add_and_activate_connection_async(connection, this.device, accessPoints[0].path, null, null);
+            this._client.add_and_activate_connection(connection, this.device, accessPoints[0].dbus_path, null);
         }
     },
 
@@ -1227,7 +1229,7 @@
     },
 
     _networkCompare: function(network, accessPoint) {
-        if (!network.ssid.equal (accessPoint.get_ssid()))
+        if (!ssidCompare(network.ssid, accessPoint.get_ssid()))
             return false;
         if (network.mode != accessPoint.mode)
             return false;
@@ -1394,7 +1396,7 @@
             let accessPoints = sortAccessPoints(accessPointObj.accessPoints);
             for (let i = 0; i < accessPoints.length; i++) {
                 if (accessPoints[i].connection_valid(connection)) {
-                    this._client.activate_connection_async(connection, this.device, accessPoints[i].path, null, null);
+                    this._client.activate_connection(connection, this.device, accessPoints[i].dbus_path, null);
                     break;
                 }
             }
@@ -1501,7 +1503,8 @@
     },
 
     _createActiveConnectionItem: function() {
-        if (this._activeConnection.connection) {
+        if (this._activeConnection._connection) {
+            let connection = this._activeConnection._connection;
             if (!this._activeNetwork) {
                 if (this.device.active_access_point) {
                     let networkPos = this._findNetwork(this.device.active_access_point);
@@ -1517,7 +1520,7 @@
             if (this._activeNetwork)
                 this._activeConnectionItem = new NMNetworkMenuItem(this._activeNetwork.accessPoints, undefined, { reactive: false });
             else
-                this._activeConnectionItem = new PopupMenu.PopupImageMenuItem(this._activeConnection.connection._name, 'network-wireless-connected', { reactive: false });
+                this._activeConnectionItem = new PopupMenu.PopupImageMenuItem(connection._name, 'network-wireless-connected', { reactive: false });
         } else {
             // We cannot read the connection (due to ACL, or API incompatibility), but we still show signal if we have it
             if (this._activeNetwork)
@@ -1533,41 +1536,25 @@
 
     _createAutomaticConnection: function(apObj) {
         let name;
-        let ssid = NM.utils_ssid_to_utf8(apObj.ssid.get_data());
+        let ssid = NetworkManager.utils_ssid_to_utf8(apObj.ssid);
         if (ssid) {
             /* TRANSLATORS: this the automatic wireless connection name (including the network name) */
             name = _("Auto %s").format(ssid);
         } else
             name = _("Auto wireless");
 
-        let connection = new NM.SimpleConnection();
-        connection.add_setting(new NM.SettingWireless());
-        let setting_conn = new NM.SettingConnection({
+        let connection = new NetworkManager.Connection();
+        connection.add_setting(new NetworkManager.SettingWireless());
+        connection.add_setting(new NetworkManager.SettingConnection({
             id: name,
             autoconnect: true, // NetworkManager will know to ignore this if appropriate
-            uuid: NM.utils_uuid_generate(),
-            type: NM.SETTING_WIRELESS_SETTING_NAME
-        });
-        if (default_to_private_connections(this._client)) {
-            setting_conn.add_permission('user', GLib.get_user_name(), null);
-            if (apObj.security == NMAccessPointSecurity.WPA2_PSK ||
-                apObj.security == NMAccessPointSecurity.WPA_PSK) {
-                connection.add_setting(new NM.SettingWirelessSecurity({
-                    psk_flags: NM.SettingSecretFlags.AGENT_OWNED
-                }));
-            }
-            if (apObj.security == NMAccessPointSecurity.WEP) {
-                connection.add_setting(new NM.SettingWirelessSecurity({
-                    wep_key_flags: NM.SettingSecretFlags.AGENT_OWNED
-                }));
-            }
-        }
-        connection.add_setting(setting_conn);
+            uuid: NetworkManager.utils_uuid_generate(),
+            type: NetworkManager.SETTING_WIRELESS_SETTING_NAME
+        }));
         return connection;
     },
 
     _createNetworkItem: function(apObj, position) {
-
         if(!apObj.accessPoints || apObj.accessPoints.length == 0) {
             // this should not happen, but I have no idea why it happens
             return;
@@ -1591,10 +1578,10 @@
                     // 802.1x-enabled APs require further configuration, so they're
                     // handled in cinnamon-settings
                     Util.spawn(['cinnamon-settings', 'network', 'connect-8021x-wifi',
-                                this.device.get_path(), accessPoints[0].path]);
+                                this.device.get_path(), accessPoints[0].dbus_path]);
                 } else {
                     let connection = this._createAutomaticConnection(apObj);
-                    this._client.add_and_activate_connection_async(connection, this.device, accessPoints[0].path, null, null);
+                    this._client.add_and_activate_connection(connection, this.device, accessPoints[0].dbus_path, null);
                 }
             }));
         }
@@ -1697,16 +1684,7 @@
             this._currentIconName = undefined;
             this._setIcon('network-offline');
 
-            NM.Client.new_async(null, Lang.bind(this, this._clientGot));
-        }
-        catch (e) {
-            global.logError(e);
-        }
-    },
-
-    _clientGot: function(obj, result) {
-        try {
-            this._client = NM.Client.new_finish(result);
+            this._client = NMClient.Client.new();
 
             this._statusSection = new PopupMenu.PopupMenuSection();
             this._statusItem = new PopupMenu.PopupMenuItem('', { style_class: 'popup-inactive-menu-item', reactive: false });
@@ -1772,7 +1750,7 @@
                 let devices = this._devices.wireless.devices;
 
                 for (let i = 0; i < devices.length; i++) {
-                    devices[i].device.request_scan(null);
+                    devices[i].device.request_scan_simple(null);
                 }
             }));
 
@@ -1797,36 +1775,42 @@
 
             // Device types
             this._dtypes = { };
-            this._dtypes[NM.DeviceType.ETHERNET] = NMDeviceWired;
-            this._dtypes[NM.DeviceType.WIFI] = NMDeviceWireless;
-            this._dtypes[NM.DeviceType.MODEM] = NMDeviceModem;
-            this._dtypes[NM.DeviceType.BT] = NMDeviceBluetooth;
+            this._dtypes[NetworkManager.DeviceType.ETHERNET] = NMDeviceWired;
+            this._dtypes[NetworkManager.DeviceType.WIFI] = NMDeviceWireless;
+            this._dtypes[NetworkManager.DeviceType.MODEM] = NMDeviceModem;
+            this._dtypes[NetworkManager.DeviceType.BT] = NMDeviceBluetooth;
+            // TODO: WiMax support
 
             // Connection types
             this._ctypes = { };
-            this._ctypes[NM.SETTING_WIRELESS_SETTING_NAME] = NMConnectionCategory.WIRELESS;
-            this._ctypes[NM.SETTING_WIRED_SETTING_NAME] = NMConnectionCategory.WIRED;
-            this._ctypes[NM.SETTING_PPPOE_SETTING_NAME] = NMConnectionCategory.WIRED;
-            this._ctypes[NM.SETTING_PPP_SETTING_NAME] = NMConnectionCategory.WIRED;
-            this._ctypes[NM.SETTING_BLUETOOTH_SETTING_NAME] = NMConnectionCategory.WWAN;
-            this._ctypes[NM.SETTING_CDMA_SETTING_NAME] = NMConnectionCategory.WWAN;
-            this._ctypes[NM.SETTING_GSM_SETTING_NAME] = NMConnectionCategory.WWAN;
-            this._ctypes[NM.SETTING_VPN_SETTING_NAME] = NMConnectionCategory.VPN;
-
-            this._readConnections();
-            this._readDevices();
-            this._syncNMState();
-
-            this._client.connect('notify::nm-running', Lang.bind(this, this._syncNMState));
-            this._client.connect('notify::networking-enabled', Lang.bind(this, this._syncNMState));
-            this._client.connect('notify::state', Lang.bind(this, this._syncNMState));
-            this._client.connect('notify::active-connections', Lang.bind(this, this._updateIcon));
-            this._client.connect('device-added', Lang.bind(this, this._deviceAdded));
-            this._client.connect('device-removed', Lang.bind(this, this._deviceRemoved));
-            this._client.connect('connection-added', Lang.bind(this, this._connectionAdded));
-            this._client.connect('connection-removed', Lang.bind(this, this._connectionRemoved));
-
-            this._inited = true;
+            this._ctypes[NetworkManager.SETTING_WIRELESS_SETTING_NAME] = NMConnectionCategory.WIRELESS;
+            this._ctypes[NetworkManager.SETTING_WIRED_SETTING_NAME] = NMConnectionCategory.WIRED;
+            this._ctypes[NetworkManager.SETTING_PPPOE_SETTING_NAME] = NMConnectionCategory.WIRED;
+            this._ctypes[NetworkManager.SETTING_PPP_SETTING_NAME] = NMConnectionCategory.WIRED;
+            this._ctypes[NetworkManager.SETTING_BLUETOOTH_SETTING_NAME] = NMConnectionCategory.WWAN;
+            this._ctypes[NetworkManager.SETTING_CDMA_SETTING_NAME] = NMConnectionCategory.WWAN;
+            this._ctypes[NetworkManager.SETTING_GSM_SETTING_NAME] = NMConnectionCategory.WWAN;
+            this._ctypes[NetworkManager.SETTING_VPN_SETTING_NAME] = NMConnectionCategory.VPN;
+
+            this._settings = NMClient.RemoteSettings.new(null);
+            this._connectionsReadId = this._settings.connect('connections-read', Lang.bind(this, function() {
+                this._readConnections();
+                this._readDevices();
+                this._syncNMState();
+
+                // Connect to signals late so that early signals don't find in inconsistent state
+                // and connect only once (this signal handler can be called again if NetworkManager goes up and down)
+                if (!this._inited) {
+                    this._inited = true;
+                    this._client.connect('notify::manager-running', Lang.bind(this, this._syncNMState));
+                    this._client.connect('notify::networking-enabled', Lang.bind(this, this._syncNMState));
+                    this._client.connect('notify::state', Lang.bind(this, this._syncNMState));
+                    this._client.connect('notify::active-connections', Lang.bind(this, this._updateIcon));
+                    this._client.connect('device-added', Lang.bind(this, this._deviceAdded));
+                    this._client.connect('device-removed', Lang.bind(this, this._deviceRemoved));
+                    this._settings.connect('new-connection', Lang.bind(this, this._newConnection));
+                }
+            }));
 
             this._periodicUpdateIcon();
 
@@ -1969,7 +1953,7 @@
 
             this._syncSectionTitle(wrapper.category);
         } else
-            log('Unknown network device type, is ' + device.get_device_type());
+            log('Invalid network device type, is ' + device.get_device_type());
     },
 
     _deviceRemoved: function(client, device) {
@@ -2027,12 +2011,24 @@
                 a._inited = true;
             }
 
-            if (!a._type) {
-                a._type = a.connection._type;
-                a._section = this._ctypes[a._type];
-                if (a._errorLogged) {
-                    log('network applet: Found connection for active');
-                    a._errorLogged = false;
+            if (!a._connection) {
+                a._connection = this._settings.get_connection_by_path(a.connection);
+
+                if (a._connection) {
+                    a._type = a._connection._type;
+                    a._section = this._ctypes[a._type];
+                    if (a._errorLogged) {
+                        log('network applet: Found connection for active');
+                        a._errorLogged = false;
+                    }
+                } else {
+                    a._connection = null;
+                    a._type = null;
+                    a._section = null;
+                    if (!a._errorLogged) {
+                        a._errorLogged = true;
+                        log('network applet: Cannot find connection for active (or connection cannot be read)');
+                    }
                 }
             }
 
@@ -2042,7 +2038,7 @@
                 continue;
             }
 
-            if (a.state == NM.ActiveConnectionState.ACTIVATED) {
+            if (a.state == NetworkManager.ActiveConnectionState.ACTIVATED) {
                 if (!default_ip4) {
                     // We didn't find the default IPV4 device yet..
                     if (!default_ip6) {
@@ -2057,7 +2053,7 @@
                     }
                 }
             }
-            if (a.state == NM.ActiveConnectionState.ACTIVATING) {
+            if (a.state == NetworkManager.ActiveConnectionState.ACTIVATING) {
                 activating = a;
             }
 
@@ -2069,7 +2065,7 @@
             }
 
             if (!a._primaryDevice) {
-                if (a._type != NM.SETTING_VPN_SETTING_NAME) {
+                if (a._type != NetworkManager.SETTING_VPN_SETTING_NAME) {
                     // find a good device to be considered primary
                     a._primaryDevice = null;
                     let devices = a.get_devices() || [ ];
@@ -2086,7 +2082,7 @@
                 if (a._primaryDevice)
                     a._primaryDevice.setActiveConnection(a);
 
-                if (a.state == NM.ActiveConnectionState.ACTIVATED &&
+                if (a.state == NetworkManager.ActiveConnectionState.ACTIVATED &&
                     a._primaryDevice && a._primaryDevice._notification) {
                         a._primaryDevice._notification.destroy();
                         a._primaryDevice._notification = null;
@@ -2098,7 +2094,7 @@
     },
 
     _notifyActivated: function(activeConnection) {
-        if (activeConnection.state == NM.ActiveConnectionState.ACTIVATED &&
+        if (activeConnection.state == NetworkManager.ActiveConnectionState.ACTIVATED &&
             activeConnection._primaryDevice && activeConnection._primaryDevice._notification) {
                 activeConnection._primaryDevice._notification.destroy();
                 activeConnection._primaryDevice._notification = null;
@@ -2108,27 +2104,29 @@
     },
 
     _readConnections: function() {
-        let connections = this._client.get_connections();
+        let connections = this._settings.list_connections();
         for (let i = 0; i < connections.length; i++) {
             let connection = connections[i];
             if (connection._uuid) {
                 // connection was already seen (for example because NetworkManager was restarted)
                 continue;
             }
-            connection._updatedId = connection.connect('changed', Lang.bind(this, this._updateConnection));
+            connection._removedId = connection.connect('removed', Lang.bind(this, this._connectionRemoved));
+            connection._updatedId = connection.connect('updated', Lang.bind(this, this._updateConnection));
 
             this._updateConnection(connection);
             this._connections.push(connection);
         }
     },
 
-    _connectionAdded: function(client, connection) {
+    _newConnection: function(settings, connection) {
         if (connection._uuid) {
             // connection was already seen
             return;
         }
 
-        connection._updatedId = connection.connect('changed', Lang.bind(this, this._updateConnection));
+        connection._removedId = connection.connect('removed', Lang.bind(this, this._connectionRemoved));
+        connection._updatedId = connection.connect('updated', Lang.bind(this, this._updateConnection));
 
         this._updateConnection(connection);
         this._connections.push(connection);
@@ -2136,7 +2134,7 @@
         this._updateIcon();
     },
 
-    _connectionRemoved: function(client, connection) {
+    _connectionRemoved: function(connection) {
         let pos = this._connections.indexOf(connection);
         if (pos != -1)
             this._connections.splice(pos);
@@ -2154,11 +2152,12 @@
         }
 
         connection._uuid = null;
+        connection.disconnect(connection._removedId);
         connection.disconnect(connection._updatedId);
     },
 
     _updateConnection: function(connection) {
-        let connectionSettings = connection.get_setting_by_name(NM.SETTING_CONNECTION_SETTING_NAME);
+        let connectionSettings = connection.get_setting_by_name(NetworkManager.SETTING_CONNECTION_SETTING_NAME);
         connection._type = connectionSettings.type;
 
         connection._section = this._ctypes[connection._type] || NMConnectionCategory.INVALID;
@@ -2204,7 +2203,7 @@
     },
 
     _syncNMState: function() {
-        if (!this._client.nm_running) {
+        if (!this._client.manager_running) {
             log('NetworkManager is not running, hiding...');
             this.menu.close();
             this.actor.hide();
@@ -2234,7 +2233,7 @@
             if (!mc) {
                 this._setIcon('network-offline');
                 this.set_applet_tooltip(_("No connection"));
-            } else if (mc.state == NM.ActiveConnectionState.ACTIVATING) {
+            } else if (mc.state == NetworkManager.ActiveConnectionState.ACTIVATING) {
                 new_delay = FAST_PERIODIC_UPDATE_FREQUENCY_SECONDS;
                 switch (mc._section) {
                 case NMConnectionCategory.WWAN:
@@ -2276,7 +2275,7 @@
                             this.set_applet_tooltip(_("Connected to the wireless network"));
                         } else {
                             this._setIcon('network-wireless-signal-' + signalToIcon(ap.strength));
-                            this.set_applet_tooltip(_("Wireless connection") + ": " + ssidToLabel(ap.get_ssid()) + " ("+ ap.strength +"%)");
+                            this.set_applet_tooltip(_("Wireless connection") + ": " + ap.get_ssid() + " ("+ ap.strength +"%)");
                         }
                     } else {
                         log('Active connection with no primary device?');
diff -Naur Cinnamon-fbb12f3/js/misc/modemManager.js Cinnamon-4.0.1/js/misc/modemManager.js
--- Cinnamon-fbb12f3/js/misc/modemManager.js	2018-12-02 09:29:17.000000000 +0100
+++ Cinnamon-4.0.1/js/misc/modemManager.js	2018-11-12 16:01:43.000000000 +0100
@@ -2,7 +2,7 @@
 
 const Gio = imports.gi.Gio;
 const Lang = imports.lang;
-const NMA = imports.gi.NMA;
+const NMGtk = imports.gi.NMGtk;
 const Signals = imports.signals;
 
 // _getMobileProvidersDatabase:
@@ -14,7 +14,7 @@
 function _getMobileProvidersDatabase() {
     if (_mpd == null) {
         try {
-            _mpd = new NMA.MobileProvidersDatabase();
+            _mpd = new NMGtk.MobileProvidersDatabase();
             _mpd.init(null);
         } catch (e) {
             log(e.message);
diff -Naur Cinnamon-fbb12f3/src/Makefile.am Cinnamon-4.0.1/src/Makefile.am
--- Cinnamon-fbb12f3/src/Makefile.am	2018-12-02 09:29:17.000000000 +0100
+++ Cinnamon-4.0.1/src/Makefile.am	2018-11-12 16:01:43.000000000 +0100
@@ -239,7 +239,7 @@
 libcinnamon_la_CPPFLAGS = $(MUFFIN_CFLAGS) $(cinnamon_cflags)
 
 Cinnamon-0.1.gir: libcinnamon.la St-1.0.gir
-Cinnamon_0_1_gir_INCLUDES = Clutter-0 ClutterX11-0 CoglPango-0 Cogl-0 Meta-Muffin.0 Soup-2.4 CMenu-3.0 NM-1.0
+Cinnamon_0_1_gir_INCLUDES = Clutter-0 ClutterX11-0 CoglPango-0 Cogl-0 Meta-Muffin.0 Soup-2.4 CMenu-3.0 NetworkManager-1.0 NMClient-1.0
 Cinnamon_0_1_gir_CFLAGS = $(libcinnamon_la_CPPFLAGS) -I $(srcdir)
 Cinnamon_0_1_gir_LIBS = libcinnamon.la
 Cinnamon_0_1_gir_FILES = $(libcinnamon_la_gir_sources)
