diff -Nur orig/files/usr/bin/cinnamon-settings-users mod/files/usr/bin/cinnamon-settings-users
--- orig/files/usr/bin/cinnamon-settings-users	2016-11-23 14:03:02.000000000 +0100
+++ mod/files/usr/bin/cinnamon-settings-users	2016-11-26 13:43:24.839523959 +0100
@@ -5,4 +5,4 @@
 
 import os
 
-os.system("gksu /usr/share/cinnamon/cinnamon-settings-users/cinnamon-settings-users.py")
+os.system("pkexec /usr/share/cinnamon/cinnamon-settings-users/cinnamon-settings-users.py")
