--- a/files/usr/share/cinnamon/cinnamon-settings/bin/Spices.py	2015-07-12 16:46:57.531901725 +0100
+++ b/files/usr/share/cinnamon/cinnamon-settings/bin/Spices.py	2015-07-12 19:07:57.780193298 +0100
@@ -387,8 +387,12 @@ class Spice_Harvester:
 
     def load_assets_thread(self, f, icon_path, url):
         valid = True
-
-        self.download(f, icon_path, url)
+        try:
+            urllib2.urlopen(url).getcode()
+        except:
+            valid = False
+        if valid:
+            self.download(f, icon_path, url)
 
         self.load_assets_done()
         thread.exit()
@@ -690,7 +694,6 @@ class Spice_Harvester:
         blockSize = 1024 * 8
         try:
             urlobj = urllib2.urlopen(url)
-            assert urlobj.getcode() == 200
         except Exception, detail:
             f.close()
             self.abort_download = ABORT_ERROR
