diff -Naur orig/imap/src/osdep/unix/env_unix.c patched/imap/src/osdep/unix/env_unix.c
--- orig/imap/src/osdep/unix/env_unix.c	2004-09-13 23:31:19.000000000 +0200
+++ patched/imap/src/osdep/unix/env_unix.c	2016-07-12 00:17:39.853624852 +0200
@@ -1074,20 +1074,7 @@
       }
       close (pi[0]); close (pi[1]);
     }
-				/* find directory/file delimiter */
-    if (s = strrchr (base->lock,'/')) {
-      *s = '\0';		/* tie off at directory */
-      sprintf(tmp,		/* generate default message */
-	      "Mailbox vulnerable - directory %.80s must have 1777 protection",
-	      base->lock);
-				/* definitely not 1777 if can't stat */
-      mask = stat (base->lock,&sb) ? 0 : (sb.st_mode & 1777);
-      *s = '/';			/* restore lock name */
-      if (mask != 1777) {	/* default warning if not 1777 */
-	if (!disableLockWarning) MM_LOG (tmp,WARN);
 	break;
-      }
-    }
   default:
     sprintf (tmp,"Mailbox vulnerable - error creating %.80s: %s",
 	     base->lock,strerror (errno));
