diff -Naur orig/pico/osdep/term.cap patched/pico/osdep/term.cap
--- orig/pico/osdep/term.cap	2004-07-22 00:06:24.000000000 +0200
+++ patched/pico/osdep/term.cap	2016-07-12 00:22:00.218113478 +0200
@@ -141,8 +141,8 @@
      */
     kpinsert("\033[4J", KEY_PGUP, 1);
     kpinsert("\033[3J", KEY_PGDN, 1);
-    kpinsert("\033[2J", KEY_HOME, 1);
-    kpinsert("\033[N",  KEY_END, 1);
+    kpinsert("\033[1~", KEY_HOME, 1);
+    kpinsert("\033[4~", KEY_END, 1);
 
     /*
      * vt220?
diff -Naur orig/pico/osdep/term.inf patched/pico/osdep/term.inf
--- orig/pico/osdep/term.inf	2004-07-22 00:06:24.000000000 +0200
+++ patched/pico/osdep/term.inf	2016-07-12 00:22:43.474705989 +0200
@@ -139,8 +139,8 @@
      */
     kpinsert("\033[4J", KEY_PGUP, 1);
     kpinsert("\033[3J", KEY_PGDN, 1);
-    kpinsert("\033[2J", KEY_HOME, 1);
-    kpinsert("\033[N",  KEY_END, 1);
+    kpinsert("\033[1~", KEY_HOME, 1);
+    kpinsert("\033[4~", KEY_END, 1);
 
     /*
      * vt220?
