diff -Naur tiff-4.0.0-orig/Makefile.in tiff-4.0.0/Makefile.in
--- tiff-4.0.0-orig/Makefile.in	2011-12-22 01:21:34.000000000 +0100
+++ tiff-4.0.0/Makefile.in	2012-03-13 18:09:35.000000000 +0100
@@ -328,8 +328,8 @@
 	nmake.opt \
 	libtiff-4.pc.in
 
-dist_doc_DATA = $(docfiles)
-SUBDIRS = port libtiff tools build contrib test man html
+dist_doc_DATA = #$(docfiles)
+SUBDIRS = port libtiff tools build contrib test man #html
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = libtiff-4.pc
 all: all-recursive
