diff -Naur wxWidgets-3.0.2-orig/src/stc/scintilla/src/Editor.cxx wxWidgets-3.0.2/src/stc/scintilla/src/Editor.cxx
--- wxWidgets-3.0.2-orig/src/stc/scintilla/src/Editor.cxx	2014-10-06 23:33:44.000000000 +0200
+++ wxWidgets-3.0.2/src/stc/scintilla/src/Editor.cxx	2016-11-11 09:24:13.729275217 +0100
@@ -10,7 +10,7 @@
 #include <stdio.h>
 #include <ctype.h>
 #include <assert.h>
-
+#include <cmath>
 #include <string>
 #include <vector>
 #include <map>
@@ -5841,9 +5841,9 @@
 }
 
 static bool Close(Point pt1, Point pt2) {
-	if (abs(pt1.x - pt2.x) > 3)
+	if (std::abs(pt1.x - pt2.x) > 3)
 		return false;
-	if (abs(pt1.y - pt2.y) > 3)
+	if (std::abs(pt1.y - pt2.y) > 3)
 		return false;
 	return true;
 }
