diff -Naur orig/data/lxappearance.desktop.in mod/data/lxappearance.desktop.in
--- orig/data/lxappearance.desktop.in	2014-10-14 17:23:11.000000000 +0200
+++ mod/data/lxappearance.desktop.in	2014-12-03 19:36:00.772092026 +0100
@@ -6,6 +6,6 @@
 _Keywords=windows;preferences;settings;theme;style;appearance;
 Icon=preferences-desktop-theme
 Exec=lxappearance
-NotShowIn=GNOME;KDE;XFCE;
+OnlyShowIn=LXDE;
 StartupNotify=true
 Categories=GTK;Settings;DesktopSettings;X-LXDE-Settings;
