diff -Naur orig/zapping.desktop.in.in patched/zapping.desktop.in.in
--- orig/zapping.desktop.in.in	2006-05-17 20:07:57.000000000 +0200
+++ patched/zapping.desktop.in.in	2016-02-20 13:44:18.808797264 +0100
@@ -1,14 +1,13 @@
 [Desktop Entry]
-Version=0.9
 Encoding=UTF-8
-_Name=Zapping TV Viewer
+_Name=Zapping
 _Comment=Lets you watch TV in a window
 Type=Application
 Exec=zapping
-Icon=zapping/gnome-television.png
+Icon=uhu/zapping.png
 Terminal=false
 StartupNotify=true
-Categories=GNOME;Application;Multimedia;AudioVideo;Video;TV;
+Categories=GNOME;Multimedia;AudioVideo;Video;TV;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=Zapping
 X-GNOME-Bugzilla-Version=@VERSION@
