diff -Naur zapping-0.10cvs6-orig/src/zvbi.c zapping-0.10cvs6/src/zvbi.c
--- zapping-0.10cvs6-orig/src/zvbi.c	2006-06-10 02:42:40.000000000 +0200
+++ zapping-0.10cvs6/src/zvbi.c	2012-05-11 09:53:14.000000000 +0200
@@ -25,7 +25,7 @@
 #include <gnome.h>
 #include <gdk-pixbuf/gdk-pixbuf.h>
 #include <libgnomevfs/gnome-vfs.h>
-#include <glib/gmessages.h>
+#include <glib.h>
 #include <pthread.h>
 #include <ctype.h>
 #include <time.h>
