--- totem.orig/data/totem.desktop.in.in.in	2009-03-16 11:40:06.000000000 +0100
+++ totem/data/totem.desktop.in.in.in	2009-03-20 22:02:43.000000000 +0100
@@ -1,12 +1,12 @@
 [Desktop Entry]
-_Name=Videos
+_Name=Totem
 _Comment=Play movies
 _Keywords=Video;Movie;Film;Clip;Series;Player;DVD;TV;Disc;
 Exec=totem %U
 Icon=totem
 Terminal=false
 Type=Application
-Categories=GTK;GNOME;AudioVideo;Player;Video;
+Categories=GTK;GNOME;AudioVideo;Player;Video;X-UHU-Menu/Multimedia/VideoPlayers;
 X-GNOME-DocPath=totem/totem.xml
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=totem
