diff -Naur multitail-5.2.8-orig/Makefile multitail-5.2.8/Makefile
--- multitail-5.2.8-orig/Makefile	2011-04-14 22:07:39.000000000 +0200
+++ multitail-5.2.8/Makefile	2011-05-13 19:23:00.962781115 +0200
@@ -27,7 +27,7 @@
 	### COPIED multitail.conf.new, YOU NEED TO REPLACE THE multitail.conf
 	### YOURSELF WITH THE NEW FILE
 	#
-	cp multitail.conf $(CONFIG_FILE).new
+	cp multitail.conf $(CONFIG_FILE)
 	mkdir -p $(DESTDIR)/etc/multitail/
 	cp convert-* colors-* $(DESTDIR)/etc/multitail/
 	rm -f $(DESTDIR)/usr/share/man/man1/multitail.1.gz
