diff -Naur libdbh2-5.0.16-orig/man/Makefile.am libdbh2-5.0.16/man/Makefile.am
--- libdbh2-5.0.16-orig/man/Makefile.am	2014-04-01 19:25:18.000000000 +0200
+++ libdbh2-5.0.16/man/Makefile.am	2015-01-11 22:13:06.272111632 +0100
@@ -51,11 +51,11 @@
 	     dbh_writeheader.3	
 
 
-if WANT_FREEBSD
-dbhman3dir = ${prefix}/man/man3
-else
+#if WANT_FREEBSD
+# dbhman3dir = ${prefix}/man/man3
+# else
 dbhman3dir = $(mandir)/man3
-endif
+# endif
 
 EXTRA_DIST =  $(dbhman3_DATA)
 
diff -Naur libdbh2-5.0.16-orig/man/Makefile.in libdbh2-5.0.16/man/Makefile.in
--- libdbh2-5.0.16-orig/man/Makefile.in	2014-10-06 22:09:06.000000000 +0200
+++ libdbh2-5.0.16/man/Makefile.in	2015-01-11 22:11:28.128351153 +0100
@@ -336,7 +336,7 @@
 	     dbh_writeheader.3	
 
 @WANT_FREEBSD_FALSE@dbhman3dir = $(mandir)/man3
-@WANT_FREEBSD_TRUE@dbhman3dir = ${prefix}/man/man3
+# @WANT_FREEBSD_TRUE@dbhman3dir = ${prefix}/man/man3
 EXTRA_DIST = $(dbhman3_DATA)
 all: all-am
 
