--- brasero/data/brasero.desktop.in.in.orig	2011-09-27 11:19:01.000000000 +0200
+++ brasero/data/brasero.desktop.in.in	2011-10-16 19:04:12.000000000 +0200
@@ -2,7 +2,7 @@
 _GenericName=Disc Burner and Copier
 _Comment=Create and copy CDs and DVDs
 _Keywords=disc;cdrom;dvd;burn;audio;video;
-Categories=GTK;GNOME;AudioVideo;Audio;Video;DiscBurning;
+Categories=GTK;GNOME;AudioVideo;Audio;Video;DiscBurning;X-UHU-Menu/Multimedia/CDwriters;
 MimeType=application/x-cd-image;application/x-cdrdao-toc;application/x-cue;application/x-toc;audio/x-scpls;audio/x-ms-asx;audio/x-mp3-playlist;audio/x-mpegurl;application/x-brasero;x-content/audio-cdda;x-content/video-dvd;x-content/video-vcd;x-content/video-svcd;x-content/image-picturecd;
 Exec=brasero %U
 Icon=brasero
