diff -Naur orig/udav/udav.desktop patched/udav/udav.desktop
--- orig/udav/udav.desktop	2016-02-13 20:03:22.000000000 +0100
+++ patched/udav/udav.desktop	2016-02-19 02:36:10.347091015 +0100
@@ -8,6 +8,7 @@
 MimeType=text/mgl;
 Comment=Data handling and plotting tool
 Comment[en_US]=Data handling and plotting tool
+Comment[hu]=Adat megjelenítő és rajzoló eszköz
 Comment[ru_RU]=Обработка и отображение данных
 Categories=Education;Science;Math;
-Keywords=Visualization;Data;Array;
\ No newline at end of file
+Keywords=Visualization;Data;Array;
