diff -Naurd a/udav/udav.desktop b/udav/udav.desktop
--- a/udav/udav.desktop	2013-11-11 11:46:33.000000000 +0100
+++ b/udav/udav.desktop	2013-11-11 19:46:27.000000000 +0100
@@ -1,5 +1,5 @@
 [Desktop Entry]
-Version=1.0
+Version=1.1
 Type=Application
 Terminal=false
 Icon=udav
@@ -8,4 +8,5 @@
 Comment=Data handling and plotting tool
 Comment[en_US]=Data handling and plotting tool
 Comment[ru_RU]=Обработка и отображение данных
-Categories=Education;Science;Math;
+Comment[hu]=Adat megjelenítő és rajzoló eszköz
+Categories=Education;Science;Math;X-UHU-Menu/Edutainment/Mathematics;
