diff -Naur orig/src-qt5/desktop-utils/lumina-pdf/mainUI.cpp patched/src-qt5/desktop-utils/lumina-pdf/mainUI.cpp
--- orig/src-qt5/desktop-utils/lumina-pdf/mainUI.cpp	2017-11-21 17:21:47.134739579 +0100
+++ patched/src-qt5/desktop-utils/lumina-pdf/mainUI.cpp	2017-11-25 16:38:08.993620249 +0100
@@ -126,7 +126,7 @@
   numPages = -1;
   DOC = TDOC; //Save this for later
   qDebug() << "Opening File:" << path;
-  this->setWindowTitle(DOC->title());
+  this->setWindowTitle(DOC->info("Title"));
   if(this->windowTitle().isEmpty()){ this->setWindowTitle(path.section("/",-1)); }
 
   //Setup the Document
