diff -Naur orig/gio/glib-compile-schemas.c patched/gio/glib-compile-schemas.c
--- orig/gio/glib-compile-schemas.c	2017-03-01 21:13:23.134741323 +0100
+++ patched/gio/glib-compile-schemas.c	2017-04-05 00:42:07.684289725 +0200
@@ -1217,12 +1217,6 @@
       return;
     }
 
-  if (path && (g_str_has_prefix (path, "/apps/") ||
-               g_str_has_prefix (path, "/desktop/") ||
-               g_str_has_prefix (path, "/system/")))
-    g_printerr ("warning: Schema '%s' has path '%s'.  Paths starting with "
-                "'/apps/', '/desktop/' or '/system/' are deprecated.\n", id, path);
-
   state->schema_state = schema_state_new (path, gettext_domain,
                                           extends, extends_name, list_of);
 
