diff -Naur orig/usr/sbin/update-ca-certificates patched/usr/sbin/update-ca-certificates
--- orig/usr/sbin/update-ca-certificates	2015-12-20 10:50:00.000000000 +0100
+++ patched/usr/sbin/update-ca-certificates	2015-12-21 00:09:47.864321545 +0100
@@ -117,7 +117,7 @@
   find . -type l -print | while read symlink
   do
     case $(readlink "$symlink") in
-      $CERTSDIR*|$LOCALCERTSDIR*) rm -f $symlink;;
+      "$CERTSDIR"*|"$LOCALCERTSDIR"*) rm -f $symlink;;
     esac
   done
   find . -type l -print | while read symlink
