diff -Naurd orig/src/default.h mod/src/default.h
--- orig/src/default.h	2009-01-25 15:39:51.000000000 +0100
+++ mod/src/default.h	2009-12-02 22:18:52.000000000 +0100
@@ -157,8 +157,8 @@
 XSV(const char *, terminalCommand,              "xterm")
 XSV(const char *, logoutCommand,                0)
 XSV(const char *, logoutCancelCommand,          0)
-XSV(const char *, shutdownCommand,              0)
-XSV(const char *, rebootCommand,                0)
+XSV(const char *, shutdownCommand,              "/usr/bin/poweroff")
+XSV(const char *, rebootCommand,                "/usr/bin/reboot")
 XIV(int, taskBarCPUDelay,                       500)
 XIV(int, taskBarNetSamples,                     20)
 XIV(int, taskBarNetDelay,                       500)
