diff -Nur orig/lib/toolbar.in mod/lib/toolbar.in
--- orig/lib/toolbar.in	2019-09-18 18:52:15.000000000 +0200
+++ mod/lib/toolbar.in	2019-11-13 17:34:03.166189274 +0100
@@ -3,6 +3,6 @@
 # Place your variants in @CFGDIR@ or in $HOME/.icewm since
 # modifications to this file will be overwritten when you reinstall icewm.
 #
-prog "@XTERMCMD@" utilities-terminal @XTERMCMD@
-prog "Web browser" web-browser xdg-open about:blank
-
+prog "Xterm terminál" utilities-terminal xterm
+prog "Web böngésző" web-browser xdg-open about:blank
+prog synaptic synaptic synaptic
