diff -Naur freetype-2.6.3-orig/include/freetype/freetype.h freetype-2.6.3/include/freetype/freetype.h
--- freetype-2.6.3-orig/include/freetype/freetype.h	2016-02-07 19:28:21.000000000 +0100
+++ freetype-2.6.3/include/freetype/freetype.h	2016-02-09 18:46:35.865734182 +0100
@@ -20,7 +20,7 @@
 #define FREETYPE_H_
 
 
-#ifndef FT_FREETYPE_H
+#if 0 /* #ifndef FT_FREETYPE_H */
 #error "`ft2build.h' hasn't been included yet!"
 #error "Please always use macros to include FreeType header files."
 #error "Example:"
