--- a/data/gnome-system-log.desktop.in.in.orig	2013-03-07 20:11:28.000000000 +0100
+++ a/data/gnome-system-log.desktop.in.in	2013-03-07 20:12:07.000000000 +0100
@@ -7,7 +7,7 @@
 Terminal=false
 Type=Application
 StartupNotify=true
-Categories=GTK;GNOME;System;Monitor;
+Categories=GTK;GNOME;System;Monitor;X-UHU-Menu/Tools/Utilities;
 NotShowIn=KDE;
 X-GNOME-DocPath=gnome-system-log/gnome-system-log.xml
 X-GNOME-Bugzilla-Bugzilla=GNOME
