diff -Naur ORIG/extra/xmoto.desktop UHU/extra/xmoto.desktop
--- ORIG/extra/xmoto.desktop	2008-11-23 10:32:34.000000000 +0100
+++ UHU/extra/xmoto.desktop	2009-05-02 12:48:54.000000000 +0200
@@ -4,6 +4,7 @@
 Name=Xmoto
 GenericName=Motocross game
 GenericName[fr_FR]=Jeu de moto
-Icon=xmoto.xpm
+Comment[hu]=Két dimenziós motokrossz
+Icon=/usr/share/pixmaps/xmoto.xpm
 Exec=xmoto
 Categories=Game;SportsGame;
