diff -Naurd a/share/applications/ecere.desktop b/share/applications/ecere.desktop
--- a/share/applications/ecere.desktop	2012-11-08 06:35:38.000000000 +0100
+++ b/share/applications/ecere.desktop	2012-11-10 15:44:27.000000000 +0100
@@ -1,11 +1,11 @@
 [Desktop Entry]
-Version=
 Name=Ecere IDE
 Type=Application
 Comment=Ecere IDE
+Comment[hu]=eC fejlesztői környezet
 Terminal=false
-Exec=ide
+Exec=ecere
 #Path=$(HOME)
 Icon=ecere
-Categories=Development
+Categories=Development;IDE;
 GenericName=Ecere IDE
