diff -Naur ecere-sdk-0.44.14-orig/compiler/bootstrap/libec/bootstrap/output.c ecere-sdk-0.44.14/compiler/bootstrap/libec/bootstrap/output.c
--- ecere-sdk-0.44.14-orig/compiler/bootstrap/libec/bootstrap/output.c	2016-01-13 22:46:41.000000000 +0100
+++ ecere-sdk-0.44.14/compiler/bootstrap/libec/bootstrap/output.c	2016-11-20 18:14:40.890465771 +0100
@@ -2009,7 +2009,7 @@
 
 __internal_ClassInst ? __internal_ClassInst->_vTbl : __ecereClass___ecereNameSpace__ecere__sys__File->_vTbl;
 })[__ecereVMethodID___ecereNameSpace__ecere__sys__File_Puts]);
-__internal_VirtualMethod ? __internal_VirtualMethod(f, " ") : (unsigned int)1;
+__internal_VirtualMethod ? __internal_VirtualMethod(f, ", ") : (unsigned int)1;
 }));
 OutputAttribute(attrib, f);
 }
diff -Naur ecere-sdk-0.44.14-orig/compiler/libec/src/output.ec ecere-sdk-0.44.14/compiler/libec/src/output.ec
--- ecere-sdk-0.44.14-orig/compiler/libec/src/output.ec	2016-01-13 22:46:41.000000000 +0100
+++ ecere-sdk-0.44.14/compiler/libec/src/output.ec	2016-11-20 18:16:50.646467292 +0100
@@ -890,7 +890,7 @@
       Attribute attrib;
       for(attrib = attr.attribs->first; attrib; attrib = attrib.next)
       {
-         if(attrib.prev) f.Puts(" ");
+         if(attrib.prev) f.Puts(", ");
          OutputAttribute(attrib, f);
       }
    }
