diff -Naupr ipvsadm-1.26.orig/Makefile ipvsadm-1.26/Makefile
--- ipvsadm-1.26.orig/Makefile	2011-02-08 01:24:23.000000000 +0100
+++ ipvsadm-1.26/Makefile	2011-07-09 23:22:54.834017156 +0200
@@ -63,15 +63,7 @@ RPMBUILD = $(shell				\
 	fi )
 
 ifeq (,$(FORCE_GETOPT))
-LIB_SEARCH = /lib64 /usr/lib64 /usr/local/lib64 /lib /usr/lib /usr/local/lib
-POPT_LIB = $(shell for i in $(LIB_SEARCH); do \
-  if [ -f $$i/libpopt.a ]; then \
-    if nm $$i/libpopt.a | fgrep -q poptGetContext; then \
-	echo "-lpopt"; \
-	break; \
-    fi; \
-  fi; \
-done)
+POPT_LIB = "-lpopt"
 endif
 
 ifneq (,$(POPT_LIB))
