diff -urN usr.orig/share/applications/flash-player-properties.desktop usr/share/applications/flash-player-properties.desktop
--- a/usr/share/applications/flash-player-properties.desktop	2011-05-06 01:27:53.000000000 +0200
+++ b/usr/share/applications/flash-player-properties.desktop	2011-05-16 12:05:44.520662675 +0200
@@ -7,6 +7,7 @@
 Comment[es]=Preferencias para Adobe Flash Player
 Comment[fr]=Préférences pour Adobe Flash Player
 Comment[gl]=Preferencias para Adobe Flash Player
+Comment[hu]=Adobe Flash Player beállításai
 Comment[nb]=Brukervalg for Adobe Flash Player
 Comment[nl]=Voorkeuren voor Adobe Flash Player
 Comment[pl]=Preferencje Adobe Flash Player
@@ -20,5 +21,5 @@
 Terminal=false
 Type=Application
 StartupNotify=true
-Categories=Settings;GNOME;GTK;X-GNOME-PersonalSettings;
+Categories=Settings;GNOME;GTK;X-GNOME-PersonalSettings;X-UHU-Menu/Settings
 NotShowIn=KDE;
