diff -Naur orig/configure patched/configure
--- orig/configure	2014-04-14 21:27:50.000000000 +0200
+++ patched/configure	2015-12-13 04:54:04.507717032 +0100
@@ -15100,7 +15100,7 @@
 	# ================================================================
 	# Terminal (vte)
 	# ================================================================
-	if `$PYTHON -c "import gi; gi.require_version('Vte', '2.90')" 2>/dev/null`;
+	if `$PYTHON -c "import gi; gi.require_version('Vte', '2.91')" 2>/dev/null`;
 	then
 		have_vte=yes
 		PLUGINS="$PLUGINS terminal"
diff -Naur orig/plugins/terminal/terminal.py patched/plugins/terminal/terminal.py
--- orig/plugins/terminal/terminal.py	2014-02-09 21:36:36.000000000 +0100
+++ patched/plugins/terminal/terminal.py	2015-12-13 04:53:20.807500334 +0100
@@ -21,7 +21,7 @@
 # Boston, MA  02110-1301  USA
 
 import gi
-gi.require_version('Vte', '2.90')
+gi.require_version('Vte', '2.91')
 from gi.repository import GObject, GLib, Gio, Pango, Gdk, Gtk, Gedit, Vte
 import os
 import gettext
@@ -36,7 +36,6 @@
 class GeditTerminal(Vte.Terminal):
 
     defaults = {
-        'emulation'             : 'xterm',
         'visible_bell'          : False,
     }
 
@@ -62,7 +61,7 @@
 
         self.reconfigure_vte()
 
-        self.fork_command_full(Vte.PtyFlags.DEFAULT, None, [Vte.get_user_shell()], None, GLib.SpawnFlags.SEARCH_PATH, None, None)
+        self.spawn_sync(Vte.PtyFlags.DEFAULT, None, [Vte.get_user_shell()], None, GLib.SpawnFlags.SEARCH_PATH, None, None, None)
 
     def do_drag_data_received(self, drag_context, x, y, data, info, time):
         if info == self.TARGET_URI_LIST:
@@ -138,7 +137,7 @@
                     palette = []
                     break
 
-        self.set_colors_rgba(fg, bg, palette)
+        self.set_colors(fg, bg, palette)
         self.set_cursor_blink_mode(self.profile_settings.get_enum("cursor-blink-mode"))
         self.set_cursor_shape(self.profile_settings.get_enum("cursor-shape"))
         self.set_audible_bell(self.profile_settings.get_boolean("audible-bell"))
@@ -146,7 +145,6 @@
         self.set_scroll_on_keystroke(self.profile_settings.get_boolean("scroll-on-keystroke"))
         self.set_scroll_on_output(self.profile_settings.get_boolean("scroll-on-output"))
         self.set_word_chars(self.profile_settings.get_string("word-chars"))
-        self.set_emulation(self.defaults['emulation'])
         self.set_visible_bell(self.defaults['visible_bell'])
 
         if self.profile_settings.get_boolean("scrollback-unlimited"):
@@ -201,7 +199,7 @@
         scrollbar.show()
         self.pack_start(scrollbar, False, False, 0)
 
-    def on_vte_child_exited(self, term):
+    def on_vte_child_exited(self, term, status):
         for child in self.get_children():
             child.destroy()
 
