diff -urN abiword-3.0.0.orig/abiword.desktop abiword-3.0.0/abiword.desktop
--- abiword-3.0.0.orig/abiword.desktop	2012-05-13 13:15:23.000000000 +0200
+++ abiword-3.0.0/abiword.desktop	2014-02-24 10:07:27.222710885 +0100
@@ -3,12 +3,14 @@
 Icon=abiword
 Terminal=false
 Type=Application
-Categories=Office;WordProcessor;GNOME;GTK;X-Red-Hat-Base;
+Categories=Office;WordProcessor;GNOME;GTK;X-Red-Hat-Base;X-UHU-Menu/Tools/Word_Processors;
 StartupNotify=true
 X-Desktop-File-Install-Version=0.9
 MimeType=application/x-abiword;text/x-abiword;text/x-xml-abiword;text/plain;application/msword;application/rtf;application/vnd.plain;application/xhtml+xml;text/html;application/x-crossmark;application/docbook+xml;application/x-t602;application/vnd.oasis.opendocument.text;application/vnd.oasis.opendocument.text-template;application/vnd.oasis.opendocument.text-web;application/vnd.sun.xml.writer;application/vnd.stardivision.writer;text/vnd.wap.wml;application/wordperfect6;application/wordperfect5.1;application/vnd.wordperfect;application/x-abicollab;
 Name=AbiWord
 GenericName=Word Processor
+GenericName[hu]=Szövegszerkesztő
+Comment[hu]=Szöveges dokumentumok készítése, szerkesztése, megjelenítése
 Comment=Compose, edit, and view documents
 Comment[de]=AbiWord Textverarbeitung
 GenericName[es]=Procesador de textos
