diff -Naur orig/src/af/util/xp/ut_jpeg.cpp patched/src/af/util/xp/ut_jpeg.cpp
--- orig/src/af/util/xp/ut_jpeg.cpp	2013-04-07 15:53:03.000000000 +0200
+++ patched/src/af/util/xp/ut_jpeg.cpp	2014-03-24 22:40:59.840097556 +0100
@@ -102,7 +102,7 @@
 	src->pub.next_input_byte = src->sourceBuf->getPointer (src->pos);
 	src->pub.bytes_in_buffer = src->sourceBuf->getLength ();
 	
-	return TRUE;
+	return (boolean)TRUE;
 }
 
 /*
@@ -161,7 +161,7 @@
 	/* set the data source */
 	_JPEG_ByteBufSrc (&cinfo, pBB);
 
-	jpeg_read_header(&cinfo, TRUE);
+	jpeg_read_header(&cinfo, (boolean)TRUE);
 	jpeg_start_decompress(&cinfo);
     iImageWidth = cinfo.output_width;
     iImageHeight = cinfo.output_height;
@@ -169,7 +169,7 @@
 	/* This is an important step since it will release a good deal of memory. */
 	jpeg_destroy_decompress(&cinfo);
 
-    return true;
+    return (boolean)TRUE;
 }
 
 // Converts the JPEG image data in pBB into raw RGB data,
@@ -189,7 +189,7 @@
 	/* set the data source */
 	_JPEG_ByteBufSrc (&cinfo, pBB);
 
-	jpeg_read_header(&cinfo, TRUE);
+	jpeg_read_header(&cinfo, (boolean)TRUE);
 	jpeg_start_decompress(&cinfo);
     
 	int row_stride = cinfo.output_width * cinfo.output_components;
@@ -266,7 +266,7 @@
 	/* This is an important step since it will release a good deal of memory. */
 	jpeg_destroy_decompress(&cinfo);
 
-    return true;
+    return (boolean)TRUE;
 }
 
 static void _JPEG_ByteBufSrc (j_decompress_ptr cinfo, const UT_ByteBuf* sourceBuf)
